---
title: "Squirrel_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---
### Load Libraries
```{r}
library('readr')
#source('http://portal.nersc.gov/dna/RD/Metagenome_RD/MetaBAT/Files/benchmark.R')
  # https://bitbucket.org/berkeleylab/metabat/wiki/Best%20Binning%20Practices
#library('limma')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
library('phangorn')
library('vegan')
library('pheatmap')
library('RColorBrewer')
library('gplots')
library('tidyr')
library('dplyr')
source('Scripts/misc.r')
theme_set(theme_bw())
set.seed(1)
```

### Load Functions
```{r}
source('Scripts/misc.R')

### DESeq2
deSEQ <- function(data, valuetest, cutoff = 0, alpha = 0.05){
  data_pruned <- prune_taxa(taxa_sums(data) > cutoff, data)
  de_data <- phyloseq_to_deseq2(data_pruned, valuetest)
  de_data2 <- DESeq(de_data, test="Wald", fitType="local")
  res_data <- results(de_data2, cooksCutoff = FALSE, pAdjustMethod = "BH")
  sig_data <- res_data[which(res_data$padj < alpha), ]
  sigtab_sherm <- cbind(as(sig_data, "data.frame"), as(tax_table(data_pruned)[rownames(sig_data), ], "matrix"))
}

### Simper
simper.pretty = function(x, metrics, interesting, perc_cutoff, low_cutoff, low_val, output_name){
  library(vegan)
  #handling otu tables for taxa levels
  save=list(0)
  if(grepl("Otu", colnames(x)[1])!=TRUE){
    #converts output from A.Neumann Taxonomy script
    save=list(58)
    x=as.data.frame(t(x))
    orig_names=colnames(x)
    new_names=list()
    l=1
    for(n in colnames(x)){
      ifelse((l<10), (colnames(x)[l]=c(paste0('Otu000',c(l)))), (colnames(x)[l]=c(paste0('Otu00',c(l))))) 
      new_names=append(new_names, colnames(x)[l])
      l=l+1
    }
    orig_names=as.data.frame(orig_names, row.names = as.character(new_names))
  }
  #running simper
  for(variables in interesting){
    test_1=with(metrics, simper(x, metrics[[variables]]))
    #parsing through simper output, saving desired info to table
    for(name in names(test_1)){
      testmx=matrix(ncol=length(interesting))
      testmx=cbind(test_1[[name]]$ord,test_1[[name]]$cusum)
      sorted=testmx[order(testmx[,1]),]
      sorted=cbind(sorted,test_1[[name]]$species)
      sorted=sorted[order(sorted[,2]),]
      t=matrix(sorted[sorted[,2]<=perc_cutoff,],ncol=3)
      i=nrow(t)
      #converting percents to percent of whole
      while(i>1){
        t[i,2]=as.character(as.numeric(t[i,2])-as.numeric(t[i-1,2]))
        i=i-1
      }
      t[,1]=name
      write.table(t,file=paste(output_name,'_simper.csv',sep=""), append=TRUE, sep=",", col.names = FALSE)
    }}
  y=read.table(paste(output_name,'_simper.csv',sep=""), header=FALSE,sep=",",fill = TRUE,row.names = NULL)
  file.remove(paste(output_name,'_simper.csv',sep = ""))
  y=y[-c(1)]
  colnames(y) = c("Comparison", "SIMPER", "OTU")
  #removing results lower than low cutoff
  if(low_cutoff=='y'){
    y=y[!(as.numeric(as.character(y$SIMPER))<low_val),]
  }
  #prevents changing of colnames if OTU table
  if(58 %in% save){
    y$OTU=orig_names[match(y$OTU, rownames(orig_names)),1]
  }
  write.csv(y,file=paste(output_name,'_clean_simper.csv', sep=''))
}


### Kruskal-Wallis for Simper
kruskal.pretty = function(otu, metrics, csv, interesting, output_name, taxonomy){
  library(vegan)
  library(dplyr)
  if(grepl("Otu", colnames(otu)[1])!=TRUE){
    #converts output from A.Neuman Taxonomy script
    otu=as.data.frame(t(otu))
  }
  #changing csv$X to rownames to allow proper splitting of comparisons
  csv$X=as.integer(rownames(csv))
  L=list()
  R=list()
  mean_L=c()
  sd_L=c()
  mean_R=c()
  sd_R=c()
  L_mean=c()
  R_mean=c()
  L_sd=c()
  R_sd=c()
  krusk=c()
  tax=c()
  L_abund=c()
  R_abund=c()
  L_abund_sd=c()
  R_abund_sd=c()
  abund=as.matrix(otu)
  abund=abund/rowSums(abund)
  for(b in levels(csv$Comparison)){
    otu_list=dplyr::filter(csv, Comparison==b) #saves otu list for current comparison
    for(i in csv$X){
      if(as.character(csv$Comparison[i])==b){  ##splitting comparisons so can call individually for table generation
        splt=as.data.frame(matrix(unlist(strsplit(as.character(csv$Comparison[i]),'_')), nrow=1, byrow=T))
        cola=as.character(splt[1,1])
        colb=as.character(splt[1,2])
        break
      }
    }
    #saving topic containing var of interest (cola/colb) (less memory intensive)
    for(topic in interesting){
      #preventing crash if there is only one topic in interesting
      if(is.null(levels(metrics[[topic]]))==TRUE){
        topic1=topic
        break
      }
      for(sbtpic in levels(metrics[[topic]])){
        if(sbtpic==cola){
          topic1=topic
          break
        }
      } 
    }
    #iterate thru rows in tpics of intrst til matches cola and colb, generates otu and metrics tbl  ##!Processing can be reduced!##
    for(rowe1 in metrics[[topic1]]){
      for(rowe2 in metrics[[topic1]]){ 
        if(rowe1==cola & rowe2==colb){ 
          listbact=otu[c(metrics[[topic1]]==cola|metrics[[topic1]]==colb),]
          listmet=metrics[c(metrics[[topic1]]==cola|metrics[[topic1]]==colb),]
          break
        }
      }
    }
    #collecting differential abundances
    sample_L=row.names(subset(metrics, metrics[[topic1]] == c(cola)))
    sample_R=row.names(subset(metrics, metrics[[topic1]] == c(colb)))
    #collecting abund values, perform/save mean and stdev calculations
    for(otus in otu_list$OTU){
      for(sample in sample_L){
        L=append(L,abund[sample,otus])
        mean_L[[otus]]=mean(as.numeric(L))
        sd_L[[otus]]=sd(as.numeric(L))
      }
      for(sample in sample_R){
        R=append(R,abund[sample,otus])
        mean_R[[otus]]=mean(as.numeric(R))
        sd_R[[otus]]=sd(as.numeric(R))
      }
      L=list()
      R=list()
    }
    #runs kruskal.test for each otu in simper csv, stores as list, also stores abundances
    for(otus in otu_list$OTU){
      result=kruskal.test(listbact[[otus]]~listmet[[topic1]])
      krusk=append(krusk, result$p.value)
      #stores taxonomic classification for each otu as list
      if(missing(taxonomy)){
        tax=append(tax, c("NA"))
      } else {
        tax=append(tax, as.character(taxonomy[otus, "Taxonomy"]))
      }
      L_mean=append(L_mean, as.character(mean_L[[otus]]))
      R_mean=append(R_mean, as.character(mean_R[[otus]]))
      L_sd=append(L_sd, as.character(sd_L[[otus]]))
      R_sd=append(R_sd, as.character(sd_R[[otus]]))
      
    }
  }
  #adjusted p-values for multiple comparisons
  fdr=p.adjust(krusk, method='fdr')
  #order csv to match 'krusk'/'fdr' list, add p.val, add taxonomy, re-ord to match orig csv, write to csv
  o_csv=dplyr::arrange(csv, Comparison)
  o_csv[,5]=krusk
  o_csv[,6]=fdr
  o_csv[,7]=tax
  o_csv[,8]=L_mean
  o_csv[,9]=L_sd
  o_csv[,10]=R_mean
  o_csv[,11]=R_sd
  o_csv=dplyr::arrange(o_csv, X)
  colnames(o_csv)[which(names(o_csv) == "V5")] <- "krusk_p.val" #changes column header
  colnames(o_csv)[which(names(o_csv) == "V6")] <- "fdr_krusk_p.val"
  colnames(o_csv)[which(names(o_csv) == "V7")] <- "Taxonomy"
  colnames(o_csv)[which(names(o_csv) == "V8")] <- "Left mean abund"
  colnames(o_csv)[which(names(o_csv) == "V9")] <- "Left stdev"
  colnames(o_csv)[which(names(o_csv) == "V10")] <- "Right mean abund"
  colnames(o_csv)[which(names(o_csv) == "V11")] <- "Right stdev"
  o_csv[,1]=NULL
  write.csv(o_csv, file=paste(output_name,"_krusk_simper.csv", sep=""))
}

### NMDS Ellipses
veganCovEllipse <- function(cov, center = c(0, 0), scale = 1, npoints = 100) {
    theta <- (0:npoints) * 2 * pi/npoints
    Circle <- cbind(cos(theta), sin(theta))
    t(center + scale * t(Circle %*% chol(cov)))
}

```

### Kaiju - Taxonomic Classification
```{r}
########## Domain ##########
# Load files
domain <- read.csv('kaiju/nohost/domain.csv', header=T)
domain$Season <- ordered(domain$Season, levels=c("Summer", "Winter", "Spring"))
meta <- read.csv('metadata.csv', header=T)
meta$Season <- ordered(meta$Season, levels=c("Summer", "Winter", "Spring"))

# Calculate relative abundance for total sequences
domain.ra <- domain
domain.ra$Archaea <- (domain.ra$Archaea / domain.ra$Total_Sequences)
domain.ra$Bacteria <- (domain.ra$Bacteria / domain.ra$Total_Sequences)
domain.ra$Eukaryota <- (domain.ra$Eukaryota / domain.ra$Total_Sequences)
domain.ra$Viruses <- (domain.ra$Viruses / domain.ra$Total_Sequences)
domain.ra$Unclassified <- (domain.ra$Total_Unclassified / domain.ra$Total_Sequences)

# Calculate relative abundance for classified sequences
domain.clas.ra <- domain
domain.clas.ra$Archaea <- (domain.clas.ra$Archaea / domain.clas.ra$Total_Classified)
domain.clas.ra$Bacteria <- (domain.clas.ra$Bacteria / domain.clas.ra$Total_Classified)
domain.clas.ra$Eukaryota <- (domain.clas.ra$Eukaryota / domain.clas.ra$Total_Classified)
domain.clas.ra$Viruses <- (domain.clas.ra$Viruses / domain.clas.ra$Total_Classified)

# Summarize all sequences
domain.sum <- domain.ra %>%
  group_by(Season) %>%
  summarize(Archaea_Mean = mean(Archaea)*100,
            Bacteria_Mean = mean(Bacteria)*100,
            Eukaryota_Mean = mean(Eukaryota)*100,
            Viruses_Mean = mean(Viruses)*100,
            Unclassified_Mean = mean(Unclassified)*100,
            Archaea_SD = sd(Archaea)*100,
            Bacteria_SD = sd(Bacteria)*100,
            Eukaryota_SD = sd(Eukaryota)*100,
            Viruses_SD = sd(Viruses)*100,
            Unclassified_SD = sd(Unclassified)*100,
            Archaea_SE = se(Archaea)*100,
            Bacteria_SE = se(Bacteria)*100,
            Eukaryota_SE = se(Eukaryota)*100,
            Viruses_SE = se(Viruses)*100,
            Unclassified_SE = se(Unclassified)*100)

# Summarize only classified sequences
domain.clas.sum <- domain.clas.ra %>%
  group_by(Season) %>%
  summarize(Archaea_Mean = mean(Archaea)*100,
            Bacteria_Mean = mean(Bacteria)*100,
            Eukaryota_Mean = mean(Eukaryota)*100,
            Viruses_Mean = mean(Viruses)*100,
            Archaea_SD = sd(Archaea)*100,
            Bacteria_SD = sd(Bacteria)*100,
            Eukaryota_SD = sd(Eukaryota)*100,
            Viruses_SD = sd(Viruses)*100,
            Archaea_SE = se(Archaea)*100,
            Bacteria_SE = se(Bacteria)*100,
            Eukaryota_SE = se(Eukaryota)*100,
            Viruses_SE = se(Viruses)*100)

# Create domain all seqs dataframe for plotting
domain.df <- data.frame(matrix(NA, nrow=15, ncol=5))
colnames(domain.df) <- c("Season", "Domain", "Mean", "SD", "SE")
domain.df$Season <- c(rep("Summer", 5), rep("Winter", 5), rep("Spring", 5))
domain.df$Season <- ordered(domain.df$Season, levels=c("Summer", "Winter", "Spring"))
domain.df$Domain <- c(rep(c("Archaea", "Bacteria", "Eukaryota", "Viruses", "Unclassified"), 3))
domain.df$Domain <- ordered(domain.df$Domain, levels=c("Archaea", "Bacteria", "Eukaryota", "Viruses",
                                                       "Unclassified"))
domain.df$Mean <- as.numeric(c(domain.sum[1,2:6], domain.sum[2, 2:6], domain.sum[3, 2:6]))
domain.df$SD <- as.numeric(c(domain.sum[1, 7:11], domain.sum[2, 7:11], domain.sum[3, 7:11]))
domain.df$SE <- as.numeric(c(domain.sum[1, 12:16], domain.sum[2, 12:16], domain.sum[3, 12:16]))

# Create domain classified seqs dataframe for plotting
domain.clas.df <- data.frame(matrix(NA, nrow=12, ncol=5))
colnames(domain.clas.df) <- c("Season", "Domain", "Mean", "SD", "SE")
domain.clas.df$Season <- c(rep("Summer", 4), rep("Winter", 4), rep("Spring", 4))
domain.clas.df$Season <- ordered(domain.clas.df$Season, levels=c("Summer", "Winter", "Spring"))
domain.clas.df$Domain <- c(rep(c("Archaea", "Bacteria", "Eukaryota", "Viruses"), 3))
domain.clas.df$Domain <- ordered(domain.clas.df$Domain, levels=c("Archaea", "Bacteria", "Eukaryota",
                                                                "Viruses"))
domain.clas.df$Mean <- as.numeric(c(domain.clas.sum[1,2:5], domain.clas.sum[2, 2:5],
                                    domain.clas.sum[3, 2:5]))
domain.clas.df$SD <- as.numeric(c(domain.clas.sum[1, 6:9], domain.clas.sum[2, 6:9],
                                  domain.clas.sum[3, 6:9]))
domain.clas.df$SE <- as.numeric(c(domain.clas.sum[1, 10:13], domain.clas.sum[2, 10:13],
                                  domain.clas.sum[3, 10:13]))

# Barplot for all sequences
ggplot(domain.df, aes(x=Domain, y=Mean, fill=Domain)) +
  geom_bar(stat="identity") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SD), width=0.25) +
  scale_y_continuous(limits=c(-0.5,60)) +
  facet_grid(Season ~ .) +
  scale_fill_brewer(palette="Set1") +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))

# Barplot for classified sequences
ggplot(domain.clas.df, aes(x=Domain, y=Mean, fill=Domain)) +
  geom_bar(stat="identity") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SD), width=0.25) +
  scale_y_continuous(limits=c(-0.5,100)) +
  facet_grid(Season ~ .) +
  scale_fill_brewer(palette="Set1") +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))








test <- read.table("kaiju/3715.kaiju.table", sep="\t", header=T)
test1 <- separate(test, col=taxon_name, into=c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Empty"), sep=";", remove=T)
  # Last 3 rows will have missing pieces because these are Viruses + unclassified seq's
test1 <- test1[,-1]
test1 <- test1[,-3]
test1 <- test1[,-10]
```









###### EVERYTHING BELOW HERE IS OLD CODE ######
### KEGG Brite
```{r}
ko.comp <- read.csv('KEGG/ko_comp.csv', fileEncoding="UTF-8-BOM")
rownames(ko.comp) <- ko.comp[,1]
ko.comp <- ko.comp[,-1]

# Remove SPAdes
kegg <- ko.comp[-which(ko.comp$Assembler == "SPAdes"),]


### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
kegg.norm <- kegg[,-26:-29]
for(i in 1:9){
  kegg.norm[i,] <- round(kegg.norm[i,]*norm.factor[i,1])
}
kegg.norm$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")




### Summarize KEGG
sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
kegg.norm.sum <- sum[4:6,]
kegg.norm.sum[4:6,] <- win[4:6,]
kegg.norm.sum[7:9,] <- spr[4:6,]
colnames(kegg.norm.sum) <- colnames(sum)
kegg.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(kegg.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")

### Check for Normal distribution
norm.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.test <- shapiro.test(kegg.norm[,i])
  norm.pvals[i,1] <- temp.test$p.value
  rownames(norm.pvals)[i] <- colnames(kegg.norm)[i]
}
length(which(norm.pvals[,1] < 0.06))
rownames(norm.pvals)[which(norm.pvals[,1] < 0.06)]

### Overall Summary
kegg.all.sum <- data.frame(999)
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.all.sum[1,i] <- sum(kegg.norm[1:9,i])/9
  kegg.all.sum[2,i] <- sd(kegg.norm[1:9,i])
  kegg.all.sum[3,i] <- sd(kegg.norm[1:9,i])/(sqrt(9))
}
colnames(kegg.all.sum) <- colnames(kegg.norm)[1:25]
rownames(kegg.all.sum) <- c("Mean", "SD", "SE")

### Overall Rel Abund
kegg.relabund <- kegg.norm[,-1]
kegg.relabund <- kegg.relabund[,-25]
for(i in 1:9){
  kegg.relabund[i+9,] <- (kegg.relabund[i,] / kegg.relabund[i,24])*100
}
kegg.relabund <- kegg.relabund[-1:-9,]
rownames(kegg.relabund) <- rownames(kegg.norm)[1:9]

kegg.relabund.sum <- data.frame(999)
for(i in 1:(ncol(kegg.relabund))){
  kegg.relabund.sum[1,i] <- sum(kegg.relabund[1:3,i])/3
  kegg.relabund.sum[2,i] <- sd(kegg.relabund[1:3,i])
  kegg.relabund.sum[3,i] <- sd(kegg.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.relabund.sum) <- colnames(kegg.relabund)[1:24]
rownames(kegg.relabund.sum) <- c("Mean", "SD", "SE")
kegg.relabund.sum <- kegg.relabund.sum[,order(kegg.relabund.sum[1,], decreasing=T)]




### Seasonal Summary
kegg.sum.sum <- data.frame(999)
kegg.sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.sum.sum[1,i] <- sum(kegg.sum[1:3,i])/3
  kegg.sum.sum[2,i] <- sd(kegg.sum[1:3,i])
  kegg.sum.sum[3,i] <- sd(kegg.sum[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.sum) <- colnames(kegg.sum)[1:25]
rownames(kegg.sum.sum) <- c("Mean", "SD", "SE")
# Winter
kegg.win.sum <- data.frame(999)
kegg.win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.win.sum[1,i] <- sum(kegg.win[1:3,i])/3
  kegg.win.sum[2,i] <- sd(kegg.win[1:3,i])
  kegg.win.sum[3,i] <- sd(kegg.win[1:3,i])/(sqrt(3))
}
colnames(kegg.win.sum) <- colnames(kegg.win)[1:25]
rownames(kegg.win.sum) <- c("Mean", "SD", "SE")
#Spring
kegg.spr.sum <- data.frame(999)
kegg.spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.spr.sum[1,i] <- sum(kegg.spr[1:3,i])/3
  kegg.spr.sum[2,i] <- sd(kegg.spr[1:3,i])
  kegg.spr.sum[3,i] <- sd(kegg.spr[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.sum) <- colnames(kegg.spr)[1:25]
rownames(kegg.spr.sum) <- c("Mean", "SD", "SE")

### Seasonal Rel Abund
kegg.seas.relabund <- kegg.norm[,-1]
kegg.seas.relabund <- kegg.seas.relabund[,-25]
for(i in 1:9){
  kegg.seas.relabund[i+9,] <- (kegg.seas.relabund[i,] / kegg.seas.relabund[i,24])*100
}
kegg.seas.relabund <- kegg.seas.relabund[-1:-9,]
rownames(kegg.seas.relabund) <- rownames(kegg.norm)[1:9]

sum <- which(kegg.norm$Season == "Summer")
win <- which(kegg.norm$Season == "Winter")
spr <- which(kegg.norm$Season == "Spring")
sum.relabund <- kegg.seas.relabund[sum,]
win.relabund <- kegg.seas.relabund[win,]
spr.relabund <- kegg.seas.relabund[spr,]

# Summer
kegg.sum.relabund.sum <- data.frame(999)
for(i in 1:(ncol(sum.relabund))){
  kegg.sum.relabund.sum[1,i] <- sum(sum.relabund[1:3,i])/3
  kegg.sum.relabund.sum[2,i] <- sd(sum.relabund[1:3,i])
  kegg.sum.relabund.sum[3,i] <- sd(sum.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.relabund.sum) <- colnames(sum.relabund)[1:24]
rownames(kegg.sum.relabund.sum) <- c("Mean", "SD", "SE")
kegg.sum.relabund.sum <- kegg.sum.relabund.sum[,order(kegg.sum.relabund.sum[1,], decreasing=T)]

# Winter
kegg.win.relabund.sum <- data.frame(999)
for(i in 1:(ncol(win.relabund))){
  kegg.win.relabund.sum[1,i] <- sum(win.relabund[1:3,i])/3
  kegg.win.relabund.sum[2,i] <- sd(win.relabund[1:3,i])
  kegg.win.relabund.sum[3,i] <- sd(win.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.win.relabund.sum) <- colnames(win.relabund)[1:24]
rownames(kegg.win.relabund.sum) <- c("Mean", "SD", "SE")
kegg.win.relabund.sum <- kegg.win.relabund.sum[,order(kegg.win.relabund.sum[1,], decreasing=T)]

# Spring
kegg.spr.relabund.sum <- data.frame(999)
for(i in 1:(ncol(spr.relabund))){
  kegg.spr.relabund.sum[1,i] <- sum(spr.relabund[1:3,i])/3
  kegg.spr.relabund.sum[2,i] <- sd(spr.relabund[1:3,i])
  kegg.spr.relabund.sum[3,i] <- sd(spr.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.relabund.sum) <- colnames(spr.relabund)[1:24]
rownames(kegg.spr.relabund.sum) <- c("Mean", "SD", "SE")
kegg.spr.relabund.sum <- kegg.spr.relabund.sum[,order(kegg.spr.relabund.sum[1,], decreasing=T)]


### Prep for barplot
sum.trim <- kegg.sum.relabund.sum
sum.trim <- data.frame(t(sum.trim))
sum.trim <- sum.trim[-1:-2,]
sum.trim <- sum.trim[-3,]
sum.trim <- sum.trim[-4,]
sum.trim <- sum.trim[-6:-20,]
sum.trim$Season <- rep("Summer",5)
sum.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
sum.trim$se.max <- sum.trim$Mean + sum.trim$SE
sum.trim$se.min <- sum.trim$Mean - sum.trim$SE
win.trim <- kegg.win.relabund.sum
win.trim <- data.frame(t(win.trim))
win.trim <- win.trim[-1:-2,]
win.trim <- win.trim[-3,]
win.trim <- win.trim[-4,]
win.trim <- win.trim[-6:-20,]
win.trim$Season <- rep("Winter",5)
win.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
win.trim$se.max <- win.trim$Mean + win.trim$SE
win.trim$se.min <- win.trim$Mean - win.trim$SE
spr.trim <- kegg.spr.relabund.sum
spr.trim <- data.frame(t(spr.trim))
spr.trim <- spr.trim[-1:-2,]
spr.trim <- spr.trim[-3,]
spr.trim <- spr.trim[-4,]
spr.trim <- spr.trim[-6:-20,]
spr.trim$Season <- rep("Spring",5)
spr.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
spr.trim$se.max <- spr.trim$Mean + spr.trim$SE
spr.trim$se.min <- spr.trim$Mean - spr.trim$SE
kegg.barplot <- sum.trim
kegg.barplot[6:10,] <- win.trim
kegg.barplot[11:15,] <- spr.trim
kegg.barplot$Pathway <- ordered(kegg.barplot$Pathway, levels=c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism"))
kegg.barplot$Season <- ordered(kegg.barplot$Season, levels=c("Summer", "Winter", "Spring"))

##### BARPLOT! #####
tiff("kegg.barplot.tiff", width = 13, height = 7, units = "in", res = 300)
ggplot(kegg.barplot, aes(x=Pathway, y=Mean, fill=Pathway)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(. ~ Season) +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values = c("lightcoral", "lightskyblue", "palegreen", "plum", "khaki1")) +
  xlab("Pathway") +
  ylab("Mean Relative Abundance") +
  scale_y_continuous(limits=c(0,55), expand=c(0,0))+
  theme(axis.text.x = element_blank(),
        panel.spacing=unit(1, "lines"),
        axis.title = element_text(size=20),
        strip.text.x = element_text(size=14),
        strip.text.y = element_text(size=14),
        axis.text.y = element_text(size=12),
        legend.title = element_text(size = 14),
        legend.text = element_text(size=12))
dev.off()



### ANOVA
kegg.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.aov <- summary(aov(kegg.norm[,i] ~ Season, data=kegg.norm))
  kegg.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(kegg.pvals)[i] <- colnames(kegg)[i]
}
sig <- rownames(kegg.pvals)[which(kegg.pvals[,1] < 0.06)]
# 
# # Tukey HSD
# tukey.pvals <- data.frame(rep(999,4))
# cat <- which(colnames(kegg) == sig[1])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,1] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[2])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,2] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[3])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,3] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[4])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,4] <- temp.tukey$Season[,4]
# colnames(tukey.pvals) <- sig
# rownames(tukey.pvals) <- rownames(temp.tukey$Season)

# tukey.pvals <- data.frame(rep(999,4))
# for(i in 1:length(sig)){
#   cat <- which(colnames(kegg) == sig[i])
#   temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
#   temp.tukey <- TukeyHSD(temp.aov)
#   tukey.pvals[(1:3),i] <- temp.tukey$Season[,4]
#   rownames(tukey.pvals)[1:3,i] <- rownames(temp.tukey$Season)
# }

```


### dbcan
```{r}
dbcan.comp <- read.csv('dbcan/dbcan_comp.csv', fileEncoding="UTF-8-BOM")
rownames(dbcan.comp) <- dbcan.comp[,1]
dbcan.comp <- dbcan.comp[,-1]

# Remove SPAdes
dbcan <- dbcan.comp[-which(dbcan.comp$Assembler == "SPAdes"),]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
dbcan.norm <- dbcan[,-9:-10]
for(i in 1:8){
  dbcan.norm[i,] <- round(dbcan.norm[i,]*norm.factor[i,1])
}
dbcan.norm$Season <- dbcan$Season

### Check for Normal distribution
dbcan.norm.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.test <- shapiro.test(dbcan.norm[,i])
  dbcan.norm.pvals[i,1] <- temp.test$p.value
  rownames(dbcan.norm.pvals)[i] <- colnames(dbcan.norm)[i]
}
  # All normal

### Summarize dbcan
sum <- dbcan.norm[which(dbcan.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- dbcan.norm[which(dbcan.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- dbcan.norm[which(dbcan.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
dbcan.norm.sum <- sum[4:6,]
dbcan.norm.sum[4:6,] <- win[4:6,]
dbcan.norm.sum[7:9,] <- spr[4:6,]
colnames(dbcan.norm.sum) <- colnames(sum)
dbcan.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(dbcan.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")


### Prep for barplot ###
# barplot.df <- dbcan.norm.sum[,-1]
# barplot.df <- barplot.df[,-4]
# barplot.df <- barplot.df[,-5:-6]
# dbcan.barplot <- t(barplot.df[1,])
# dbcan.barplot[6:10,] <- t(barplot.df[4,])

##### BARPLOT! #####
dbcan.barplot <- read.csv('dbcan/barplot_df.csv', fileEncoding="UTF-8-BOM")
dbcan.barplot$Season <- ordered(dbcan.barplot$Season, levels=c("Summer", "Winter", "Spring"))
cbm <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate-Binding Modules"),]
ce <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate Esterases"),]
gh <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Glycoside Hydrolases"),]
pl <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Polysaccharide Lyases"),]

cbm.barplot <- ggplot(cbm, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate-Binding Modules") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) + #,
        #legend.title = element_text(size=16),
        #legend.text = element_text(size=14)) +
        guides(fill = F)

g_legend <- function(a.gplot){
   tmp <- ggplot_gtable(ggplot_build(a.gplot))
   leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
   legend <- tmp$grobs[[leg]]
   return(legend)}
legend <- g_legend(cbm.barplot)

ce.barplot <- ggplot(ce, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate Esterases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

gh.barplot <- ggplot(gh, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Glycoside Hydrolases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

pl.barplot <- ggplot(pl, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Polysaccharide Lyases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)



tiff("dbcan.barplot.tiff", width = 13, height = 8, units = "in", res = 300)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(2,3, widths=c(1,1,0.3), heights=c(1,1)))))
print(cbm.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(ce.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
print(gh.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=1))
print(pl.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=2))
legend$vp <- viewport(layout.pos.row = 1:2, layout.pos.col = 3)
grid.draw(legend)
dev.off()

### Overall Summary
dbcan.all.sum <- data.frame(999)
for(i in 1:(ncol(dbcan.norm)-1)){
  dbcan.all.sum[1,i] <- sum(dbcan.norm[1:9,i])/9
  dbcan.all.sum[2,i] <- sd(dbcan.norm[1:9,i])
  dbcan.all.sum[3,i] <- sd(dbcan.norm[1:9,i])/(sqrt(9))
}
colnames(dbcan.all.sum) <- colnames(dbcan.norm)[1:6]
rownames(dbcan.all.sum) <- c("Mean", "SD", "SE")

### ANOVA
dbcan.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.aov <- summary(aov(dbcan.norm[,i] ~ Season, data=dbcan.norm))
  dbcan.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(dbcan.pvals)[i] <- colnames(dbcan)[i]
}
sig <- rownames(dbcan.pvals)[which(dbcan.pvals[,1] < 0.06)]

```































### KEGG Pathways
```{r}
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(class) <- class[,1]
class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

### Calculate proportions
path.prop <- path.norm
for(i in 1:9){
  path.prop[,i] <- (path.prop[,i] / colSums(path.prop)[i])
}
rem <- which(rowSums(path.prop) < 0.001)
path.prop <- path.prop[-rem,]

kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(kend) <- kend[,1]
kend[,2:9] <- NA
colnames(kend) <- kend[,1]
kend[,1] <- NA

kend[1,1] <- cor(path.prop[,1], path.prop[,1], method = "kendall")
kend[1,2] <- cor(path.prop[,1], path.prop[,2], method = "kendall")
kend[1,3] <- cor(path.prop[,1], path.prop[,3], method = "kendall")
kend[1,4] <- cor(path.prop[,1], path.prop[,4], method = "kendall")
kend[1,5] <- cor(path.prop[,1], path.prop[,5], method = "kendall")
kend[1,6] <- cor(path.prop[,1], path.prop[,6], method = "kendall")
kend[1,7] <- cor(path.prop[,1], path.prop[,7], method = "kendall")
kend[1,8] <- cor(path.prop[,1], path.prop[,8], method = "kendall")
kend[1,9] <- cor(path.prop[,1], path.prop[,9], method = "kendall")
kend[2,1] <- cor(path.prop[,2], path.prop[,1], method = "kendall")
kend[2,2] <- cor(path.prop[,2], path.prop[,2], method = "kendall")
kend[2,3] <- cor(path.prop[,2], path.prop[,3], method = "kendall")
kend[2,4] <- cor(path.prop[,2], path.prop[,4], method = "kendall")
kend[2,5] <- cor(path.prop[,2], path.prop[,5], method = "kendall")
kend[2,6] <- cor(path.prop[,2], path.prop[,6], method = "kendall")
kend[2,7] <- cor(path.prop[,2], path.prop[,7], method = "kendall")
kend[2,8] <- cor(path.prop[,2], path.prop[,8], method = "kendall")
kend[2,9] <- cor(path.prop[,2], path.prop[,9], method = "kendall")
kend[3,1] <- cor(path.prop[,3], path.prop[,1], method = "kendall")
kend[3,2] <- cor(path.prop[,3], path.prop[,2], method = "kendall")
kend[3,3] <- cor(path.prop[,3], path.prop[,3], method = "kendall")
kend[3,4] <- cor(path.prop[,3], path.prop[,4], method = "kendall")
kend[3,5] <- cor(path.prop[,3], path.prop[,5], method = "kendall")
kend[3,6] <- cor(path.prop[,3], path.prop[,6], method = "kendall")
kend[3,7] <- cor(path.prop[,3], path.prop[,7], method = "kendall")
kend[3,8] <- cor(path.prop[,3], path.prop[,8], method = "kendall")
kend[3,9] <- cor(path.prop[,3], path.prop[,9], method = "kendall")
kend[4,1] <- cor(path.prop[,4], path.prop[,1], method = "kendall")
kend[4,2] <- cor(path.prop[,4], path.prop[,2], method = "kendall")
kend[4,3] <- cor(path.prop[,4], path.prop[,3], method = "kendall")
kend[4,4] <- cor(path.prop[,4], path.prop[,4], method = "kendall")
kend[4,5] <- cor(path.prop[,4], path.prop[,5], method = "kendall")
kend[4,6] <- cor(path.prop[,4], path.prop[,6], method = "kendall")
kend[4,7] <- cor(path.prop[,4], path.prop[,7], method = "kendall")
kend[4,8] <- cor(path.prop[,4], path.prop[,8], method = "kendall")
kend[4,9] <- cor(path.prop[,4], path.prop[,9], method = "kendall")
kend[5,1] <- cor(path.prop[,5], path.prop[,1], method = "kendall")
kend[5,2] <- cor(path.prop[,5], path.prop[,2], method = "kendall")
kend[5,3] <- cor(path.prop[,5], path.prop[,3], method = "kendall")
kend[5,4] <- cor(path.prop[,5], path.prop[,4], method = "kendall")
kend[5,5] <- cor(path.prop[,5], path.prop[,5], method = "kendall")
kend[5,6] <- cor(path.prop[,5], path.prop[,6], method = "kendall")
kend[5,7] <- cor(path.prop[,5], path.prop[,7], method = "kendall")
kend[5,8] <- cor(path.prop[,5], path.prop[,8], method = "kendall")
kend[5,9] <- cor(path.prop[,5], path.prop[,9], method = "kendall")
kend[6,1] <- cor(path.prop[,6], path.prop[,1], method = "kendall")
kend[6,2] <- cor(path.prop[,6], path.prop[,2], method = "kendall")
kend[6,3] <- cor(path.prop[,6], path.prop[,3], method = "kendall")
kend[6,4] <- cor(path.prop[,6], path.prop[,4], method = "kendall")
kend[6,5] <- cor(path.prop[,6], path.prop[,5], method = "kendall")
kend[6,6] <- cor(path.prop[,6], path.prop[,6], method = "kendall")
kend[6,7] <- cor(path.prop[,6], path.prop[,7], method = "kendall")
kend[6,8] <- cor(path.prop[,6], path.prop[,8], method = "kendall")
kend[6,9] <- cor(path.prop[,6], path.prop[,9], method = "kendall")
kend[7,1] <- cor(path.prop[,7], path.prop[,1], method = "kendall")
kend[7,2] <- cor(path.prop[,7], path.prop[,2], method = "kendall")
kend[7,3] <- cor(path.prop[,7], path.prop[,3], method = "kendall")
kend[7,4] <- cor(path.prop[,7], path.prop[,4], method = "kendall")
kend[7,5] <- cor(path.prop[,7], path.prop[,5], method = "kendall")
kend[7,6] <- cor(path.prop[,7], path.prop[,6], method = "kendall")
kend[7,7] <- cor(path.prop[,7], path.prop[,7], method = "kendall")
kend[7,8] <- cor(path.prop[,7], path.prop[,8], method = "kendall")
kend[7,9] <- cor(path.prop[,7], path.prop[,9], method = "kendall")
kend[8,1] <- cor(path.prop[,8], path.prop[,1], method = "kendall")
kend[8,2] <- cor(path.prop[,8], path.prop[,2], method = "kendall")
kend[8,3] <- cor(path.prop[,8], path.prop[,3], method = "kendall")
kend[8,4] <- cor(path.prop[,8], path.prop[,4], method = "kendall")
kend[8,5] <- cor(path.prop[,8], path.prop[,5], method = "kendall")
kend[8,6] <- cor(path.prop[,8], path.prop[,6], method = "kendall")
kend[8,7] <- cor(path.prop[,8], path.prop[,7], method = "kendall")
kend[8,8] <- cor(path.prop[,8], path.prop[,8], method = "kendall")
kend[8,9] <- cor(path.prop[,8], path.prop[,9], method = "kendall")
kend[9,1] <- cor(path.prop[,9], path.prop[,1], method = "kendall")
kend[9,2] <- cor(path.prop[,9], path.prop[,2], method = "kendall")
kend[9,3] <- cor(path.prop[,9], path.prop[,3], method = "kendall")
kend[9,4] <- cor(path.prop[,9], path.prop[,4], method = "kendall")
kend[9,5] <- cor(path.prop[,9], path.prop[,5], method = "kendall")
kend[9,6] <- cor(path.prop[,9], path.prop[,6], method = "kendall")
kend[9,7] <- cor(path.prop[,9], path.prop[,7], method = "kendall")
kend[9,8] <- cor(path.prop[,9], path.prop[,8], method = "kendall")
kend[9,9] <- cor(path.prop[,9], path.prop[,9], method = "kendall")

#write.table(kend, file="kegg.kendall.txt", sep="\t")

```

### Pathway UPGMA
```{r}
kend.fix <- kend
kend.fix[1,1] <- 0
kend.fix[2,2] <- 0
kend.fix[3,3] <- 0
kend.fix[4,4] <- 0
kend.fix[5,5] <- 0
kend.fix[6,6] <- 0
kend.fix[7,7] <- 0
kend.fix[8,8] <- 0
kend.fix[9,9] <- 0
path.upgma <- upgma(kend.fix)
plot(path.upgma)


path.nj <- NJ(data.frame(kend.fix))

```




### Pathway Bray-Curtis
```{r}
# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]
kegg.class <- kegg.class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
path.otu <- otu_table(path.norm, taxa_are_rows = T)
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

nmds.kegg <- ordinate(physeq=path.physeq, method="NMDS", distance="bray")
  # stress = 0.01481376

plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Season")
plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Diet")
plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Sex")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.bray<- phyloseq::distance(physeq=path.physeq, method="bray")

# Season
adonis(path.bray~ Season, data=path.sampdf)
  # p-value = 0.333
beta.tax = betadisper(d=path.bray, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.304

# Diet
adonis(path.bray~ Diet, data=path.sampdf)
  # p-value = 0.515
beta.tax = betadisper(d=path.bray, group=path.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.368

# Sex
adonis(path.bray~ Sex, data=path.sampdf)
  # p-value = 0.241
beta.tax = betadisper(d=path.bray, group=path.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.592

### ANOSIM
anosim(path.bray, meta$Season, permutations=1000)
  # p-value = 0.13986
anosim(path.bray, meta$Diet, permutations=1000)
  # p-value =  0.50849
anosim(path.bray, meta$Sex, permutations=1000)
  # p-value = 0.52747
```

### Pathway Jaccard
```{r}
jac.kegg <- ordinate(physeq=path.physeq, method="NMDS", distance="jaccard")
  # stress = 0.02570319
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Season")
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Diet")
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Sex")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.jac <- phyloseq::distance(physeq=path.physeq, method="jaccard")

# Season
adonis(path.jac~ Season, data=path.sampdf)
  # p-value = 0.235
beta.tax = betadisper(d=path.jac, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.367

# Diet
adonis(path.jac~ Diet, data=path.sampdf)
  # p-value = 0.456
beta.tax = betadisper(d=path.jac, group=path.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.372

# Diet
adonis(path.jac~ Sex, data=path.sampdf)
  # p-value = 0.248
beta.tax = betadisper(d=path.jac, group=path.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.61

### ANOSIM
anosim(path.jac, path.sampdf$Season, permutations=1000)
  # p-value = 0.14386
anosim(path.jac, path.sampdf$Diet, permutations=1000)
  # p-value = 0.5045
anosim(path.jac, path.sampdf$Sex, permutations=1000)
  # 0.52348
```

### Pathway SIMPER
```{r}
simper.pretty(path.norm, metrics = meta, interesting = c('Season','Diet', 'Sex'), perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'kegg')
path.simper <- read.csv("kegg_clean_simper.csv")
kruskal.pretty(otu = path.norm, metrics = meta, csv =  path.simper, interesting = c('Season','Diet', 'Sex'), output_name = 'kegg')
path.simper.kw <- read.csv("kegg_krusk_simper.csv")
```

### Pathway DESeq
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]

path.otu <- otu_table(path, taxa_are_rows = T)
#path.otu <- path.otu + 1
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.path <- rownames(test[sum,])
  # "ko04370" "ko04145"
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])

# Spring vs. Summer
ss.physeq <- subset_samples(path.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
  # "ko00650" "ko01051" "ko04122" "ko02010"
sum <- which(test$log2FoldChange > 0)
sum.path <- rownames(test[sum,])
  # "ko00061" "ko00540" "ko00740" "ko00130"

# Spring vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
  # "ko00650" "ko01051" "ko04122" "ko02010"
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])
  # "ko00061" "ko00540" "ko00740" "ko00130"

# Feed vs Fast
test <- deSEQ(data=path.physeq, valuetest= ~Diet)
fast <- which(test$log2FoldChange < 0)
fast.path <- rownames(test[spr,])
  # "ko00040"
feed <- which(test$log2FoldChange > 0)
feed.path <- rownames(test[win,])
  # "ko00190" "ko00940" "ko00405" "ko04145"

### Identify Overrep kegg families
kegg.feed.id <- data.frame(feed.path)
for(i in 1:length(feed.path)){
  test <- which(feed.path[i] == kegg.class[,1])
  kegg.feed.id[i,2] <- kegg.class[test,2]
  kegg.feed.id[i,3] <- kegg.class[test,3]
  kegg.feed.id[i,4] <- kegg.class[test,4]
}
#write.csv(kegg.feed.id, 'KEGG/kegg.feed.id.csv')
kegg.fast.id <- data.frame(fast.path[1])
test <- which(fast.path[1] == kegg.class[,1])
kegg.fast.id[1,2] <- kegg.class[test,2]
kegg.fast.id[1,3] <- kegg.class[test,3]
kegg.fast.id[1,4] <- kegg.class[test,4]
#write.csv(kegg.fast.id, 'KEGG/kegg.fast.id.csv')
  
# for(i in 1:length(fast.path[1])){
#   test <- which(fast.path[i] == kegg.class[,1])
#   kegg.fast.id[i,2] <- kegg.class[test,2]
#   kegg.fast.id[i,3] <- kegg.class[test,3]
#   kegg.fast.id[i,4] <- kegg.class[test,4]
# }
#write.csv(kegg.fast.id, 'kegg/kegg.fast.id.csv')

# Female vs Male
test <- deSEQ(data=path.physeq, valuetest= ~Sex)
fem <- which(test$log2FoldChange < 0)
fem.path <- rownames(test[fem,])
  # "ko00232" "ko03040" "ko03013" "ko03015" "ko03008" "ko04141" "ko03460" "ko04014" "ko04015" "ko04010" "ko04011" "ko04310" "ko04340" "ko04390" "ko04392" "ko04371" "ko04064" "ko04668" "ko04068" "ko04020" "ko04072" "ko04071" "ko04024" "ko04022" "ko04150" "ko04144" "ko04145" "ko04139" "ko04110" "ko04111" "ko04113" "ko04210" "ko04810"
mal <- which(test$log2FoldChange > 0)
mal.path <- rownames(test[mal,])
  # "ko00190" "ko00940" "ko00405" "ko04145"
```


### KEGG Heatmap
```{r}
# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]

path.otu <- otu_table(path, taxa_are_rows = T)
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
kegg.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)


kegg.de.data <- phyloseq_to_deseq2(kegg.physeq, ~Season)
kegg.de <- DESeq(kegg.de.data, test="Wald", fitType="local")

kegg.res <- results(kegg.de, pAdjustMethod = "BH")

### Extracting results: https://support.bioconductor.org/p/72765/

# Summer vs. Winter
kegg.res.su.wi <- results(kegg.de, contrast=c("Season", "Summer", "Winter"), pAdjustMethod = "BH")
kegg.su.wi.sig <- kegg.res.su.wi[which(kegg.res.su.wi$padj < 0.05),]
  # No sig!!!
# kegg.su.wi.sig.df <- cbind(as(kegg.su.wi.sig, "data.frame"),
#                           as(tax_table(kegg.physeq)[rownames(kegg.su.wi.sig), ], "matrix"))
# kegg.su.wi.sum.sig <- which(kegg.su.wi.sig.df$log2FoldChange < 0)
# kegg.su.wi.sum <- rownames(kegg.su.wi.sig[kegg.su.wi.sum.sig,])
#   # Summer
# kegg.su.wi.win.sig <- which(kegg.su.wi.sig.df$log2FoldChange > 0)
# kegg.su.wi.win <- rownames(kegg.su.wi.sig[kegg.su.wi.win.sig,])
#   # Winter

# Summer vs. Spring
kegg.res.su.sp <- results(kegg.de, contrast=c("Season", "Spring", "Summer"), pAdjustMethod = "BH")
kegg.su.sp.sig <- kegg.res.su.sp[which(kegg.res.su.sp$padj < 0.05),]
kegg.su.sp.sig.df <- cbind(as(kegg.su.sp.sig, "data.frame"),
                          as(tax_table(kegg.physeq)[rownames(kegg.su.sp.sig), ], "matrix"))
kegg.su.sp.spr.sig <- which(kegg.su.sp.sig.df$log2FoldChange < 0)
kegg.su.sp.spr <- rownames(kegg.su.sp.sig[kegg.su.sp.spr.sig,])
  # Spring
  # "ko00061" "ko00540" "ko00740" "ko00130"
kegg.su.sp.sum.sig <- which(kegg.su.sp.sig.df$log2FoldChange > 0)
kegg.su.sp.sum <- rownames(kegg.su.sp.sig[kegg.su.sp.sum.sig,])
  # Summer
  #  "ko00030" "ko00650" "ko01051" "ko04122" "ko02010"

# Spring vs Winter
kegg.res.sp.wi <- results(kegg.de, contrast=c("Season", "Spring", "Winter"), pAdjustMethod = "BH")
kegg.sp.wi.sig <- kegg.res.sp.wi[which(kegg.res.sp.wi$padj < 0.05),]
kegg.sp.wi.sig.df <- cbind(as(kegg.sp.wi.sig, "data.frame"),
                          as(tax_table(kegg.physeq)[rownames(kegg.sp.wi.sig), ], "matrix"))
kegg.sp.wi.spr.sig <- which(kegg.sp.wi.sig.df$log2FoldChange < 0)
kegg.sp.wi.spr <- rownames(kegg.sp.wi.sig[kegg.sp.wi.spr.sig,])
  # Spring
  # "ko00190" "ko00410" "ko00531" "ko00604" "ko00540" "ko00780" "ko00790" "ko00130"
kegg.sp.wi.win.sig <- which(kegg.sp.wi.sig.df$log2FoldChange > 0)
kegg.sp.wi.win <- rownames(kegg.sp.wi.sig[kegg.sp.wi.win.sig,])
  # Winter
  # "ko00040" "ko00051" "ko00500" "ko00660" "ko00290" "ko00460" "ko00940" "ko00405" "ko02010" "ko02060" "ko04145"

# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

### Pull out top 2 (largest log fold change while sig) for each comparison
# kegg.su.wi.sum.df <- kegg.su.wi.sig.df[kegg.su.wi.sum.sig,]
# kegg.su.wi.sum.df <- kegg.su.wi.sum.df[order(kegg.su.wi.sum.df$log2FoldChange),]
# kegg.su.wi.win.df <- kegg.su.wi.sig.df[kegg.su.wi.win.sig,]
# kegg.su.wi.win.df <- kegg.su.wi.win.df[order(kegg.su.wi.win.df$log2FoldChange),]

kegg.su.sp.spr.df <- kegg.su.sp.sig.df[kegg.su.sp.spr.sig,]
kegg.su.sp.spr.df <- kegg.su.sp.spr.df[order(kegg.su.sp.spr.df$log2FoldChange),]
kegg.su.sp.sum.df <- kegg.su.sp.sig.df[kegg.su.sp.sum.sig,]
kegg.su.sp.sum.df <- kegg.su.sp.sum.df[order(kegg.su.sp.sum.df$log2FoldChange),]
kegg.sp.wi.spr.df <- kegg.sp.wi.sig.df[kegg.sp.wi.spr.sig,]
kegg.sp.wi.spr.df <- kegg.sp.wi.spr.df[order(kegg.sp.wi.spr.df$log2FoldChange),]
kegg.sp.wi.win.df <- kegg.sp.wi.sig.df[kegg.sp.wi.win.sig,]
kegg.sp.wi.win.df <- kegg.sp.wi.win.df[order(kegg.sp.wi.win.df$log2FoldChange),]

kegg.sig <- rbind(kegg.su.sp.spr.df, kegg.su.sp.sum.df, kegg.sp.wi.spr.df, kegg.sp.wi.win.df)
kegg.sig$log2FoldChange[which(kegg.sig$log2FoldChange < 0)] <-
  kegg.sig$log2FoldChange[which(kegg.sig$log2FoldChange < 0)] * -1
kegg.sig <- kegg.sig[order(kegg.sig$log2FoldChange, decreasing=T),]
#kegg.sig <- kegg.sig[order(kegg.sig$padj),]
kegg.use <- rownames(kegg.sig)[1:10] 
kegg.use[5] <- "ko00540"
kegg.use <- kegg.use[-6]
kegg.use <- kegg.use[-7]
kegg.use[9:10] <- rownames(kegg.sig)[11:12]
kegg.use[10] <- rownames(kegg.sig)[13]
kegg.use <- kegg.use[-9]
kegg.use[10] <- rownames(kegg.sig)[14]

#select <- order(rowMeans(counts(kegg.de, normalized=T)), decreasing=T)[1:20]
  # For top 20 keggs
kegg.rows <- rownames(kegg.de)
kegg.select <- rep(1,10)
for(i in 1:length(kegg.use)){
  save <- which(kegg.rows == kegg.use[i])
  kegg.select[i] <- save
}
kegg.de.df <- data.frame(c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"))
rownames(kegg.de.df) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")
colnames(kegg.de.df) <- "Season"
kegg.de.df$Season <- ordered(kegg.de.df$Season, levels=c("Summer", "Winter", "Spring"))
kegg.de$X <- ordered(kegg.de$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))
kegg.ntd <- normTransform(kegg.de)
  # DESeq test automatically transforms data
  # To visualize our data, we need to run this step separately
  # I use normTransform because I run DESeq with local fit
      # Local fit does a transformation similar to normTransform
  # Otherwise you should use vst instead
  # See link above for more info


kegg.ntd.assay <- data.frame(assay(kegg.ntd)[kegg.select,])
kegg.ntd.assay.order <- kegg.ntd.assay
kegg.ntd.assay.order[,1] <- kegg.ntd.assay$X3715
kegg.ntd.assay.order[,2] <- kegg.ntd.assay$X3717
kegg.ntd.assay.order[,3] <- kegg.ntd.assay$X3744
kegg.ntd.assay.order[,4] <- kegg.ntd.assay$X3723
kegg.ntd.assay.order[,5] <- kegg.ntd.assay$X3733
kegg.ntd.assay.order[,6] <- kegg.ntd.assay$X3772
kegg.ntd.assay.order[,7] <- kegg.ntd.assay$X3734
kegg.ntd.assay.order[,8] <- kegg.ntd.assay$X3773
kegg.ntd.assay.order[,9] <- kegg.ntd.assay$X3775
colnames(kegg.ntd.assay.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

tiff("kegg_DESeq.tiff", width=6, height=5.5, units="in", res=600)
pheatmap(kegg.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, treeheight_col = 0,
         show_colnames=F, border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255),
         labels_col = c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"),
         annotation_col=kegg.de.df, annotation_legend = F,
         annotation_colors = list(Season = c(Summer = "#00cccc", Winter = "#008484", Spring = "#004f4f"))
)
dev.off()


tiff("kegg_DESeq1.tiff", width=3, height=2.25, units="in", res=600)
pheatmap(kegg.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, show_colnames=F, 
         border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
)
dev.off()


pheatmap(counts(cog.de)[kegg.select,], cluster_rows=F, show_rownames=T, cluster_cols=F, show_colnames=T, 
+          border_color = "gray40", fontsize = 14, width = 3, height=3,
+          color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
+ )



































# Prepare phyloseq object
kegg.size.fac <- estimateSizeFactors(kegg.de)
kegg.norm.deseq <- counts(kegg.size.fac, normalized=TRUE)
kegg.otu.deseq <- otu_table(kegg.norm.deseq, taxa_are_rows = T)
kegg.samp <- sample_data(meta)
kegg.tax <- tax_table(kegg.class)
rownames(kegg.tax) <- rownames(kegg.class)
kegg.deseq <- merge_phyloseq(kegg.otu.deseq, kegg.samp, kegg.tax)
sample_data(kegg.deseq)$Season <- ordered(sample_data(kegg.deseq)$Season, levels=c("Summer", "Winter", "Spring"))

nmds.kegg.deseq <- ordinate(physeq=kegg.deseq, method="NMDS", distance="bray")
  # stress = 0.03492425

### SEASON ELLIPSES
sample_data(kegg.deseq)$Elip <- paste(sample_data(kegg.deseq)$Season)
kegg.deseq.elip.sum <- data.frame(MDS1 = nmds.kegg.deseq$points[,1], MDS2 = nmds.kegg.deseq$points[,2],
                           Group = sample_data(kegg.deseq)$Season)
plot.new()
kegg.deseq.elip.ord <- ordiellipse(nmds.kegg.deseq, sample_data(kegg.deseq)$Elip, display="sites", kind="se", conf=0.95, label=T)
kegg.deseq.elip.sum.df <- data.frame()
for(i in levels(kegg.deseq.elip.sum$Group)){
  kegg.deseq.elip.sum.df <- rbind(kegg.deseq.elip.sum.df,
                           cbind(as.data.frame(with(kegg.deseq.elip.sum[kegg.deseq.elip.sum$Group==i,],
                                                    veganCovEllipse(kegg.deseq.elip.ord[[i]]$cov,
                                                                    kegg.deseq.elip.ord[[i]]$center,
                                                                    kegg.deseq.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
tiff("kegg_NMDS.tiff", width=3.5, height=3.5, units="in", res=600)
plot_ordination(physeq=kegg.deseq, ordination=nmds.kegg.deseq, color="Season") +
  geom_path(data=kegg.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
    theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
dev.off()

kegg.nmds.plot <- plot_ordination(physeq=kegg.deseq, ordination=nmds.kegg.deseq, color="Season") +
  geom_path(data=kegg.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
    theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5),
        plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
  guides(color=F)


### PERMANOVA
kegg.deseq.sampdf <- data.frame(sample_data(kegg.deseq))
kegg.deseq.bray<- phyloseq::distance(physeq=kegg.deseq, method="bray")

## Season
adonis(kegg.deseq.bray~ Season, data=kegg.deseq.sampdf)
  # p-value = 0.075
beta.tax = betadisper(d=kegg.deseq.bray, group=kegg.deseq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.457

## Diet
adonis(kegg.deseq.bray~ Diet, data=kegg.deseq.sampdf)
  # p-value = 0.371
beta.tax = betadisper(d=kegg.deseq.bray, group=kegg.deseq.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.052

```
### dbcan family prep
```{r}
### 3715
dbcan.3715 <- read.csv("dbcan/3715.idba.dbCAN.parsed.csv")
dbcan.3717 <- read.csv("dbcan/3717.idba.dbCAN.parsed.csv")
dbcan.3723 <- read.csv("dbcan/3723.idba.dbCAN.parsed.csv")
dbcan.3733 <- read.csv("dbcan/3733.idba.dbCAN.parsed.csv")
dbcan.3734 <- read.csv("dbcan/3734.idba.dbCAN.parsed.csv")
dbcan.3744 <- read.csv("dbcan/3744.idba.dbCAN.parsed.csv")
dbcan.3772 <- read.csv("dbcan/3772.idba.dbCAN.parsed.csv")
dbcan.3773 <- read.csv("dbcan/3773.idba.dbCAN.parsed.csv")
dbcan.3775 <- read.csv("dbcan/3775.idba.dbCAN.parsed.csv")


# Isolate family names
fam.3715 <- substr(dbcan.3715$predicted_protein, 1, 5)
fam.3717 <- substr(dbcan.3717$predicted_protein, 1, 5)
fam.3723 <- substr(dbcan.3723$predicted_protein, 1, 5)
fam.3733 <- substr(dbcan.3733$predicted_protein, 1, 5)
fam.3734 <- substr(dbcan.3734$predicted_protein, 1, 5)
fam.3744 <- substr(dbcan.3744$predicted_protein, 1, 5)
fam.3772 <- substr(dbcan.3772$predicted_protein, 1, 5)
fam.3773 <- substr(dbcan.3773$predicted_protein, 1, 5)
fam.3775 <- substr(dbcan.3775$predicted_protein, 1, 5)


# Remove unwanted categories
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "AA")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "GT")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "SL")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "co")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "do")]

fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "AA")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "GT")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "SL")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "co")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "do")]

fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "AA")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "GT")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "SL")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "co")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "do")]

fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "AA")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "GT")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "SL")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "co")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "do")]

fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "AA")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "GT")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "SL")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "co")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "do")]

fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "AA")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "GT")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "SL")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "co")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "do")]

fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "AA")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "GT")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "SL")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "co")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "do")]

fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "AA")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "GT")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "SL")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "co")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "do")]

fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "AA")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "GT")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "SL")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "co")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "do")]

# Pull out unique families
fam.names.3715 <- unique(fam.3715)
fam.names.3717 <- unique(fam.3717)
fam.names.3723 <- unique(fam.3723)
fam.names.3733 <- unique(fam.3733)
fam.names.3734 <- unique(fam.3734)
fam.names.3744 <- unique(fam.3744)
fam.names.3772 <- unique(fam.3772)
fam.names.3773 <- unique(fam.3773)
fam.names.3775 <- unique(fam.3775)

# Create df
fam.df.3715 <- data.frame(fam.names.3715)
fam.df.3715$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3715)){
  x <- length(which(substr(fam.3715, 1, 5) == fam.names.3715[i]))
  fam.df.3715[i,2] <- x
}

fam.df.3717 <- data.frame(fam.names.3717)
fam.df.3717$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3717)){
  x <- length(which(substr(fam.3717, 1, 5) == fam.names.3717[i]))
  fam.df.3717[i,2] <- x
}

fam.df.3723 <- data.frame(fam.names.3723)
fam.df.3723$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3723)){
  x <- length(which(substr(fam.3723, 1, 5) == fam.names.3723[i]))
  fam.df.3723[i,2] <- x
}

fam.df.3733 <- data.frame(fam.names.3733)
fam.df.3733$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3733)){
  x <- length(which(substr(fam.3733, 1, 5) == fam.names.3733[i]))
  fam.df.3733[i,2] <- x
}

fam.df.3734 <- data.frame(fam.names.3734)
fam.df.3734$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3734)){
  x <- length(which(substr(fam.3734, 1, 5) == fam.names.3734[i]))
  fam.df.3734[i,2] <- x
}

fam.df.3744 <- data.frame(fam.names.3744)
fam.df.3744$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3744)){
  x <- length(which(substr(fam.3744, 1, 5) == fam.names.3744[i]))
  fam.df.3744[i,2] <- x
}

fam.df.3772 <- data.frame(fam.names.3772)
fam.df.3772$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3772)){
  x <- length(which(substr(fam.3772, 1, 5) == fam.names.3772[i]))
  fam.df.3772[i,2] <- x
}

fam.df.3773 <- data.frame(fam.names.3773)
fam.df.3773$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3773)){
  x <- length(which(substr(fam.3773, 1, 5) == fam.names.3773[i]))
  fam.df.3773[i,2] <- x
}

fam.df.3775 <- data.frame(fam.names.3775)
fam.df.3775$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3775)){
  x <- length(which(substr(fam.3775, 1, 5) == fam.names.3775[i]))
  fam.df.3775[i,2] <- x
}

# Save dataframe
#write.csv(fam.df.3715, "3715.dbcan.fam.csv")
#write.csv(fam.df.3717, "3717.dbcan.fam.csv")
#write.csv(fam.df.3723, "3723.dbcan.fam.csv")
#write.csv(fam.df.3733, "3733.dbcan.fam.csv")
#write.csv(fam.df.3734, "3734.dbcan.fam.csv")
#write.csv(fam.df.3744, "3744.dbcan.fam.csv")
#write.csv(fam.df.3772, "3772.dbcan.fam.csv")
#write.csv(fam.df.3773, "3773.dbcan.fam.csv")
#write.csv(fam.df.3775, "3775.dbcan.fam.csv")


```

### dbCAN Family
```{r}
fam.count <- read.csv("dbcan.fam.counts.csv")
row.names(fam.count) <- fam.count[,1]
fam.count <- fam.count[,-1]
colnames(fam.count) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

class <- read.csv('dbcan/classification.csv')
row.names(class) <- class[,1]
class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
fam.count.norm <- fam.count
for(i in 1:9){
  fam.count.norm[,i] <- round(fam.count.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(fam.count.norm)){
  print(sum(fam.count.norm[i,]))
}
  # All > 0

### Calculate proportions
fam.prop <- fam.count.norm
for(i in 1:9){
  fam.prop[,i] <- (fam.prop[,i] / colSums(fam.prop)[i])
}
rem <- which(rowSums(fam.prop) < 0.001)
fam.prop <- fam.prop[-rem,]

kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(kend) <- kend[,1]
kend[,2:9] <- NA
colnames(kend) <- kend[,1]
kend[,1] <- NA

kend[1,1] <- cor(fam.prop[,1], fam.prop[,1], method = "kendall")
kend[1,2] <- cor(fam.prop[,1], fam.prop[,2], method = "kendall")
kend[1,3] <- cor(fam.prop[,1], fam.prop[,3], method = "kendall")
kend[1,4] <- cor(fam.prop[,1], fam.prop[,4], method = "kendall")
kend[1,5] <- cor(fam.prop[,1], fam.prop[,5], method = "kendall")
kend[1,6] <- cor(fam.prop[,1], fam.prop[,6], method = "kendall")
kend[1,7] <- cor(fam.prop[,1], fam.prop[,7], method = "kendall")
kend[1,8] <- cor(fam.prop[,1], fam.prop[,8], method = "kendall")
kend[1,9] <- cor(fam.prop[,1], fam.prop[,9], method = "kendall")
kend[2,1] <- cor(fam.prop[,2], fam.prop[,1], method = "kendall")
kend[2,2] <- cor(fam.prop[,2], fam.prop[,2], method = "kendall")
kend[2,3] <- cor(fam.prop[,2], fam.prop[,3], method = "kendall")
kend[2,4] <- cor(fam.prop[,2], fam.prop[,4], method = "kendall")
kend[2,5] <- cor(fam.prop[,2], fam.prop[,5], method = "kendall")
kend[2,6] <- cor(fam.prop[,2], fam.prop[,6], method = "kendall")
kend[2,7] <- cor(fam.prop[,2], fam.prop[,7], method = "kendall")
kend[2,8] <- cor(fam.prop[,2], fam.prop[,8], method = "kendall")
kend[2,9] <- cor(fam.prop[,2], fam.prop[,9], method = "kendall")
kend[3,1] <- cor(fam.prop[,3], fam.prop[,1], method = "kendall")
kend[3,2] <- cor(fam.prop[,3], fam.prop[,2], method = "kendall")
kend[3,3] <- cor(fam.prop[,3], fam.prop[,3], method = "kendall")
kend[3,4] <- cor(fam.prop[,3], fam.prop[,4], method = "kendall")
kend[3,5] <- cor(fam.prop[,3], fam.prop[,5], method = "kendall")
kend[3,6] <- cor(fam.prop[,3], fam.prop[,6], method = "kendall")
kend[3,7] <- cor(fam.prop[,3], fam.prop[,7], method = "kendall")
kend[3,8] <- cor(fam.prop[,3], fam.prop[,8], method = "kendall")
kend[3,9] <- cor(fam.prop[,3], fam.prop[,9], method = "kendall")
kend[4,1] <- cor(fam.prop[,4], fam.prop[,1], method = "kendall")
kend[4,2] <- cor(fam.prop[,4], fam.prop[,2], method = "kendall")
kend[4,3] <- cor(fam.prop[,4], fam.prop[,3], method = "kendall")
kend[4,4] <- cor(fam.prop[,4], fam.prop[,4], method = "kendall")
kend[4,5] <- cor(fam.prop[,4], fam.prop[,5], method = "kendall")
kend[4,6] <- cor(fam.prop[,4], fam.prop[,6], method = "kendall")
kend[4,7] <- cor(fam.prop[,4], fam.prop[,7], method = "kendall")
kend[4,8] <- cor(fam.prop[,4], fam.prop[,8], method = "kendall")
kend[4,9] <- cor(fam.prop[,4], fam.prop[,9], method = "kendall")
kend[5,1] <- cor(fam.prop[,5], fam.prop[,1], method = "kendall")
kend[5,2] <- cor(fam.prop[,5], fam.prop[,2], method = "kendall")
kend[5,3] <- cor(fam.prop[,5], fam.prop[,3], method = "kendall")
kend[5,4] <- cor(fam.prop[,5], fam.prop[,4], method = "kendall")
kend[5,5] <- cor(fam.prop[,5], fam.prop[,5], method = "kendall")
kend[5,6] <- cor(fam.prop[,5], fam.prop[,6], method = "kendall")
kend[5,7] <- cor(fam.prop[,5], fam.prop[,7], method = "kendall")
kend[5,8] <- cor(fam.prop[,5], fam.prop[,8], method = "kendall")
kend[5,9] <- cor(fam.prop[,5], fam.prop[,9], method = "kendall")
kend[6,1] <- cor(fam.prop[,6], fam.prop[,1], method = "kendall")
kend[6,2] <- cor(fam.prop[,6], fam.prop[,2], method = "kendall")
kend[6,3] <- cor(fam.prop[,6], fam.prop[,3], method = "kendall")
kend[6,4] <- cor(fam.prop[,6], fam.prop[,4], method = "kendall")
kend[6,5] <- cor(fam.prop[,6], fam.prop[,5], method = "kendall")
kend[6,6] <- cor(fam.prop[,6], fam.prop[,6], method = "kendall")
kend[6,7] <- cor(fam.prop[,6], fam.prop[,7], method = "kendall")
kend[6,8] <- cor(fam.prop[,6], fam.prop[,8], method = "kendall")
kend[6,9] <- cor(fam.prop[,6], fam.prop[,9], method = "kendall")
kend[7,1] <- cor(fam.prop[,7], fam.prop[,1], method = "kendall")
kend[7,2] <- cor(fam.prop[,7], fam.prop[,2], method = "kendall")
kend[7,3] <- cor(fam.prop[,7], fam.prop[,3], method = "kendall")
kend[7,4] <- cor(fam.prop[,7], fam.prop[,4], method = "kendall")
kend[7,5] <- cor(fam.prop[,7], fam.prop[,5], method = "kendall")
kend[7,6] <- cor(fam.prop[,7], fam.prop[,6], method = "kendall")
kend[7,7] <- cor(fam.prop[,7], fam.prop[,7], method = "kendall")
kend[7,8] <- cor(fam.prop[,7], fam.prop[,8], method = "kendall")
kend[7,9] <- cor(fam.prop[,7], fam.prop[,9], method = "kendall")
kend[8,1] <- cor(fam.prop[,8], fam.prop[,1], method = "kendall")
kend[8,2] <- cor(fam.prop[,8], fam.prop[,2], method = "kendall")
kend[8,3] <- cor(fam.prop[,8], fam.prop[,3], method = "kendall")
kend[8,4] <- cor(fam.prop[,8], fam.prop[,4], method = "kendall")
kend[8,5] <- cor(fam.prop[,8], fam.prop[,5], method = "kendall")
kend[8,6] <- cor(fam.prop[,8], fam.prop[,6], method = "kendall")
kend[8,7] <- cor(fam.prop[,8], fam.prop[,7], method = "kendall")
kend[8,8] <- cor(fam.prop[,8], fam.prop[,8], method = "kendall")
kend[8,9] <- cor(fam.prop[,8], fam.prop[,9], method = "kendall")
kend[9,1] <- cor(fam.prop[,9], fam.prop[,1], method = "kendall")
kend[9,2] <- cor(fam.prop[,9], fam.prop[,2], method = "kendall")
kend[9,3] <- cor(fam.prop[,9], fam.prop[,3], method = "kendall")
kend[9,4] <- cor(fam.prop[,9], fam.prop[,4], method = "kendall")
kend[9,5] <- cor(fam.prop[,9], fam.prop[,5], method = "kendall")
kend[9,6] <- cor(fam.prop[,9], fam.prop[,6], method = "kendall")
kend[9,7] <- cor(fam.prop[,9], fam.prop[,7], method = "kendall")
kend[9,8] <- cor(fam.prop[,9], fam.prop[,8], method = "kendall")
kend[9,9] <- cor(fam.prop[,9], fam.prop[,9], method = "kendall")

#write.table(kend, file="dbcan.kendall.txt", sep="\t")

```
### dbCAN UPGMA
```{r}
kend.fix <- kend
kend.fix[1,1] <- 0
kend.fix[2,2] <- 0
kend.fix[3,3] <- 0
kend.fix[4,4] <- 0
kend.fix[5,5] <- 0
kend.fix[6,6] <- 0
kend.fix[7,7] <- 0
kend.fix[8,8] <- 0
kend.fix[9,9] <- 0
fam.upgma <- upgma(kend.fix)
plot(fam.upgma)


fam.nj <- NJ(data.frame(kend.fix))

```
### dbCAN DESeq
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
fam.count <- read.csv("dbcan.fam.counts.csv")
row.names(fam.count) <- fam.count[,1]
fam.count <- fam.count[,-1]
colnames(fam.count) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

fam.class <- read.csv('dbcan/classification.csv')
row.names(fam.class) <- fam.class[,1]
fam.class <- fam.class[,-1]

fam.otu <- otu_table(fam.count, taxa_are_rows = T)
fam.otu <- fam.otu + 1
fam.samp <- sample_data(meta)
fam.tax <- tax_table(fam.class)
row.names(fam.tax) <- row.names(fam.class)
fam.physeq <- merge_phyloseq(fam.otu, fam.samp, fam.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(fam.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.fam <- rownames(test[sum,])
  # "CBM37" "CE12"  "GH115" "GH28"  "GH43"  "PL1"   "PL11"  "PL9"
win <- which(test$log2FoldChange > 0)
win.fam <- rownames(test[win,])
  # "GH20" "GH33" "GH84"

# Spring vs. Summer
ss.physeq <- subset_samples(fam.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.fam <- rownames(test[spr,])
  # GH1
sum <- which(test$log2FoldChange > 0)
sum.fam <- rownames(test[sum,])

# Spring vs. Winter
sw.physeq <- subset_samples(fam.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.fam <- rownames(test[spr,])
  # GH1
win <- which(test$log2FoldChange > 0)
win.fam <- rownames(test[win,])

# Fasting vs Feeding
test <- deSEQ(data=fam.physeq, valuetest= ~Diet)
fast <- which(test$log2FoldChange < 0)
fast.fam <- rownames(test[fast,])
  # "CBM32" "CBM44" "CE11"  "GH102" "GH103" "GH110" "GH16"  "GH20"  "GH33"  "GH84"  "GH85"  "GH89" 
feed <- which(test$log2FoldChange > 0)
feed.fam <- rownames(test[feed,])
  # "CBM13" "CBM37" "CE12"  "GH105" "GH115" "GH25"  "GH28"  "GH32"  "GH39"  "GH42"  "GH43"  "GH51" "GH78"  "PL1"   "PL11"  "PL26" 

### Identify Overrep dbCAN families
dbcan.feed.id <- data.frame(feed.fam)
for(i in 1:length(feed.fam)){
  test <- which(feed.fam[i] == fam.class[,2])
  dbcan.feed.id[i,2] <- fam.class[test,1]
}
#write.csv(dbcan.feed.id, 'dbcan/dbcan.feed.id.csv')
dbcan.fast.id <- data.frame(fast.fam)
for(i in 1:length(fast.fam)){
  test <- which(fast.fam[i] == fam.class[,2])
  dbcan.fast.id[i,2] <- fam.class[test,1]
}
#write.csv(dbcan.fast.id, 'dbcan/dbcan.fast.id.csv')

# Female vs Male <-- STILL NEED TO RUN!!!
test <- deSEQ(data=fam.physeq, valuetest= ~Sex)
fem <- which(test$log2FoldChange < 0)
fem.fam <- rownames(test[fem,])
mal <- which(test$log2FoldChange > 0)
mal.fam <- rownames(test[mal,])

```

### dbcan Heatmap
```{r}
# Prepare physeq object
fam.count <- read.csv("dbcan.fam.counts.csv")
row.names(fam.count) <- fam.count[,1]
fam.count <- fam.count[,-1]
colnames(fam.count) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

fam.class <- read.csv('dbcan/classification.csv')
row.names(fam.class) <- fam.class[,1]
fam.class <- fam.class[,-1]

fam.otu <- otu_table(fam.count, taxa_are_rows = T)
fam.samp <- sample_data(meta)
fam.tax <- tax_table(fam.class)
row.names(fam.tax) <- row.names(fam.class)
dbcan.physeq <- merge_phyloseq(fam.otu, fam.samp, fam.tax)



dbcan.de.data <- phyloseq_to_deseq2(dbcan.physeq, ~Season)
dbcan.de <- DESeq(dbcan.de.data, test="Wald", fitType="local")

dbcan.res <- results(dbcan.de, pAdjustMethod = "BH")

### Extracting results: https://support.bioconductor.org/p/72765/

# Summer vs. Winter
dbcan.res.su.wi <- results(dbcan.de, contrast=c("Season", "Summer", "Winter"), pAdjustMethod = "BH")
dbcan.su.wi.sig <- dbcan.res.su.wi[which(dbcan.res.su.wi$padj < 0.05),]
dbcan.su.wi.sig.df <- cbind(as(dbcan.su.wi.sig, "data.frame"),
                          as(tax_table(dbcan.physeq)[rownames(dbcan.su.wi.sig), ], "matrix"))
dbcan.su.wi.sum.sig <- which(dbcan.su.wi.sig.df$log2FoldChange < 0)
dbcan.su.wi.sum <- rownames(dbcan.su.wi.sig[dbcan.su.wi.sum.sig,])
  # Summer
  # "GH33" "GH84"
dbcan.su.wi.win.sig <- which(dbcan.su.wi.sig.df$log2FoldChange > 0)
dbcan.su.wi.win <- rownames(dbcan.su.wi.sig[dbcan.su.wi.win.sig,])
  # Winter
  # "CBM37" "CE12"  "GH115" "GH43"  "PL1" 

# Summer vs. Spring
dbcan.res.su.sp <- results(dbcan.de, contrast=c("Season", "Spring", "Summer"), pAdjustMethod = "BH")
dbcan.su.sp.sig <- dbcan.res.su.sp[which(dbcan.res.su.sp$padj < 0.05),]
  # No sig!!!
# dbcan.su.sp.sig.df <- cbind(as(dbcan.su.sp.sig, "data.frame"),
#                           as(tax_table(dbcan.physeq)[rownames(dbcan.su.sp.sig), ], "matrix"))
# dbcan.su.sp.spr.sig <- which(dbcan.su.sp.sig.df$log2FoldChange < 0)
# dbcan.su.sp.spr <- rownames(dbcan.su.sp.sig[dbcan.su.sp.spr.sig,])
#   # Spring
# dbcan.su.sp.sum.sig <- which(dbcan.su.sp.sig.df$log2FoldChange > 0)
# dbcan.su.sp.sum <- rownames(dbcan.su.sp.sig[dbcan.su.sp.sum.sig,])
#   # Summer

# Spring vs Winter
dbcan.res.sp.wi <- results(dbcan.de, contrast=c("Season", "Spring", "Winter"), pAdjustMethod = "BH")
dbcan.sp.wi.sig <- dbcan.res.sp.wi[which(dbcan.res.sp.wi$padj < 0.05),]
dbcan.sp.wi.sig.df <- cbind(as(dbcan.sp.wi.sig, "data.frame"),
                          as(tax_table(dbcan.physeq)[rownames(dbcan.sp.wi.sig), ], "matrix"))
dbcan.sp.wi.spr.sig <- which(dbcan.sp.wi.sig.df$log2FoldChange < 0)
dbcan.sp.wi.spr <- rownames(dbcan.sp.wi.sig[dbcan.sp.wi.spr.sig,])
  # Spring
  # "CBM32" "CBM9"  "CE11"  "GH102" "GH103" "GH16"  "GH20"  "GH23"  "GH24"  "GH33"  "GH57"  "GH84" "GH89"  "GH92"  "PL22" 
dbcan.sp.wi.win.sig <- which(dbcan.sp.wi.sig.df$log2FoldChange > 0)
dbcan.sp.wi.win <- rownames(dbcan.sp.wi.sig[dbcan.sp.wi.win.sig,])
  # Winter
  # "CBM37" "CE12"  "GH1"   "GH112" "GH115" "GH129" "GH25"  "GH3"   "GH32"  "GH39"  "GH42"  "GH43"  "GH5"   "GH51"  "GH59"  "GH78"  "PL1"

# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

### Pull out top 2 (largest log fold change while sig) for each comparison
dbcan.su.wi.sum.df <- dbcan.su.wi.sig.df[dbcan.su.wi.sum.sig,]
dbcan.su.wi.sum.df <- dbcan.su.wi.sum.df[order(dbcan.su.wi.sum.df$log2FoldChange),]
dbcan.su.wi.win.df <- dbcan.su.wi.sig.df[dbcan.su.wi.win.sig,]
dbcan.su.wi.win.df <- dbcan.su.wi.win.df[order(dbcan.su.wi.win.df$log2FoldChange),]

# dbcan.su.sp.spr.df <- dbcan.su.sp.sig.df[dbcan.su.sp.spr.sig,]
# dbcan.su.sp.spr.df <- dbcan.su.sp.spr.df[order(dbcan.su.sp.spr.df$log2FoldChange),]
# dbcan.su.sp.sum.df <- dbcan.su.sp.sig.df[dbcan.su.sp.sum.sig,]
# dbcan.su.sp.sum.df <- dbcan.su.sp.sum.df[order(dbcan.su.sp.sum.df$log2FoldChange),]
dbcan.sp.wi.spr.df <- dbcan.sp.wi.sig.df[dbcan.sp.wi.spr.sig,]
dbcan.sp.wi.spr.df <- dbcan.sp.wi.spr.df[order(dbcan.sp.wi.spr.df$log2FoldChange),]
dbcan.sp.wi.win.df <- dbcan.sp.wi.sig.df[dbcan.sp.wi.win.sig,]
dbcan.sp.wi.win.df <- dbcan.sp.wi.win.df[order(dbcan.sp.wi.win.df$log2FoldChange),]

dbcan.sig <- rbind(dbcan.su.wi.sum.df, dbcan.su.wi.win.df, dbcan.sp.wi.spr.df, dbcan.sp.wi.win.df)
dbcan.sig <- dbcan.sig[order(dbcan.sig$log2FoldChange),]
#dbcan.sig <- dbcan.sig[order(dbcan.sig$padj),]
dbcan.use <-rownames(dbcan.sig)[1:10] 
dbcan.use <- dbcan.use[-3]
dbcan.use[10] <- rownames(dbcan.sig)[12]
  # Somehow a "1" is being added to some rownames
  # I will investigate when I have time


#select <- order(rowMeans(counts(dbcan.de, normalized=T)), decreasing=T)[1:20]
  # For top 20 dbcans
dbcan.rows <- rownames(dbcan.de)
dbcan.select <- rep(1,10)
for(i in 1:length(dbcan.use)){
  save <- which(dbcan.rows == dbcan.use[i])
  dbcan.select[i] <- save
}
dbcan.de.df <- data.frame(c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"))
rownames(dbcan.de.df) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")
colnames(dbcan.de.df) <- "Season"
dbcan.de.df$Season <- ordered(dbcan.de.df$Season, levels=c("Summer", "Winter", "Spring"))
dbcan.de$X <- ordered(dbcan.de$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))
dbcan.ntd <- normTransform(dbcan.de)
  # DESeq test automatically transforms data
  # To visualize our data, we need to run this step separately
  # I use normTransform because I run DESeq with local fit
      # Local fit does a transformation similar to normTransform
  # Otherwise you should use vst instead
  # See link above for more info
dbcan.ntd$X <- ordered(dbcan.ntd$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))
tiff("dbcan_DESeq.tiff", width=4, height=4, units="in", res=600)
test <- pheatmap(assay(dbcan.ntd)[dbcan.select,], cluster_rows=F, show_rownames=T, cluster_cols=F, treeheight_col = 5,labels_col = c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring"),
         border_color="gray40", fontsize = 16, width = 4, height=4,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255))
test$tree_col$order <- c(1,2,4,5,7,3,6,8,9)
test
dev.off()

```
### dbCAN Bray-Curtis
```{r}
# Prepare physeq object
fam.count <- read.csv("dbcan.fam.counts.csv")
row.names(fam.count) <- fam.count[,1]
fam.count <- fam.count[,-1]
colnames(fam.count) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]
fam.class <- read.csv('dbcan/classification.csv')
row.names(fam.class) <- fam.class[,1]
fam.class <- fam.class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
fam.norm <- fam.count
for(i in 1:9){
  fam.norm[,i] <- round(fam.norm[,i]*norm.factor[i,1])
}
# Remove famways = 0
for(i in 1:nrow(fam.norm)){
  print(sum(fam.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
fam.otu <- otu_table(fam.norm, taxa_are_rows = T)
fam.samp <- sample_data(meta)
fam.tax <- tax_table(fam.class)
row.names(fam.tax) <- row.names(fam.class)
fam.physeq <- merge_phyloseq(fam.otu, fam.samp, fam.tax)

nmds <- ordinate(physeq=fam.physeq, method="NMDS", distance="bray")
  # stress = 0.07162816 

plot_ordination(physeq=fam.physeq, ordination=nmds, color="Season")
plot_ordination(physeq=fam.physeq, ordination=nmds, color="Diet")
plot_ordination(physeq=fam.physeq, ordination=nmds, color="Sex")

### PERMANOVA
fam.sampdf <- data.frame(sample_data(fam.physeq))
fam.bray<- phyloseq::distance(physeq=fam.physeq, method="bray")

# Season
adonis(fam.bray~ Season, data=fam.sampdf)
  # p-value = 0.085
beta.tax = betadisper(d=fam.bray, group=fam.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.701

# Diet
adonis(fam.bray~ Diet, data=fam.sampdf)
  # p-value = 0.044
beta.tax = betadisper(d=fam.bray, group=fam.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.534

# Sex
adonis(fam.bray~ Sex, data=fam.sampdf)
  # p-value = 0.833
beta.tax = betadisper(d=fam.bray, group=fam.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.313

### ANOSIM
anosim(fam.bray, meta$Season, permutations=1000)
  # p-value = 0.014
anosim(fam.bray, meta$Diet, permutations=1000)
  # p-value = 0.050949
anosim(fam.bray, meta$Sex, permutations=1000)
  # p-value = 0.76224

```

### dbCAN Jaccard
```{r}
jac.dbcan <- ordinate(physeq=fam.physeq, method="NMDS", distance="jaccard")
  # stress = 0.08480145
plot_ordination(physeq=fam.physeq, ordination=jac.dbcan, color="Season")
plot_ordination(physeq=fam.physeq, ordination=jac.dbcan, color="Diet")
plot_ordination(physeq=fam.physeq, ordination=jac.dbcan, color="Sex")


### PERMANOVA
fam.sampdf <- data.frame(sample_data(fam.physeq))
fam.jac <- phyloseq::distance(physeq=fam.physeq, method="jaccard")

# Season
adonis(fam.jac~ Season, data=fam.sampdf)
  # p-value = 0.053
beta.tax = betadisper(d=fam.jac, group=fam.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.643

# Diet
adonis(fam.jac~ Diet, data=fam.sampdf)
  # p-value = 0.045
beta.tax = betadisper(d=fam.jac, group=fam.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.536

# Sex
adonis(fam.jac~ Sex, data=fam.sampdf)
  # p-value = 0.779
beta.tax = betadisper(d=fam.jac, group=fam.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.267

### ANOSIM
anosim(fam.jac, fam.sampdf$Season, permutations=1000)
  # p-value = 0.012987
anosim(fam.jac, fam.sampdf$Diet, permutations=1000)
  # p-value = 0.041958
anosim(fam.jac, fam.sampdf$Sex, permutations=1000)
  # p-value = 0.78422

```

### dbCAN SIMPER
```{r}
simper.pretty(fam.norm, metrics = meta, interesting = c('Season', 'Diet', 'Sex'), perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'dbcan')

dbcan.simper <- read.csv("dbcan_clean_simper.csv")

kruskal.pretty(otu = fam.norm, metrics = meta, csv =  dbcan.simper, interesting = c('Season', 'Diet', 'Sex'), output_name = 'dbcan')

dbcan.simper.kw <- read.csv("dbcan_krusk_simper.csv")
```

### COG Functional Groups
```{r}
cog.fun <- read.csv("COG/cog_func.csv")
rownames(cog.fun) <- cog.fun[,1]
cog.fun <- cog.fun[,-1]
colnames(cog.fun) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
cog.fun.norm <- cog.fun
for(i in 1:9){
  cog.fun.norm[,i] <- round(cog.fun.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(cog.fun.norm)){
  print(sum(cog.fun.norm[i,]))
}
  # All > 0

### Summarize
cog.fun.norm.t <- data.frame(t(cog.fun.norm))
sum.cog <- cog.fun.norm.t[which(meta$Season == "Summer"),]
for(i in 1:(ncol(sum.cog))){
  sum.cog[4,i] <- sum(sum.cog[1:3,i])/3
  sum.cog[5,i] <- sd(sum.cog[1:3,i])
  sum.cog[6,i] <- sd(sum.cog[1:3,i])/(sqrt(3))
}
rownames(sum.cog)[4:6] <- c("Mean", "SD", "SE")
spr.cog <- cog.fun.norm.t[which(meta$Season == "Spring"),]
for(i in 1:(ncol(spr.cog))){
  spr.cog[4,i] <- sum(spr.cog[1:3,i])/3
  spr.cog[5,i] <- sd(spr.cog[1:3,i])
  spr.cog[6,i] <- sd(spr.cog[1:3,i])/(sqrt(3))
}
rownames(spr.cog)[4:6] <- c("Mean", "SD", "SE")
win.cog <- cog.fun.norm.t[which(meta$Season == "Winter"),]
for(i in 1:(ncol(win.cog))){
  win.cog[4,i] <- sum(win.cog[1:3,i])/3
  win.cog[5,i] <- sd(win.cog[1:3,i])
  win.cog[6,i] <- sd(win.cog[1:3,i])/(sqrt(3))
}
rownames(win.cog)[4:6] <- c("Mean", "SD", "SE")
cog.sum <- sum.cog[4:6,]
cog.sum[4:6,] <- win.cog[4:6,]
cog.sum[7:9,] <- spr.cog[4:6,]
colnames(cog.sum) <- colnames(sum.cog)
rownames(cog.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")


### Check for Normal distribution
norm.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(cog.fun.norm))){
  temp.test <- shapiro.test(cog.fun.norm[,i])
  norm.pvals[i,1] <- temp.test$p.value
  rownames(norm.pvals)[i] <- colnames(cog.fun.norm)[i]
}
length(which(norm.pvals[,1] < 0.06))
rownames(norm.pvals)[which(norm.pvals[,1] < 0.06)]
  # 3733, 3734, 3744, and 3773 aren't normal

### Overall Summary
cog.fun.sum <- data.frame(999)
for(i in 1:(ncol(cog.fun.norm.t))){
  cog.fun.sum[1,i] <- sum(cog.fun.norm.t[1:ncol(cog.fun.norm.t),i])/9
  cog.fun.sum[2,i] <- sd(cog.fun.norm.t[1:ncol(cog.fun.norm.t),i])
  cog.fun.sum[3,i] <- sd(cog.fun.norm.t[1:ncol(cog.fun.norm.t),i])/(sqrt(9))
}
colnames(cog.fun.sum) <- colnames(cog.fun.norm.t)[1:25]
rownames(cog.fun.sum) <- c("Mean", "SD", "SE")

### Kruskal-Wallis
# Seasons
cog.fun.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(cog.fun.norm.t))){
  temp.kw <- kruskal.test(formula = cog.fun.norm.t[,i] ~ meta$Season, data=cog.fun.norm.t)
  cog.fun.pvals[i,1] <- temp.kw[3]
  rownames(cog.fun.pvals)[i] <- colnames(cog.fun.norm.t)[i]
}
sig <- rownames(cog.fun.pvals)[which(cog.fun.pvals[,1] < 0.06)]
  # None

# Active vs. Hib
cog.fun.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(cog.fun.norm.t))){
  temp.kw <- kruskal.test(formula = cog.fun.norm.t[,i] ~ meta$SubSeason, data=cog.fun.norm.t)
  cog.fun.pvals[i,1] <- temp.kw[3]
  rownames(cog.fun.pvals)[i] <- colnames(cog.fun.norm.t)[i]
}
sig <- rownames(cog.fun.pvals)[which(cog.fun.pvals[,1] < 0.06)]


```


### COG Count Prep
```{r}
### Count Counts
# Load tables
cog.3715 <- read.table("COG/3715.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3717 <- read.table("COG/3717.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3723 <- read.table("COG/3723.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3733 <- read.table("COG/3733.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3734 <- read.table("COG/3734.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3744 <- read.table("COG/3744.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3772 <- read.table("COG/3772.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3773 <- read.table("COG/3773.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3775 <- read.table("COG/3775.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)

# Create COG count table
cog.names <- paste(c(cog.3715[,1], cog.3717[,1], cog.3723[,1], cog.3733[,1], cog.3734[,1], cog.3744[,1], cog.3772[,1], cog.3773[,1], cog.3775[,1]))
cog.uniq <- unique(cog.names)
cog.all <- data.frame(cog.uniq)
for(i in 1:nrow(cog.3715)){
  test <- which(cog.all[,1] == cog.3715[i,1])
  cog.all[test,2] <- cog.3715[i,3]
}
for(i in 1:nrow(cog.3717)){
  test <- which(cog.all[,1] == cog.3717[i,1])
  cog.all[test,3] <- cog.3717[i,3]
}
for(i in 1:nrow(cog.3723)){
  test <- which(cog.all[,1] == cog.3723[i,1])
  cog.all[test,4] <- cog.3723[i,3]
}
for(i in 1:nrow(cog.3733)){
  test <- which(cog.all[,1] == cog.3733[i,1])
  cog.all[test,5] <- cog.3733[i,3]
}
for(i in 1:nrow(cog.3734)){
  test <- which(cog.all[,1] == cog.3734[i,1])
  cog.all[test,6] <- cog.3734[i,3]
}
for(i in 1:nrow(cog.3744)){
  test <- which(cog.all[,1] == cog.3744[i,1])
  cog.all[test,7] <- cog.3744[i,3]
}
for(i in 1:nrow(cog.3772)){
  test <- which(cog.all[,1] == cog.3772[i,1])
  cog.all[test,8] <- cog.3772[i,3]
}
for(i in 1:nrow(cog.3773)){
  test <- which(cog.all[,1] == cog.3773[i,1])
  cog.all[test,9] <- cog.3773[i,3]
}
for(i in 1:nrow(cog.3775)){
  test <- which(cog.all[,1] == cog.3775[i,1])
  cog.all[test,10] <- cog.3775[i,3]
}
rownames(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.all[is.na(cog.all)] <- 0
  # Replaces all NA's w/ 0


# Save COG count table
#write.csv(cog.all, "COG/cog_counts.csv")

### Create COG Classification table
# Temp version
cog.class <- data.frame(cog.uniq)
#write.csv(cog.class, "COG/cog_classification.csv")


# Load tables
cog.prot.3715 <- read.table("COG/3715.results/protein-id_cog.edit.txt", sep="\t", quote="", stringsAsFactors = F, na = NA)
test <- unique(cog.prot.3715)
test1 <- cog.prot.3715[,2]
test2 <- which(rownames(cog.all) == test1[1])
test3 <- cog.all

cog..prot.3717 <- read.table("COG/3717.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3723 <- read.table("COG/3723.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3733 <- read.table("COG/3733.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3734 <- read.table("COG/3734.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3744 <- read.table("COG/3744.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3772 <- read.table("COG/3772.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3773 <- read.table("COG/3773.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3775 <- read.table("COG/3775.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
```

### COG Kendall Correlation
```{r}
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('COG/cog_classification.csv')
rownames(class) <- class[,1]
#class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
cog.all.norm <- cog.all
for(i in 1:9){
  cog.all.norm[,i] <- round(cog.all.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(cog.all.norm)){
  print(sum(cog.all.norm[i,]))
}
  # All > 0

### Calculate proportions
cog.all.prop <- cog.all.norm
for(i in 1:9){
  cog.all.prop[,i] <- (cog.all.prop[,i] / colSums(cog.all.prop)[i])
}
rem <- which(rowSums(cog.all.prop) < 0.001)
cog.all.prop <- cog.all.prop[-rem,]

cog.kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(cog.kend) <- cog.kend[,1]
cog.kend[,2:9] <- NA
colnames(cog.kend) <- cog.kend[,1]
cog.kend[,1] <- NA

cog.kend[1,1] <- cor(cog.all.prop[,1], cog.all.prop[,1], method = "kendall")
cog.kend[1,2] <- cor(cog.all.prop[,1], cog.all.prop[,2], method = "kendall")
cog.kend[1,3] <- cor(cog.all.prop[,1], cog.all.prop[,3], method = "kendall")
cog.kend[1,4] <- cor(cog.all.prop[,1], cog.all.prop[,4], method = "kendall")
cog.kend[1,5] <- cor(cog.all.prop[,1], cog.all.prop[,5], method = "kendall")
cog.kend[1,6] <- cor(cog.all.prop[,1], cog.all.prop[,6], method = "kendall")
cog.kend[1,7] <- cor(cog.all.prop[,1], cog.all.prop[,7], method = "kendall")
cog.kend[1,8] <- cor(cog.all.prop[,1], cog.all.prop[,8], method = "kendall")
cog.kend[1,9] <- cor(cog.all.prop[,1], cog.all.prop[,9], method = "kendall")
cog.kend[2,1] <- cor(cog.all.prop[,2], cog.all.prop[,1], method = "kendall")
cog.kend[2,2] <- cor(cog.all.prop[,2], cog.all.prop[,2], method = "kendall")
cog.kend[2,3] <- cor(cog.all.prop[,2], cog.all.prop[,3], method = "kendall")
cog.kend[2,4] <- cor(cog.all.prop[,2], cog.all.prop[,4], method = "kendall")
cog.kend[2,5] <- cor(cog.all.prop[,2], cog.all.prop[,5], method = "kendall")
cog.kend[2,6] <- cor(cog.all.prop[,2], cog.all.prop[,6], method = "kendall")
cog.kend[2,7] <- cor(cog.all.prop[,2], cog.all.prop[,7], method = "kendall")
cog.kend[2,8] <- cor(cog.all.prop[,2], cog.all.prop[,8], method = "kendall")
cog.kend[2,9] <- cor(cog.all.prop[,2], cog.all.prop[,9], method = "kendall")
cog.kend[3,1] <- cor(cog.all.prop[,3], cog.all.prop[,1], method = "kendall")
cog.kend[3,2] <- cor(cog.all.prop[,3], cog.all.prop[,2], method = "kendall")
cog.kend[3,3] <- cor(cog.all.prop[,3], cog.all.prop[,3], method = "kendall")
cog.kend[3,4] <- cor(cog.all.prop[,3], cog.all.prop[,4], method = "kendall")
cog.kend[3,5] <- cor(cog.all.prop[,3], cog.all.prop[,5], method = "kendall")
cog.kend[3,6] <- cor(cog.all.prop[,3], cog.all.prop[,6], method = "kendall")
cog.kend[3,7] <- cor(cog.all.prop[,3], cog.all.prop[,7], method = "kendall")
cog.kend[3,8] <- cor(cog.all.prop[,3], cog.all.prop[,8], method = "kendall")
cog.kend[3,9] <- cor(cog.all.prop[,3], cog.all.prop[,9], method = "kendall")
cog.kend[4,1] <- cor(cog.all.prop[,4], cog.all.prop[,1], method = "kendall")
cog.kend[4,2] <- cor(cog.all.prop[,4], cog.all.prop[,2], method = "kendall")
cog.kend[4,3] <- cor(cog.all.prop[,4], cog.all.prop[,3], method = "kendall")
cog.kend[4,4] <- cor(cog.all.prop[,4], cog.all.prop[,4], method = "kendall")
cog.kend[4,5] <- cor(cog.all.prop[,4], cog.all.prop[,5], method = "kendall")
cog.kend[4,6] <- cor(cog.all.prop[,4], cog.all.prop[,6], method = "kendall")
cog.kend[4,7] <- cor(cog.all.prop[,4], cog.all.prop[,7], method = "kendall")
cog.kend[4,8] <- cor(cog.all.prop[,4], cog.all.prop[,8], method = "kendall")
cog.kend[4,9] <- cor(cog.all.prop[,4], cog.all.prop[,9], method = "kendall")
cog.kend[5,1] <- cor(cog.all.prop[,5], cog.all.prop[,1], method = "kendall")
cog.kend[5,2] <- cor(cog.all.prop[,5], cog.all.prop[,2], method = "kendall")
cog.kend[5,3] <- cor(cog.all.prop[,5], cog.all.prop[,3], method = "kendall")
cog.kend[5,4] <- cor(cog.all.prop[,5], cog.all.prop[,4], method = "kendall")
cog.kend[5,5] <- cor(cog.all.prop[,5], cog.all.prop[,5], method = "kendall")
cog.kend[5,6] <- cor(cog.all.prop[,5], cog.all.prop[,6], method = "kendall")
cog.kend[5,7] <- cor(cog.all.prop[,5], cog.all.prop[,7], method = "kendall")
cog.kend[5,8] <- cor(cog.all.prop[,5], cog.all.prop[,8], method = "kendall")
cog.kend[5,9] <- cor(cog.all.prop[,5], cog.all.prop[,9], method = "kendall")
cog.kend[6,1] <- cor(cog.all.prop[,6], cog.all.prop[,1], method = "kendall")
cog.kend[6,2] <- cor(cog.all.prop[,6], cog.all.prop[,2], method = "kendall")
cog.kend[6,3] <- cor(cog.all.prop[,6], cog.all.prop[,3], method = "kendall")
cog.kend[6,4] <- cor(cog.all.prop[,6], cog.all.prop[,4], method = "kendall")
cog.kend[6,5] <- cor(cog.all.prop[,6], cog.all.prop[,5], method = "kendall")
cog.kend[6,6] <- cor(cog.all.prop[,6], cog.all.prop[,6], method = "kendall")
cog.kend[6,7] <- cor(cog.all.prop[,6], cog.all.prop[,7], method = "kendall")
cog.kend[6,8] <- cor(cog.all.prop[,6], cog.all.prop[,8], method = "kendall")
cog.kend[6,9] <- cor(cog.all.prop[,6], cog.all.prop[,9], method = "kendall")
cog.kend[7,1] <- cor(cog.all.prop[,7], cog.all.prop[,1], method = "kendall")
cog.kend[7,2] <- cor(cog.all.prop[,7], cog.all.prop[,2], method = "kendall")
cog.kend[7,3] <- cor(cog.all.prop[,7], cog.all.prop[,3], method = "kendall")
cog.kend[7,4] <- cor(cog.all.prop[,7], cog.all.prop[,4], method = "kendall")
cog.kend[7,5] <- cor(cog.all.prop[,7], cog.all.prop[,5], method = "kendall")
cog.kend[7,6] <- cor(cog.all.prop[,7], cog.all.prop[,6], method = "kendall")
cog.kend[7,7] <- cor(cog.all.prop[,7], cog.all.prop[,7], method = "kendall")
cog.kend[7,8] <- cor(cog.all.prop[,7], cog.all.prop[,8], method = "kendall")
cog.kend[7,9] <- cor(cog.all.prop[,7], cog.all.prop[,9], method = "kendall")
cog.kend[8,1] <- cor(cog.all.prop[,8], cog.all.prop[,1], method = "kendall")
cog.kend[8,2] <- cor(cog.all.prop[,8], cog.all.prop[,2], method = "kendall")
cog.kend[8,3] <- cor(cog.all.prop[,8], cog.all.prop[,3], method = "kendall")
cog.kend[8,4] <- cor(cog.all.prop[,8], cog.all.prop[,4], method = "kendall")
cog.kend[8,5] <- cor(cog.all.prop[,8], cog.all.prop[,5], method = "kendall")
cog.kend[8,6] <- cor(cog.all.prop[,8], cog.all.prop[,6], method = "kendall")
cog.kend[8,7] <- cor(cog.all.prop[,8], cog.all.prop[,7], method = "kendall")
cog.kend[8,8] <- cor(cog.all.prop[,8], cog.all.prop[,8], method = "kendall")
cog.kend[8,9] <- cor(cog.all.prop[,8], cog.all.prop[,9], method = "kendall")
cog.kend[9,1] <- cor(cog.all.prop[,9], cog.all.prop[,1], method = "kendall")
cog.kend[9,2] <- cor(cog.all.prop[,9], cog.all.prop[,2], method = "kendall")
cog.kend[9,3] <- cor(cog.all.prop[,9], cog.all.prop[,3], method = "kendall")
cog.kend[9,4] <- cor(cog.all.prop[,9], cog.all.prop[,4], method = "kendall")
cog.kend[9,5] <- cor(cog.all.prop[,9], cog.all.prop[,5], method = "kendall")
cog.kend[9,6] <- cor(cog.all.prop[,9], cog.all.prop[,6], method = "kendall")
cog.kend[9,7] <- cor(cog.all.prop[,9], cog.all.prop[,7], method = "kendall")
cog.kend[9,8] <- cor(cog.all.prop[,9], cog.all.prop[,8], method = "kendall")
cog.kend[9,9] <- cor(cog.all.prop[,9], cog.all.prop[,9], method = "kendall")

#write.table(cog.kend, file="COG/cog.kendall.txt", sep="\t")
```

### COG UPGMA
```{r}
cog.kend <- read.table("COG/cog.kendall.txt", sep="\t")
cog.kend.fix <- cog.kend
cog.kend.fix[1,1] <- 0
cog.kend.fix[2,2] <- 0
cog.kend.fix[3,3] <- 0
cog.kend.fix[4,4] <- 0
cog.kend.fix[5,5] <- 0
cog.kend.fix[6,6] <- 0
cog.kend.fix[7,7] <- 0
cog.kend.fix[8,8] <- 0
cog.kend.fix[9,9] <- 0
cog.upgma <- upgma(cog.kend.fix)
plot(cog.upgma)


cog.nj <- NJ(data.frame(cog.kend.fix))
```
### COG DESeq2
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

cog.otu <- otu_table(cog.all, taxa_are_rows = T)
#cog.otu <- cog.otu + 1
cog.samp <- sample_data(meta)
rownames(cog.class) <- cog.class[,2]
cog.tax <- tax_table(cog.class)
row.names(cog.tax) <- row.names(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(cog.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.cog <- rownames(test[sum,])
  # "COG0243" "COG1874" "COG2207" "COG2730" "COG3507" "COG3534" "COG3693" "COG3757" "COG3866" "COG4677" "COG5434"
win <- which(test$log2FoldChange > 0)
win.cog <- rownames(test[win,])
  # "COG1775" "COG3246"

# Spring vs. Summer
ss.physeq <- subset_samples(cog.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.cog <- rownames(test[spr,])
  # "COG0395" "COG0444" "COG1129" "COG1172" "COG1175" "COG3958" "COG3959" "COG4166"
sum <- which(test$log2FoldChange > 0)
sum.cog <- rownames(test[sum,])
  # "COG0243" "COG0437" "COG0457" "COG0810" "COG1409" "COG1538" "COG1596" "COG1629" "COG2885"

# Spring vs. Winter
sw.physeq <- subset_samples(cog.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.cog <- rownames(test[spr,])
  # "COG0395" "COG0444" "COG1129" "COG1172" "COG1175" "COG3958" "COG3959" "COG4166"
win <- which(test$log2FoldChange > 0)
win.cog <- rownames(test[win,])
  # "COG0243" "COG0437" "COG0457" "COG0810" "COG1538" "COG1596" "COG1629" "COG2885"

# Fasting vs Feeding
test <- deSEQ(data=sw.physeq, valuetest= ~Diet)
fast <- which(test$log2FoldChange < 0)
fast.cog <- rownames(test[fast,])
  # "COG0156" "COG0247" "COG0339" "COG0457" "COG0612" "COG0659" "COG0760" "COG0767" "COG0774" "COG0776" "COG0795" "COG0810" "COG0811" "COG0841" "COG0845" "COG0847" "COG0848" "COG0859" "COG1043" "COG1137" "COG1409" "COG1452" "COG1463" "COG1538" "COG1566" "COG1596" "COG1629" "COG1663" "COG2067" "COG2095" "COG2259" "COG2814" "COG2825" "COG2834" "COG2885" "COG2982" "COG3169" "COG3176" "COG3187" "COG3203" "COG3468" "COG3525" "COG3637" "COG3712" "COG4105" "COG4409" "COG4566" "COG4775" "COG4783" "COG5010"
feed <- which(test$log2FoldChange > 0)
feed.cog <- rownames(test[feed,])
  # "COG0004" "COG0028" "COG0246" "COG0372" "COG0395" "COG0407" "COG0448" "COG0456" "COG0554" "COG0561" "COG0675" "COG0722" "COG0745" "COG0855" "COG1026" "COG1061" "COG1070" "COG1104" "COG1105" "COG1131" "COG1132" "COG1134" "COG1136" "COG1175" "COG1302" "COG1307" "COG1349" "COG1461" "COG1472" "COG1511" "COG1609" "COG1621" "COG1653" "COG1662" "COG1686" "COG1874" "COG1882" "COG1925" "COG2182" "COG2190" "COG2199" "COG2200" "COG2207" "COG2252" "COG2357" "COG2407" "COG2508" "COG2730" "COG2972" "COG3044" "COG3279" "COG3331" "COG3345" "COG3437" "COG3459" "COG3481" "COG3507" "COG3534" "COG3635" "COG3664" "COG3693" "COG3757" "COG3835" "COG3857" "COG3887" "COG3894" "COG3940" "COG3958" "COG3959" "COG4166" "COG4189" "COG4209" "COG4284" "COG4465" "COG4468" "COG4485" "COG4509" "COG4626" "COG4709" "COG4753" "COG4805" "COG4870" "COG5578"

### Identify Overrep COGs
cog.feed.id <- data.frame(feed.cog)
cog.feed.id$ID <- NA
for(i in 1:length(feed.cog)){
  test <- which(feed.cog[i] == cog.3715[,1])
  cog.feed.id[i,2] <- cog.3715[test,2]
}
#write.csv(cog.feed.id, 'COG/cog.feed.id.csv')
cog.fast.id <- data.frame(fast.cog)
for(i in 1:length(fast.cog)){
  test <- which(fast.cog[i] == cog.3723[,1])
  cog.fast.id[i,2] <- cog.3723[test,2]
}
#write.csv(cog.fast.id, 'COG/cog.fast.id.csv')

# Female vs Male
test <- deSEQ(data=cog.physeq, valuetest= ~Sex)
fem <- which(test$log2FoldChange < 0)
fem.cog <- rownames(test[fem,])
  # "COG0639" "COG0724" "COG1025" "COG1224" "COG1643" "COG2319" "COG5049" "COG5077" "COG5078" "COG5099" "COG5206" "COG5207" "COG5272" "COG5307" "COG5647"
feed <- which(test$log2FoldChange > 0)
feed.cog <- rownames(test[feed,])
  # "COG5184" "COG5279"
```





### COG Heatmap
```{r}
# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
cog.all.order <- cog.all
cog.all.order$X3715 <- cog.all$X3715
cog.all.order$X3717 <- cog.all$X3717
cog.all.order$X3744 <- cog.all$X3744
cog.all.order$X3723 <- cog.all$X3723
cog.all.order$X3733 <- cog.all$X3733
cog.all.order$X3772 <- cog.all$X3772
cog.all.order$X3734 <- cog.all$X3734
cog.all.order$X3773 <- cog.all$X3773
cog.all.order$X3775 <- cog.all$X3775
colnames(cog.all.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

meta.order <- read.csv('metadata.order.csv')
row.names(meta.order) <- meta.order[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

cog.otu <- otu_table(cog.all.order, taxa_are_rows = T)
#cog.otu <- cog.otu + 1
cog.samp <- sample_data(meta.order)
rownames(cog.class) <- cog.class[,2]
cog.tax <- tax_table(cog.class)
row.names(cog.tax) <- row.names(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)



cog.de.data <- phyloseq_to_deseq2(cog.physeq, ~Season)
cog.de <- DESeq(cog.de.data, test="Wald", fitType="local")

cog.res <- results(cog.de, pAdjustMethod = "BH")

### Extracting results: https://support.bioconductor.org/p/72765/

# Summer vs. Winter
cog.res.su.wi <- results(cog.de, contrast=c("Season", "Summer", "Winter"), pAdjustMethod = "BH")
cog.su.wi.sig <- cog.res.su.wi[which(cog.res.su.wi$padj < 0.05),]
cog.su.wi.sig.df <- cbind(as(cog.su.wi.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.su.wi.sig), ], "matrix"))
cog.su.wi.sum.sig <- which(cog.su.wi.sig.df$log2FoldChange < 0)
cog.su.wi.sum <- rownames(cog.su.wi.sig[cog.su.wi.sum.sig,])
  # Summer
  # "COG1775" "COG3246"
cog.su.wi.win.sig <- which(cog.su.wi.sig.df$log2FoldChange > 0)
cog.su.wi.win <- rownames(cog.su.wi.sig[cog.su.wi.win.sig,])
  # Winter
  # "COG1472" "COG1874" "COG2207" "COG2730" "COG3507" "COG3534" "COG3693" "COG3757" "COG3866" "COG3940" "COG4677" "COG5434"

# Summer vs. Spring
cog.res.su.sp <- results(cog.de, contrast=c("Season", "Spring", "Summer"), pAdjustMethod = "BH")
cog.su.sp.sig <- cog.res.su.sp[which(cog.res.su.sp$padj < 0.05),]
cog.su.sp.sig.df <- cbind(as(cog.su.sp.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.su.sp.sig), ], "matrix"))
cog.su.sp.spr.sig <- which(cog.su.sp.sig.df$log2FoldChange < 0)
cog.su.sp.spr <- rownames(cog.su.sp.sig[cog.su.sp.spr.sig,])
  # Spring
  # "COG0243" "COG0437" "COG0457" "COG0545" "COG0810" "COG0811" "COG1409" "COG1452" "COG1538" "COG1566" "COG1596" "COG1629" "COG2067" "COG2814" "COG2825" "COG2885" "COG3176" "COG3187" "COG4206" "COG4235" "COG4775"
cog.su.sp.sum.sig <- which(cog.su.sp.sig.df$log2FoldChange > 0)
cog.su.sp.sum <- rownames(cog.su.sp.sig[cog.su.sp.sum.sig,])
  # Summer
  # "COG0183" "COG0395" "COG0411" "COG0444" "COG0448" "COG1126" "COG1172" "COG1175" "COG1744" "COG1775" "COG3958" "COG3959" "COG4166" "COG4905"

# Spring vs Winter
cog.res.sp.wi <- results(cog.de, contrast=c("Season", "Spring", "Winter"), pAdjustMethod = "BH")
cog.sp.wi.sig <- cog.res.sp.wi[which(cog.res.sp.wi$padj < 0.05),]
cog.sp.wi.sig.df <- cbind(as(cog.sp.wi.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.sp.wi.sig), ], "matrix"))
cog.sp.wi.spr.sig <- which(cog.sp.wi.sig.df$log2FoldChange < 0)
cog.sp.wi.spr <- rownames(cog.sp.wi.sig[cog.sp.wi.spr.sig,])
  # Spring
  # "COG0156" "COG0247" "COG0306" "COG0457" "COG0612" "COG0659" "COG0760" "COG0767" "COG0774" "COG0795" "COG0810" "COG0811" "COG0841" "COG0845" "COG0847" "COG0848" "COG0859" "COG1043" "COG1127" "COG1137" "COG1409" "COG1452" "COG1463" "COG1538" "COG1596" "COG1629" "COG1663" "COG1678" "COG2067" "COG2095" "COG2151" "COG2209" "COG2259" "COG2825" "COG2834" "COG2885" "COG2911" "COG2982" "COG3169" "COG3176" "COG3187" "COG3203" "COG3468" "COG3525" "COG3636" "COG3637" "COG3657" "COG4105" "COG4232" "COG4235" "COG4409" "COG4566" "COG4775" "COG4783"
cog.sp.wi.win.sig <- which(cog.sp.wi.sig.df$log2FoldChange > 0)
cog.sp.wi.win <- rownames(cog.sp.wi.sig[cog.sp.wi.win.sig,])
  # Winter
  # "COG0004" "COG0028" "COG0119" "COG0246" "COG0395" "COG0407" "COG0554" "COG0561" "COG0722" "COG0745" "COG0855" "COG1026" "COG1051" "COG1061" "COG1070" "COG1104" "COG1105" "COG1131" "COG1132" "COG1136" "COG1161" "COG1172" "COG1175" "COG1263" "COG1302" "COG1307" "COG1349" "COG1472" "COG1511" "COG1609" "COG1621" "COG1653" "COG1874" "COG1879" "COG1882" "COG1925" "COG2182" "COG2190" "COG2199" "COG2207" "COG2252" "COG2357" "COG2407" "COG2723" "COG2730" "COG2816" "COG2972" "COG3044" "COG3153" "COG3279" "COG3331" "COG3345" "COG3481" "COG3507" "COG3534" "COG3635" "COG3664" "COG3693" "COG3757" "COG3835" "COG3857" "COG3866" "COG3887" "COG3894" "COG3940" "COG3959" "COG3973" "COG4086" "COG4166" "COG4189" "COG4209" "COG4284" "COG4465" "COG4468" "COG4485" "COG4509" "COG4709" "COG4753" "COG4805" "COG4862" "COG4870" "COG4962" "COG5578"

# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

### Pull out top 2 (largest log fold change while sig) for each comparison
cog.su.wi.sum.df <- cog.su.wi.sig.df[cog.su.wi.sum.sig,]
cog.su.wi.sum.df <- cog.su.wi.sum.df[order(cog.su.wi.sum.df$log2FoldChange),]

cog.su.wi.win.df <- cog.su.wi.sig.df[cog.su.wi.win.sig,]
cog.su.wi.win.df <- cog.su.wi.win.df[order(cog.su.wi.win.df$log2FoldChange),]

cog.su.sp.spr.df <- cog.su.sp.sig.df[cog.su.sp.spr.sig,]
cog.su.sp.spr.df <- cog.su.sp.spr.df[order(cog.su.sp.spr.df$log2FoldChange),]
cog.su.sp.sum.df <- cog.su.sp.sig.df[cog.su.sp.sum.sig,]
cog.su.sp.sum.df <- cog.su.sp.sum.df[order(cog.su.sp.sum.df$log2FoldChange),]
cog.sp.wi.spr.df <- cog.sp.wi.sig.df[cog.sp.wi.spr.sig,]
cog.sp.wi.spr.df <- cog.sp.wi.spr.df[order(cog.sp.wi.spr.df$log2FoldChange),]
cog.sp.wi.win.df <- cog.sp.wi.sig.df[cog.sp.wi.win.sig,]
cog.sp.wi.win.df <- cog.sp.wi.win.df[order(cog.sp.wi.win.df$log2FoldChange),]

cog.sig <- rbind(cog.su.wi.sum.df, cog.su.wi.win.df, cog.su.sp.spr.df, cog.su.sp.sum.df, cog.sp.wi.spr.df, cog.sp.wi.win.df)
cog.sig$log2FoldChange[which(cog.sig$log2FoldChange < 0)] <-
  cog.sig$log2FoldChange[which(cog.sig$log2FoldChange < 0)] * -1
cog.sig <- cog.sig[order(cog.sig$log2FoldChange, decreasing=T),]
cog.use <-rownames(cog.sig)[1:10] 


#select <- order(rowMeans(counts(cog.de, normalized=T)), decreasing=T)[1:20]
  # For top 20 COGs
cog.rows <- rownames(cog.de)
select <- rep(1,10)
for(i in 1:length(cog.use)){
  save <- which(cog.rows == cog.use[i])
  select[i] <- save
}
cog.de.df <- as.data.frame(colData(cog.de)[,c("Season")])
rownames(cog.de.df) <- meta$X
colnames(cog.de.df) <- "Season"
cog.de.df$Season <- ordered(cog.de.df$Season, levels=c("Summer", "Winter", "Spring"))


cog.de.df <- data.frame(c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"))
rownames(cog.de.df) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")
colnames(cog.de.df) <- "Season"
cog.de.df$Season <- ordered(cog.de.df$Season, levels=c("Summer", "Winter", "Spring"))
cog.de$X <- ordered(cog.de$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))


cog.ntd <- normTransform(cog.de)
  # DESeq test automatically transforms data
  # To visualize our data, we need to run this step separately
  # I use normTransform because I run DESeq with local fit
      # Local fit does a transformation similar to normTransform
  # Otherwise you should use vst instead
  # See link above for more info

cog.ntd.assay <- data.frame(assay(cog.ntd)[select,])
cog.ntd.assay.order <- cog.ntd.assay
cog.ntd.assay.order[,1] <- cog.ntd.assay$X3715
cog.ntd.assay.order[,2] <- cog.ntd.assay$X3717
cog.ntd.assay.order[,3] <- cog.ntd.assay$X3744
cog.ntd.assay.order[,4] <- cog.ntd.assay$X3723
cog.ntd.assay.order[,5] <- cog.ntd.assay$X3733
cog.ntd.assay.order[,6] <- cog.ntd.assay$X3772
cog.ntd.assay.order[,7] <- cog.ntd.assay$X3734
cog.ntd.assay.order[,8] <- cog.ntd.assay$X3773
cog.ntd.assay.order[,9] <- cog.ntd.assay$X3775
colnames(cog.ntd.assay.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

tiff("COG_DESeq.tiff", width=6.2, height=5.5, units="in", res=600)
pheatmap(cog.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, treeheight_col = 0,
         show_colnames=F, border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255),
         labels_col = c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"), annotation_legend = F,
         annotation_col=cog.de.df,
         annotation_colors = list(Season = c(Summer = "#00cccc", Winter = "#008484", Spring = "#004f4f"))
)
dev.off()


tiff("COG_DESeq1.tiff", width=3, height=2.25, units="in", res=600)
pheatmap(cog.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, show_colnames=F, 
         border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
)
dev.off()




### Sample Clustering: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

cog.kend <- read.table("COG/cog.kendall.txt", sep="\t")
cog.kend.fix <- cog.kend
cog.kend.fix[1,1] <- 0
cog.kend.fix[2,2] <- 0
cog.kend.fix[3,3] <- 0
cog.kend.fix[4,4] <- 0
cog.kend.fix[5,5] <- 0
cog.kend.fix[6,6] <- 0
cog.kend.fix[7,7] <- 0
cog.kend.fix[8,8] <- 0
cog.kend.fix[9,9] <- 0
pheatmap(cog.kend.fix, clustering_distance_rows=cog.kend.fix, clustering_distance_cols=cog.kend.fix)

cog.dist <- dist(t(assay(cog.ntd)))
cog.dist.matrix <- as.matrix(cog.dist)
rownames(cog.dist.matrix) <- paste(cog.ntd$Season)
colnames(cog.dist.matrix) <- NULL


pheatmap(cog.dist.matrix, clustering_distance_rows=cog.dist, clustering_distance_cols=cog.dist,
         col=colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(100), annotation_rows=T, annotation_cols=T)
  # Default clustering method = complete


cog.pca <- plotPCA(cog.ntd, intgroup="Season", returnData=T)
cog.pca.var <- round(100 * attr(cog.pca, "Percent_Variation"))
ggplot(cog.pca, aes(x=PC1, y=PC2, color=Season)) +
  geom_point(size=2) +
  xlab(paste0("PC1: ", cog.pca.var[1], "% variance")) +
  ylab(paste0("PC2: ", cog.pca.var[2], "% variance")) +
  coord_fixed()

# Prepare phyloseq object
cog.size.fac <- estimateSizeFactors(cog.de)
cog.norm.deseq <- counts(cog.size.fac, normalized=TRUE)
cog.otu.deseq <- otu_table(cog.norm.deseq, taxa_are_rows = T)
cog.samp <- sample_data(meta)
cog.tax <- tax_table(cog.class)
rownames(cog.tax) <- rownames(cog.class)
cog.deseq <- merge_phyloseq(cog.otu.deseq, cog.samp, cog.tax)
sample_data(cog.deseq)$Season <- ordered(sample_data(cog.deseq)$Season, levels=c("Summer", "Winter", "Spring"))

nmds.cog.deseq <- ordinate(physeq=cog.deseq, method="NMDS", distance="bray")
  # Run 20 stress 0.06470567

### SEASON ELLIPSES
sample_data(cog.deseq)$Elip <- paste(sample_data(cog.deseq)$Season)
cog.deseq.elip.sum <- data.frame(MDS1 = nmds.cog.deseq$points[,1], MDS2 = nmds.cog.deseq$points[,2],
                           Group = sample_data(cog.deseq)$Season)
plot.new()
cog.deseq.elip.ord <- ordiellipse(nmds.cog.deseq, sample_data(cog.deseq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.deseq.elip.sum.df <- data.frame()
for(i in levels(cog.deseq.elip.sum$Group)){
  cog.deseq.elip.sum.df <- rbind(cog.deseq.elip.sum.df,
                           cbind(as.data.frame(with(cog.deseq.elip.sum[cog.deseq.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.deseq.elip.ord[[i]]$cov,
                                                                    cog.deseq.elip.ord[[i]]$center,
                                                                    cog.deseq.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
#tiff("COG_NMDS.tiff", width=3.5, height=3.5, units="in", res=600)
plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
#dev.off()


cog.nmds.plot <- plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
cog.nmds.legend <- g_legend(cog.nmds.plot)
cog.nmds.plot <- plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5),
        plot.margin = unit(c(0.1, 1, 0.1, 0.1), "cm")) +
  guides(color=F)


### PERMANOVA
cog.deseq.sampdf <- data.frame(sample_data(cog.deseq))
cog.deseq.bray<- phyloseq::distance(physeq=cog.deseq, method="bray")

## Season
adonis(cog.deseq.bray~ Season, data=cog.deseq.sampdf)
  # p-value = 0.003
beta.tax = betadisper(d=cog.deseq.bray, group=cog.deseq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.721


## Diet
adonis(cog.deseq.bray~ Diet, data=cog.deseq.sampdf)
  # p-value = 0.137
beta.tax = betadisper(d=cog.deseq.bray, group=cog.deseq.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.12




```

### COG Bray-Curtis
```{r}
# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
cog.all.norm <- cog.all
for(i in 1:9){
  cog.all.norm[,i] <- round(cog.all.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(cog.all.norm)){
  print(sum(cog.all.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
cog.otu <- otu_table(cog.all, taxa_are_rows = T)
cog.samp <- sample_data(meta)
cog.tax <- tax_table(cog.class)
rownames(cog.tax) <- rownames(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)
sample_data(cog.physeq)$Season <- ordered(sample_data(cog.physeq)$Season, levels=c("Summer", "Winter", "Spring"))


nmds.cog <- ordinate(physeq=cog.physeq, method="NMDS", distance="bray")
  # stress = 0.05592207 

plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Season")
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Diet")
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Sex")

### DIET ELLIPSES
sample_data(cog.physeq)$Elip <- paste(sample_data(cog.physeq)$Diet)
cog.elip.sum <- data.frame(MDS1 = nmds.cog$points[,1], MDS2 = nmds.cog$points[,2],
                           Group = sample_data(cog.physeq)$Elip)
plot.new()
cog.elip.ord <- ordiellipse(nmds.cog, sample_data(cog.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.elip.sum.df <- data.frame()
for(i in levels(cog.elip.sum$Group)){
  cog.elip.sum.df <- rbind(cog.elip.sum.df,
                           cbind(as.data.frame(with(cog.elip.sum[cog.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.elip.ord[[i]]$cov,
                                                                    cog.elip.ord[[i]]$center,
                                                                    cog.elip.ord[[i]]$scale))), group=i
                                 ))

}


### SEASON ELLIPSES
sample_data(cog.physeq)$Elip <- paste(sample_data(cog.physeq)$Season)
cog.elip.sum <- data.frame(MDS1 = nmds.cog$points[,1], MDS2 = nmds.cog$points[,2],
                           Group = sample_data(cog.physeq)$Season)
plot.new()
cog.elip.ord <- ordiellipse(nmds.cog, sample_data(cog.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.elip.sum.df <- data.frame()
for(i in levels(cog.elip.sum$Group)){
  cog.elip.sum.df <- rbind(cog.elip.sum.df,
                           cbind(as.data.frame(with(cog.elip.sum[cog.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.elip.ord[[i]]$cov,
                                                                    cog.elip.ord[[i]]$center,
                                                                    cog.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Season") +
  geom_path(data=cog.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("sd", "asdf", "Asdf"))




### PERMANOVA
cog.sampdf <- data.frame(sample_data(cog.physeq))
cog.bray<- phyloseq::distance(physeq=cog.physeq, method="bray")

## Season
adonis(cog.bray~ Season, data=cog.sampdf)
  # p-value = 0.164
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.877

## Diet
adonis(cog.bray~ Diet, data=cog.sampdf)
  # p-value = 0.237
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.426

## Sex
adonis(cog.bray~ Sex, data=cog.sampdf)
  # p-value = 0.372
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.58

### ANOSIM
anosim(cog.bray, meta$Season, permutations=1000)
  # p-value = 0.1852
anosim(cog.bray, meta$Diet, permutations=1000)
  # p-value = 0.1049
anosim(cog.bray, meta$Sex, permutations=1000)
  # p-value = 0.92308
```

### COG Jaccard
```{r}
jac.cog <- ordinate(physeq=cog.physeq, method="NMDS", distance="jaccard")
  # stress = 0.04125187
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Season")
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Diet")
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Sex")

### PERMANOVA
cog.sampdf <- data.frame(sample_data(cog.physeq))
cog.jac <- phyloseq::distance(physeq=cog.physeq, method="jaccard")

## Season
adonis(cog.jac~ Season, data=cog.sampdf)
  # p-value = 0.148
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.867

## Diet
adonis(cog.jac~ Diet, data=cog.sampdf)
  # p-value = 0.17
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.398

## Sex
adonis(cog.jac~ Sex, data=cog.sampdf)
  # p-value = 0.509
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.585

### ANOSIM
anosim(cog.jac, cog.sampdf$Season, permutations=1000)
  # p-value = 0.14386
anosim(cog.jac, cog.sampdf$Diet, permutations=1000)
  # p-value = 0.25874
anosim(cog.jac, cog.sampdf$Sex, permutations=1000)
  # p-value = 0.9001
```

### COG SIMPER
```{r}
### Separate out pairw
simper.pretty(cog.all.norm, metrics = meta, interesting = c('Season', 'Diet', 'Sex'), perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'cog')

cog.simper <- read.csv("cog_clean_simper.csv")

kruskal.pretty(otu = cog.all.norm, metrics = meta, csv = cog.simper, interesting = c('Season', 'Diet', 'Sex'), output_name = 'cog.kw')

cog.simper.kw <- read.csv("cog.kw_krusk_simper.csv")
```



### Poster combined NMDS (COG & KEGG)
```{r}
tiff("NMDS_cog_kegg.tiff", width = 7.5, height = 3, units = "in", res = 600)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(1,3, widths=c(1,1,0.3)))))
print(cog.nmds.plot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(kegg.nmds.plot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
cog.nmds.legend$vp <- viewport(layout.pos.row = 1, layout.pos.col = 3)
grid.draw(cog.nmds.legend)
dev.off()
```