---
title: "Squirrel_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---

```{r}
library('readr')
source('http://portal.nersc.gov/dna/RD/Metagenome_RD/MetaBAT/Files/benchmark.R')
  # https://bitbucket.org/berkeleylab/metabat/wiki/Best%20Binning%20Practices
#library('limma')
library('pgirmess')
set.seed(1)
```

### No Host ###

### 3715 idba metabat
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.idba.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.nohost.idba.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.nohost.idba.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.nohost.idba.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.idba.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.nohost.idba.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```

### 3715 SPAdes metabat
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.nohost.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.nohost.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.nohost.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.nohost.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.nohost.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.nohost.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3715 nohost idba centrifuge
```{r}
# Load in .tsv files
nohost.3715.idba.output <- read_tsv('3715.nohost.idba.125.output.tsv')

# Calculate proportion classified & unclassified
nohost.3715.idba.prop <- data.frame(matrix(1,2))
unclass.3715.idba <- length(which(nohost.3715.idba.output$seqID == "unclassified"))/nrow(nohost.3715.idba.output)

# Number of species
nohost.3715.idba.report <- read_tsv('3715.nohost.idba.125.report.tsv')
nohost.3715.idba.abund <- nohost.3715.idba.report[which(nohost.3715.idba.report$abundance>0),]
num.species.3715.idba <- nrow(nohost.3715.idba.abund)
```

### 3715 nohost SPAdes centrifuge
```{r}
# Load in .tsv files
nohost.3715.spades.output <- read_tsv('3715.nohost.spades.125.output.tsv')

# Calculate proportion classified & unclassified
nohost.3715.spades.prop <- data.frame(matrix(1,2))
unclass.3715.spades <- length(which(nohost.3715.spades.output$seqID == "unclassified"))/nrow(nohost.3715.spades.output)

# Number of species
nohost.3715.spades.report <- read_tsv('3715.nohost.spades.125.report.tsv')
nohost.3715.spades.abund <- nohost.3715.spades.report[which(nohost.3715.spades.report$abundance>0),]
num.species.3715.spades <- nrow(nohost.3715.spades.abund)
```

### 3715 nohost megahit centrifuge
```{r}
# Load in .tsv files
nohost.3715.megahit.output <- read_tsv('3715.nohost.megahit.125.output.tsv')

# Calculate proportion classified & unclassified
nohost.3715.megahit.prop <- data.frame(matrix(1,2))
unclass.3715.megahit <- length(which(nohost.3715.megahit.output$seqID == "unclassified"))/nrow(nohost.3715.megahit.output)

# Number of species
nohost.3715.megahit.report <- read_tsv('3715.nohost.megahit.125.report.tsv')
nohost.3715.megahit.abund <- nohost.3715.megahit.report[which(nohost.3715.megahit.report$abundance>0),]
num.species.3715.megahit <- nrow(nohost.3715.megahit.abund)
```

### 3715 nohost unassembled centrifuge
```{r}
# Load in .tsv files
nohost.3715.output <- read_tsv('3715.nohost.output2.tsv')

# Calculate proportion classified & unclassified
nohost.3715.prop <- data.frame(matrix(1,2))
unclass.nohost.3715 <- length(which(nohost.3715.output$seqID == "unclassified"))/nrow(nohost.3715.output)

# Number of species
nohost.3715.report <- read_tsv('3715.nohost.report2.tsv')
nohost.3715.abund <- nohost.3715.report[which(nohost.3715.report$abundance>0),]
num.species.3715 <- nrow(nohost.3715.abund)


```


### 3733 nohost SPAdes centrifuge
```{r}
# Load in .tsv files
nohost.3733.spades.output <- read_tsv('3733.nohost.spades.127.output.tsv')

# Calculate proportion classified & unclassified
nohost.3733.spades.prop <- data.frame(matrix(1,2))
unclass.3733.spades <- length(which(nohost.3733.spades.output$seqID == "unclassified"))/nrow(nohost.3733.spades.output)

# Number of species
nohost.3733.spades.report <- read_tsv('3733.nohost.spades.127.report.tsv')
nohost.3733.spades.abund <- nohost.3733.spades.report[which(nohost.3733.spades.report$abundance>0),]
num.species.3733.spades <- nrow(nohost.3733.spades.abund)
```



### 3733 nohost idba centrifuge
```{r}
# Load in .tsv files
nohost.3733.idba.output <- read_tsv('3733.nohost.idba.127.output.tsv')

# Calculate proportion classified & unclassified
nohost.3733.idba.prop <- data.frame(matrix(1,2))
unclass.3733.idba <- length(which(nohost.3733.idba.output$seqID == "unclassified"))/nrow(nohost.3733.idba.output)

# Number of species
nohost.3733.idba.report <- read_tsv('3733.nohost.idba.127.report.tsv')
nohost.3733.idba.abund <- nohost.3733.idba.report[which(nohost.3733.idba.report$abundance>0),]
num.species.3733.idba <- nrow(nohost.3733.idba.abund)
```

### 3733 nohost unassembled centrifuge
```{r}
# Load in .tsv files
nohost.3733.output <- read_tsv('3733.nohost.output.tsv')

# Calculate proportion classified & unclassified
nohost.3733.prop <- data.frame(matrix(1,2))
unclass.nohost.3733 <- length(which(nohost.3733.output$seqID == "unclassified"))/nrow(nohost.3733.output)

# Number of species
nohost.3733.report <- read_tsv('3733.nohost.report.tsv')
nohost.3733.abund <- nohost.3733.report[which(nohost.3733.report$abundance>0),]
num.species.3733 <- nrow(nohost.3733.abund)
```


### 3775 nohost SPAdes centrifuge
```{r}
# Load in .tsv files
nohost.3775.spades.output <- read_tsv('3775.nohost.spades.127.output.tsv')

# Calculate proportion classified & unclassified
nohost.3775.spades.prop <- data.frame(matrix(1,2))
unclass.3775.spades <- length(which(nohost.3775.spades.output$seqID == "unclassified"))/nrow(nohost.3775.spades.output)

# Number of species
nohost.3775.spades.report <- read_tsv('3775.nohost.spades.127.report.tsv')
nohost.3775.spades.abund <- nohost.3775.spades.report[which(nohost.3775.spades.report$abundance>0),]
num.species.3775.spades <- nrow(nohost.3775.spades.abund)
```

### 3775 nohost idba centrifuge
```{r}
# Load in .tsv files
nohost.3775.idba.output <- read_tsv('3775.nohost.idba.127.output.tsv')

# Calculate proportion classified & unclassified
nohost.3775.idba.prop <- data.frame(matrix(1,2))
unclass.3775.idba <- length(which(nohost.3775.idba.output$seqID == "unclassified"))/nrow(nohost.3775.idba.output)

# Number of species
nohost.3775.idba.report <- read_tsv('3775.nohost.idba.127.report.tsv')
nohost.3775.idba.abund <- nohost.3775.idba.report[which(nohost.3775.idba.report$abundance>0),]
num.species.3775.idba <- nrow(nohost.3775.idba.abund)
```


### 3775 nohost unassembled centrifuge
```{r}
# Load in .tsv files
nohost.3775.output <- read_tsv('3775.nohost.output.tsv')

# Calculate proportion classified & unclassified
nohost.3775.prop <- data.frame(matrix(1,2))
unclass.nohost.3775 <- length(which(nohost.3775.output$seqID == "unclassified"))/nrow(nohost.3775.output)

# Number of species
nohost.3775.report <- read_tsv('3775.nohost.report.tsv')
nohost.3775.abund <- nohost.3775.report[which(nohost.3775.report$abundance>0),]
num.species.3775 <- nrow(nohost.3775.abund)
```










### ISME ###
# All SPAdes 125

### 3715
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3717
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3717.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3717.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3717.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3717.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3717.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3717.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3723
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3723.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3723.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3723.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3723.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3723.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3723.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3733
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3733.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3733.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3733.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3733.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3733.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3733.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```

### 3734
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3734.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3734.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3734.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3734.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3734.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3734.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3744
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3744.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3744.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3744.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3744.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3744.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3744.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3772
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3772.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3772.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3772.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3772.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3772.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3772.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3773
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3773.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3773.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3773.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3773.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3773.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3773.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### 3775
```{r}
### Job 1
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("isme/checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("isme/checkm.3715.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

```


### KEGG Outputs (metaSPAdes) ###
```{r}
kegg <- read.csv('isme/KEGG/KEGG_results.csv', fileEncoding="UTF-8-BOM")
kegg.idba <

# Calculate proportion of each pathway in each metagenome
kegg.prop <- kegg
for(i in 1:nrow(kegg.prop)){
  for(j in 4:ncol(kegg.prop)){
    kegg.prop[i,j] <- (kegg.prop[i,j]/kegg.prop[i,3])*100
  }
}


# Kruskal-Wallis
kruskal.test(formula= CarbMet ~ Season, data=kegg.prop)
  # p-value = 0.6703
kruskal.test(formula= FAMet ~ Season, data=kegg.prop)
  # p-value = 0.1479
kruskal.test(formula= NRGMet ~ Season, data=kegg.prop)
  # p-value = 0.4298
kruskal.test(formula= LipMet ~ Season, data=kegg.prop)
  # p-value = 0.1133
kruskal.test(formula= NucMet ~ Season, data=kegg.prop)
  # p-value = 0.1133
kruskal.test(formula= AAMet ~ Season, data=kegg.prop)
  # p-value = 0.5611
kruskal.test(formula= GlycMet ~ Season, data=kegg.prop)
  # p-value = 0.06081
kruskal.test(formula= CofVitMet ~ Season, data=kegg.prop)
  # p-value = 0.06646
kruskal.test(formula= TerPolMet ~ Season, data=kegg.prop)
  # p-value = 0.3932
kruskal.test(formula= SecMet ~ Season, data=kegg.prop)
  # p-value = 0.4298
kruskal.test(formula= Xen ~ Season, data=kegg.prop)
  # p-value = 0.2521
kruskal.test(formula= Met_Tot ~ Season, data=kegg.prop)
  # p-value = 0.1931
kruskal.test(formula= Transcription ~ Season, data=kegg.prop)
  # p-value = 0.5611
kruskal.test(formula= Translation ~ Season, data=kegg.prop)
  # p-value = 0.2881
kruskal.test(formula= FoldSortDegr ~ Season, data=kegg.prop)
  # p-value = 0.09915
kruskal.test(formula= RepRep ~ Season, data=kegg.prop)
  # p-value = 0.1767
kruskal.test(formula= GeneticInfo_Tot ~ Season, data=kegg.prop)
  # p-value = 0.3932
kruskal.test(formula= MemTrans ~ Season, data=kegg.prop)
  # p-value = 0.06646
kruskal.test(formula= SigTransd ~ Season, data=kegg.prop)
  # p-value = 0.1931
kruskal.test(formula= SigMol ~ Season, data=kegg.prop)
  # p-value = 0.3292
kruskal.test(formula= EnvironInfoProc_Tot ~ Season, data=kegg.prop)
  # p-value = 0.02732
kruskal.test(formula= Other_Tot ~ Season, data=kegg.prop)
  # p-value = 0.3292

# Can I find pairwise trends?
kruskalmc(resp = GlycMet ~ Season, data =kegg.prop)
kruskalmc(resp = CofVitMet ~ Season, data =kegg.prop)
kruskalmc(resp = FoldSortDegr ~ Season, data =kegg.prop)
kruskalmc(resp = MemTrans ~ Season, data =kegg.prop)
kruskalmc(resp = EnvironInfoProc_Tot ~ Season, data =kegg.prop)
  # Spring differs from winter
  # No diff btwn spr/sum & sum/win


kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.04953
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="less")
  # Sum vs. Win
  # Sum < Win
  # p-value = 0.05
kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.5127
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Winter")
  # Sum vs. Spr
  # p-value = 0.7
kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.04953
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="less")
  # Sum vs. Win
  # Sum < Win
  # p-value = 0.05
kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.8273
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Winter")
  # Sum vs. Spr
  # p-value = 1
kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.5127
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Spring")
  # Sum vs. Win
  # p-value = 0.7
kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.1266
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="less")
  # Sum vs. Spr
  # Spr < Sum (weak trend)
  # p-value = 0.1
kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.8273
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Spring")
  # Sum vs. Win
  # p-value = 1
kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.04953
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="greater")
  # Sum vs. Spr
  # Spr > Sum
  # p-value = 0.05
kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="greater")
  # Win vs. Spr
  # Spr > Win
  # p-value = 0.05


kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.04953
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="greater")
  # Sum vs. Win
  # Sum > Win
  # p-value = 0.05
kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.04953
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="greater")
  # Sum vs. Spr
  # Spr > Sum
  # p-value = 0.05
kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="greater")
  # Win vs. Spr
  # Spr > Win
  # p-value = 0.05
```



### KEGG Outputs (IDBA) ###
```{r}
kegg <- read.csv('isme/KEGG/KEGG_results.csv', fileEncoding="UTF-8-BOM")
kegg <- kegg[10:17,]

# Calculate proportion of each pathway in each metagenome
kegg.prop <- kegg
for(i in 1:nrow(kegg.prop)){
  for(j in 4:ncol(kegg.prop)){
    kegg.prop[i,j] <- (kegg.prop[i,j]/kegg.prop[i,3])*100
  }
}


# Kruskal-Wallis
kruskal.test(formula= CarbMet ~ Season, data=kegg.prop)
  # p-value = 0.8948
kruskal.test(formula= FAMet ~ Season, data=kegg.prop)
  # p-value = 0.08208
kruskal.test(formula= NRGMet ~ Season, data=kegg.prop)
  # p-value = 0.2494
kruskal.test(formula= LipMet ~ Season, data=kegg.prop)
  # p-value = 0.06218
kruskal.test(formula= NucMet ~ Season, data=kegg.prop)
  # p-value = 0.1492
kruskal.test(formula= AAMet ~ Season, data=kegg.prop)
  # p-value = 0.1997
kruskal.test(formula= GlycMet ~ Season, data=kegg.prop)
  # p-value = 0.08208
kruskal.test(formula= CofVitMet ~ Season, data=kegg.prop)
  # p-value = 0.08208
kruskal.test(formula= TerPolMet ~ Season, data=kegg.prop)
  # p-value = 0.4857
kruskal.test(formula= SecMet ~ Season, data=kegg.prop)
  # p-value = 0.1194
kruskal.test(formula= Xen ~ Season, data=kegg.prop)
  # p-value = 0.1512
kruskal.test(formula= Met_Tot ~ Season, data=kegg.prop)
  # p-value = 0.1931
kruskal.test(formula= Transcription ~ Season, data=kegg.prop)
  # p-value = 0.2494
kruskal.test(formula= Translation ~ Season, data=kegg.prop)
  # p-value = 0.26
kruskal.test(formula= FoldSortDegr ~ Season, data=kegg.prop)
  # p-value = 0.08208
kruskal.test(formula= RepRep ~ Season, data=kegg.prop)
  # p-value = 0.3889
kruskal.test(formula= GeneticInfo_Tot ~ Season, data=kegg.prop)
  # p-value = 0.3889
kruskal.test(formula= MemTrans ~ Season, data=kegg.prop)
  # p-value = 0.07658
kruskal.test(formula= SigTransd ~ Season, data=kegg.prop)
  # p-value = 0.1863
kruskal.test(formula= SigMol ~ Season, data=kegg.prop)
  # p-value = 0.8948
kruskal.test(formula= EnvironInfoProc_Tot ~ Season, data=kegg.prop)
  # p-value = 0.07658
kruskal.test(formula= Other_Tot ~ Season, data=kegg.prop)
  # p-value = 0.3679

# Can I find pairwise trends?
kruskalmc(resp = LipMet ~ Season, data =kegg.prop)
kruskalmc(resp = GlycMet ~ Season, data =kegg.prop)
kruskalmc(resp = CofVitMet ~ Season, data =kegg.prop)
kruskalmc(resp = FoldSortDegr ~ Season, data =kegg.prop)
kruskalmc(resp = MemTrans ~ Season, data =kegg.prop)
kruskalmc(resp = EnvironInfoProc_Tot ~ Season, data =kegg.prop)


kruskal.test(formula = LipMet ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.2482
wilcox.test(LipMet ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="less")
  # Sum vs. Win
  # p-value = 0.9
kruskal.test(formula = LipMet ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.08326
wilcox.test(LipMet ~ Season, data=kegg.prop, subset= Season!="Winter")
  # Sum vs. Spr
  # p-value = 0.2
kruskal.test(formula = LipMet ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(LipMet ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05

kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 1
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="less")
  # Sum vs. Win
  # Sum < Win
  # p-value = 0.6
kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.08326
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Winter")
  # Sum vs. Spr
  # p-value = 0.2
kruskal.test(formula = GlycMet ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(GlycMet ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 1
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="less")
  # Sum vs. Win
  # Sum < Win
  # p-value = 0.6
kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.8326
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Winter")
  # Sum vs. Spr
  # p-value = 0.2
kruskal.test(formula = CofVitMet ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(CofVitMet ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.08326
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Spring")
  # Sum vs. Win
  # p-value = 0.2
kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 1
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="less")
  # Sum vs. Spr
  # Spr < Sum (weak trend)
  # p-value = 0.6
kruskal.test(formula = FoldSortDegr ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(FoldSortDegr ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="less")
  # Win vs. Spr
  # Spr < Win
  # p-value = 0.05


kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.5637
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Spring")
  # Sum vs. Win
  # p-value = 0.8
kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.08326
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="greater")
  # Sum vs. Spr
  # Spr > Sum
  # p-value = 0.1
kruskal.test(formula = MemTrans ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(MemTrans ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="greater")
  # Win vs. Spr
  # Spr > Win
  # p-value = 0.05


kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Spring")
  # Sum vs. Win
  # p-value = 0.5637
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Spring", alternative="greater")
  # Sum vs. Win
  # Sum > Win
  # p-value = 0.8
kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Winter")
  # Sum vs. Spr
  # p-value = 0.08326
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Winter", alternative="greater")
  # Sum vs. Spr
  # Spr > Sum
  # p-value = 0.1
kruskal.test(formula = EnvironInfoProc_Tot ~ Season, data = kegg.prop, subset= Season!= "Summer")
  # Win vs. Spr
  # p-value = 0.04953
wilcox.test(EnvironInfoProc_Tot ~ Season, data=kegg.prop, subset= Season!="Summer", alternative="greater")
  # Win vs. Spr
  # Spr > Win
  # p-value = 0.05
```



















##### OLD #####

### 3715 IDBA 111
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.111.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.idba.111.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.idba.111.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.idba.111.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.111.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.idba.111.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```


### 3715 IDBA 95
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=F), calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.95.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.idba.95.job2.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.idba.95.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```


### 3715 SPAdes 95
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.95.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.95.job2.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.95.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```



### 3715 SPAdes 105
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.105.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.105.job2.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.105.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```




### 3715 SPAdes 105
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.115.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.115.job2.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.115.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```





### 3715 SPAdes 125
```{r}
### Job 1
printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 2
printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

### Job 3
printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=F)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

printPerf(list(calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=T)), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))


### Comparisons
diffPerf(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.125.job1.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=F), 
         calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=F), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))

diffPerf(calcPerfBySCG("checkm.3715.spades.125.job2.txt", removeStrain=T), 
         calcPerfBySCG("checkm.3715.spades.125.job3.txt", removeStrain=T), rec=c(seq(.1,.9,.1),.95), prec=c(seq(.6,.9,.1),.95,.99))
```





###### OLD ######









### 3715 Untrimmed SPAdes (Old)
```{r}
# Load in .tsv files
untrim.3715.55 <- read_tsv('3715.untrim/old/3715.untrimmed.55.output.tsv')
untrim.3715.75 <- read_tsv('3715.untrim/old/3715.untrimmed.75.output.tsv')
untrim.3715.95 <- read_tsv('3715.untrim/old/3715.untrimmed.95.output.tsv')
untrim.3715.111 <- read_tsv('3715.untrim/old/3715.untrimmed.111.output.tsv')
untrim.3715.127 <- read_tsv('3715.untrim/old/3715.untrimmed.127.output.tsv')

untrim.3715.55.sq <- read_tsv('3715.untrim/old/3715.untrimmed.55.sq.output.tsv')
untrim.3715.75.sq <- read_tsv('3715.untrim/old/3715.untrimmed.75.sq.output.tsv')
untrim.3715.95.sq <- read_tsv('3715.untrim/old/3715.untrimmed.95.sq.output.tsv')
untrim.3715.111.sq <- read_tsv('3715.untrim/old/3715.untrimmed.111.sq.output.tsv')
untrim.3715.127.sq <- read_tsv('3715.untrim/old/3715.untrimmed.127.sq.output.tsv')

# Calculate unclassified vs. classified
untrim.3715.prop <- data.frame(matrix(5,5))
untrim.3715.prop$Kmer <- c(55,75,95,111,127)
untrim.3715.prop$Unclassified <- c(
  length(which(untrim.3715.55$seqID =="unclassified"))/nrow(untrim.3715.55),
  length(which(untrim.3715.55$seqID =="unclassified"))/nrow(untrim.3715.55),
  length(which(untrim.3715.95$seqID =="unclassified"))/nrow(untrim.3715.95),
  length(which(untrim.3715.111$seqID =="unclassified"))/nrow(untrim.3715.111),
  length(which(untrim.3715.127$seqID =="unclassified"))/nrow(untrim.3715.127)
)
untrim.3715.prop <- untrim.3715.prop[,-1]
untrim.3715.prop$Bact.Arch <- c(
  length(which(untrim.3715.55$seqID !="unclassified"))/nrow(untrim.3715.55),
  length(which(untrim.3715.55$seqID !="unclassified"))/nrow(untrim.3715.55),
  length(which(untrim.3715.95$seqID !="unclassified"))/nrow(untrim.3715.95),
  length(which(untrim.3715.111$seqID !="unclassified"))/nrow(untrim.3715.111),
  length(which(untrim.3715.127$seqID !="unclassified"))/nrow(untrim.3715.127)
)
untrim.3715.prop$Squirrel <- c(
  length(which(untrim.3715.55.sq$seqID !="unclassified"))/nrow(untrim.3715.55.sq),
  length(which(untrim.3715.55.sq$seqID !="unclassified"))/nrow(untrim.3715.55.sq),
  length(which(untrim.3715.95.sq$seqID != "unclassified"))/nrow(untrim.3715.95.sq),
  length(which(untrim.3715.111.sq$seqID !="unclassified"))/nrow(untrim.3715.111.sq),
  length(which(untrim.3715.127.sq$seqID !="unclassified"))/nrow(untrim.3715.127.sq)
)
untrim.3715.prop$Unclassified <- 1 - untrim.3715.prop$Bact.Arch - untrim.3715.prop$Squirrel

# Number of classified bact/arch
untrim.3715.55.report <- read_tsv('3715.untrim/old/3715.untrimmed.55.report.tsv')
untrim.3715.75.report <- read_tsv('3715.untrim/old/3715.untrimmed.75.report.tsv')
untrim.3715.95.report <- read_tsv('3715.untrim/old/3715.untrimmed.95.report.tsv')
untrim.3715.111.report <- read_tsv('3715.untrim/old/3715.untrimmed.111.report.tsv')
untrim.3715.127.report <- read_tsv('3715.untrim/old/3715.untrimmed.127.report.tsv')

untrim.3715.55.abund <- untrim.3715.55.report[which(untrim.3715.55.report$abundance>0),]
untrim.3715.75.abund <- untrim.3715.55.report[which(untrim.3715.75.report$abundance>0),]
untrim.3715.95.abund <- untrim.3715.55.report[which(untrim.3715.95.report$abundance>0),]
untrim.3715.111.abund <- untrim.3715.55.report[which(untrim.3715.111.report$abundance>0),]
untrim.3715.127.abund <- untrim.3715.55.report[which(untrim.3715.127.report$abundance>0),]

untrim.3715.prop$Num.Species <- c(nrow(untrim.3715.55.abund),nrow(untrim.3715.75.abund),
                                nrow(untrim.3715.95.abund),nrow(untrim.3715.111.abund),
                                nrow(untrim.3715.127.abund))

# Quast N50
untrim.3715.prop$N50 <- c(2505,2959,3296,3545,4801)
untrim.3715.prop$Contigs <- c(111864,108790,105754,103161,91748)      
untrim.3715.prop$Largest.Contig <- c(115865,207778,399291,518071,1334334)
```

### 3715 Trimmed SPAdes (Old)
```{r}
# Load in .tsv files
trim.3715.55 <- read_tsv('3715.trimmed/old/3715.trimmed.55.output.tsv')
trim.3715.75 <- read_tsv('3715.trimmed/old/3715.trimmed.75.output.tsv')
trim.3715.95 <- read_tsv('3715.trimmed/old/3715.trimmed.95.output.tsv')
trim.3715.111 <- read_tsv('3715.trimmed/old/3715.trimmed.111.output.tsv')
trim.3715.127 <- read_tsv('3715.trimmed/old/3715.trimmed.127.output.tsv')

trim.3715.55.sq <- read_tsv('3715.trimmed/old/3715.trimmed.55.sq.output.tsv')
trim.3715.75.sq <- read_tsv('3715.trimmed/old/3715.trimmed.75.sq.output.tsv')
trim.3715.95.sq <- read_tsv('3715.trimmed/old/3715.trimmed.95.sq.output.tsv')
trim.3715.111.sq <- read_tsv('3715.trimmed/old/3715.trimmed.111.sq.output.tsv')
trim.3715.127.sq <- read_tsv('3715.trimmed/old/3715.trimmed.127.sq.output.tsv')

# Calculate unclassified vs. classified
trim.3715.prop <- data.frame(matrix(5,5))
trim.3715.prop$Kmer <- c(55,75,95,111,127)
trim.3715.prop$Unclassified <- c(
  length(which(trim.3715.55$seqID =="unclassified"))/nrow(trim.3715.55),
  length(which(trim.3715.75$seqID =="unclassified"))/nrow(trim.3715.75),
  length(which(trim.3715.95$seqID =="unclassified"))/nrow(trim.3715.95),
  length(which(trim.3715.111$seqID =="unclassified"))/nrow(trim.3715.111),
  length(which(trim.3715.127$seqID =="unclassified"))/nrow(trim.3715.127)
  )
trim.3715.prop <- trim.3715.prop[,-1]
trim.3715.prop$Bact.Arch <- c(
  length(which(trim.3715.55$seqID !="unclassified"))/nrow(trim.3715.55),
  length(which(trim.3715.55$seqID !="unclassified"))/nrow(trim.3715.55),
  length(which(trim.3715.95$seqID !="unclassified"))/nrow(trim.3715.95),
  length(which(trim.3715.111$seqID !="unclassified"))/nrow(trim.3715.111),
  length(which(trim.3715.127$seqID !="unclassified"))/nrow(trim.3715.127)
)
trim.3715.prop$Squirrel <- c(
  length(which(trim.3715.55.sq$seqID != "unclassified"))/nrow(trim.3715.55.sq),
  length(which(trim.3715.55.sq$seqID != "unclassified"))/nrow(trim.3715.55.sq),
  length(which(trim.3715.95.sq$seqID !="unclassified"))/nrow(trim.3715.95.sq),
  length(which(trim.3715.111.sq$seqID !="unclassified"))/nrow(trim.3715.111.sq),
  length(which(trim.3715.127.sq$seqID !="unclassified"))/nrow(trim.3715.127.sq)
)
trim.3715.prop$Unclassified <- 1 - trim.3715.prop$Bact.Arch - trim.3715.prop$Squirrel

# Number of classified bact/arch
trim.3715.55.report <- read_tsv('3715.trimmed/old/3715.trimmed.55.report.tsv')
trim.3715.75.report <- read_tsv('3715.trimmed/old/3715.trimmed.75.report.tsv')
trim.3715.95.report <- read_tsv('3715.trimmed/old/3715.trimmed.95.report.tsv')
trim.3715.111.report <- read_tsv('3715.trimmed/old/3715.trimmed.111.report.tsv')
trim.3715.127.report <- read_tsv('3715.trimmed/old/3715.trimmed.127.report.tsv')

trim.3715.55.abund <- trim.3715.55.report[which(trim.3715.55.report$abundance>0),]
trim.3715.75.abund <- trim.3715.55.report[which(trim.3715.75.report$abundance>0),]
trim.3715.95.abund <- trim.3715.55.report[which(trim.3715.95.report$abundance>0),]
trim.3715.111.abund <- trim.3715.55.report[which(trim.3715.111.report$abundance>0),]
trim.3715.127.abund <- trim.3715.55.report[which(trim.3715.127.report$abundance>0),]

trim.3715.prop$Num.Species <- c(nrow(trim.3715.55.abund),nrow(trim.3715.75.abund),
                                nrow(trim.3715.95.abund),nrow(trim.3715.111.abund),
                                nrow(trim.3715.127.abund))

# Quast N50
trim.3715.prop$N50 <- c(2532,2979,3299,3541,4812)
trim.3715.prop$Contigs <- c(110852,108041,105253,102685,91594)      
trim.3715.prop$Largest.Contig <- c(195880,207778,399402,518333,1334344)
```


### 3715 Untrimmed SPAdes
```{r}
# Load in .tsv files
untrim.3715.55 <- read_tsv('3715.untrim/3715.spades.untrim.55.output.tsv')
untrim.3715.75 <- read_tsv('3715.untrim/3715.spades.untrim.75.output.tsv')
untrim.3715.95 <- read_tsv('3715.untrim/3715.spades.untrim.95.output.tsv')
untrim.3715.111 <- read_tsv('3715.untrim/3715.spades.untrim.111.output.tsv')
untrim.3715.127 <- read_tsv('3715.untrim/3715.spades.untrim.127.output.tsv')

# Calculate unclassified vs. classified
untrim.3715.prop <- data.frame(matrix(5,5))
untrim.3715.prop$Kmer <- c(55,75,95,111,127)
untrim.3715.prop$Unclassified <- c(
  length(which(untrim.3715.55$taxID=="0"))/nrow(untrim.3715.55),
  length(which(untrim.3715.75$taxID=="0"))/nrow(untrim.3715.75),
  length(which(untrim.3715.95$taxID=="0"))/nrow(untrim.3715.95),
  length(which(untrim.3715.111$taxID=="0"))/nrow(untrim.3715.111),
  length(which(untrim.3715.127$taxID=="0"))/nrow(untrim.3715.127)
)
untrim.3715.prop <- untrim.3715.prop[,-1]
untrim.3715.prop$Microbial <- c(
  length(which(untrim.3715.55$taxID != "0" &
                 untrim.3715.55$taxID != "43179"))/nrow(untrim.3715.55),
  length(which(untrim.3715.75$taxID != "0" &
                 untrim.3715.75$taxID != "43179"))/nrow(untrim.3715.75),
  length(which(untrim.3715.95$taxID != "0" &
                 untrim.3715.95$taxID != "43179"))/nrow(untrim.3715.95),
  length(which(untrim.3715.111$taxID != "0" &
                 untrim.3715.111$taxID != "43179"))/nrow(untrim.3715.111),
  length(which(untrim.3715.127$taxID != "0" &
                 untrim.3715.127$taxID != "43179"))/nrow(untrim.3715.127)
)
untrim.3715.prop$Squirrel <- c(
  length(which(untrim.3715.55$taxID == "43179"))/nrow(untrim.3715.55),
  length(which(untrim.3715.75$taxID == "43179"))/nrow(untrim.3715.75),
  length(which(untrim.3715.95$taxID == "43179"))/nrow(untrim.3715.95),
  length(which(untrim.3715.111$taxID == "43179"))/nrow(untrim.3715.111),
  length(which(untrim.3715.127$taxID == "43179"))/nrow(untrim.3715.127)
)

# Number of classified Microbes
untrim.3715.55.report <- read_tsv('3715.untrim/3715.spades.untrim.55.report.tsv')
untrim.3715.75.report <- read_tsv('3715.untrim/3715.spades.untrim.75.report.tsv')
untrim.3715.95.report <- read_tsv('3715.untrim/3715.spades.untrim.95.report.tsv')
untrim.3715.111.report <- read_tsv('3715.untrim/3715.spades.untrim.111.report.tsv')
untrim.3715.127.report <- read_tsv('3715.untrim/3715.spades.untrim.127.report.tsv')

untrim.3715.55.abund <- untrim.3715.55.report[which(untrim.3715.55.report$abundance>0),]
untrim.3715.75.abund <- untrim.3715.55.report[which(untrim.3715.75.report$abundance>0),]
untrim.3715.95.abund <- untrim.3715.55.report[which(untrim.3715.95.report$abundance>0),]
untrim.3715.111.abund <- untrim.3715.55.report[which(untrim.3715.111.report$abundance>0),]
untrim.3715.127.abund <- untrim.3715.55.report[which(untrim.3715.127.report$abundance>0),]

untrim.3715.prop$Num.Species <- c(nrow(untrim.3715.55.abund), nrow(untrim.3715.75.abund),
                                nrow(untrim.3715.95.abund), nrow(untrim.3715.111.abund),
                                nrow(untrim.3715.127.abund))

# Quast Stats
untrim.3715.prop$N50 <- c(2505,2959,3296,3545,4801)
untrim.3715.prop$Contigs <- c(111864,108790,105754,103161,91748)      
untrim.3715.prop$Largest.Contig <- c(115865,207778,399291,518071,1334334)
```

### 3715 Trimmed SPAdes
```{r}
# Load in .tsv files
trim.3715.55 <- read_tsv('3715.trim/3715.spades.trim.55.output.tsv')
trim.3715.75 <- read_tsv('3715.trim/3715.spades.trim.75.output.tsv')
trim.3715.95 <- read_tsv('3715.trim/3715.spades.trim.95.output.tsv')
trim.3715.111 <- read_tsv('3715.trim/3715.spades.trim.111.output.tsv')
trim.3715.127 <- read_tsv('3715.trim/3715.spades.trim.127.output.tsv')

# Calculate unclassified vs. classified
trim.3715.prop <- data.frame(matrix(5,5))
trim.3715.prop$Kmer <- c(55,75,95,111,127)
trim.3715.prop$Unclassified <- c(
  length(which(trim.3715.55$taxID=="0"))/nrow(trim.3715.55),
  length(which(trim.3715.75$taxID=="0"))/nrow(trim.3715.75),
  length(which(trim.3715.95$taxID=="0"))/nrow(trim.3715.95),
  length(which(trim.3715.111$taxID=="0"))/nrow(trim.3715.111),
  length(which(trim.3715.127$taxID=="0"))/nrow(trim.3715.127)
)
trim.3715.prop <- trim.3715.prop[,-1]
trim.3715.prop$Microbial <- c(
  length(which(trim.3715.55$taxID != "0" &
                 trim.3715.55$taxID != "43179"))/nrow(trim.3715.55),
  length(which(trim.3715.75$taxID != "0" &
                 trim.3715.75$taxID != "43179"))/nrow(trim.3715.75),
  length(which(trim.3715.95$taxID != "0" &
                 trim.3715.95$taxID != "43179"))/nrow(trim.3715.95),
  length(which(trim.3715.111$taxID != "0" &
                 trim.3715.111$taxID != "43179"))/nrow(trim.3715.111),
  length(which(trim.3715.127$taxID != "0" &
                 trim.3715.127$taxID != "43179"))/nrow(trim.3715.127)
)
trim.3715.prop$Squirrel <- c(
  length(which(trim.3715.55$taxID == "43179"))/nrow(trim.3715.55),
  length(which(trim.3715.75$taxID == "43179"))/nrow(trim.3715.75),
  length(which(trim.3715.95$taxID == "43179"))/nrow(trim.3715.95),
  length(which(trim.3715.111$taxID == "43179"))/nrow(trim.3715.111),
  length(which(trim.3715.127$taxID == "43179"))/nrow(trim.3715.127)
)

# Number of classified Microbes
trim.3715.55.report <- read_tsv('3715.trim/3715.spades.trim.55.report.tsv')
trim.3715.75.report <- read_tsv('3715.trim/3715.spades.trim.75.report.tsv')
trim.3715.95.report <- read_tsv('3715.trim/3715.spades.trim.95.report.tsv')
trim.3715.111.report <- read_tsv('3715.trim/3715.spades.trim.111.report.tsv')
trim.3715.127.report <- read_tsv('3715.trim/3715.spades.trim.127.report.tsv')

trim.3715.55.abund <- trim.3715.55.report[which(trim.3715.55.report$abundance>0),]
trim.3715.75.abund <- trim.3715.55.report[which(trim.3715.75.report$abundance>0),]
trim.3715.95.abund <- trim.3715.55.report[which(trim.3715.95.report$abundance>0),]
trim.3715.111.abund <- trim.3715.55.report[which(trim.3715.111.report$abundance>0),]
trim.3715.127.abund <- trim.3715.55.report[which(trim.3715.127.report$abundance>0),]

trim.3715.prop$Num.Species <- c(nrow(trim.3715.55.abund), nrow(trim.3715.75.abund),
                                nrow(trim.3715.95.abund), nrow(trim.3715.111.abund),
                                nrow(trim.3715.127.abund))

# Quast Stats
trim.3715.prop$N50 <- c(2532,2979,3299,3541,4812)
trim.3715.prop$Contigs <- c(110852,108041,105253,102685,91594)      
trim.3715.prop$Largest.Contig <- c(195880,207778,399402,518333,1334344)
```

### 3715 Normalized SPAdes
```{r}
# Load in .tsv files
norm.3715.55 <- read_tsv('3715.norm/3715.norm.spades.55.output.tsv')
norm.3715.75 <- read_tsv('3715.norm/3715.norm.spades.75.output.tsv')
norm.3715.95 <- read_tsv('3715.norm/3715.norm.spades.95.output.tsv')
norm.3715.111 <- read_tsv('3715.norm/3715.norm.spades.111.output.tsv')
norm.3715.127 <- read_tsv('3715.norm/3715.norm.spades.127.output.tsv')

# Calculate unclassified vs. classified
norm.3715.prop <- data.frame(matrix(5,5))
norm.3715.prop$Kmer <- c(55,75,95,111,127)
norm.3715.prop$Unclassified <- c(
  length(which(norm.3715.55$taxID == "0"))/nrow(norm.3715.55),
  length(which(norm.3715.75$taxID == "0"))/nrow(norm.3715.75),
  length(which(norm.3715.95$taxID == "0"))/nrow(norm.3715.95),
  length(which(norm.3715.111$taxID == "0"))/nrow(norm.3715.111),
  length(which(norm.3715.127$taxID == "0"))/nrow(norm.3715.127)
)
norm.3715.prop <- norm.3715.prop[,-1]
norm.3715.prop$Microbial <- c(
  length(which(norm.3715.55$taxID != "0" &
                 norm.3715.55$taxID != "43179"))/nrow(norm.3715.55),
  length(which(norm.3715.75$taxID != "0" &
                 norm.3715.75$taxID != "43179"))/nrow(norm.3715.75),
  length(which(norm.3715.95$taxID != "0" &
                 norm.3715.95$taxID != "43179"))/nrow(norm.3715.95),
  length(which(norm.3715.111$taxID != "0" &
                 norm.3715.111$taxID != "43179"))/nrow(norm.3715.111),
  length(which(norm.3715.127$taxID != "0" &
                 norm.3715.127$taxID != "43179"))/nrow(norm.3715.127)
)
norm.3715.prop$Squirrel <- c(
  length(which(norm.3715.55$taxID == "43179"))/nrow(norm.3715.55),
  length(which(norm.3715.75$taxID == "43179"))/nrow(norm.3715.75),
  length(which(norm.3715.95$taxID == "43179"))/nrow(norm.3715.95),
  length(which(norm.3715.111$taxID == "43179"))/nrow(norm.3715.111),
  length(which(norm.3715.127$taxID == "43179"))/nrow(norm.3715.127)
)

# Number of classified Microbes
norm.3715.55.report <- read_tsv('3715.norm/3715.norm.spades.55.report.tsv')
norm.3715.75.report <- read_tsv('3715.norm/3715.norm.spades.75.report.tsv')
norm.3715.95.report <- read_tsv('3715.norm/3715.norm.spades.95.report.tsv')
norm.3715.111.report <- read_tsv('3715.norm/3715.norm.spades.111.report.tsv')
norm.3715.127.report <- read_tsv('3715.norm/3715.norm.spades.127.report.tsv')

norm.3715.55.abund <- norm.3715.55.report[which(norm.3715.55.report$abundance>0),]
norm.3715.75.abund <- norm.3715.75.report[which(norm.3715.75.report$abundance>0),]
norm.3715.95.abund <- norm.3715.95.report[which(norm.3715.95.report$abundance>0),]
norm.3715.111.abund <- norm.3715.111.report[which(norm.3715.111.report$abundance>0),]
norm.3715.127.abund <- norm.3715.55.report[which(norm.3715.127.report$abundance>0),]

norm.3715.prop$Num.Species <- c(nrow(norm.3715.55.abund), nrow(norm.3715.75.abund),
                                nrow(norm.3715.95.abund), nrow(norm.3715.111.abund),
                                nrow(norm.3715.127.abund))

# Quast Stats
norm.3715.prop$N50 <- c(2531, 2933, 3224, 3479, 4661)
norm.3715.prop$Contigs <- c(110791, 108451, 105581, 102245, 91899)
norm.3715.prop$Largest.Contig <- c(195880, 207776, 399550, 540728, 1178389)


```

### 3715 Sickle Normalized SPades
```{r}
# Load in .tsv files
sickle.spa.3715.55 <- read_tsv('3715.sickle/3715.sickle.spades.55.output.tsv')
sickle.spa.3715.75 <- read_tsv('3715.sickle/3715.sickle.spades.75.output.tsv')
sickle.spa.3715.95 <- read_tsv('3715.sickle/3715.sickle.spades.95.output.tsv')
sickle.spa.3715.111 <- read_tsv('3715.sickle/3715.sickle.spades.111.output.tsv')
sickle.spa.3715.127 <- read_tsv('3715.sickle/3715.sickle.spades.127.output.tsv')

# Calculate unclassified vs. classified
sickle.spa.3715.prop <- data.frame(matrix(5,5))
sickle.spa.3715.prop$Kmer <- c(55,75,95,111,127)
sickle.spa.3715.prop$Unclassified <- c(
  length(which(sickle.spa.3715.55$taxID == "0"))/nrow(sickle.spa.3715.55),
  length(which(sickle.spa.3715.75$taxID == "0"))/nrow(sickle.spa.3715.75),
  length(which(sickle.spa.3715.95$taxID == "0"))/nrow(sickle.spa.3715.95),
  length(which(sickle.spa.3715.111$taxID == "0"))/nrow(sickle.spa.3715.111),
  length(which(sickle.spa.3715.127$taxID == "0"))/nrow(sickle.spa.3715.127)
)
sickle.spa.3715.prop <- sickle.spa.3715.prop[,-1]
sickle.spa.3715.prop$Microbial <- c(
  length(which(sickle.spa.3715.55$taxID != "0" &
                 sickle.spa.3715.55$taxID != "43179"))/nrow(sickle.spa.3715.55),
  length(which(sickle.spa.3715.75$taxID != "0" &
                 sickle.spa.3715.75$taxID != "43179"))/nrow(sickle.spa.3715.75),
  length(which(sickle.spa.3715.95$taxID != "0" &
                 sickle.spa.3715.95$taxID != "43179"))/nrow(sickle.spa.3715.95),
  length(which(sickle.spa.3715.111$taxID != "0" &
                 sickle.spa.3715.111$taxID != "43179"))/nrow(sickle.spa.3715.111),
  length(which(sickle.spa.3715.127$taxID != "0" &
                 sickle.spa.3715.127$taxID != "43179"))/nrow(sickle.spa.3715.127)
)
sickle.spa.3715.prop$Squirrel <- c(
  length(which(sickle.spa.3715.55$taxID == "43179"))/nrow(sickle.spa.3715.55),
  length(which(sickle.spa.3715.75$taxID == "43179"))/nrow(sickle.spa.3715.75),
  length(which(sickle.spa.3715.95$taxID == "43179"))/nrow(sickle.spa.3715.95),
  length(which(sickle.spa.3715.111$taxID == "43179"))/nrow(sickle.spa.3715.111),
  length(which(sickle.spa.3715.127$taxID == "43179"))/nrow(sickle.spa.3715.127)
)

# Number of classified Microbes
sickle.spa.3715.55.report <- read_tsv('3715.sickle/3715.sickle.spades.55.report.tsv')
sickle.spa.3715.75.report <- read_tsv('3715.sickle/3715.sickle.spades.75.report.tsv')
sickle.spa.3715.95.report <- read_tsv('3715.sickle/3715.sickle.spades.95.report.tsv')
sickle.spa.3715.111.report <- read_tsv('3715.sickle/3715.sickle.spades.111.report.tsv')
sickle.spa.3715.127.report <- read_tsv('3715.sickle/3715.sickle.spades.127.report.tsv')

sickle.spa.3715.55.abund <-
  sickle.spa.3715.55.report[which(sickle.spa.3715.55.report$abundance>0),]
sickle.spa.3715.75.abund <-
  sickle.spa.3715.75.report[which(sickle.spa.3715.75.report$abundance>0),]
sickle.spa.3715.95.abund <-
  sickle.spa.3715.95.report[which(sickle.spa.3715.95.report$abundance>0),]
sickle.spa.3715.111.abund <-
  sickle.spa.3715.111.report[which(sickle.spa.3715.111.report$abundance>0),]
sickle.spa.3715.127.abund <-
  sickle.spa.3715.127.report[which(sickle.spa.3715.127.report$abundance>0),]

sickle.spa.3715.prop$Num.Species <- c(nrow(sickle.spa.3715.55.abund), 
                                      nrow(sickle.spa.3715.75.abund),
                                      nrow(sickle.spa.3715.95.abund),
                                      nrow(sickle.spa.3715.111.abund),
                                      nrow(sickle.spa.3715.127.abund))

# Quast Stats
sickle.spa.3715.prop$N50 <- c(2534, 2942, 3233, 3494, 4626)
sickle.spa.3715.prop$Contigs <- c(110398, 108014, 105149, 101852, 91882)
sickle.spa.3715.prop$Largest.Contig <- c(195880, 207776, 399550, 540728, 1178389)

```
### 3715 Sickle Raw SPAdes
```{r}
# Load in .tsv files
sickle.raw.spa.3715.55 <- read_tsv('3715.sickle/3715.sickle.raw.spades.55.output.tsv')
sickle.raw.spa.3715.75 <- read_tsv('3715.sickle/3715.sickle.raw.spades.75.output.tsv')
sickle.raw.spa.3715.95 <- read_tsv('3715.sickle/3715.sickle.raw.spades.95.output.tsv')
sickle.raw.spa.3715.111 <- read_tsv('3715.sickle/3715.sickle.raw.spades.111.output.tsv')
sickle.raw.spa.3715.127 <- read_tsv('3715.sickle/3715.sickle.raw.spades.127.output.tsv')

# Calculate unclassified vs. classified
sickle.raw.spa.3715.prop <- data.frame(matrix(5,5))
sickle.raw.spa.3715.prop$Kmer <- c(55,75,95,111,127)
sickle.raw.spa.3715.prop$Unclassified <- c(
  length(which(sickle.raw.spa.3715.55$taxID == "0"))/nrow(sickle.raw.spa.3715.55),
  length(which(sickle.raw.spa.3715.75$taxID == "0"))/nrow(sickle.raw.spa.3715.75),
  length(which(sickle.raw.spa.3715.95$taxID == "0"))/nrow(sickle.raw.spa.3715.95),
  length(which(sickle.raw.spa.3715.111$taxID == "0"))/nrow(sickle.raw.spa.3715.111),
  length(which(sickle.raw.spa.3715.127$taxID == "0"))/nrow(sickle.raw.spa.3715.127)
)
sickle.raw.spa.3715.prop <- sickle.raw.spa.3715.prop[,-1]
sickle.raw.spa.3715.prop$Microbial <- c(
  length(which(sickle.raw.spa.3715.55$taxID != "0" &
                 sickle.raw.spa.3715.55$taxID != "43179"))/nrow(sickle.raw.spa.3715.55),
  length(which(sickle.raw.spa.3715.75$taxID != "0" &
                 sickle.raw.spa.3715.75$taxID != "43179"))/nrow(sickle.raw.spa.3715.75),
  length(which(sickle.raw.spa.3715.95$taxID != "0" &
                 sickle.raw.spa.3715.95$taxID != "43179"))/nrow(sickle.raw.spa.3715.95),
  length(which(sickle.raw.spa.3715.111$taxID != "0" &
                 sickle.raw.spa.3715.111$taxID != "43179"))/nrow(sickle.raw.spa.3715.111),
  length(which(sickle.raw.spa.3715.127$taxID != "0" &
                 sickle.raw.spa.3715.127$taxID != "43179"))/nrow(sickle.raw.spa.3715.127)
)
sickle.raw.spa.3715.prop$Squirrel <- c(
  length(which(sickle.raw.spa.3715.55$taxID == "43179"))/nrow(sickle.raw.spa.3715.55),
  length(which(sickle.raw.spa.3715.75$taxID == "43179"))/nrow(sickle.raw.spa.3715.75),
  length(which(sickle.raw.spa.3715.95$taxID == "43179"))/nrow(sickle.raw.spa.3715.95),
  length(which(sickle.raw.spa.3715.111$taxID == "43179"))/nrow(sickle.raw.spa.3715.111),
  length(which(sickle.raw.spa.3715.127$taxID == "43179"))/nrow(sickle.raw.spa.3715.127)
)

# Number of classified Microbes
sickle.raw.spa.3715.55.report <- read_tsv('3715.sickle/3715.sickle.raw.spades.55.report.tsv')
sickle.raw.spa.3715.75.report <- read_tsv('3715.sickle/3715.sickle.raw.spades.75.report.tsv')
sickle.raw.spa.3715.95.report <- read_tsv('3715.sickle/3715.sickle.raw.spades.95.report.tsv')
sickle.raw.spa.3715.111.report <- read_tsv('3715.sickle/3715.sickle.raw.spades.111.report.tsv')
sickle.raw.spa.3715.127.report <- read_tsv('3715.sickle/3715.sickle.raw.spades.127.report.tsv')

sickle.raw.spa.3715.55.abund <-
  sickle.raw.spa.3715.55.report[which(sickle.raw.spa.3715.55.report$abundance>0),]
sickle.raw.spa.3715.75.abund <-
  sickle.raw.spa.3715.75.report[which(sickle.raw.spa.3715.75.report$abundance>0),]
sickle.raw.spa.3715.95.abund <-
  sickle.raw.spa.3715.95.report[which(sickle.raw.spa.3715.95.report$abundance>0),]
sickle.raw.spa.3715.111.abund <-
  sickle.raw.spa.3715.111.report[which(sickle.raw.spa.3715.111.report$abundance>0),]
sickle.raw.spa.3715.127.abund <-
  sickle.raw.spa.3715.127.report[which(sickle.raw.spa.3715.127.report$abundance>0),]

sickle.raw.spa.3715.prop$Num.Species <- c(nrow(sickle.raw.spa.3715.55.abund), 
                                      nrow(sickle.raw.spa.3715.75.abund),
                                      nrow(sickle.raw.spa.3715.95.abund),
                                      nrow(sickle.raw.spa.3715.111.abund),
                                      nrow(sickle.raw.spa.3715.127.abund))

# Quast Stats
sickle.raw.spa.3715.prop$N50 <- c(2519, 2973, 3308, 3556, 4785)
sickle.raw.spa.3715.prop$Contigs <- c(111036, 107972, 105008, 102411, 91644)
sickle.raw.spa.3715.prop$Largest.Contig <- c(172139, 207778, 399402, 51833, 1334344)
```
### 3715 Megahit
```{r}
# Load in .tsv files
untrim.3715.mega <- read_tsv('3715.megahit/3715.untrim.mega.output.tsv')
untrim.3715.mega.sq <- read_tsv('3715.megahit/3715.untrim.mega.sq.output.tsv')
trim.3715.mega <- read_tsv('3715.megahit/3715.trim.mega.output.tsv')
trim.3715.mega.sq <- read_tsv('3715.megahit/3715.trim.mega.sq.output.tsv')

# Calculate unclassified vs. classified
mega.3715.prop <- data.frame(matrix(2,2))
mega.3715.prop$Data <- c("Trimmed", "Untrimmed")
mega.3715.prop$Unclassified <- c(
  length(which(trim.3715.mega$seqID == "unclassified"))/nrow(trim.3715.mega),
  length(which(untrim.3715.mega$seqID =="unclassified"))/nrow(untrim.3715.mega))
mega.3715.prop$Bact.Arch <- c(
  length(which(trim.3715.mega$seqID !="unclassified"))/nrow(trim.3715.mega),
  length(which(untrim.3715.mega$seqID !="unclassified"))/nrow(untrim.3715.mega))
mega.3715.prop$Squirrel <- c(
  length(which(trim.3715.mega.sq$seqID !="unclassified"))/nrow(trim.3715.mega.sq),
  length(which(untrim.3715.mega.sq$seqID !="unclassified"))/nrow(untrim.3715.mega.sq))
mega.3715.prop$Unclassified <- 1 - mega.3715.prop$Bact.Arch - mega.3715.prop$Squirrel
mega.3715.prop <- mega.3715.prop[,-1]

# Number of classified bact/arch
mega.3715.trim.report <- read_tsv('3715.megahit/3715.trim.mega.report.tsv')
mega.3715.untrim.report <- read_tsv('3715.megahit/3715.untrim.mega.report.tsv')

mega.3715.trim.abund <- 
  mega.3715.trim.report[which(mega.3715.trim.report$abundance>0),]
mega.3715.untrim.abund <-
  mega.3715.trim.report[which(mega.3715.untrim.report$abundance>0),]

mega.3715.prop$Num.Species <- c(nrow(mega.3715.trim.abund), nrow(mega.3715.untrim.abund))

# Quast N50
mega.3715.prop$N50 <- c(3071, 3079)
mega.3715.prop$Contigs <- c(103691,104554)
mega.3715.prop$Largest.Contig <- c(498993,498993)
```

### 3715 IDBA Untrimmed
```{r}
# Load in .tsv files
idba.untrim.55 <- read_tsv('3715.untrim/3715.idba.untrim.55.output.tsv')
idba.untrim.75 <- read_tsv('3715.untrim/3715.idba.untrim.75.output.tsv')
idba.untrim.95 <- read_tsv('3715.untrim/3715.idba.untrim.95.output.tsv')
idba.untrim.111 <- read_tsv('3715.untrim/3715.idba.untrim.111.output.tsv')
idba.untrim.127 <- read_tsv('3715.untrim/3715.idba.untrim.127.output.tsv')

# Calculate unclassified vs. classified
idba.untrim.3715.prop <- data.frame(matrix(5,5))
idba.untrim.3715.prop$Kmer <- c(55,75,85,111,127)
idba.untrim.3715.prop$Unclassified <- c(
  length(which(idba.untrim.55$taxID == "0"))/nrow(idba.untrim.55),
  length(which(idba.untrim.75$taxID == "0"))/nrow(idba.untrim.75),
  length(which(idba.untrim.95$taxID == "0"))/nrow(idba.untrim.95),
  length(which(idba.untrim.111$taxID == "0"))/nrow(idba.untrim.111),
  length(which(idba.untrim.127$taxID == "0"))/nrow(idba.untrim.127)
)
idba.untrim.3715.prop <- idba.untrim.3715.prop[,-1]
idba.untrim.3715.prop$Microbial <- c(
  length(which(idba.untrim.55$taxID != "0" &
                 idba.untrim.55$taxID != "43179"))/nrow(idba.untrim.55),
  length(which(idba.untrim.75$taxID != "0" &
                 idba.untrim.75$taxID != "43179"))/nrow(idba.untrim.75),
  length(which(idba.untrim.95$taxID != "0" &
                 idba.untrim.95$taxID != "43179"))/nrow(idba.untrim.95),
  length(which(idba.untrim.111$taxID != "0" &
                 idba.untrim.111$taxID != "43179"))/nrow(idba.untrim.111),
  length(which(idba.untrim.127$taxID != "0" &
                 idba.untrim.127$taxID != "43179"))/nrow(idba.untrim.127)
)
idba.untrim.3715.prop$Squirrel <- c(
  length(which(idba.untrim.55$taxID == "43179"))/nrow(idba.untrim.55),
  length(which(idba.untrim.75$taxID == "43179"))/nrow(idba.untrim.75),
  length(which(idba.untrim.95$taxID == "43179"))/nrow(idba.untrim.95),
  length(which(idba.untrim.111$taxID == "43179"))/nrow(idba.untrim.111),
  length(which(idba.untrim.127$taxID == "43179"))/nrow(idba.untrim.127)
)

# Number of classified Microbes
idba.untrim.3715.55.report <- read_tsv('3715.untrim/3715.idba.untrim.55.report.tsv')
idba.untrim.3715.75.report <- read_tsv('3715.untrim/3715.idba.untrim.75.report.tsv')
idba.untrim.3715.95.report <- read_tsv('3715.untrim/3715.idba.untrim.95.report.tsv')
idba.untrim.3715.111.report <- read_tsv('3715.untrim/3715.idba.untrim.111.report.tsv')
idba.untrim.3715.127.report <- read_tsv('3715.untrim/3715.idba.untrim.127.report.tsv')

idba.untrim.3715.55.abund <- 
  idba.untrim.3715.55.report[which(idba.untrim.3715.55.report$abundance>0),]
idba.untrim.3715.75.abund <- 
  idba.untrim.3715.75.report[which(idba.untrim.3715.75.report$abundance>0),]
idba.untrim.3715.95.abund <- 
  idba.untrim.3715.95.report[which(idba.untrim.3715.95.report$abundance>0),]
idba.untrim.3715.111.abund <- 
  idba.untrim.3715.111.report[which(idba.untrim.3715.111.report$abundance>0),]
idba.untrim.3715.127.abund <- 
  idba.untrim.3715.127.report[which(idba.untrim.3715.127.report$abundance>0),]

idba.untrim.3715.prop$Num.Species <- c(nrow(idba.untrim.3715.55.abund),
                                       nrow(idba.untrim.3715.75.abund),
                                       nrow(idba.untrim.3715.95.abund),
                                       nrow(idba.untrim.3715.111.abund),
                                       nrow(idba.untrim.3715.127.abund))

# Quast
idba.untrim.3715.prop$N50 <- c(1942, 3135, 3726, 6523, 4324)
idba.untrim.3715.prop$Contigs <- c(102225, 95431, 91776, 28390, 15392)
idba.untrim.3715.prop$Largest.Contig <- c(195878, 220158, 340857, 322916, 450474)

```

### 3715 IDBA Trimmed
```{r}
# Load in .tsv files
idba.trim.55 <- read_tsv('3715.trim/3715.idba.trim.55.output.tsv')
idba.trim.75 <- read_tsv('3715.trim/3715.idba.trim.75.output.tsv')
idba.trim.95 <- read_tsv('3715.trim/3715.idba.trim.95.output.tsv')
idba.trim.111 <- read_tsv('3715.trim/3715.idba.trim.111.output.tsv')
idba.trim.127 <- read_tsv('3715.trim/3715.idba.trim.127.output.tsv')

# Calculate unclassified vs. classified
idba.trim.3715.prop <- data.frame(matrix(5,5))
idba.trim.3715.prop$Kmer <- c(55,75,85,111,127)
idba.trim.3715.prop$Unclassified <- c(
  length(which(idba.trim.55$taxID == "0"))/nrow(idba.trim.55),
  length(which(idba.trim.75$taxID == "0"))/nrow(idba.trim.75),
  length(which(idba.trim.95$taxID == "0"))/nrow(idba.trim.95),
  length(which(idba.trim.111$taxID == "0"))/nrow(idba.trim.111),
  length(which(idba.trim.127$taxID == "0"))/nrow(idba.trim.127)
)
idba.trim.3715.prop <- idba.trim.3715.prop[,-1]
idba.trim.3715.prop$Microbial <- c(
  length(which(idba.trim.55$taxID != "0" &
                 idba.trim.55$taxID != "43179"))/nrow(idba.trim.55),
  length(which(idba.trim.75$taxID != "0" &
                 idba.trim.75$taxID != "43179"))/nrow(idba.trim.75),
  length(which(idba.trim.95$taxID != "0" &
                 idba.trim.95$taxID != "43179"))/nrow(idba.trim.95),
  length(which(idba.trim.111$taxID != "0" &
                 idba.trim.111$taxID != "43179"))/nrow(idba.trim.111),
  length(which(idba.trim.127$taxID != "0" &
                 idba.trim.127$taxID != "43179"))/nrow(idba.trim.127)
)
idba.trim.3715.prop$Squirrel <- c(
  length(which(idba.trim.55$taxID == "43179"))/nrow(idba.trim.55),
  length(which(idba.trim.75$taxID == "43179"))/nrow(idba.trim.75),
  length(which(idba.trim.95$taxID == "43179"))/nrow(idba.trim.95),
  length(which(idba.trim.111$taxID == "43179"))/nrow(idba.trim.111),
  length(which(idba.trim.127$taxID == "43179"))/nrow(idba.trim.127)
)

# Number of classified Microbes
idba.trim.3715.55.report <- read_tsv('3715.trim/3715.idba.trim.55.report.tsv')
idba.trim.3715.75.report <- read_tsv('3715.trim/3715.idba.trim.75.report.tsv')
idba.trim.3715.95.report <- read_tsv('3715.trim/3715.idba.trim.95.report.tsv')
idba.trim.3715.111.report <- read_tsv('3715.trim/3715.idba.trim.111.report.tsv')
idba.trim.3715.127.report <- read_tsv('3715.trim/3715.idba.trim.127.report.tsv')

idba.trim.3715.55.abund <- 
  idba.trim.3715.55.report[which(idba.trim.3715.55.report$abundance>0),]
idba.trim.3715.75.abund <- 
  idba.trim.3715.75.report[which(idba.trim.3715.75.report$abundance>0),]
idba.trim.3715.95.abund <- 
  idba.trim.3715.95.report[which(idba.trim.3715.95.report$abundance>0),]
idba.trim.3715.111.abund <- 
  idba.trim.3715.111.report[which(idba.trim.3715.111.report$abundance>0),]
idba.trim.3715.127.abund <- 
  idba.trim.3715.127.report[which(idba.trim.3715.127.report$abundance>0),]

idba.trim.3715.prop$Num.Species <- c(nrow(idba.trim.3715.55.abund),
                                       nrow(idba.trim.3715.75.abund),
                                       nrow(idba.trim.3715.95.abund),
                                       nrow(idba.trim.3715.111.abund),
                                       nrow(idba.trim.3715.127.abund))

# Quast
idba.trim.3715.prop$N50 <- c(1942, 3149, 3652, 6823, 4103)
idba.trim.3715.prop$Contigs <- c(101789, 95023, 91384, 27939, 15565)
idba.trim.3715.prop$Largest.Contig <- c(19878, 220158, 340857, 322916, 488581)
```


### 3715 IDBA Normalized
```{r}
# Load in .tsv files
norm.idba.3715.55 <- read_tsv('3715.norm/3715.norm.idba.55.output.tsv')
norm.idba.3715.75 <- read_tsv('3715.norm/3715.norm.idba.75.output.tsv')
norm.idba.3715.95 <- read_tsv('3715.norm/3715.norm.idba.95.output.tsv')
norm.idba.3715.111 <- read_tsv('3715.norm/3715.norm.idba.111.output.tsv')
norm.idba.3715.127 <- read_tsv('3715.norm/3715.norm.idba.127.output.tsv')

# Calculate unclassified vs. classified
norm.idba.3715.prop <- data.frame(matrix(5,5))
norm.idba.3715.prop$Kmer <- c(55,75,95,111,127)
norm.idba.3715.prop$Unclassified <- c(
  length(which(norm.idba.3715.55$taxID == "0"))/nrow(norm.idba.3715.55),
  length(which(norm.idba.3715.75$taxID == "0"))/nrow(norm.idba.3715.75),
  length(which(norm.idba.3715.95$taxID == "0"))/nrow(norm.idba.3715.95),
  length(which(norm.idba.3715.111$taxID == "0"))/nrow(norm.idba.3715.111),
  length(which(norm.idba.3715.127$taxID == "0"))/nrow(norm.idba.3715.127)
)
norm.idba.3715.prop <- norm.idba.3715.prop[,-1]
norm.idba.3715.prop$Microbial <- c(
  length(which(norm.idba.3715.55$taxID != "0" &
                 norm.idba.3715.55$taxID != "43179"))/nrow(norm.idba.3715.55),
  length(which(norm.idba.3715.75$taxID != "0" &
                 norm.idba.3715.75$taxID != "43179"))/nrow(norm.idba.3715.75),
  length(which(norm.idba.3715.95$taxID != "0" &
                 norm.idba.3715.95$taxID != "43179"))/nrow(norm.idba.3715.95),
  length(which(norm.idba.3715.111$taxID != "0" &
                 norm.idba.3715.111$taxID != "43179"))/nrow(norm.idba.3715.111),
  length(which(norm.idba.3715.127$taxID != "0" &
                 norm.idba.3715.127$taxID != "43179"))/nrow(norm.idba.3715.127)
)
norm.idba.3715.prop$Unclassified <- c(
  length(which(norm.idba.3715.55$taxID == "0"))/nrow(norm.idba.3715.55),
  length(which(norm.idba.3715.75$taxID == "0"))/nrow(norm.idba.3715.75),
  length(which(norm.idba.3715.95$taxID == "0"))/nrow(norm.idba.3715.95),
  length(which(norm.idba.3715.111$taxID == "0"))/nrow(norm.idba.3715.111),
  length(which(norm.idba.3715.127$taxID == "0"))/nrow(norm.idba.3715.127)
)
norm.idba.3715.prop$Squirrel <- c(
  length(which(norm.idba.3715.55$taxID == "43179"))/nrow(norm.idba.3715.55),
  length(which(norm.idba.3715.75$taxID == "43179"))/nrow(norm.idba.3715.75),
  length(which(norm.idba.3715.95$taxID == "43179"))/nrow(norm.idba.3715.95),
  length(which(norm.idba.3715.111$taxID == "43179"))/nrow(norm.idba.3715.111),
  length(which(norm.idba.3715.127$taxID == "43179"))/nrow(norm.idba.3715.127)
)

# Number of classified Microbes
norm.idba.3715.55.report <- read_tsv('3715.norm/3715.norm.idba.55.report.tsv')
norm.idba.3715.75.report <- read_tsv('3715.norm/3715.norm.idba.75.report.tsv')
norm.idba.3715.95.report <- read_tsv('3715.norm/3715.norm.idba.95.report.tsv')
norm.idba.3715.111.report <- read_tsv('3715.norm/3715.norm.idba.111.report.tsv')
norm.idba.3715.127.report <- read_tsv('3715.norm/3715.norm.idba.127.report.tsv')

norm.idba.3715.55.abund <-
  norm.idba.3715.55.report[which(norm.idba.3715.55.report$abundance>0),]
norm.idba.3715.75.abund <-
  norm.idba.3715.75.report[which(norm.idba.3715.75.report$abundance>0),]
norm.idba.3715.95.abund <-
  norm.idba.3715.95.report[which(norm.idba.3715.95.report$abundance>0),]
norm.idba.3715.111.abund <-
  norm.idba.3715.111.report[which(norm.idba.3715.111.report$abundance>0),]
norm.idba.3715.127.abund <-
  norm.idba.3715.55.report[which(norm.idba.3715.127.report$abundance>0),]

norm.idba.3715.prop$Num.Species <- c(nrow(norm.idba.3715.55.abund),
                                     nrow(norm.idba.3715.75.abund),
                                     nrow(norm.idba.3715.95.abund),
                                     nrow(norm.idba.3715.111.abund),
                                     nrow(norm.idba.3715.127.abund))

# Quast Stats
norm.idba.3715.prop$N50 <- c(1942, 3149, 3652, 6823, 4103)
norm.idba.3715.prop$Contigs <- c(101789, 95023, 91384, 27939, 15565)
norm.idba.3715.prop$Largest.Contig <- c(195878, 220158, 340857, 322916, 448581)

```

### 3715 IDBA Sickle Normalized
```{r}
# Load in .tsv files
sickle.idba.3715.55 <- read_tsv('3715.sickle/3715.sickle.idba.55.output.tsv')
sickle.idba.3715.75 <- read_tsv('3715.sickle/3715.sickle.idba.75.output.tsv')
sickle.idba.3715.95 <- read_tsv('3715.sickle/3715.sickle.idba.95.output.tsv')
sickle.idba.3715.111 <- read_tsv('3715.sickle/3715.sickle.idba.111.output.tsv')
sickle.idba.3715.127 <- read_tsv('3715.sickle/3715.sickle.idba.127.output.tsv')

# Calculate unclassified vs. classified
sickle.idba.3715.prop <- data.frame(matrix(5,5))
sickle.idba.3715.prop$Kmer <- c(55,75,95,111,127)
sickle.idba.3715.prop$Unclassified <- c(
  length(which(sickle.idba.3715.55$taxID == "0"))/nrow(sickle.idba.3715.55),
  length(which(sickle.idba.3715.75$taxID == "0"))/nrow(sickle.idba.3715.75),
  length(which(sickle.idba.3715.95$taxID == "0"))/nrow(sickle.idba.3715.95),
  length(which(sickle.idba.3715.111$taxID == "0"))/nrow(sickle.idba.3715.111),
  length(which(sickle.idba.3715.127$taxID == "0"))/nrow(sickle.idba.3715.127)
)
sickle.idba.3715.prop <- sickle.idba.3715.prop[,-1]
sickle.idba.3715.prop$Microbial <- c(
  length(which(sickle.idba.3715.55$taxID != "0" &
                 sickle.idba.3715.55$taxID != "43179"))/nrow(sickle.idba.3715.55),
  length(which(sickle.idba.3715.75$taxID != "0" &
                 sickle.idba.3715.75$taxID != "43179"))/nrow(sickle.idba.3715.75),
  length(which(sickle.idba.3715.95$taxID != "0" &
                 sickle.idba.3715.95$taxID != "43179"))/nrow(sickle.idba.3715.95),
  length(which(sickle.idba.3715.111$taxID != "0" &
                 sickle.idba.3715.111$taxID != "43179"))/nrow(sickle.idba.3715.111),
  length(which(sickle.idba.3715.127$taxID != "0" &
                 sickle.idba.3715.127$taxID != "43179"))/nrow(sickle.idba.3715.127)
)
sickle.idba.3715.prop$Squirrel <- c(
  length(which(sickle.idba.3715.55$taxID == "43179"))/nrow(sickle.idba.3715.55),
  length(which(sickle.idba.3715.75$taxID == "43179"))/nrow(sickle.idba.3715.75),
  length(which(sickle.idba.3715.95$taxID == "43179"))/nrow(sickle.idba.3715.95),
  length(which(sickle.idba.3715.111$taxID == "43179"))/nrow(sickle.idba.3715.111),
  length(which(sickle.idba.3715.127$taxID == "43179"))/nrow(sickle.idba.3715.127)
)

# Number of classified Microbes
sickle.idba.3715.55.report <- read_tsv('3715.sickle/3715.sickle.idba.55.report.tsv')
sickle.idba.3715.75.report <- read_tsv('3715.sickle/3715.sickle.idba.75.report.tsv')
sickle.idba.3715.95.report <- read_tsv('3715.sickle/3715.sickle.idba.95.report.tsv')
sickle.idba.3715.111.report <- read_tsv('3715.sickle/3715.sickle.idba.111.report.tsv')
sickle.idba.3715.127.report <- read_tsv('3715.sickle/3715.sickle.idba.127.report.tsv')

sickle.idba.3715.55.abund <-
  sickle.idba.3715.55.report[which(sickle.idba.3715.55.report$abundance>0),]
sickle.idba.3715.75.abund <-
  sickle.idba.3715.75.report[which(sickle.idba.3715.75.report$abundance>0),]
sickle.idba.3715.95.abund <-
  sickle.idba.3715.95.report[which(sickle.idba.3715.95.report$abundance>0),]
sickle.idba.3715.111.abund <-
  sickle.idba.3715.111.report[which(sickle.idba.3715.111.report$abundance>0),]
sickle.idba.3715.127.abund <-
  sickle.idba.3715.127.report[which(sickle.idba.3715.127.report$abundance>0),]

sickle.idba.3715.prop$Num.Species <- c(nrow(sickle.idba.3715.55.abund), 
                                      nrow(sickle.idba.3715.75.abund),
                                      nrow(sickle.idba.3715.95.abund),
                                      nrow(sickle.idba.3715.111.abund),
                                      nrow(sickle.idba.3715.127.abund))

# Quast Stats
sickle.idba.3715.prop$N50 <- c(1945, 3107, 3614, 3568, 758)
sickle.idba.3715.prop$Contigs <- c(100369, 94302, 90960, 30033, 27657)
sickle.idba.3715.prop$Largest.Contig <- c(195878, 220158, 340857, 82715, 3664)

```


### 3715 IDBA Sickle Raw
```{r}
# Load in .tsv files
sickle.raw.idba.3715.55 <- read_tsv('3715.sickle/3715.sickle.raw.idba.55.output.tsv')
sickle.raw.idba.3715.75 <- read_tsv('3715.sickle/3715.sickle.raw.idba.75.output.tsv')
sickle.raw.idba.3715.95 <- read_tsv('3715.sickle/3715.sickle.raw.idba.95.output.tsv')
sickle.raw.idba.3715.111 <- read_tsv('3715.sickle/3715.sickle.raw.idba.111.output.tsv')
sickle.raw.idba.3715.127 <- read_tsv('3715.sickle/3715.sickle.raw.idba.127.output.tsv')

# Calculate unclassified vs. classified
sickle.raw.idba.3715.prop <- data.frame(matrix(5,5))
sickle.raw.idba.3715.prop$Kmer <- c(55,75,95,111,127)
sickle.raw.idba.3715.prop$Unclassified <- c(
  length(which(sickle.raw.idba.3715.55$taxID == "0"))/nrow(sickle.raw.idba.3715.55),
  length(which(sickle.raw.idba.3715.75$taxID == "0"))/nrow(sickle.raw.idba.3715.75),
  length(which(sickle.raw.idba.3715.95$taxID == "0"))/nrow(sickle.raw.idba.3715.95),
  length(which(sickle.raw.idba.3715.111$taxID == "0"))/nrow(sickle.raw.idba.3715.111),
  length(which(sickle.raw.idba.3715.127$taxID == "0"))/nrow(sickle.raw.idba.3715.127)
)
sickle.raw.idba.3715.prop <- sickle.raw.idba.3715.prop[,-1]
sickle.raw.idba.3715.prop$Microbial <- c(
  length(which(sickle.raw.idba.3715.55$taxID != "0" &
                 sickle.raw.idba.3715.55$taxID != "43179"))/nrow(sickle.raw.idba.3715.55),
  length(which(sickle.raw.idba.3715.75$taxID != "0" &
                 sickle.raw.idba.3715.75$taxID != "43179"))/nrow(sickle.raw.idba.3715.75),
  length(which(sickle.raw.idba.3715.95$taxID != "0" &
                 sickle.raw.idba.3715.95$taxID != "43179"))/nrow(sickle.raw.idba.3715.95),
  length(which(sickle.raw.idba.3715.111$taxID != "0" &
                 sickle.raw.idba.3715.111$taxID != "43179"))/nrow(sickle.raw.idba.3715.111),
  length(which(sickle.raw.idba.3715.127$taxID != "0" &
                 sickle.raw.idba.3715.127$taxID != "43179"))/nrow(sickle.raw.idba.3715.127)
)
sickle.raw.idba.3715.prop$Squirrel <- c(
  length(which(sickle.raw.idba.3715.55$taxID == "43179"))/nrow(sickle.raw.idba.3715.55),
  length(which(sickle.raw.idba.3715.75$taxID == "43179"))/nrow(sickle.raw.idba.3715.75),
  length(which(sickle.raw.idba.3715.95$taxID == "43179"))/nrow(sickle.raw.idba.3715.95),
  length(which(sickle.raw.idba.3715.111$taxID == "43179"))/nrow(sickle.raw.idba.3715.111),
  length(which(sickle.raw.idba.3715.127$taxID == "43179"))/nrow(sickle.raw.idba.3715.127)
)

# Number of classified Microbes
sickle.raw.idba.3715.55.report <- read_tsv('3715.sickle/3715.sickle.raw.idba.55.report.tsv')
sickle.raw.idba.3715.75.report <- read_tsv('3715.sickle/3715.sickle.raw.idba.75.report.tsv')
sickle.raw.idba.3715.95.report <- read_tsv('3715.sickle/3715.sickle.raw.idba.95.report.tsv')
sickle.raw.idba.3715.111.report <- read_tsv('3715.sickle/3715.sickle.raw.idba.111.report.tsv')
sickle.raw.idba.3715.127.report <- read_tsv('3715.sickle/3715.sickle.raw.idba.127.report.tsv')

sickle.raw.idba.3715.55.abund <-
  sickle.raw.idba.3715.55.report[which(sickle.raw.idba.3715.55.report$abundance>0),]
sickle.raw.idba.3715.75.abund <-
  sickle.raw.idba.3715.75.report[which(sickle.raw.idba.3715.75.report$abundance>0),]
sickle.raw.idba.3715.95.abund <-
  sickle.raw.idba.3715.95.report[which(sickle.raw.idba.3715.95.report$abundance>0),]
sickle.raw.idba.3715.111.abund <-
  sickle.raw.idba.3715.111.report[which(sickle.raw.idba.3715.111.report$abundance>0),]
sickle.raw.idba.3715.127.abund <-
  sickle.raw.idba.3715.127.report[which(sickle.raw.idba.3715.127.report$abundance>0),]

sickle.raw.idba.3715.prop$Num.Species <- c(nrow(sickle.raw.idba.3715.55.abund), 
                                      nrow(sickle.raw.idba.3715.75.abund),
                                      nrow(sickle.raw.idba.3715.95.abund),
                                      nrow(sickle.raw.idba.3715.111.abund),
                                      nrow(sickle.raw.idba.3715.127.abund))

# Quast Stats
sickle.raw.idba.3715.prop$N50 <- c(1936, 3134, 3638, 6771, 4134)
sickle.raw.idba.3715.prop$Contigs <- c(101720, 94935, 91315, 27838, 15498)
sickle.raw.idba.3715.prop$Largest.Contig <- c(195878, 220158, 340857, 322916, 450474)

```
### 3733 SPAdes
```{r}
# Load in .tsv files
hib.3733.51 <- read_tsv('3733/51.output.tsv')
hib.3733.65 <- read_tsv('3733/65.output.tsv')
hib.3733.71 <- read_tsv('3733/71.output.tsv')
hib.3733.75 <- read_tsv('3733/75.output.tsv')
hib.3733.77 <- read_tsv('3733/77.output.tsv')
hib.3733.85 <- read_tsv('3733/85.2.output.tsv')
hib.3733.95 <- read_tsv('3733/95.2.output.tsv')
hib.3733.127 <- read_tsv('3733/127.2.output.tsv')

hib.3733.51.sq <- read_tsv('3733/51.sq.output.tsv')
hib.3733.65.sq <- read_tsv('3733/65.sq.output.tsv')
hib.3733.71.sq <- read_tsv('3733/71.sq.output.tsv')
hib.3733.75.sq <- read_tsv('3733/75.sq.output.tsv')
hib.3733.77.sq <- read_tsv('3733/77.sq.output.tsv')
hib.3733.85.sq <- read_tsv('3733/85.sq.output.tsv')
hib.3733.95.sq <- read_tsv('3733/95.sq.output.tsv')
hib.3733.127.sq <- read_tsv('3733/127.sq.output.tsv')

# Calculate proportion classified & unclassified
hib.3733.prop <- data.frame(matrix(8,8))
hib.3733.prop$Kmer <- c(51,65,71,75,77,85,95,127)
hib.3733.prop$Unclassified <- c(
  length(which(hib.3733.51$seqID == "unclassified"))/nrow(hib.3733.51),
  length(which(hib.3733.65$seqID == "unclassified"))/nrow(hib.3733.65),
  length(which(hib.3733.71$seqID == "unclassified"))/nrow(hib.3733.71),
  length(which(hib.3733.75$seqID == "unclassified"))/nrow(hib.3733.75),
  length(which(hib.3733.77$seqID == "unclassified"))/nrow(hib.3733.77),
  length(which(hib.3733.85$seqID == "unclassified"))/nrow(hib.3733.85),
  length(which(hib.3733.95$seqID == "unclassified"))/nrow(hib.3733.95),
  length(which(hib.3733.127$seqID == "unclassified"))/nrow(hib.3733.127)
)
hib.3733.prop$Bact.Arch <- c(
  length(which(hib.3733.51$seqID != "unclassified"))/nrow(hib.3733.51),
  length(which(hib.3733.65$seqID != "unclassified"))/nrow(hib.3733.65),
  length(which(hib.3733.71$seqID != "unclassified"))/nrow(hib.3733.71),
  length(which(hib.3733.75$seqID != "unclassified"))/nrow(hib.3733.75),
  length(which(hib.3733.77$seqID != "unclassified"))/nrow(hib.3733.77),
  length(which(hib.3733.85$seqID != "unclassified"))/nrow(hib.3733.85),
  length(which(hib.3733.95$seqID != "unclassified"))/nrow(hib.3733.95),
  length(which(hib.3733.127$seqID != "unclassified"))/nrow(hib.3733.127)
)
hib.3733.prop <- hib.3733.prop[,-1]
hib.3733.prop$Squirrel <- c(
  length(which(hib.3733.51.sq$seqID != "unclassified"))/nrow(hib.3733.51.sq),
  length(which(hib.3733.65.sq$seqID != "unclassified"))/nrow(hib.3733.65.sq),
  length(which(hib.3733.71.sq$seqID != "unclassified"))/nrow(hib.3733.71.sq),
  length(which(hib.3733.75.sq$seqID != "unclassified"))/nrow(hib.3733.75.sq),
  length(which(hib.3733.77.sq$seqID != "unclassified"))/nrow(hib.3733.77.sq),
  length(which(hib.3733.85.sq$seqID != "unclassified"))/nrow(hib.3733.85.sq),
  length(which(hib.3733.95.sq$seqID != "unclassified"))/nrow(hib.3733.95.sq),
  length(which(hib.3733.127.sq$seqID != "unclassified"))/nrow(hib.3733.127.sq)
)
hib.3733.prop$Unclassified <- 1 - hib.3733.prop$Bact.Arch - hib.3733.prop$Squirrel

# Number of species
hib.3733.51.report <- read_tsv('3733/51.report.tsv')
hib.3733.65.report <- read_tsv('3733/65.report.tsv')
hib.3733.71.report <- read_tsv('3733/71.report.tsv')
hib.3733.75.report <- read_tsv('3733/75.report.tsv')
hib.3733.77.report <- read_tsv('3733/77.report.tsv')
hib.3733.85.report <- read_tsv('3733/85.2.report.tsv')
hib.3733.95.report <- read_tsv('3733/95.2.report.tsv')
hib.3733.127.report <- read_tsv('3733/127.2.report.tsv')

hib.3733.51.abund <- hib.3733.51.report[which(hib.3733.51.report$abundance>0),]
hib.3733.65.abund <- hib.3733.65.report[which(hib.3733.65.report$abundance>0),]
hib.3733.71.abund <- hib.3733.71.report[which(hib.3733.71.report$abundance>0),]
hib.3733.75.abund <- hib.3733.75.report[which(hib.3733.75.report$abundance>0),]
hib.3733.77.abund <- hib.3733.77.report[which(hib.3733.77.report$abundance>0),]
hib.3733.85.abund <- hib.3733.85.report[which(hib.3733.85.report$abundance>0),]
hib.3733.95.abund <- hib.3733.95.report[which(hib.3733.95.report$abundance>0),]
hib.3733.127.abund <- hib.3733.127.report[which(hib.3733.127.report$abundance>0),]

hib.3733.prop$Num.Species <- c(nrow(hib.3733.51.abund),nrow(hib.3733.65.abund),
                               nrow(hib.3733.71.abund),nrow(hib.3733.75.abund),
                               nrow(hib.3733.77.abund),nrow(hib.3733.85.abund),
                               nrow(hib.3733.95.abund),nrow(hib.3733.127.abund))

hib.3733.prop$N50 <- c(2741,4357,4522,4089,NA,5749,6187,12299)

```

### 3734 SPAdes
```{r}
# Load in .tsv files
sp.3734.51 <- read_tsv('3734/3734.51.output.tsv')
sp.3734.65 <- read_tsv('3734/3734.65.output.tsv')
sp.3734.71 <- read_tsv('3734/3734.71.output.tsv')
sp.3734.75 <- read_tsv('3734/3734.75.output.tsv')
sp.3734.77 <- read_tsv('3734/3734.77.output.tsv')
sp.3734.85 <- read_tsv('3734/3734.85.output.tsv')
sp.3734.95 <- read_tsv('3734/3734.95.output.tsv')
sp.3734.105 <- read_tsv('3734/3734.105.output.tsv')
sp.3734.115 <- read_tsv('3734/3734.115.output.tsv')
sp.3734.127 <- read_tsv('3734/3734.127.output.tsv')


# Calculate proportion classified & unclassified
sp.3734.prop <- data.frame(matrix(10,10))
sp.3734.prop$Kmer <- c(51,65,71,75,77,85,95,105,115,127)
sp.3734.prop$Unclassified <- c(
  length(which(sp.3734.51$seqID == "unclassified"))/nrow(sp.3734.51),
  length(which(sp.3734.65$seqID == "unclassified"))/nrow(sp.3734.65),
  length(which(sp.3734.71$seqID == "unclassified"))/nrow(sp.3734.71),
  length(which(sp.3734.75$seqID == "unclassified"))/nrow(sp.3734.75),
  length(which(sp.3734.77$seqID == "unclassified"))/nrow(sp.3734.77),
  length(which(sp.3734.85$seqID == "unclassified"))/nrow(sp.3734.85),
  length(which(sp.3734.95$seqID == "unclassified"))/nrow(sp.3734.95),
  length(which(sp.3734.105$seqID == "unclassified"))/nrow(sp.3734.105),
  length(which(sp.3734.115$seqID == "unclassified"))/nrow(sp.3734.115),
  length(which(sp.3734.127$seqID == "unclassified"))/nrow(sp.3734.127)
)
sp.3734.prop$Bact.Arch <- c(
  length(which(sp.3734.51$seqID != "unclassified"))/nrow(sp.3734.51),
  length(which(sp.3734.65$seqID != "unclassified"))/nrow(sp.3734.65),
  length(which(sp.3734.71$seqID != "unclassified"))/nrow(sp.3734.71),
  length(which(sp.3734.75$seqID != "unclassified"))/nrow(sp.3734.75),
  length(which(sp.3734.77$seqID != "unclassified"))/nrow(sp.3734.77),
  length(which(sp.3734.85$seqID != "unclassified"))/nrow(sp.3734.85),
  length(which(sp.3734.95$seqID != "unclassified"))/nrow(sp.3734.95),
  length(which(sp.3734.105$seqID != "unclassified"))/nrow(sp.3734.105),
  length(which(sp.3734.115$seqID != "unclassified"))/nrow(sp.3734.115),
  length(which(sp.3734.127$seqID != "unclassified"))/nrow(sp.3734.127)
)
sp.3734.prop <- sp.3734.prop[,-1]

# Number of species
sp.3734.51.report <- read_tsv('3734/3734.51.report.tsv')
sp.3734.65.report <- read_tsv('3734/3734.65.report.tsv')
sp.3734.71.report <- read_tsv('3734/3734.71.report.tsv')
sp.3734.75.report <- read_tsv('3734/3734.75.report.tsv')
sp.3734.77.report <- read_tsv('3734/3734.77.report.tsv')
sp.3734.85.report <- read_tsv('3734/3734.85.report.tsv')
sp.3734.95.report <- read_tsv('3734/3734.95.report.tsv')
sp.3734.105.report <- read_tsv('3734/3734.105.report.tsv')
sp.3734.115.report <- read_tsv('3734/3734.115.report.tsv')
sp.3734.127.report <- read_tsv('3734/3734.127.report.tsv')

sp.3734.51.abund <- sp.3734.51.report[which(sp.3734.51.report$abundance>0),]
sp.3734.65.abund <- sp.3734.65.report[which(sp.3734.65.report$abundance>0),]
sp.3734.71.abund <- sp.3734.71.report[which(sp.3734.71.report$abundance>0),]
sp.3734.75.abund <- sp.3734.75.report[which(sp.3734.75.report$abundance>0),]
sp.3734.77.abund <- sp.3734.77.report[which(sp.3734.77.report$abundance>0),]
sp.3734.85.abund <- sp.3734.85.report[which(sp.3734.85.report$abundance>0),]
sp.3734.95.abund <- sp.3734.95.report[which(sp.3734.95.report$abundance>0),]
sp.3734.105.abund <- sp.3734.95.report[which(sp.3734.95.report$abundance>0),]
sp.3734.115.abund <- sp.3734.95.report[which(sp.3734.95.report$abundance>0),]
sp.3734.127.abund <- sp.3734.127.report[which(sp.3734.127.report$abundance>0),]

sp.3734.prop$Num.Species <- c(nrow(sp.3734.51.abund),nrow(sp.3734.65.abund),
                               nrow(sp.3734.71.abund),nrow(sp.3734.75.abund),
                               nrow(sp.3734.77.abund),nrow(sp.3734.85.abund),
                               nrow(sp.3734.95.abund),nrow(sp.3734.105.abund),
                               nrow(sp.3734.115.abund),nrow(sp.3734.127.abund))

sp.3734.prop$N50 <- c(1622,2016,2051,2861,2083,2124,2671,2899,2937,5293)
```







### 3715 IDBA 111 Bin (Default)
```{r}
# printPerf function is to show cumulative number of bins fulfilling given recall (completeness or sensitivity) and precision (purity or specificity) cutoffs.

# 9 bins total

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                             removeStrain=F)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                             removeStrain=T)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

# The difference between two tables is whether to remove strain level contamination in CheckM results. If removeStrain is set to TRUE, contamination by different strains is ignored for precision calculation and the number of bins fulfilling the same precision cutoff will increase.

# Compare the 2 tables by looking at the last row (specificity = 0.99)
# The increase in table 1 to table 2 will be due to strain contamination (mixture of different strains)
```

### 3715 IDBA 111 Bin (Contig min = 1500)
```{r}
# printPerf function is to show cumulative number of bins fulfilling given recall (completeness or sensitivity) and precision (purity or specificity) cutoffs.

# 12 bins total

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                             removeStrain=F)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                             removeStrain=T)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

# The difference between two tables is whether to remove strain level contamination in CheckM results. If removeStrain is set to TRUE, contamination by different strains is ignored for precision calculation and the number of bins fulfilling the same precision cutoff will increase.

# Compare the 2 tables by looking at the last row (specificity = 0.99)
# The increase in table 1 to table 2 will be due to strain contamination (mixture of different strains)
```

### 3715 IDBA 111 Bin (Bin size min = 100k)
```{r}
# printPerf function is to show cumulative number of bins fulfilling given recall (completeness or sensitivity) and precision (purity or specificity) cutoffs.

# 15 bins total

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                             removeStrain=F)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                             removeStrain=T)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

# The difference between two tables is whether to remove strain level contamination in CheckM results. If removeStrain is set to TRUE, contamination by different strains is ignored for precision calculation and the number of bins fulfilling the same precision cutoff will increase.

# Compare the 2 tables by looking at the last row (specificity = 0.99)
# The increase in table 1 to table 2 will be due to strain contamination (mixture of different strains)
```

### 3715 IDBA 111 Bin (Bin size min = 50k)
```{r}
# printPerf function is to show cumulative number of bins fulfilling given recall (completeness or sensitivity) and precision (purity or specificity) cutoffs.

# 22 bins total

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                             removeStrain=F)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                             removeStrain=T)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

# The difference between two tables is whether to remove strain level contamination in CheckM results. If removeStrain is set to TRUE, contamination by different strains is ignored for precision calculation and the number of bins fulfilling the same precision cutoff will increase.

# Compare the 2 tables by looking at the last row (specificity = 0.99)
# The increase in table 1 to table 2 will be due to strain contamination (mixture of different strains)
```

### 3715 IDBA 111 Bin (Bin size min = 10k)
```{r}
# printPerf function is to show cumulative number of bins fulfilling given recall (completeness or sensitivity) and precision (purity or specificity) cutoffs.

# 57 bins total

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                             removeStrain=F)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

printPerf(list(calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                             removeStrain=T)), rec=c(seq(.1,.9,.1),.95),
          prec=c(seq(.6,.9,.1),.95,.99))
  # Recall = completeness/sensitivity
  # Precision = purity/specificity

# The difference between two tables is whether to remove strain level contamination in CheckM results. If removeStrain is set to TRUE, contamination by different strains is ignored for precision calculation and the number of bins fulfilling the same precision cutoff will increase.

# Compare the 2 tables by looking at the last row (specificity = 0.99)
# The increase in table 1 to table 2 will be due to strain contamination (mixture of different strains)
```

### Compare 3715 IDBA 111 Metabat Outputs
```{r}
# Calculates difference between 2 independent runs of CheckM results.
# Positive = first result has more bins fulfilling cutoffs; more positive = better
# You want more positive #'s in bottom right --> more complete & precise bins

diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # 3715.bin.checkm is much worse at low completion precision bins, but is barely better than 3715.lowcontig.chekm at high precision & completion bins
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # Pretty much the same
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # Meh
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # Pretty much the same
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.bin.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # 3715.lowcontig is better at low completeness & precision, but worse at high precision & completness
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # Pretty much the same
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # 3715.lowcontig is better at low completeness & precision, but worse at high precision & completness
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.lowcontig.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # 100 just slightly better
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # Pretty much the same
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.100.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=F),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=F),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))
  # 50 is worse than 10
diffPerf(calcPerfBySCG("3715.untrim/idba.111.counts/3715.50.checkm.txt",
                       removeStrain=T),
         calcPerfBySCG("3715.untrim/idba.111.counts/3715.10.checkm.txt",
                       removeStrain=T),
         rec=c(seq(.1,.9,.1),.95),
         prec=c(seq(.6,.9,.1), .95, .99))



# No huge difference between the different metabat runs, but if I had to pick, I'd go with 3715.100
```




