---
title: "OG_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---
### Load Libraries
```{r}
library('readr')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
library('phangorn')
library('vegan')
library('pheatmap')
library('RColorBrewer')
library('gplots')
library('tidyr')
library('dplyr')
library('stringr')
library('lme4')
library('picante')
library('ggdendro')
theme_set(theme_bw())
set.seed(1)
```

### Load Functions
```{r}
source('scripts/misc.R')
source('scripts/simper.R')
source('scripts/pairwise.adonis.R')
```

### MicrobeCensus - Average Genome Size
```{r}
# Load data
mc <- read.csv('MicrobeCensus/Summary.csv', row.names = 1)
mc.ma <- as.matrix(mc)

# Test normality
hist(mc.ma[2,])
qqnorm(mc.ma[2,])
qqline(mc.ma[2,])
shapiro.test(mc.ma[2,])
  # p-value = 0.03487
  # Not Normal

# Format for Kruskal-Wallis
mc <- t(mc)
mc <- data.frame(mc)
mc$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
mc$Season <- ordered(mc$Season, levels=c("Summer", "Winter", "Spring"))

# Kruskal-Wallis
kruskal.test(formula = AGS ~ Season, data = mc)
  # p-value = 0.4298
```


### dbCAN - Proportion CAZymes from ORFs
```{r}
prodigalCounts <- read.csv('dbcan/prodigalCounts.csv', row.names=1)
prodigalCounts <- as.matrix(prodigalCounts)
# Test normality
hist(prodigalCounts[3,])
qqnorm(prodigalCounts[3,])
qqline(prodigalCounts[3,])
shapiro.test(prodigalCounts[3,])
  # p-value = 0.001575
  # Not Normal
# Format for Kruskal-Wallis
prodigalCounts <- read.csv('dbcan/prodigalCounts.csv', row.names=1)
prodigalCounts <- t(prodigalCounts)
prodigalCounts <- data.frame(prodigalCounts)
prodigalCounts$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
prodigalCounts$Season <- ordered(prodigalCounts$Season, levels=c("Summer", "Winter", "Spring"))
prodigalCounts$Diet <- c("Feeding", "Feeding", "Fasting", "Fasting", "Feeding", "Feeding", "Fasting", "Feeding", "Feeding")
prodigalCounts$Diet <- ordered(prodigalCounts$Diet, levels=c("Feeding", "Fasting"))
colnames(prodigalCounts) <- c("Total_ORFs", "Total_CAZymes", "CAZymePercentage", "Season", "Diet")
# Test w/ Kruskal-Wallis
kruskal.test(formula = CAZymePercentage ~ Season, data = prodigalCounts)
  # p-value = 0.5611
kruskal.test(formula = CAZymePercentage ~ Diet, data = prodigalCounts)
  # p-value = 0.4386
### Dot plot
#tiff("dbcan.prop.tiff", width=5, height=4, units="in", res= 300)
ggplot(prodigalCounts, aes(x=Season, y=CAZymePercentage, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as CAZyme (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(2.0, 2.6, 3.2, 3.8, 4.4), limits=c(2.0, 4.4)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
#dev.off()
```
### dbCAN - Create phyloseq object
```{r}
# Load data
otu <- read.csv('dbcan/dbcan_otu_table.csv', row.names=1)
colnames(otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
otu.tab <- otu_table(otu, taxa_are_rows=T)
tax <- read.csv('dbcan/dbcan_tax_table.csv', row.names=1)
tax <- tax_table(tax)
colnames(tax) <- c("Class", "Family")
taxa_names(tax) <- tax[,2]
meta <- read.csv('dbcan/dbcan.spades.metadata.csv', row.names=1)
meta.ord <- meta

# Create phyloseq object
dbcan.phy <- merge_phyloseq(otu.tab, tax, sample_data(meta.ord))

# Trim CAZymes w/ count < 4
# 3 samples per season, so 4 seemed like a somewhat reasonable arbitrary cutoff
# Cutoff = 4 also ensures that no normalized counts < 3, so no further cutoffs needed
dbcan.cut.phy <- prune_taxa(taxa_sums(dbcan.phy) > 3, dbcan.phy)

# Identify trimmed CAZymes (count < 4)
#all <- rownames(tax_table(dbcan.phy))
#remain <- rownames(tax_table(dbcan.cut.phy))
#for (i in all){
#  if (i %in% remain == F){
#    print(i)
#  }
#}
# GH12, GH17, GH34, GH52, GH100, GH107, GH118, GH149, GH153, GT29, GT37, GT42, GT52, GT58, GT60, GT67, GT68, GT77, GT85, GT93, GT97, CE5, CE13, PL2, PL14, PL16, PL18, PL28


# Normalized Counts At Family-level
# https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
#otu.cut <- data.frame(otu_table(dbcan.cut.phy))
#dbcan.dds <- DESeqDataSetFromMatrix(countData = otu.cut, colData = meta, design = ~ Season)
#dbcan.dds <- estimateSizeFactors(dbcan.dds)
#dbcan.norm <- counts(dbcan.dds, normalized=T)
#write.csv(dbcan.norm, file="dbcan/dbcan.Normalized.Counts.Family.csv")

dbcan.fam.norm <- read.csv('dbcan/dbcan.Normalized.Counts.Family.csv', row.names=1)
dbcan.fam.norm.otu <- otu_table(dbcan.fam.norm, taxa_are_rows=T)
sample_names(dbcan.fam.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
dbcan.fam.norm.phy <- merge_phyloseq(dbcan.fam.norm.otu, sample_data(dbcan.phy), tax_table(dbcan.phy))

# Normalize Counts to Class-Level
#dbcan.class <- tax_glom(dbcan.cut.phy, taxrank="Class")
#dbcan.class.otu <- data.frame(otu_table(dbcan.class))
#dbcan.dds <- DESeqDataSetFromMatrix(countData = dbcan.class.otu, colData = meta, design = ~ Season)
#dbcan.dds <- estimateSizeFactors(dbcan.dds)
#dbcan.norm <- counts(dbcan.dds, normalized=T)
#write.csv(dbcan.norm, file="dbcan/dbcan.Normalized.Counts.Class.csv")

dbcan.class.norm <- read.csv('dbcan/dbcan.Normalized.Counts.Class.csv', row.names=1)
rownames(dbcan.class.norm) <- c("Polysaccharide Lyase", "Glycosyltransferase", "Glycoside Hydrolase", "Carbohydrate Esterase")

# Pairwise season physeq objects
#dbcan.sw <- subset_samples(dbcan99, Season != "Spring")
  # Summer & Winter
#dbcan.sp <- subset_samples(dbcan99, Season != "Winter")
  # Summer & Spring
#dbcan.wp <- subset_samples(dbcan99, Season != "Summer")
  # Winter & Spring
```
### dbCAN - Count unique CAZyme families
```{r}
# Subset out samples by season
phy.sum <- subset_samples(dbcan.cut.phy, Season == "Summer")
phy.win <- subset_samples(dbcan.cut.phy, Season == "Winter")
phy.spr <- subset_samples(dbcan.cut.phy, Season == "Spring")

# Remove CAZyme families with no counts
phy.sum <- prune_taxa(taxa_sums(phy.sum) > 0, phy.sum)
phy.win <- prune_taxa(taxa_sums(phy.win) > 0, phy.win)
phy.spr <- prune_taxa(taxa_sums(phy.spr) > 0, phy.spr)


# Pull out unique CAZyme families in each season and order alphabetically
sum.fam <- get_taxa_unique(phy.sum, "Family")
sum.fam <- sort(sum.fam)
win.fam <- get_taxa_unique(phy.win, "Family")
win.fam <- sort(win.fam)
spr.fam <- get_taxa_unique(phy.spr, "Family")
spr.fam <- sort(spr.fam)


# Summer - Count # of CAZymes in each class
class <- substr(sum.fam,1,2)
length(which(class == "GH"))
  # 122
length(which(class == "GT"))
  # 60
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 26


# Winter - Count # of CAZymes in each class
class <- substr(win.fam,1,2)
length(which(class == "GH"))
  # 117
length(which(class == "GT"))
  # 60
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 25


# Spring - Count # of CAZymes in each class
class <- substr(spr.fam,1,2)
length(which(class == "GH"))
  # 124
length(which(class == "GT"))
  # 63
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 26

```

### dbCAN - Class-level bar plot
```{r}
# Create phyloseq object to summarize data
class.tax <- data.frame(c("Carbohydrate Esterase", "Glycoside Hydrolase", "Glycosyltransferase", 
                          "Polysaccharide Lyase"))
class.tax <- tax_table(class.tax)
row.names(class.tax) <- class.tax[,1]
colnames(class.tax) <- "Class"
dbcan.class.norm.otu <- otu_table(dbcan.class.norm, taxa_are_rows=T)
colnames(dbcan.class.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
dbcan.class.phy <- merge_phyloseq(dbcan.class.norm.otu, class.tax, sample_data(meta.ord))

# Summarize data
dbcan.class.df <- psmelt(dbcan.class.phy)
dbcan.class.sum <- dbcan.class.df %>%
  group_by(Class, Season) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
dbcan.class.sum$Class <- ordered(dbcan.class.sum$Class, c("Glycoside Hydrolase",
                                                                  "Glycosyltransferase",
                                                                  "Carbohydrate Esterase",
                                                                  "Polysaccharide Lyase"))
dbcan.class.sum$Season <- ordered(dbcan.class.sum$Season, levels = c("Summer", "Winter", "Spring"))

# Plot barplot
#tiff("dbcan.class.tiff", width=10, height=4, units="in", res= 300)
ggplot(dbcan.class.sum, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,9500)) +
  facet_grid(. ~ Class) +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab("CAZyme Class") +
  ylab("Mean Normalized Counts") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"),
        panel.spacing = unit(0.5, "lines"),
        axis.title.x = element_text(margin = margin(t = 20)))
#dev.off()




# # Summarize data
# dbcan.class.diet.sum <- dbcan.class.df %>%
#   group_by(Class, Diet) %>%
#   summarize(Mean = mean(Abundance),
#             SD = sd(Abundance),
#             SE = se(Abundance))
# dbcan.class.diet.sum$Class <- ordered(dbcan.class.diet.sum$Class, c("Glycoside Hydrolase",
#                                                                   "Glycosyltransferase",
#                                                                   "Carbohydrate Esterase",
#                                                                   "Polysaccharide Lyase"))
# dbcan.class.diet.sum$Diet <- ordered(dbcan.class.diet.sum$Diet, levels=c("Feeding", "Fasting"))
# 
# # Plot barplot
# ggplot(dbcan.class.diet.sum, aes(x=Diet, y=Mean, fill=Diet)) +
#   geom_bar(stat="identity", color="black") +
#   geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
#   scale_y_continuous(limits=c(0,9500)) +
#   facet_grid(. ~ Class) +
#   scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3")) +
#   xlab("CAZyme Class") +
#   ylab("Mean Normalized Counts") +
#   theme(panel.grid.major = element_blank(),
#         panel.grid.minor = element_blank(),
#         strip.background = element_blank(),
#         strip.text = element_text(face="bold"),
#         strip.text.x = element_text(size=11),
#         legend.title = element_text(face="bold"),
#         panel.spacing = unit(0.5, "lines"),
#         axis.title.x = element_text(margin = margin(t = 15)))
```

### dbCAN - Test class-level
```{r}
# Same results if you compare Diet (Feeding vs. Fasting)
### CE
ce <- dbcan.class.df[which(dbcan.class.df$Class == "Carbohydrate Esterase"),]

# Test normality
hist(ce$Abundance)
qqnorm(ce$Abundance)
qqline(ce$Abundance)
shapiro.test(ce$Abundance)
  # p-value = 0.6811
  # Normal

# Test CE w/ ANOVA
summary(aov(Abundance ~ Season, data = ce))
  # p-value = 0.332



### GH
gh <- dbcan.class.df[which(dbcan.class.df$Class == "Glycoside Hydrolase"),]

# Test normality
hist(gh$Abundance)
qqnorm(gh$Abundance)
qqline(gh$Abundance)
shapiro.test(gh$Abundance)
  # p-value = 0.4749
  # Normal

# Test GH w/ ANOVA
summary(aov(Abundance ~ Season, data = gh))
  # p-value = 0.00789

# GH posthoc test w/ Tukey's
TukeyHSD(aov(Abundance ~ Season, data = gh))
  # Summer-Spring = 0.0297554*
  # Winter-Spring = 0.0076492**
  # Winter-Summer = 0.4762540



### GT
gt <- dbcan.class.df[which(dbcan.class.df$Class == "Glycosyltransferase"),]

# Test normality
hist(gt$Abundance)
qqnorm(gt$Abundance)
qqline(gt$Abundance)
shapiro.test(gt$Abundance)
  # p-value = 0.4515
  # Normal

# Test GT w/ ANOVA
summary(aov(Abundance ~ Season, data = gt))
  # p-value = 0.0074

# GT posthoc test w/ Tukey's
TukeyHSD(aov(Abundance ~ Season, data = gt))
  # Summer-Spring = 0.3591323
  # Winter-Spring = 0.0067560**
  # Winter-Summer = 0.0347394*



### PL
pl <- dbcan.class.df[which(dbcan.class.df$Class == "Polysaccharide Lyase"),]

# Test normality
hist(pl$Abundance)
qqnorm(pl$Abundance)
qqline(pl$Abundance)
shapiro.test(pl$Abundance)
  # p-value = 0.08667
  # Normal

# Test PL w/ ANOVA
summary(aov(Abundance ~ Season, data = pl))
  # p-value = 0.116




### Adjust Pvals
p.adjust(c(0.332, 0.00789, 0.0074, 0.116), method="BH")
  # CE = 0.33200
  # GH = 0.01578 *
  # GT = 0.01578 *
  # PL = 0.15467


```












### dbCAN - Create DESeq Object
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
# https://compbiocore.github.io/deseq-workshop-1/assets/deseq_workshop_1.html
# https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
# Use UN-NORMALIZED data!!!
  # DESeq accounts for library size


### Test DESeq2 to determine fit type
# https://support.bioconductor.org/p/81094/
#dbcan.dds <- phyloseq_to_deseq2(dbcan.cut.phy, ~Season)
#dbcan.dds$Season <- factor(dbcan.dds$Season, levels=c("Summer", "Winter", "Spring"))
#dbcan.dds <- dbcan.dds[rowSums(counts(dbcan.dds)) > 1,]
#dbcan.local <- DESeq(dbcan.dds, fitType = "local")
#dbcan.par <- DESeq(dbcan.dds, fitType = "parametric")
#mad(log(mcols(dbcan.local)$dispGeneEst) - log(mcols(dbcan.local)$dispFit))
  # 9.744673
#mad(log(mcols(dbcan.par)$dispGeneEst) - log(mcols(dbcan.par)$dispFit))
  # 7.288668
# Parametric is a better fit (lower median absolute residual) than local

### Convert phyloseq object to deseq2 object
dbcan.dds <- phyloseq_to_deseq2(dbcan.cut.phy, ~Season)
  # Design variable must be unordered otherwise you get an error
dbcan.dds$Season <- factor(dbcan.dds$Season, levels=c("Summer", "Winter", "Spring"))

# DESeq DataStructure for analyses
dbcan.dds <- DESeq(dbcan.dds)

# vst-transformed data for visualizations (VST = variance stabilizing transformation)
  # Choosing vst instead of rlog due to these two threads:
    # https://support.bioconductor.org/p/104615/#104650
    # https://support.bioconductor.org/p/123969/
dbcan.vsd <- varianceStabilizingTransformation(dbcan.dds)
```
### dbCAN - DESeq - find overrep genes
```{r}
# Pull out results from pairwise comparison
dbcan.sw <- results(dbcan.dds, contrast=c("Season", "Summer", "Winter"))
dbcan.sw.sig <- dbcan.sw[which(dbcan.sw$padj < 0.05),]
  # Enriched in Summer compared to Winter (positive log2FoldChange)
    # GT101, PL1, GH43, GH115, CE12, GH10, GH59
dbcan.sp <- results(dbcan.dds, contrast=c("Season", "Summer", "Spring"))
dbcan.sp.sig <- dbcan.sp[which(dbcan.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # GT101
dbcan.wp <- results(dbcan.dds, contrast=c("Season", "Winter", "Spring"))
dbcan.wp.sig <- dbcan.wp[which(dbcan.wp$padj < 0.05),]
dbcan.wp.sig[which(dbcan.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
      # GT41, GH20, GT4, GT9, CE11, GT3, GH84
dbcan.wp.sig[which(dbcan.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH3, GH32, GH43, GH5, GH78, GH115, GH42, CE12, GH106, GH10, GH39
```

### dbCAN - DESeq - Diet
```{r}
dbcan.diet.dds <- phyloseq_to_deseq2(dbcan.cut.phy, ~Diet)
dbcan.diet.dds <- DESeq(dbcan.diet.dds)
dbcan.diet.results <- results(dbcan.diet.dds)
dbcan.diet.sig <- dbcan.diet.results[which(dbcan.diet.results$padj < 0.05),]
dbcan.diet.sig[which(dbcan.diet.sig$log2FoldChange > 0),]
  # Enriched in feeding compared to fasting
  # PL1, GH43, GH115, CE12, GH106, GH10, GH39, GH59
dbcan.diet.sig[which(dbcan.diet.sig$log2FoldChange < 0),]
  # Enriched in fasting compared to feeding
  # GT4, GT9, CE11, GH84
```
### dbCAN - DESeq heatmap of normalized counts
```{r}
# Note: I tried merging the samples by season so the heatmap only displays Summer, Winter, and Spring, but I get an error that say I have insufficient data. Therefore, I display each sample separately.

# Excluding some families because they has such high counts that it makes the heatmap hard to interpret visually
# I'll make a separate plot for just GH43, GH3, GT4, and GT41

dbcan.sig <- subset_taxa(dbcan.heatmap.phy, Family == "GT101" | Family == "GH59" | Family == "PL1" |
                           Family == "GH10" | Family == "GH115" | Family == "CE12" | Family == "GH5" |
                           Family == "GH32" | Family == "GH39" | Family == "GH42" | Family == "GH78" |
                           Family == "GH106" | Family == "GH20" | Family == "GH84" | Family == "GT3" |
                           Family == "GH9" | Family == "CE11")



#tiff("dbcan.heatmap.tiff", width=9, height=10, units="in", res=300)
plot_heatmap(dbcan.sig, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("CE11", "GH9", "GT3", "GH84", "GH20", "GH106", "GH78", "GH42", "GH39", "GH32", "GH5", "CE12", "GH115", "GH10", "PL1", "GH59", "GT101")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("CAZyme Family\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(7,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
#dev.off()

### GH43, GH3, GT4, and GT41 plot
dbcan.high <- subset_taxa(dbcan.fam.norm.phy, Family == "GH43" | Family == "GH3" | Family == "GT4" |
                            Family == "GT41")
#tiff("dbcan.high.heatmap.tiff", width=9, height=3.3, units="in", res=300)
plot_heatmap(dbcan.high, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("GT41", "GT4", "GH3", "GH43")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("CAZyme Family\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"), limits=c(300,1200),
                       breaks=c(300, 600, 900, 1200), 
                       labels=c("300", "600", "900", "1200"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
   theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(2,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
#dev.off()

```





### dbCAN - DESeq sample dist heatmap / dendrogram
``` {r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#exploring-and-exporting-results
# http://bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

dbcan.dist <- dist(t(assay(dbcan.vsd)))
dbcan.dist.matrix <- as.matrix(dbcan.dist)
rownames(dbcan.dist.matrix) <- dbcan.vsd$Season
colnames(dbcan.dist.matrix) <- dbcan.vsd$Season

# Heatmap
pheatmap(dbcan.dist.matrix, clustering_distance_rows = dbcan.dist, clustering_distance_cols = dbcan.dist,
         col = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c", "#fc4e2a", "#e31a1c",
                 "#bd0026", "#800026"))

# Dendrogram
dbcan.hclust <- hclust(dbcan.dist)
dbcan.den <- as.dendrogram(dbcan.hclust)
dbcan.den.data <- dendro_data(dbcan.den, type="rectangle")
dbcan.den.data$labels[,3] <- c("Summer", "Winter", "Winter", "Winter", "Summer", "Spring", "Spring", "Spring", "Summer")
dbcan.den.data$labels[,3] <- ordered(dbcan.den.data$labels[,3], levels=c("Summer", "Winter", "Spring"))

#tiff("dbcan.dendro.tiff", width=10, height=5, units="in", res=300)
ggplot(dbcan.den.data$segments) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  geom_text(data = dbcan.den.data$labels, aes(x, y, label=label, colour=label, fontface = "bold"),
            hjust=0, size=5) +
  ylab("Euclidean Distance") +
  xlab("Season") +
  labs(colour = "Season") +
  coord_flip() +
  scale_y_reverse() +
  scale_color_manual(values=c("darkgoldenrod3", "dodgerblue3", "green4")) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_text(size=20),
        panel.border = element_blank(),
        axis.line.x = element_line(),
        legend.title = element_text(face="bold"),
        axis.title.x = element_text(face="bold", size=20),
        axis.title.y = element_text(face="bold", size=20))
#dev.off()


```

### dbCAN - sample dist PCA & PERMANOVA
```{r}
dbcan.pca.data <- plotPCA(dbcan.vsd, intgroup="Season", returnData=T)
dbcan.pca.perc <- round(100*attr(dbcan.pca.data, "percentVar"))
# PCA variance explained = 62%

#tiff("dbcan.pca.tiff", width=8, height=6, units="in", res=300)
ggplot(dbcan.pca.data, aes(PC1, PC2, color=Season)) +
  geom_point(size=5) +
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab(paste0("PC1: ", dbcan.pca.perc[1], "% variance")) +
  ylab(paste0("PC2: ", dbcan.pca.perc[2], "% variance")) +
  ggtitle("Euclidean Distance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20),
        legend.text = element_text(size=18),
        plot.title = element_text(hjust=0.5, size=20))
#dev.off()


### Test normality
hist(dbcan.dist)
qqnorm(dbcan.dist)
qqline(dbcan.dist)
shapiro.test(dbcan.dist)
  # p-value = 0.5035
  # Normal

### Run PERMANOVA on DESEq Euclidean distance
dbcan.dist <- dist(t(assay(dbcan.vsd)))
adonis(dbcan.dist ~ Season, data=meta, permutations = 99999)
  # p-value = 0.00721
beta.tax = betadisper(d=dbcan.dist, group=meta$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.652



### Pairwise PERMANOVA Comparisons
dbcan.dist.matrix <- as.matrix(dbcan.dist)
dbcan.dist.df <- data.frame(dbcan.dist.matrix)

# Summer vs. Winter
dbcan.sw.dist.df <- dbcan.dist.df[,-5]
dbcan.sw.dist.df <- dbcan.sw.dist.df[,-7:-8]
dbcan.sw.dist.df <- dbcan.sw.dist.df[-5,]
dbcan.sw.dist.df <- dbcan.sw.dist.df[-7:-8,]
dbcan.sw.dist <- as.dist(dbcan.sw.dist.df)
meta.sw <- meta[-which(meta$Season == "Spring"),]
adonis(dbcan.sw.dist ~ Season, data=meta.sw, permutations= 99999)
  # p-value = 0.1

# Summer vs. Spring
dbcan.sp.dist.df <- dbcan.dist.df[,-3:-4]
dbcan.sp.dist.df <- dbcan.sp.dist.df[,-5]
dbcan.sp.dist.df <- dbcan.sp.dist.df[-3:-4,]
dbcan.sp.dist.df <- dbcan.sp.dist.df[-5,]
dbcan.sp.dist <- as.dist(dbcan.sp.dist.df)
meta.sp <- meta[-which(meta$Season == "Winter"),]
adonis(dbcan.sp.dist ~ Season, data=meta.sp, permutations= 99999)
  # p-value = 0.2

# Spring vs. Winter
dbcan.wp.dist.df <- dbcan.dist.df[,-3:-4]
dbcan.wp.dist.df <- dbcan.wp.dist.df[,-5]
dbcan.wp.dist.df <- dbcan.wp.dist.df[-3:-4,]
dbcan.wp.dist.df <- dbcan.wp.dist.df[-5,]
dbcan.wp.dist <- as.dist(dbcan.wp.dist.df)
meta.wp <- meta[-which(meta$Season == "Summer"),]
adonis(dbcan.wp.dist ~ Season, data=meta.wp, permutations= 99999)
  # p-value = 0.2


### Pairwise Welch's T-Test Comparisons

# Summer vs. Winter
meta.sw$Season <- ordered(meta.sw$Season, levels=c("Summer","Winter"))
WT.test(dbcan.sw.dist, meta.sw$Season)
  # p-value = 0.123

# Summer vs. Spring
meta.sp$Season <- ordered(meta.sp$Season, levels=c("Summer","Spring"))
WT.test(dbcan.sp.dist, meta.sp$Season)
  # P-value = 0.205

# Winter vs. Spring
meta.wp$Season <- ordered(meta.wp$Season, levels=c("Winter","Springr"))
WT.test(dbcan.sp.dist, meta.sp$Season)
  # p-value = 0.205
```

### dbCAN - Mucin-degrading CAZymes
```{r}
# List of mucin-degrading CAZymes
muc.enz <- c("GH2", "GH3", "GH4", "GH16", "GH18", "GH20", "GH27", "GH29", "GH31", "GH33", "GH35", "GH36", "GH38", "GH42", "GH67", "GH76", "GH84", "GH85", "GH89", "GH92", "GH95", "GH97", "GH98", "GH101", "GH109", "GH110", "GH112", "GH123", "GH125", "GH129", "GH130", "GH163")



# Pull out mucin-degrading CAZymes
for (i in 1:length(muc.enz)){
  
  if (i == 1){
    enz <- which(rownames(dbcan.fam.norm) == muc.enz[i])
    muc.df <- dbcan.fam.norm[enz,]
  }
  
  enz <- which(rownames(dbcan.fam.norm) == muc.enz[i])
  muc.df[i,] <- dbcan.fam.norm[enz,]
}
colnames(muc.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Sum of mucin-degrading CAZymes
for (i in 1:ncol(muc.df)) {
  if (i == 1){
    muc.sum <- sum(muc.df[,i])
    muc.sum.df <- data.frame(muc.sum)
    colnames(muc.sum.df)[i] <- colnames(muc.df)[i]
  }
  muc.sum <- sum(muc.df[,i])
  muc.sum.df[,i] <- muc.sum
  colnames(muc.sum.df)[i] <- colnames(muc.df)[i]
}
muc.sum.df <- data.frame(t(muc.sum.df))
colnames(muc.sum.df) <- "Total_MucinDeg"
muc.sum.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
muc.sum.df$Season <- ordered(muc.sum.df$Season, levels=c("Summer", "Winter", "Spring"))


# Test normality
hist(muc.sum.df[,1])
qqnorm(muc.sum.df[,1])
qqline(muc.sum.df[,1])
shapiro.test(muc.sum.df[,1])
  # p-value = 0.979
  # Normal

# Test w/ ANOVA
summary(aov(Total_MucinDeg ~ Season, data = muc.sum.df))
  # p-value = 0.0263

# Posthoc test w/ Tukey's
TukeyHSD(aov(Total_MucinDeg ~ Season, data = muc.sum.df))
  # Winter-Summer = 0.3489136
  # Spring-Summer = 0.0223984*
  # Spring-Winter = 0.1447197


### Boxplot
ggplot(muc.sum.df, aes(x=Season, y=Total_MucinDeg, fill=Season)) +
  geom_boxplot(position=position_dodge(1)) +
  geom_point(position=position_dodge(width = 0.2), size=2) +
  ylab("Normalized Counts of Mucin-degrading CAZymes") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(3400, 3600, 3800, 4000, 4200), limits=c(3400, 4200)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())


### DESeq2
dbcan.muc <- subset_taxa(dbcan.phy, Family == "GH2" | Family == "GH3" | Family == "GH4" | 
                           Family == "GH16" | Family == "GH18" | Family == "GH20" | Family == "GH27" |
                           Family == "GH29" | Family == "GH31" | Family == "GH33" | Family == "GH35" |
                           Family == "GH36" | Family == "GH38" | Family == "GH42" | Family == "GH67" |
                           Family == "GH76" | Family == "GH84" | Family == "GH85" | Family == "GH89" |
                           Family == "GH92" | Family == "GH95" | Family == "GH97" | Family == "GH98" |
                           Family == "GH101" | Family == "GH109" | Family == "GH110" |
                           Family == "GH112" | Family == "GH123" | Family == "GH125" |
                           Family == "GH129" | Family == "GH130" | Family == "GH163")




dbcan.muc.dds <- phyloseq_to_deseq2(dbcan.muc, ~Season)
  # Design variable must be unordered otherwise you get an error
dbcan.muc.dds$Season <- factor(dbcan.muc.dds$Season, levels=c("Summer", "Winter", "Spring"))
dbcan.muc.dds <- dbcan.muc.dds[rowSums(counts(dbcan.muc.dds)) > 1,]
dbcan.muc.dds <- DESeq(dbcan.muc.dds)

# Pull out results from pairwise comparisons
dbcan.muc.sw <- results(dbcan.muc.dds, contrast=c("Season", "Summer", "Winter"))
dbcan.muc.sw.sig <- dbcan.muc.sw[which(dbcan.muc.sw$padj < 0.05),]
  # None
dbcan.muc.sp <- results(dbcan.muc.dds, contrast=c("Season", "Summer", "Spring"))
dbcan.muc.sp.sig <- dbcan.muc.sp[which(dbcan.muc.sp$padj < 0.05),]
  # None
dbcan.muc.wp <- results(dbcan.muc.dds, contrast=c("Season", "Winter", "Spring"))
dbcan.muc.wp.sig <- dbcan.muc.wp[which(dbcan.muc.wp$padj < 0.05),]
dbcan.muc.wp.sig[which(dbcan.muc.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # GH20, GH84
dbcan.muc.wp.sig[which(dbcan.muc.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH3, GH36, GH42, GH129
```



### Mucin - Proportion mucin-degrading genes from ORFs
```{r}
prodigalCounts <- read.csv('mucin/prodigalCounts.csv', row.names=1)
prodigalCounts <- as.matrix(prodigalCounts)
# Test normality
hist(prodigalCounts[3,])
qqnorm(prodigalCounts[3,])
qqline(prodigalCounts[3,])
shapiro.test(prodigalCounts[3,])
  # p-value = 0.05926
  # Normal
# Format for ANOVA
prodigalCounts <- read.csv('mucin/prodigalCounts.csv', row.names=1)
prodigalCounts <- t(prodigalCounts)
prodigalCounts <- data.frame(prodigalCounts)
prodigalCounts$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
prodigalCounts$Season <- ordered(prodigalCounts$Season, levels=c("Summer", "Winter", "Spring"))
colnames(prodigalCounts) <- c("Total_ORFs", "Total_MucinGenes", "MucinGenePercentage", "Season")
# Test w/ ANOVA
summary(aov(MucinGenePercentage ~ Season, data = prodigalCounts))
  # p-value = 0.224
### Dot plot
#tiff("mucin.prop.tiff", width=5, height=4, units="in", res= 300)
ggplot(prodigalCounts, aes(x=Season, y=MucinGenePercentage, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as Mucin-Degrading Gene (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(0.06, 0.07, 0.08, 0.09), limits=c(0.06, 0.09)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
#dev.off()
```
### Mucin - Create phyloseq object
```{r}
# Load data
muc.otu <- read.csv('mucin/mucinCount.csv', row.names=1)
colnames(muc.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
muc.otu.tab <- otu_table(muc.otu, taxa_are_rows=T)
muc.tax <- read.csv('mucin/mucinClassification.csv', row.names=1)
muc.tax <- tax_table(muc.tax)
colnames(muc.tax) <- c("GH_Class", "GH_Family", "EC1", "EC2", "EC3", "EC4", "FullName")
taxa_names(muc.tax) <- muc.tax[,7]
meta <- read.csv('dbcan/dbcan.spades.metadata.csv', row.names=1)
meta.ord <- meta


# Create phyloseq object
mucin.phy <- merge_phyloseq(muc.otu.tab, muc.tax, sample_data(meta.ord))

# Remove CAZymes w/ count < 4
# 4 to stay consistent with how I trimmed all CAZymes
mucin.cut.phy <- prune_taxa(taxa_sums(mucin.phy) > 3, mucin.phy)

# Identify trimmed CAZymes (count < 4)
#all <- rownames(tax_table(mucin.phy))
#remain <- rownames(tax_table(mucin.cut.phy))
#for (i in all){
#  if (i %in% remain == F){
#    print(i)
#  }
#}
# GH16_3.2.1.23, GH16_3.2.1.103, GH4_3.2.1.24, 3.1.1.2, 3.1.6.3, 3.1.6.4, 3.1.6.8, 3.1.6.14


# Normalized Counts
#muc.cut.otu <- data.frame(otu_table(mucin.cut.phy))
#mucin.dds <- DESeqDataSetFromMatrix(countData = muc.cut.otu, colData = meta, design = ~ Season)
#mucin.dds <- estimateSizeFactors(mucin.dds)
#mucin.norm <- counts(mucin.dds, normalized=T)
#write.csv(mucin.norm, file = "mucin/mucin.normalized.counts.csv")
mucin.norm <- read.csv('mucin/mucin.normalized.counts.csv', row.names=1)
muc.norm.otu <- otu_table(mucin.norm, taxa_are_rows=T)
sample_names(muc.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
muc.norm.phy <- merge_phyloseq(muc.norm.otu, sample_data(mucin.phy), tax_table(mucin.phy))

```
### Mucin - Count unique mucin-deg genes
```{r}
# Subset out samples by season
phy.sum <- subset_samples(mucin.cut.phy, Season == "Summer")
phy.win <- subset_samples(mucin.cut.phy, Season == "Winter")
phy.spr <- subset_samples(mucin.cut.phy, Season == "Spring")

# Remove annotations with no counts
phy.sum <- prune_taxa(taxa_sums(phy.sum) > 0, phy.sum)
phy.win <- prune_taxa(taxa_sums(phy.win) > 0, phy.win)
phy.spr <- prune_taxa(taxa_sums(phy.spr) > 0, phy.spr)


# Pull out unique muc-deg gene annotations in each season and order alphabetically
sum.gene <- get_taxa_unique(phy.sum, "FullName")
sum.gene <- sort(sum.gene)
win.gene <- get_taxa_unique(phy.win, "FullName")
win.gene <- sort(win.gene)
spr.gene <- get_taxa_unique(phy.spr, "FullName")
spr.gene <- sort(spr.gene)

# Count unique annotations
length(sum.gene)
  # 20
length(win.gene)
  # 21
length(spr.gene)
  #21

```
### Mucin - Kaiju
```{r}
########## Phylum ##########
muc.phy.sum <- import.phylum.sum('kaiju/mucin_all/phylum',7)

# Summarize phyla to get top 10
muc.phy.sum.sum <- muc.phy.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
muc.cla.sum <- import.class.sum('kaiju/mucin_all/class', 11)

# Summarize classes to get top 10
muc.cla.sum.sum <- muc.cla.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
muc.ord.sum <- import.order.sum('kaiju/mucin_all/order', 14)

# Summarize orders to get top 10
muc.ord.sum.sum <- muc.ord.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
muc.fam.sum <- import.family.sum('kaiju/mucin_all/family', 27)

# Summarize families to get top 10
muc.fam.sum.sum <- muc.fam.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
muc.gen.sum <- import.genus.sum('kaiju/mucin_all/genus',35)

# Summarize genera to get top 10
muc.gen.sum.sum <- muc.gen.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 11 genera (excluding unclassified, but using the % calculated with unclassified
top.muc.genus <- c("Bacteroides", "Prevotella", "Parabacteroides", "Clostridium", "Alistipes", "Roseburia", "Blautia", "Paenibacillus", "Faecalibacterium", "Ruminococcus", "Akkermansia")



# Pull out top 11 genera
muc.gen.total <- data.frame(matrix(nrow=33, ncol=5))
colnames(muc.gen.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.muc.genus)){
  top <- which(muc.gen.sum$Genus == top.muc.genus[i])
  for (j in 1:length(top)){
    muc.gen.total$Season[(j + 3*(i-1))] <- as.character(muc.gen.sum$Season[top[j]])
    muc.gen.total$Genus[(j + 3*(i-1))] <- muc.gen.sum$Genus[top[j]]
    muc.gen.total$Mean[(j + 3*(i-1))] <- muc.gen.sum$Mean[top[j]]
    muc.gen.total$SD[(j + 3*(i-1))] <- muc.gen.sum$SD[top[j]]
    muc.gen.total$SE[(j + 3*(i-1))] <- muc.gen.sum$SE[top[j]]
  }
}
muc.gen.total$Genus <- ordered(muc.gen.total$Genus, levels=c("Bacteroides", "Prevotella", "Parabacteroides", "Clostridium", "Alistipes", "Roseburia", "Blautia", "Paenibacillus", "Faecalibacterium", "Ruminococcus", "Akkermansia"))
muc.gen.total$Season <- ordered(muc.gen.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 11 genera
ggplot(muc.gen.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-5,45)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush", "lightpink")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 5 genera - All seasons combined
muc.gen.combined <- muc.gen.total %>%
  group_by(Genus) %>%
    summarize(Mean = mean(Mean),
              SD = sd(Mean),
              SE = se(Mean))
muc.gen.combined$SD[1] = sd(muc.gen.total$Mean[1:3])
muc.gen.combined$SD[2] = sd(muc.gen.total$Mean[4:6])
muc.gen.combined$SD[3] = sd(muc.gen.total$Mean[7:9])
muc.gen.combined$SD[4] = sd(muc.gen.total$Mean[10:12])
muc.gen.combined$SD[5] = sd(muc.gen.total$Mean[13:15])
muc.gen.combined$SD[6] = sd(muc.gen.total$Mean[16:18])
muc.gen.combined$SD[7] = sd(muc.gen.total$Mean[19:21])
muc.gen.combined$SD[8] = sd(muc.gen.total$Mean[22:24])
muc.gen.combined$SD[9] = sd(muc.gen.total$Mean[25:27])
muc.gen.combined$SD[10] = sd(muc.gen.total$Mean[28:30])
muc.gen.combined$SD[11] = sd(muc.gen.total$Mean[31:33])
muc.gen.combined$SE[1] = (sd(muc.gen.total$Mean[1:3])/3)
muc.gen.combined$SE[2] = (sd(muc.gen.total$Mean[4:6])/3)
muc.gen.combined$SE[3] = (sd(muc.gen.total$Mean[7:9])/3)
muc.gen.combined$SE[4] = (sd(muc.gen.total$Mean[10:12])/3)
muc.gen.combined$SE[5] = (sd(muc.gen.total$Mean[13:15])/3)
muc.gen.combined$SE[6] = (sd(muc.gen.total$Mean[16:18])/3)
muc.gen.combined$SE[7] = (sd(muc.gen.total$Mean[19:21])/3)
muc.gen.combined$SE[8] = (sd(muc.gen.total$Mean[22:24])/3)
muc.gen.combined$SE[9] = (sd(muc.gen.total$Mean[25:27])/3)
muc.gen.combined$SE[10] = (sd(muc.gen.total$Mean[28:30])/3)
muc.gen.combined$SE[11] = (sd(muc.gen.total$Mean[31:33])/3)
  
# Barplot for top 5 genera - All seasons combined
tiff("muc.gen.tiff", width=9, height=5, units="in", res= 300)
ggplot(muc.gen.combined, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,26), breaks=c(0,6.5,13,19.5,26)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush", "lightpink")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance (%)") +
  ggtitle("Mucin-Degradation Genes Genus-level Classification") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"),
        plot.title=element_text(hjust=0.5, face="bold"),
        axis.text.x = element_text(angle=30, hjust=1))
dev.off()


########## Species ##########
muc.spe.sum <- import.species.sum('kaiju/mucin_all/species',87)

# Summarize species to get top 10
muc.spe.sum.sum <- muc.spe.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```
### Mucin - Enzyme bar plots
```{r}
# Summarize data
mucin.df <- psmelt(muc.ra.phy)
mucin.sum <- mucin.df %>%
  group_by(FullName, Season) %>%
  summarize(Mean = mean(Abundance),
            SD= sd(Abundance),
            SE = se(Abundance))
mucin.sum$FullName = ordered(mucin.sum$FullName, c("GH2_3.2.1.23", "GH2_3.2.1.25", "GH3_3.2.1.52", "GH4_3.2.1.22", "GH20_3.2.1.52", "GH27_3.2.1.22", "GH29_3.2.1.51", "GH31_3.2.1.20", "GH35_3.2.1.23", "GH36_3.2.1.22", "GH38_3.2.1.24", "GH42_3.2.1.23", "GH67_3.2.1.139", "GH84_3.2.1.169", "GH95_3.2.1.51", "GH101_3.2.1.97", "GH110_3.2.1.22", "GH112_2.4.1.211", "GH130_2.4.1.281", "GH130_2.4.1.320", "4.1.3.3"))
mucin.sum$Season = ordered(mucin.sum$Season, levels=c("Summer", "Winter", "Spring"))


# Plot barplot
#tiff("mucin.family.tiff", width=33, height=7, units="in", res= 300)
ggplot(mucin.sum, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,2e-2)) +
  facet_grid(. ~ FullName) +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab("Mucin-Degrading Gene Classification") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"),
        panel.spacing = unit(0.5, "lines"),
        axis.title.x = element_text(margin = margin(t = 15)))
#dev.off()

```

### Mucin - Create DESeq Object
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
# https://compbiocore.github.io/deseq-workshop-1/assets/deseq_workshop_1.html
# Use UN-NORMALIZED data!!!
  # DESeq accounts for library size


### Test DESeq2 to determine fit type
# https://support.bioconductor.org/p/81094/
#mucin.dds <- phyloseq_to_deseq2(mucin.cut.phy, ~Season)
#mucin.dds$Season <- factor(mucin.dds$Season, levels=c("Summer", "Winter", "Spring"))
#mucin.dds <- mucin.dds[rowSums(counts(mucin.dds)) > 1,]
#mucin.local <- DESeq(mucin.dds, fitType = "local")
#mucin.par <- DESeq(mucin.dds, fitType = "parametric")
#mad(log(mcols(mucin.local)$dispGeneEst) - log(mcols(mucin.local)$dispFit))
  # 0.8548652
#mad(log(mcols(mucin.par)$dispGeneEst) - log(mcols(mucin.par)$dispFit))
  # 0.8548652
# No difference in fit type, but DESeq says local is better



### Convert phyloseq object to deseq2 object
mucin.dds <- phyloseq_to_deseq2(mucin.cut.phy, ~Season)
  # Design variable must be unordered otherwise you get an error
mucin.dds$Season <- factor(mucin.dds$Season, levels=c("Summer", "Winter", "Spring"))
mucin.dds <- mucin.dds[rowSums(counts(mucin.dds)) > 1,]


# DESeq DataStructure for analyses
mucin.dds <- DESeq(mucin.dds)
  # DESeq choose fitType = local

# vst-transformed data for visualizations
muc.vsd <- varianceStabilizingTransformation(mucin.dds)
```

### Mucin - DESeq - find overrep genes
```{r}
# Pull out results from pairwise comparison
mucin.sw <- results(mucin.dds, contrast=c("Season", "Summer", "Winter"))
mucin.sw.sig <- mucin.sw[which(mucin.sw$padj < 0.05),]
  # None
mucin.sp <- results(mucin.dds, contrast=c("Season", "Summer", "Spring"))
mucin.sp.sig <- mucin.sp[which(mucin.sp$padj < 0.05),]
  # Enriched in Spring compared to Summer (negative log2FoldChange)
    # GH112_2.4.1.211
mucin.wp <- results(mucin.dds, contrast=c("Season", "Winter", "Spring"))
mucin.wp.sig <- mucin.wp[which(mucin.wp$padj < 0.05),]
mucin.wp.sig[which(mucin.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # GH110_3.2.1.22, GH84_3.2.1.169
mucin.wp.sig[which(mucin.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH36_3.2.1.22, GH42_3.2.1.23
```



### Mucin - DESeq - Diet
```{r}
muc.diet.dds <- phyloseq_to_deseq2(mucin.cut.phy, ~Diet)
muc.diet.dds <- DESeq(muc.diet.dds)
muc.diet.results <- results(muc.diet.dds)
muc.diet.sig <- muc.diet.results[which(muc.diet.results$padj < 0.05),]
muc.diet.sig[which(muc.diet.sig$log2FoldChange > 0),]
  # Enriched in feeding compared to fasting
  # GH36_3.2.1.22
```
### Mucin - Heatmap of normalized counts
```{r}
mucin.sig <- subset_taxa(muc.norm.phy, FullName == "GH112_2.4.1.211" | FullName == "GH36_3.2.1.22" |
                           FullName == "GH42_3.2.1.23" | FullName == "GH84_3.2.1.169" |
                           FullName == "GH110_3.2.1.22")
                           
  


#tiff("mucin.heatmap.tiff", width=15, height=7, units="in", res=300)
plot_heatmap(mucin.sig, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("GH110_3.2.1.22", "GH84_3.2.1.169", "GH42_3.2.1.23", "GH36_3.2.1.22", "GH112_2.4.1.211")) +
  geom_tile(colour="gray40") +
  geom_text(aes(label=sprintf("%0.2f", round(Abundance, digits=4))), size = 7) +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("Mucin-Degrading Enzyme Gene\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"), limits=c(0, 100),
                       breaks=c(0, 25, 50, 75, 100), 
                       labels=c("0", "25", "50", "75", "100"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(5,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
#dev.off()


```






### Mucin - DESeq sample dist heatmap / dendrogram
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#exploring-and-exporting-results
# http://bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

muc.dist <- dist(t(assay(muc.vsd)))
muc.dist.matrix <- as.matrix(muc.dist)
rownames(muc.dist.matrix) <- muc.vsd$Season
colnames(muc.dist.matrix) <- muc.vsd$Season

# Heatmap
pheatmap(muc.dist.matrix, clustering_distance_rows = muc.dist, clustering_distance_cols = muc.dist,
         col = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c", "#fc4e2a", "#e31a1c",
                 "#bd0026", "#800026"))

# Dendrogram
muc.hclust <- hclust(muc.dist)
muc.den <- as.dendrogram(muc.hclust)
muc.den.data <- dendro_data(muc.den, type="rectangle")
muc.den.data$labels[,3] <- c("Spring", "Spring", "Summer", "Summer", "Summer", "Spring", "Winter", "Winter", "Winter")
muc.den.data$labels[,3] <- ordered(muc.den.data$labels[,3], levels=c("Summer", "Winter", "Spring"))

#tiff("mucin.dendro.tiff", width=10, height=5, units="in", res=300)
ggplot(muc.den.data$segments) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  geom_text(data = muc.den.data$labels, aes(x, y, label=label, colour=label, fontface = "bold"),
            hjust=0, size=5) +
  ylab("Euclidean Distance") +
  xlab("Season") +
  labs(colour = "Season") +
  coord_flip() +
  scale_y_reverse() +
  scale_color_manual(values=c("darkgoldenrod3", "dodgerblue3", "green4")) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_text(size=20),
        panel.border = element_blank(),
        axis.line.x = element_line(),
        legend.title = element_text(face="bold"),
        axis.title.x = element_text(face="bold", size=20),
        axis.title.y = element_text(face="bold", size=20))
#dev.off()

```

### Mucin - DESeq sample dist PCA & PERMANOVA
```{r}
muc.pca.data <- plotPCA(muc.vsd, intgroup="Season", returnData=T)
muc.pca.perc <- round(100*attr(muc.pca.data, "percentVar"))
# PCA variance explained = 71%

#tiff("mucin.pca.tiff", width=8, height=6, units="in", res=300)
ggplot(muc.pca.data, aes(PC1, PC2, color=Season)) +
  geom_point(size=5) +
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab(paste0("PC1: ", muc.pca.perc[1], "% variance")) +
  ylab(paste0("PC2: ", muc.pca.perc[2], "% variance")) +
  ggtitle("Euclidean Distance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20),
        legend.text = element_text(size=18),
        plot.title = element_text(hjust=0.5, size=20))
#dev.off()


### Test normality
hist(muc.dist)
qqnorm(muc.dist)
qqline(muc.dist)
shapiro.test(muc.dist)
  # p-value = 0.2705
  # Normal


### Run PERMANOVA on DESEq Euclidean distance
muc.dist <- dist(t(assay(muc.vsd)))
adonis(muc.dist ~ Season, data=meta, permutations = 99999)
  # p-value = 0.00366
beta.tax = betadisper(d=muc.dist, group=meta$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.812



### Pairwise PERMANOVA Comparisons
muc.dist.matrix <- as.matrix(muc.dist)
muc.dist.df <- data.frame(muc.dist.matrix)

# Summer vs. Winter
muc.sw.dist.df <- muc.dist.df[,-5]
muc.sw.dist.df <- muc.sw.dist.df[,-7:-8]
muc.sw.dist.df <- muc.sw.dist.df[-5,]
muc.sw.dist.df <- muc.sw.dist.df[-7:-8,]
muc.sw.dist <- as.dist(muc.sw.dist.df)
meta.sw <- meta[-which(meta$Season == "Spring"),]
adonis(muc.sw.dist ~ Season, data=meta.sw, permutations= 99999)
  # p-value = 0.1

# Summer vs. Spring
muc.sp.dist.df <- muc.dist.df[,-3:-4]
muc.sp.dist.df <- muc.sp.dist.df[,-5]
muc.sp.dist.df <- muc.sp.dist.df[-3:-4,]
muc.sp.dist.df <- muc.sp.dist.df[-5,]
muc.sp.dist <- as.dist(muc.sp.dist.df)
meta.sp <- meta[-which(meta$Season == "Winter"),]
adonis(muc.sp.dist ~ Season, data=meta.sp, permutations= 99999)
  # p-value = 0.1

# Spring vs. Winter
muc.wp.dist.df <- muc.dist.df[,-3:-4]
muc.wp.dist.df <- muc.wp.dist.df[,-5]
muc.wp.dist.df <- muc.wp.dist.df[-3:-4,]
muc.wp.dist.df <- muc.wp.dist.df[-5,]
muc.wp.dist <- as.dist(muc.wp.dist.df)
meta.wp <- meta[-which(meta$Season == "Summer"),]
adonis(muc.wp.dist ~ Season, data=meta.wp, permutations= 99999)
  # p-value = 0.1


### Pairwise Welch's T-Test Comparisons

# Summer vs. Winter
meta.sw$Season <- ordered(meta.sw$Season, levels=c("Summer","Winter"))
WT.test(muc.sw.dist, meta.sw$Season)
  # p-value = 0.107

# Summer vs. Spring
meta.sp$Season <- ordered(meta.sp$Season, levels=c("Summer","Spring"))
WT.test(muc.sp.dist, meta.sp$Season)
  # P-value = 0.099

# Winter vs. Spring
meta.wp$Season <- ordered(meta.wp$Season, levels=c("Winter","Springr"))
WT.test(muc.sp.dist, meta.sp$Season)
  # p-value = 0.098
```
### Mucin - Family-Level for each mucin enzyme
```{r}
# I chose Family-level because that was the taxon that gave the most taxonomic resolution without too many being unclassified

### 4.1.3.3
fam.4.1.3.3.sum <- import.family.sum('kaiju/mucin_sep/4.1.3.3/family',2)
# Summarize families
fam.4.1.3.3.sum.sum <- fam.4.1.3.3.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH2_3.2.1.23
fam.GH2_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH2_3.2.1.23/family',10)
# Summarize families
fam.GH2_3.2.1.23.sum.sum <- fam.GH2_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH2_3.2.1.25
fam.GH2_3.2.1.25.sum <- import.family.sum('kaiju/mucin_sep/GH2_3.2.1.25/family',3)
# Summarize families
fam.GH2_3.2.1.25.sum.sum <- fam.GH2_3.2.1.25.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH3_3.2.1.52
# Not found in some samples

### GH4_3.2.1.22
# Not found in some samples

### GH20_3.2.1.52
# Not found in some samples

### GH27_3.2.1.22
fam.GH27_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH27_3.2.1.22/family',4)
# Summarize families
fam.GH27_3.2.1.22.sum.sum <- fam.GH27_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH29_3.2.1.51
fam.GH29_3.2.1.51.sum <- import.family.sum('kaiju/mucin_sep/GH29_3.2.1.51/family',7)
# Summarize families
fam.GH29_3.2.1.51.sum.sum <- fam.GH29_3.2.1.51.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH31_3.2.1.20
fam.GH31_3.2.1.20.sum <- import.family.sum('kaiju/mucin_sep/GH31_3.2.1.20/family',6)
# Summarize families
fam.GH31_3.2.1.20.sum.sum <- fam.GH31_3.2.1.20.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH35_3.2.1.23
fam.GH35_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH35_3.2.1.23/family',5)
# Summarize families
fam.GH35_3.2.1.23.sum.sum <- fam.GH35_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH36_3.2.1.22
fam.GH36_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH36_3.2.1.22/family',13)
# Summarize families
fam.GH36_3.2.1.22.sum.sum <- fam.GH36_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH38_3.2.1.24
# Not found in some samples

### GH42_3.2.1.23
fam.GH42_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH42_3.2.1.23/family',4)
# Summarize families
fam.GH42_3.2.1.23.sum.sum <- fam.GH42_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH67_3.2.1.139
fam.GH67_3.2.1.139.sum <- import.family.sum('kaiju/mucin_sep/GH67_3.2.1.139/family',7)
# Summarize families
fam.GH67_3.2.1.139.sum.sum <- fam.GH67_3.2.1.139.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH84_3.2.1.169
fam.GH84_3.2.1.169.sum <- import.family.sum('kaiju/mucin_sep/GH84_3.2.1.169/family',10)
# Summarize families
fam.GH84_3.2.1.169.sum.sum <- fam.GH84_3.2.1.169.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH95_3.2.1.51
fam.GH95_3.2.1.51.sum <- import.family.sum('kaiju/mucin_sep/GH95_3.2.1.51/family',7)
# Summarize families
fam.GH95_3.2.1.51.sum.sum <- fam.GH95_3.2.1.51.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH101_3.2.1.97
# Not found in some samples

### GH110_3.2.1.22
fam.GH110_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH110_3.2.1.22/family',6)
# Summarize families
fam.GH110_3.2.1.22.sum.sum <- fam.GH110_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH112_2.4.1.211
fam.GH112_2.4.1.211.sum <- import.family.sum('kaiju/mucin_sep/GH112_2.4.1.211/family',6)
# Summarize families
fam.GH112_2.4.1.211.sum.sum <- fam.GH112_2.4.1.211.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH130_2.4.1.281
fam.GH130_2.4.1.281.sum <- import.family.sum('kaiju/mucin_sep/GH130_2.4.1.281/family',8)
# Summarize families
fam.GH130_2.4.1.281.sum.sum <- fam.GH130_2.4.1.281.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH130_2.4.1.320
fam.GH130_2.4.1.320.sum <- import.family.sum('kaiju/mucin_sep/GH130_2.4.1.320/family',7)
# Summarize families
fam.GH130_2.4.1.320.sum.sum <- fam.GH130_2.4.1.320.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))
```





### COG - Normalize by Universal Single-Copy Genes (Don't run!)
```{r}
# Import raw COG count table
cog.counts <- read.csv('cog/COG.table.tab', header = T, sep = "\t")
rownames(cog.counts) <- cog.counts$COG
cog.counts <- cog.counts[,-1]
colnames(cog.counts) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Prepare universal single-copy COGs
cog.usg <- c("COG0012", "COG0016", "COG0048", "COG0049", "COG0052", "COG0080", "COG0081", "COG0085", "COG0087", "COG0088", "COG0090", "COG0091", "COG0092", "COG0093", "COG0094", "COG0096", "COG0097", "COG0098", "COG0099", "COG0100", "COG0102", "COG0103", "COG0124", "COG0184", "COG0185", "COG0186", "COG0197", "COG0200", "COG0201", "COG0256", "COG0495", "COG0522", "COG0525", "COG0533", "COG0541")
cog.usg.df <- data.frame(matrix(NA, nrow = length(cog.usg), ncol = 9))
rownames(cog.usg.df) <- cog.usg
colnames(cog.usg.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Pull out universal single-copy COGs
for(i in 1:nrow(cog.counts)){
  match <- sum(which(cog.usg == rownames(cog.counts)[i]))
  
  if (match > 0){
    rowNum <- which(rownames(cog.usg.df) == rownames(cog.counts)[i])
    cog.usg.df[rowNum,] <- cog.counts[i,]
  }
}

# Average universal single-copy COGs
cog.usg.avg <- data.frame(matrix(nrow = 1, ncol = 9))
colnames(cog.usg.avg) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
for(i in 1:ncol(cog.usg.df)){
  usgNum <- nrow(cog.usg.df)
  sum <- sum(cog.usg.df[,i])
  avg <- sum/usgNum
  cog.usg.avg[,i] <- avg
}

# Normalize COG counts by dividing by avg universal single-copy COGs for each sample
cog.norm <- data.frame(matrix(nrow = nrow(cog.counts), ncol = ncol(cog.counts)))
rownames(cog.norm) <- rownames(cog.counts)
colnames(cog.norm) <- colnames(cog.counts)
for(i in 1:ncol(cog.counts)){
 cog.norm[,i] = cog.counts[,i] / cog.usg.avg[,i]
}

# Save normalized COG counts
#write.table(cog.norm, file = "cog/cog.normalized.tab", sep = "\t")

# Normalize COG category counts by dividing by avg universal single-copy COGs for each sample
cog.cat <- read.csv('cog/cog.category.csv', header = T)
rownames(cog.cat) <- cog.cat[,1]
cog.cat <- cog.cat[,-1]
colnames(cog.cat) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.cat.norm <- data.frame(matrix(nrow = 25, ncol = 9))
rownames(cog.cat.norm) <- rownames(cog.cat)
colnames(cog.cat.norm) <- colnames(cog.cat)
for(i in 1:ncol(cog.cat)){
 cog.cat.norm[,i] = cog.cat[,i] / cog.usg.avg[,i]
}

# Save normalized COG category counts
#write.csv(cog.cat.norm, file = "cog/cog.category.normalized.csv")

```

### COG Category - Kruskal-Wallis
```{r}
# Load data
cog.cat.norm <- read.csv('cog/cog.category.normalized.csv', header = T)
rownames(cog.cat.norm) <- cog.cat.norm[,1]
cog.cat.norm <- cog.cat.norm[,-1]
colnames(cog.cat.norm) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Format data
cog.cat.df <- data.frame(t(cog.cat.norm))
cog.cat.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
cog.cat.df$Season <- ordered(cog.cat.df$Season, levels=c("Summer", "Winter", "Spring"))

# Run Kruskal-Wallis on all COGs
cog.cat.kw <- data.frame(matrix(nrow = 1, ncol = nrow(cog.cat.norm)))
colnames(cog.cat.kw) <- rownames(cog.cat.norm)
for(i in 1:nrow(cog.cat.norm)){
  cogName <- colnames(cog.cat.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", cogName), data = cog.cat.df)
  cog.cat.kw[,i] <- kw$p.value
}
cog.cat.kw[2,] <- p.adjust(cog.cat.kw[1,], method = "fdr")
which(cog.cat.kw[2,] < 0.05)
```

### COG - Kruskal-Wallis
```{r}
# Load data
cog.norm <- read.csv('cog/cog.normalized.tab', header = T, sep = "\t")

# Identify COGs that appear in only 1 sample
cog.rem <- list()
for(i in 1:nrow(cog.norm)){
  empty <- length(which(cog.norm[i,] == 0))
    # Identifies number of samples without COG
  if(empty > 7){
    # If COG appears in only 1 sample
    cog.rem <- append(cog.rem, i)
      # Create list of COGs that appear in only 1 sample
  }
}
cog.rem.row <- as.numeric(cog.rem)
  # Convert list to numeric

# Remove COGs that appear in only 1 sample
cog.norm.rem <- cog.norm[-cog.rem.row,]

# Format data
cog.df <- data.frame(t(cog.norm.rem))
rownames(cog.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
cog.df$Season <- ordered(cog.df$Season, levels=c("Summer", "Winter", "Spring"))

# Run Kruskal-Wallis on all COGs
cog.kw <- data.frame(matrix(nrow = 1, ncol = nrow(cog.norm.rem)))
colnames(cog.kw) <- rownames(cog.norm.rem)
for(i in 1:nrow(cog.norm.rem)){
  cogName <- colnames(cog.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", cogName), data = cog.df)
  cog.kw[,i] <- kw$p.value
}
cog.kw[2,] <- p.adjust(cog.kw[1,], method = "fdr")
which(cog.kw[2,] < 0.05)

```

### KEGG - Kruskal-Wallis
```{r}
# Load data
kegg1 <- read.csv('KEGG/musicc_outputs/KEGG_MUSiCC_all1.tab', header=T, sep = "\t")
kegg2 <- read.csv('KEGG/musicc_outputs/KEGG_MUSiCC_all2.tab', header=T, sep = "\t")
kegg3 <- read.csv('KEGG/musicc_outputs/KEGG_MUSiCC_all3.tab', header=T, sep = "\t")
kegg4 <- read.csv('KEGG/musicc_outputs/KEGG_MUSiCC_all4.tab', header=T, sep = "\t")

# Format data
kegg1.df <- data.frame(t(kegg1))
colnames(kegg1.df) <- kegg1.df[1,]
kegg1.df <- kegg1.df[-1,]
rownames(kegg1.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
kegg1.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
kegg1.df$Season <- ordered(kegg1.df$Season, levels=c("Summer", "Winter", "Spring"))

kegg2.df <- data.frame(t(kegg2))
colnames(kegg2.df) <- kegg2.df[1,]
kegg2.df <- kegg2.df[-1,]
rownames(kegg2.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
kegg2.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
kegg2.df$Season <- ordered(kegg2.df$Season, levels=c("Summer", "Winter", "Spring"))

kegg3.df <- data.frame(t(kegg3))
colnames(kegg3.df) <- kegg3.df[1,]
kegg3.df <- kegg3.df[-1,]
rownames(kegg3.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
kegg3.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
kegg3.df$Season <- ordered(kegg3.df$Season, levels=c("Summer", "Winter", "Spring"))

kegg4.df <- data.frame(t(kegg4))
colnames(kegg4.df) <- kegg4.df[1,]
kegg4.df <- kegg4.df[-1,]
rownames(kegg4.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
kegg4.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
kegg4.df$Season <- ordered(kegg4.df$Season, levels=c("Summer", "Winter", "Spring"))


# Run Kruskal-Wallis on normalized KEGG counts
kegg1.kw <- data.frame(matrix(nrow = 1, ncol = nrow(kegg1)))
colnames(kegg1.kw) <- kegg1[,1]
for(i in 1:nrow(kegg1.df)){
  koid <- colnames(kegg1.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", koid), data = kegg1.df)
  kegg1.kw[,i] <- kw$p.value
}
kegg1.kw[2,] <- p.adjust(kegg1.kw[1,], method = "fdr")
which(kegg1.kw[2,] < 0.05)

kegg2.kw <- data.frame(matrix(nrow = 1, ncol = nrow(kegg2)))
colnames(kegg2.kw) <- kegg2[,1]
for(i in 1:nrow(kegg2.df)){
  koid <- colnames(kegg2.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", koid), data = kegg2.df)
  kegg2.kw[,i] <- kw$p.value
}
kegg2.kw[2,] <- p.adjust(kegg2.kw[1,], method = "fdr")
which(kegg2.kw[2,] < 0.05)

kegg3.kw <- data.frame(matrix(nrow = 1, ncol = nrow(kegg3)))
colnames(kegg3.kw) <- kegg3[,1]
for(i in 1:nrow(kegg3.df)){
  koid <- colnames(kegg3.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", koid), data = kegg3.df)
  kegg3.kw[,i] <- kw$p.value
}
kegg3.kw[2,] <- p.adjust(kegg3.kw[1,], method = "fdr")
which(kegg3.kw[2,] < 0.05)

kegg4.kw <- data.frame(matrix(nrow = 1, ncol = nrow(kegg4)))
colnames(kegg4.kw) <- kegg4[,1]
for(i in 1:nrow(kegg4.df)){
  koid <- colnames(kegg4.df)[i]
  kw <- kruskal.test(formula = reformulate("Season", koid), data = kegg4.df)
  kegg4.kw[,i] <- kw$p.value
}
kegg4.kw[2,] <- p.adjust(kegg4.kw[1,], method = "fdr")
which(kegg4.kw[2,] < 0.05)
```