---
title: "COG Code"
author: "Edna Chiang"
date: "2/9/2021"
output: html_document
---

### Load Libraries
```{r}
library('readr')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
library('phangorn')
library('vegan')
library('pheatmap')
library('RColorBrewer')
library('gplots')
library('tidyr')
library('dplyr')
library('stringr')
library('lme4')
library('picante')
library('ggdendro')
library('fitdistrplus')
library('amap')
library('labdsv')
library('NBPSeq')
theme_set(theme_bw())
set.seed(1)
```

### Load Functions
```{r}
source('scripts/misc.R')
source('scripts/simper.R')
source('scripts/pairwise.adonis.R')
```

### COG - Create phyloseq objects (DON'T RUN!)
```{r}
### Count-level ###

# Load data
cog.counts <- read.csv('cog/cog.counts.tab', header = T, sep = "\t", row.names = 1)


# Identify COGs that appear in only 1 sample
cog.rem <- list()
for(i in 1:nrow(cog.counts)){
  empty <- length(which(cog.counts[i,] == 0))
    # Identifies number of samples without COG
  if(empty > 7){
    # If COG appears in only 1 sample
    cog.rem <- append(cog.rem, i)
      # Create list of COGs that appear in only 1 sample
  }
}
cog.rem.row <- as.numeric(cog.rem)
  # Convert list to numeric


# Pull out removed COGs, their category classification, and their sole sample so I can update the COG category counts
cog.rem.names <- rownames(cog.counts[cog.rem.row,])
cog.cat <- read.csv('cog/cog.classifications.tab', header = T, sep = "\t")
cog.rem.cat <- data.frame(matrix(nrow = length(cog.rem.row), ncol = 3))
colnames(cog.rem.cat) <- c("COG", "Category", "Sample")
for(j in 1:length(cog.rem.names)){
  matchName = which(cog.cat[,1] == cog.rem.names[j])
  cog.rem.cat[j,1:2] <- cog.cat[matchName,1:2]
  
  matchSamp = which(rownames(cog.counts) == cog.rem.names[j])
  yes <- which(cog.counts[matchSamp,] != 0)
  cog.rem.cat[j,3] <- substr(colnames(cog.counts)[yes], 2, 5)
}

cog.rem.cat.counts <- data.frame(matrix(nrow = 25, ncol = 11))
colnames(cog.rem.cat.counts) <- c("Category", "Count", "3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.rem.cat.counts[,1] <- c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "Y", "Z")
cog.rem.cat.counts[,2:11] <- 0
for(k in 1:nrow(cog.rem.cat)){
  match <- length(which(grepl(cog.rem.cat[k,2], cog.rem.cat.counts[,1])))

  if(match > 0){
    matchRow <- which(grepl(cog.rem.cat[k,2], cog.rem.cat.counts[,1]))
    cog.rem.cat.counts[matchRow, 2] <- cog.rem.cat.counts[matchRow, 2] + 1
    sampCol <- which(colnames(cog.rem.cat.counts) == cog.rem.cat$Sample[k])
    cog.rem.cat.counts[matchRow, sampCol] <- cog.rem.cat.counts[matchRow, sampCol] + 1
  }
}


# Remove COGs that appear in only 1 sample
cog.rem <- cog.counts[-cog.rem.row,]
colnames(cog.rem) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Format data
cog.df <- data.frame(t(cog.rem))
rownames(cog.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
cog.df$Season <- ordered(cog.df$Season, levels=c("Summer", "Winter", "Spring"))

# Load COG "tax_table"
cog.cat <- read.csv('cog/cog.classifications.tab', header = T, sep = "\t")
cog.tax <- data.frame(cog.cat[,-2:-3])

# Prepare phyloseq object inputs
cog.otu <- otu_table(cog.rem, taxa_are_rows = T)
cog.tax <- tax_table(cog.tax)
taxa_names(cog.tax) <- cog.tax[,1]
meta <- read.csv('metadata.csv', row.names=1)

# Create phyloseq object
cog.phy <- merge_phyloseq(cog.otu, cog.tax, sample_data(meta))
colnames(tax_table(cog.phy)) <- "COG"
#save(cog.phy, file = "cog.phy.Rdata")



### Category-level ###

# Load data
cog.cat <- read.csv('cog/cog.category.csv', header = T, row.names = 1)
cog.cat.rem <- cog.rem.cat.counts[,3:11]
cog.cat.trim <- cog.cat - cog.cat.rem
cog.cat.otu <- otu_table(cog.cat.trim, taxa_are_rows = T)
colnames(cog.cat.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")


# Create phyloseq object
cog.cat.tax <- data.frame(rownames(cog.cat.trim))
rownames(cog.cat.tax) <- cog.cat.tax[,1]
colnames(cog.cat.tax) <- "Category"
cog.cat.tax <- tax_table(cog.cat.tax)
taxa_names(cog.cat.tax) <- cog.cat.tax[,1]
colnames(cog.cat.tax) <- "Category"
cog.cat.phy <- merge_phyloseq(cog.cat.otu, cog.cat.tax, sample_data(meta))
#save(cog.cat.phy, file = "cog.cat.phy.Rdata")



### Category - Normalized Counts ###
cog.cat.dds <- phyloseq_to_deseq2(cog.cat.phy, ~Season)
cog.cat.dds <- estimateSizeFactors(cog.cat.dds)
cog.cat.norm <- counts(cog.cat.dds, normalized = T)
cog.cat.norm.phy <- merge_phyloseq(otu_table(cog.cat.norm, taxa_are_rows = T), tax_table(cog.cat.phy), sample_data(cog.cat.phy))
#save(cog.cat.norm.phy, file = "cog.cat.norm.phy.Rdata")
```
### COG - Load physeq objects
```{r}
load("cog.phy.Rdata")
load("cog.cat.phy.Rdata")
load("cog.cat.norm.phy.Rdata")
meta <- read.csv('metadata.csv', row.names=1)
```


### COG - Calculate % ORFs classified to COG
```{r}
prodigalCounts <- read.csv('cog/prodigalCounts.csv', row.names = 1)
prodigalCounts <- as.matrix(prodigalCounts)

# Test normality
hist(prodigalCounts[3,])
qqnorm(prodigalCounts[3,])
qqline(prodigalCounts[3,])
shapiro.test(prodigalCounts[3,])
  # p-value = 0.108
  # Normal

# Format for ANOVA
prodigalCounts <- read.csv('cog/prodigalCounts.csv', row.names=1)
prodigalCounts <- t(prodigalCounts)
prodigalCounts <- data.frame(prodigalCounts)
prodigalCounts$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
prodigalCounts$Season <- ordered(prodigalCounts$Season, levels=c("Summer", "Winter", "Spring"))
prodigalCounts$Diet <- c("Feeding", "Feeding", "Fasting", "Fasting", "Feeding", "Feeding", "Fasting", "Feeding", "Feeding")
prodigalCounts$Diet <- ordered(prodigalCounts$Diet, levels=c("Feeding", "Fasting"))
colnames(prodigalCounts) <- c("Total_ORFs", "Total_COGs", "COGPercentage", "Season", "Diet")

# Test w/ ANOVA
summary(aov(COGPercentage ~ Season, data = prodigalCounts))
  # p-value = 0.634
summary(aov(COGPercentage ~ Diet, data = prodigalCounts))
  # p-value = 0.37

### Dot plot
#tiff("cog.prop.tiff", width=5, height=4, units="in", res= 300)
ggplot(prodigalCounts, aes(x=Season, y=COGPercentage, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as COG (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(37, 40, 43, 46, 49), limits=c(37, 49)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
#dev.off()
```
### COG - Number of unique COGs
```{r}
phy.sum <- subset_samples(cog.phy, Season == "Summer")
phy.win <- subset_samples(cog.phy, Season == "Winter")
phy.spr <- subset_samples(cog.phy, Season == "Spring")


# Count # unique COGs in each season
sum.cog <- data.frame(otu_table(phy.sum))
sum.rem <- which(rowSums(sum.cog) == 0)
sum.cog <- sum.cog[-sum.rem,]
nrow(sum.cog)
  # 3673
win.cog <- data.frame(otu_table(phy.win))
win.rem <- which(rowSums(win.cog) == 0)
win.cog <- win.cog[-win.rem,]
nrow(win.cog)
  # 3544
spr.cog <- data.frame(otu_table(phy.spr))
spr.rem <- which(rowSums(spr.cog) == 0)
spr.cog <- spr.cog[-spr.rem,]
nrow(spr.cog)
  # 3698

```
### COG - General COG Stats
```{r}
# Load data
cog.stats <- read.csv('cog/cog.stats.csv', row.names = 1, header = T)
cog.stats <- as.matrix(cog.stats)

# Test normality - % COGs from ORFs
hist(cog.stats[3,])
qqnorm(cog.stats[3,])
qqline(cog.stats[3,])
shapiro.test(cog.stats[3,])
  # p-value = 0.108
  # Normal

# Test normality - Unique COGs
hist(cog.stats[4,])
qqnorm(cog.stats[4,])
qqline(cog.stats[4,])
shapiro.test(cog.stats[4,])
  # p-value = 0.1712
  # Normal

# Format for ANOVA
cog.stats <- data.frame(t(cog.stats))
rownames(cog.stats) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.stats$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")

# Test w/ ANOVA
summary(aov(Prop_COGs ~ Season, data = cog.stats))
  # p-value = 0.634
summary(aov(Unique_COGs ~ Season, data = cog.stats))
  # p-value = 0.51

ggplot(cog.stats, aes(x=Season, y=Prop_COGs, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as COG (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(37, 40, 43, 46, 49), limits=c(37, 49)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))

ggplot(cog.stats, aes(x=Season, y=Unique_COGs, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("Unique COGs") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(3200, 3400, 3600), limits=c(3200, 3600)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
```

### COG Category - Create DESeq object
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
# https://compbiocore.github.io/deseq-workshop-1/assets/deseq_workshop_1.html
# https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html

### Convert phyloseq object to deseq2 object
cog.cat.dds <- phyloseq_to_deseq2(cog.cat.phy, ~Season)
  # Design variable must be unordered otherwise you get an error
cog.cat.dds$Season <- factor(cog.cat.dds$Season, levels=c("Summer", "Winter", "Spring"))

# DESeq DataStructure for analyses
cog.cat.dds <- DESeq(cog.cat.dds, fitType = 'local')

# Hellinger-transformed data for visualizations
cog.cat.hell <- hellinger(t(assay(cog.cat.dds)))
```

### COG Category - DESeq - find overrep genes
```{r}
# Pull out results from pairwise comparison
cog.cat.sw <- results(cog.cat.dds, contrast=c("Season", "Summer", "Winter"))
cog.cat.sw.sig <- cog.cat.sw[which(cog.cat.sw$padj < 0.05),]
  # None
cog.cat.sp <- results(cog.cat.dds, contrast=c("Season", "Summer", "Spring"))
cog.cat.sp.sig <- cog.cat.sp[which(cog.cat.sp$padj < 0.05),]
  # None
cog.cat.wp <- results(cog.cat.dds, contrast=c("Season", "Winter", "Spring"))
cog.cat.wp.sig <- cog.cat.wp[which(cog.cat.wp$padj < 0.05),]
cog.cat.wp.sig[which(cog.cat.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
      # M (Cell wall/membrane/envelope biogenesis)
cog.cat.wp.sig[which(cog.cat.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # G (carb transport & metab), K (transcription), T (signal transduction mechanisms), V (defense mechanisms)
```

### COG Category - Sig Category Dot Plot
```{r}
# Prepare data
cog.cat.norm.df <- psmelt(cog.cat.norm.phy)
m <- cog.cat.norm.df[which(cog.cat.norm.df$Category == "M"),]
g <- cog.cat.norm.df[which(cog.cat.norm.df$Category == "G"),]
k <- cog.cat.norm.df[which(cog.cat.norm.df$Category == "K"),]
t <- cog.cat.norm.df[which(cog.cat.norm.df$Category == "T"),]
v <- cog.cat.norm.df[which(cog.cat.norm.df$Category == "V"),]
cog.cat.norm.sig <- rbind(m,g,k,t,v)
cog.cat.norm.sig$Season <- ordered(cog.cat.norm.sig$Season,
                                   levels=c("Summer", "Winter", "Spring"))


# Create cat labels
glab <- expression("                 G\nCarbohydrate transport\n     and metabolism")
klab <- expression("         K\nTranscription\n")
mlab <- expression("              M\nCell wall/membrane/\nenvelope biogenesis")
tlab <- expression("              T\nSignal transduction\n     mechanisms")
vlab <- expression("               V\nDefense mechanisms\n")
newLab <- list("G" = glab, "K" = klab, "M" = mlab, "T" = tlab, "V" = vlab)
newLab_labeller <- function(variable,value){
  return(newLab[value])
}

#tiff("cog.cat.sig.tiff", width = 12, height = 4, units = "in", res = 300)
ggplot(cog.cat.norm.sig, aes(x = Season, y = Abundance, fill = Season)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5, stackratio = 1.15) +
  facet_grid(.~Category, labeller = newLab_labeller) +
  ylab("Normalized counts") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(6300, 10975, 15650, 20325, 25000),
                     limits=c(6300, 25000)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=12, margin = margin(t = 25)),
        strip.background = element_blank(),
        legend.title = element_text(size = 14,face="bold"),
        legend.text = element_text(size=12),
        axis.title.y = element_text(size=14, face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=14, face="bold"),
        axis.text.x = element_text(size = 12, angle = 15, hjust = 0.5, vjust = 0.9))
#dev.off()
```
### COG Category - DESeq heatmap of normalized + transformed counts
```{r}
# Prepare data
cog.cat.hell <- otu_table(t(cog.cat.hell), taxa_are_rows = T)
colnames(cog.cat.hell) <- c("3715", "3717", "3723", "3733", "3734", "3744",
                            "3772", "3773" , "3775")
cog.cat.all <- merge_phyloseq(cog.cat.hell, tax_table(cog.cat.phy), sample_data(cog.cat.phy))

# Rename COG categories to be informative
cog.cat.all.plot <- cog.cat.all
taxa_names(cog.cat.all.plot) <- c("A (RNA processing and modification)",
                                  "B (Chromatin structure and dynamics)",
                                  "C (Energy production and conversion)",
                                  "D (Cell cycle control, cell division, chromosome partitioning)",
                                  "E (Amino acid transport and metabolism)",
                                  "F (Nucleotide transport and metabolism)",
                                  "G (Carbohydrate transport and metabolism)",
                                  "H (Coenzyme transport and metabolism)",
                                  "I (Lipid transport and metabolism)", 
                                  "J (Translation, ribosomal structure, and biogenesis)", 
                                  "K (Transcription)",
                                  "L (Replication, recombination, and repair)",
                                  "M (Cell wall/membrane/envelope biogenesis)",
                                  "N (Cell motility)",
                                  "O (Posttranslational modification, protein turnover, chaperones)",
                                  "P (Inorganic ion transport and metabolism)",
                                  "Q (Secondary metabolites biosynthesis, transport, and catabolism)",
                                  "R (General function predicted only)",
                                  "S (Function unknown)",
                                  "T (Signal transduction mechanisms)",
                                  "U (Intracellular trafficking, secretion, and vesicular transport)",
                                  "V (Defense mechanisms)",
                                  "W (Extracellular structures)",
                                  "Y (Nuclear structure)",
                                  "Z (Cytoskeleton)")


#tiff("cog.cat.all.heatmap.tiff", width=10.25, height=9, units="in", res=300)
plot_heatmap(cog.cat.all.plot, sample.label = "Season",
             sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734",
                              "3773", "3775"),
             taxa.order = c("Z (Cytoskeleton)", "Y (Nuclear structure)",
                            "W (Extracellular structures)", "V (Defense mechanisms)",
                            "U (Intracellular trafficking, secretion, and vesicular transport)",
                            "T (Signal transduction mechanisms)",
                            "S (Function unknown)",
                            "R (General function predicted only)",
                            "Q (Secondary metabolites biosynthesis, transport, and catabolism)",
                            "P (Inorganic ion transport and metabolism)",
                            "O (Posttranslational modification, protein turnover, chaperones)",
                            "N (Cell motility)",
                            "M (Cell wall/membrane/envelope biogenesis)",
                            "L (Replication, recombination, and repair)",
                            "K (Transcription)",
                            "J (Translation, ribosomal structure, and biogenesis)",
                            "I (Lipid transport and metabolism)",
                            "H (Coenzyme transport and metabolism)",
                            "G (Carbohydrate transport and metabolism)",
                            "F (Nucleotide transport and metabolism)",
                            "E (Amino acid transport and metabolism)",
                            "D (Cell cycle control, cell division, chromosome partitioning)",
                            "C (Energy production and conversion)",
                            "B (Chromatin structure and dynamics)",
                            "A (RNA processing and modification)")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season") +
  ylab("COG Category\n") +
  scale_fill_gradientn(name=expression("Hellinger-\nTransformed\nNormalized\nCounts\n"), limits = c(0.00, 0.35), breaks = c(0.00, 0.07, 0.14, 0.21, 0.28, 0.35),
                       labels = c("0.00", "0.07", "0,14", "0.21", "0.28", "0.35"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976",
                                   "#feb24c", "#fd8d3c", "#fc4e2a", "#e31a1c",
                                   "#bd0026", "#800026", "#33000f")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12, color = "black"),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border = element_blank(),
        axis.title.x = element_text(size=14, face="bold"),
        axis.title.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size=12)) +
  guides(fill = guide_colorbar(barheight = unit(7,"in"),
                               frame.colour="black",
                               frame.linewidth=1)) +
  annotate("rect", xmin = 0.5, xmax = 3.5, ymin = 25.6, ymax = 26.5,
           fill="darkgoldenrod2") +
  annotate("text", x = 1.99, y = 26.07, label = "Summer", size = 5.5) +
  annotate("rect", xmin = 3.55, xmax = 6.5, ymin = 25.6, ymax = 26.5,
           fill="#5DA5ED") +
  annotate("text", x = 4.99, y = 26.07, label = "Winter", size = 5.5) +
  annotate("rect", xmin = 6.55, xmax = 9.5, ymin = 25.6, ymax = 26.5,
           fill="green3") +
  annotate("text", x = 7.99, y = 26.07, label = "Spring", size = 5.5)
#dev.off()
```


### COG Category - Clustering
```{r}
cog.cat.dist <- dist(cog.cat.hell, method = "euclidean")
cog.cat.dist.matrix <- as.matrix(cog.cat.dist)
rownames(cog.cat.dist.matrix) <- meta$Season
colnames(cog.cat.dist.matrix) <- meta$Season


# Dendrogram
cog.cat.hclust <- hclust(cog.cat.dist, method = "average")
cog.cat.den <- as.dendrogram(cog.cat.hclust)
cog.cat.den.data <- dendro_data(cog.cat.den, type = "rectangle")
cog.cat.den.data$labels[,3] <- c("Summer", "Winter", "Winter", "Winter", "Summer", "Spring", "Spring", "Spring", "Summer")
cog.cat.den.data$labels[,3] <- ordered(cog.cat.den.data$labels[,3], levels=c("Summer", "Winter", "Spring"))


#tiff("cog.cat.dendro.tiff", width=10, height=5, units="in", res=300)
ggplot(cog.cat.den.data$segments) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  geom_text(data = cog.cat.den.data$labels, aes(x, y, label=label, colour=label, fontface = "bold"),
            hjust=0, size=5) +
  ylab("Hellinger Distance") +
  xlab("Season") +
  labs(colour = "Season") +
  coord_flip() +
  scale_y_reverse() +
  scale_color_manual(values=c("darkgoldenrod3", "dodgerblue3", "green4")) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_text(size=20),
        panel.border = element_blank(),
        axis.line.x = element_line(),
        legend.title = element_text(face="bold"),
        axis.title.x = element_text(face="bold", size=20),
        axis.title.y = element_text(face="bold", size=20))
#dev.off()
```


### COG Category - PCoA & PERMANOVA

### COG Within-Category - Separate Phyloseq Object
```{r}
# Load COG classification data
cog.class <- read.csv('cog/cog.classifications.tab', header = T, sep = "\t", row.names = 1)

# Pull out COGs in each category
mCOGs <- rownames(cog.class)[which(cog.class$Category == "M")]
gCOGs <- rownames(cog.class)[which(cog.class$Category == "G")]
kCOGs <- rownames(cog.class)[which(cog.class$Category == "K")]
tCOGs <- rownames(cog.class)[which(cog.class$Category == "T")]
vCOGs <- rownames(cog.class)[which(cog.class$Category == "V")]

# Note: the length of each string of COGs != number of COGs in the below physeq objects
# This is because not all of the COGs in a category are found in our data
# To see what COGs are missing from our data, use this:
  # setdiff(mCOGs, data.frame(tax_table(m.phy))$COG)

# Create physeq object for COGs in each category
m.phy <- prune_taxa(mCOGs, cog.phy)
g.phy <- prune_taxa(gCOGs, cog.phy)
k.phy <- prune_taxa(kCOGs, cog.phy)
t.phy <- prune_taxa(tCOGs, cog.phy)
v.phy <- prune_taxa(vCOGs, cog.phy)
```


### COG Within-Category - DESeq
```{r}
### Convert phyloseq object to deseq2 object
m.dds <- phyloseq_to_deseq2(m.phy, ~Season)
g.dds <- phyloseq_to_deseq2(g.phy, ~Season)
k.dds <- phyloseq_to_deseq2(k.phy, ~Season)
t.dds <- phyloseq_to_deseq2(t.phy, ~Season)
v.dds <- phyloseq_to_deseq2(v.phy, ~Season)
m.dds$Season <- factor(m.dds$Season, levels = c("Summer", "Winter", "Spring"))
g.dds$Season <- factor(g.dds$Season, levels = c("Summer", "Winter", "Spring"))
k.dds$Season <- factor(k.dds$Season, levels = c("Summer", "Winter", "Spring"))
t.dds$Season <- factor(t.dds$Season, levels = c("Summer", "Winter", "Spring"))
v.dds$Season <- factor(v.dds$Season, levels = c("Summer", "Winter", "Spring"))

# DESeq DataStructure for analyses
m.dds <- DESeq(m.dds)
g.dds <- DESeq(g.dds)
k.dds <- DESeq(k.dds)
t.dds <- DESeq(t.dds)
v.dds <- DESeq(v.dds)


# Hellinger-transformed data for visualizations
m.hell <- hellinger(t(assay(m.dds)))
g.hell <- hellinger(t(assay(g.dds)))
k.hell <- hellinger(t(assay(k.dds)))
t.hell <- hellinger(t(assay(t.dds)))
v.hell <- hellinger(t(assay(v.dds)))
```


### COG Within-Category - DESeq - find overrep genes
```{r}
##### M #####
# Pull out results from pairwise comparison
m.sw <- results(m.dds, contrast = c("Season", "Summer", "Winter"))
m.sw.sig <- m.sw[which(m.sw$padj < 0.05),]
  # Enriched in Summer compared to Winter  (positive log2FoldChange)
    # COG5434
m.sp <- results(m.dds, contrast = c("Season", "Summer", "Spring"))
m.sp.sig <- m.sp[which(m.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
  # COG2825, COG2885
m.wp <- results(m.dds, contrast = c("Season", "Winter", "Spring"))
m.wp.sig <- m.wp[which(m.wp$padj < 0.05),]
m.wp.sig[which(m.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # COG0774, COG1452, COG2335, COG2885, COG2982, COG3203, COG3637, COG4775
m.wp.sig[which(m.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # COG1686, COG1887, COG3757, COG3944



##### G #####
# Pull out results from pairwise comparison
g.sw <- results(g.dds, contrast = c("Season", "Summer", "Winter"))
g.sw.sig <- g.sw[which(g.sw$padj < 0.05),]
  # Enriched in Summer compared to Winter (positive log2FoldChange)
    # COG1105, COG2730, COG3507, COG3664, COG3693, COG3866, COG4284, COG4677
g.sp <- results(g.dds, contrast = c("Season", "Summer", "Spring"))
g.sp.sig <- g.sp[which(g.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # COG2814
g.wp <- results(g.dds, contrast = c("Season", "Winter", "Spring"))
g.wp.sig <- g.wp[which(g.wp$padj < 0.05),]
g.wp.sig[which(g.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # COG0158, COG2814, COG3525, COG4409
g.wp.sig[which(g.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # COG1105, COG1175, COG1653, COG1925, COG2182, COG2190, COG2730, COG3664, COG3693, COG3866, COG4284, COG4813



##### K #####
# Pull out results from pairwise comparison
k.sw <- results(k.dds, contrast = c("Season", "Summer", "Winter"))
k.sw.sig <- k.sw[which(k.sw$padj < 0.05),]
  # Enriched in Summer compared to Winter  (positive log2FoldChange)
    # COG2207
k.sp <- results(k.dds, contrast = c("Season", "Summer", "Spring"))
k.sp.sig <- k.sp[which(k.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # COG3636
k.wp <- results(k.dds, contrast = c("Season", "Winter", "Spring"))
k.wp.sig <- k.wp[which(k.wp$padj < 0.05),]
k.wp.sig[which(k.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # COG3636
k.wp.sig[which(k.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # COG1609



##### T #####
# Pull out results from pairwise comparison
t.sw <- results(t.dds, contrast = c("Season", "Summer", "Winter"))
t.sw.sig <- t.sw[which(t.sw$padj < 0.05),]
  # None
t.sp <- results(t.dds, contrast = c("Season", "Summer", "Spring"))
t.sp.sig <- t.sp[which(t.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # COG0589
t.wp <- results(t.dds, contrast = c("Season", "Winter", "Spring"))
t.wp.sig <- t.wp[which(t.wp$padj < 0.05),]
  # None




##### V #####
# Pull out results from pairwise comparison
v.sw <- results(v.dds, contrast = c("Season", "Summer", "Winter"))
v.sw.sig <- v.sw[which(v.sw$padj < 0.05),]
  # Enriched in Winter compared to Summer (negative log2FoldChange)
    # COG0610
v.sp <- results(v.dds, contrast = c("Season", "Summer", "Spring"))
v.sp.sig <- v.sp[which(v.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # COG0842, COG3023
v.wp <- results(v.dds, contrast = c("Season", "Winter", "Spring"))
v.wp.sig <- v.wp[which(v.wp$padj < 0.05),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # COG1132
```

### COG Within Category - DESeq heatmap of normalized + transformed counts
```{r}
##### M #####
m.hell <- otu_table(t(m.hell), taxa_are_rows = T)
colnames(m.hell) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773",
                      "3775")
m.hell.all <- merge_phyloseq(m.hell, tax_table(m.phy), sample_data(m.phy))
m.sig <- c("COG2825", "COG2885", "COG5434", "COG1686", "COG1887", "COG3757", "COG3944",
           "COG0774", "COG1452", "COG2335", "COG2982", "COG3203", "COG3637", "COG4775")
m.hell.sig <- prune_taxa(m.sig, m.hell.all)



#tiff("cog.m.heatmap.tiff", width=8, height=9, units="in", res=300)
plot_heatmap(m.hell.sig, sample.label = "Season",
             sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734",
                              "3773", "3775"),
             taxa.order = c("COG4775", "COG3637", "COG3203", "COG2982", "COG2335",
                            "COG1452", "COG0774", "COG3944", "COG3757", "COG1887",
                            "COG1686", "COG5434", "COG2885", "COG2825")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  scale_y_discrete(labels = c("COG4775 ()", "COG3637 ()", "COG3203 ()", "COG2982 ()", "COG2335 ()",
                            "COG1452 ()", "COG0774 ()", "COG3944 ()", "COG3757 ()", "COG1887 ()",
                            "COG1686 ()", "COG5434 ()", "COG2885 ()", "COG2825 ()")
                     
                     
                     
                     
                     c("1","2","3","4","5","6","7","8","9","10","11","12","13","14")) +
  xlab("Season") +
  ylab("Significant COGs in Category M\n") +
  scale_fill_gradientn(name=expression("Hellinger-\nTransformed\nNormalized\nCounts\n"), limits = c(0.00, 0.16), breaks = c(0.00, 0.04, 0.08, 0.12, 0.16),
                       labels = c("0.00", "0.04", "0.08", "0.12", "0.16"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976",
                                   "#feb24c", "#fd8d3c", "#fc4e2a", "#e31a1c",
                                   "#bd0026", "#800026", "#33000f")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12, color = "black"),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border = element_blank(),
        axis.title.x = element_text(size=14, face="bold"),
        axis.title.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size=12)) +
  guides(fill = guide_colorbar(barheight = unit(7,"in"),
                               frame.colour="black",
                               frame.linewidth=1)) +
  annotate("rect", xmin = 0.5, xmax = 3.5, ymin = 14.6, ymax = 15.5,
           fill="darkgoldenrod2") +
  annotate("text", x = 1.99, y = 15.07, label = "Summer", size = 5.5) +
  annotate("rect", xmin = 3.55, xmax = 6.5, ymin = 14.6, ymax = 15.5,
           fill="#5DA5ED") +
  annotate("text", x = 4.99, y = 15.07, label = "Winter", size = 5.5) +
  annotate("rect", xmin = 6.55, xmax = 9.5, ymin = 14.6, ymax = 15.5,
           fill="green3") +
  annotate("text", x = 7.99, y = 15.07, label = "Spring", size = 5.5)
#dev.off()







##### G #####
g.hell <- otu_table(t(g.hell), taxa_are_rows = T)
colnames(g.hell) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773",
                      "3775")
g.hell.all <- merge_phyloseq(g.hell, tax_table(g.phy), sample_data(g.phy))
g.sig <- c("COG2814", "COG1105", "COG2730", "COG3507", "COG3664", "COG3693", "COG3866",
           "COG4284", "COG4677", "COG1175", "COG1653" , "COG1925", "COG2182",
           "COG2190", "COG4813", "COG0158", "COG3525", "COG4409")
g.hell.sig <- prune_taxa(g.sig, g.hell.all)





tiff("cog.g.heatmap.tiff", width=8, height=9, units="in", res=300)
plot_heatmap(g.hell.sig, sample.label = "Season",
             sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734",
                              "3773", "3775"),
             taxa.order = c("COG4409", "COG3525", "COG0158", "COG4813", "COG2190",
                            "COG2182", "COG1925", "COG1653", "COG1175", "COG4677",
                            "COG4284", "COG3866", "COG3693", "COG3664", "COG3507",
                            "COG2730", "COG1105", "COG2814")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  scale_y_discrete(labels = c()) +
  xlab("Season") +
  ylab("Significant COGs in Category G\n") +
  scale_fill_gradientn(name=expression("Hellinger-\nTransformed\nNormalized\nCounts\n"), limits = c(0.00, 0.16), breaks = c(0.00, 0.04, 0.08, 0.12, 0.16),
                       labels = c("0.00", "0.04", "0.08", "0.12", "0.16"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976",
                                   "#feb24c", "#fd8d3c", "#fc4e2a", "#e31a1c",
                                   "#bd0026", "#800026", "#33000f")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12, color = "black"),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border = element_blank(),
        axis.title.x = element_text(size=14, face="bold"),
        axis.title.y = element_text(size=14, face="bold"),
        legend.title = element_text(size=14, face="bold"),
        legend.text = element_text(size=12)) +
  guides(fill = guide_colorbar(barheight = unit(7,"in"),
                               frame.colour="black",
                               frame.linewidth=1)) +
  annotate("rect", xmin = 0.5, xmax = 3.5, ymin = 14.6, ymax = 15.5,
           fill="darkgoldenrod2") +
  annotate("text", x = 1.99, y = 15.07, label = "Summer", size = 5.5) +
  annotate("rect", xmin = 3.55, xmax = 6.5, ymin = 14.6, ymax = 15.5,
           fill="#5DA5ED") +
  annotate("text", x = 4.99, y = 15.07, label = "Winter", size = 5.5) +
  annotate("rect", xmin = 6.55, xmax = 9.5, ymin = 14.6, ymax = 15.5,
           fill="green3") +
  annotate("text", x = 7.99, y = 15.07, label = "Spring", size = 5.5)
dev.off()
```







### COG All - DESeq
```{r}
### Convert phyloseq object to deseq2 object
cog.dds <- phyloseq_to_deseq2(cog.phy, ~Season)
cog.dds$Season <- factor(cog.dds$Season, levels = c("Summer", "Winter", "Spring"))

# DESeq DataStructure for analyses
cog.dds <- DESeq(cog.dds)

# Hellinger-transformed data for visualizations
cog.hell <- hellinger(t(assay(cog.dds)))
```


### COG All - DESeq - find overrep genes
```{r}
# Pull out results from pairwise comparison
cog.sw <- results(cog.dds, contrast = c("Season", "Summer", "Winter"))
cog.sw.sig <- cog.sw[which(cog.sw$padj < 0.05),]
cog.sw.sig[which(cog.sw.sig$log2FoldChange > 0),]
  # Enriched in Summer compared to Winter (positive log2FoldChange)
    # COG2207, COG3345, COG3507, COG3693, COG3866, COG3940
cog.sw.sig[which(cog.sw.sig$log2FoldChange < 0),]
  # Enriched in Winter compared to Summer (negative log2FoldChange)
    # COG3246


cog.sp <- results(cog.dds, contrast = c("Season", "Summer", "Spring"))
cog.sp.sig <- cog.sp[which(cog.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # COG0243, COG0437, COG0457, COG1240, COG1629, COG2825, COG2885


cog.wp <- results(cog.dds, contrast = c("Season", "Winter", "Spring"))
cog.wp.sig <- cog.wp[which(cog.wp$padj < 0.05),]
cog.wp.sig[which(cog.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # COG0457, COG0760, COG0774, COG1452, COG1463, COG1629, COG2825, COG2885, COG3637, COG4775
cog.wp.sig[which(cog.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # COG0004, COG0246, COG0395, COG0561, COG1026, COG1061, COG1070, COG1105, COG1131, COG1132, COG1136, COG1175, COG1472, COG1511, COG1609, COG1653, COG1874, COG1925, COG2182, COG2190, COG2207, COG2730, COG2972, COG3044, COG3345, COG3664, COG3693, COG3757, COG3866, COG3940, COG3973, COG4284, COG4485, COG4709, COG4753, COG5578

```



### COG - Clustering
```{r}
# https://sites.google.com/site/mb3gustame/reference/dissimilarity
cog <- data.frame(otu_table(cog.phy))
colnames(cog) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", 
                        "3773", "3775")
cog.hell <- hellinger(t(cog))
cog.dist <- dist(cog.hell, method = "euclidean")
cog.dist.matrix <- as.matrix(cog.dist)
rownames(cog.dist.matrix) <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
colnames(cog.dist.matrix) <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")

# Dendrogram
cog.hclust <- hclust(cog.dist, method = "average")
cog.den <- as.dendrogram(cog.hclust)
cog.den.data <- dendro_data(cog.den, type = "rectangle")
cog.den.data$labels[,3] <- c("Summer", "Winter", "Winter", "Winter", "Summer", "Spring", "Spring", "Spring", "Summer")
cog.den.data$labels[,3] <- ordered(cog.den.data$labels[,3], levels=c("Summer", "Winter", "Spring"))


#tiff("cog.dendro.tiff", width=10, height=5, units="in", res=300)
ggplot(cog.den.data$segments) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  geom_text(data = cog.den.data$labels, aes(x, y, label=label, colour=label, fontface = "bold"),
            hjust=0, size=5) +
  ylab("Hellinger Distance") +
  xlab("Season") +
  labs(colour = "Season") +
  coord_flip() +
  scale_y_reverse() +
  scale_color_manual(values=c("darkgoldenrod3", "dodgerblue3", "green4")) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_text(size=20),
        panel.border = element_blank(),
        axis.line.x = element_line(),
        legend.title = element_text(face="bold"),
        axis.title.x = element_text(face="bold", size=20),
        axis.title.y = element_text(face="bold", size=20))
#dev.off()







### PCoA
cog.pcoa <- pcoa(cog.dist)
  # Total variance explained = 62.4
cog.phy.ord <- cog.phy
sample_data(cog.phy.ord)$Season <- ordered(sample_data(cog.phy.ord)$Season, levels = c("Summer", "Winter", "Spring"))


#tiff("cog.pcoa.tiff", width=8, height=6, units="in", res=300)
plot_ordination(physeq = cog.phy.ord, ordination = cog.pcoa, axes = c(1,2),
                color = "Season") +
  geom_point(size=5) +
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Hellinger Distance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20),
        legend.text = element_text(size=18),
        plot.title = element_text(hjust=0.5, size=20))
#dev.off()






### Test normality
hist(cog.dist)
qqnorm(cog.dist)
qqline(cog.dist)
shapiro.test(cog.dist)
  # p-value = 0.05018
  # Normal

### Run PERMANOVA on Hellinger distance
adonis(cog.dist ~ Season, data=meta, permutations = 99999)
  # p-value = 0.02111
beta.tax = betadisper(d=cog.dist, group=meta$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.738




### Pairwise PERMANOVA comparisons
cog.dist.matrix <- as.matrix(cog.dist)
cog.dist.df <- data.frame(cog.dist.matrix)


# Summer vs. Winter
cog.sw.dist.df <- cog.dist.df[,-5]
cog.sw.dist.df <- cog.sw.dist.df[,-7:-8]
cog.sw.dist.df <- cog.sw.dist.df[-5,]
cog.sw.dist.df <- cog.sw.dist.df[-7:-8,]
cog.sw.dist <- as.dist(cog.sw.dist.df)
meta.sw <- meta[-which(meta$Season == "Spring"),]
adonis2(cog.sw.dist ~ Season, data=meta.sw, permutations= 99999)
  # p-value = 0.2

# Summer vs. Spring
cog.sp.dist.df <- cog.dist.df[,-3:-4]
cog.sp.dist.df <- cog.sp.dist.df[,-5]
cog.sp.dist.df <- cog.sp.dist.df[-3:-4,]
cog.sp.dist.df <- cog.sp.dist.df[-5,]
cog.sp.dist <- as.dist(cog.sp.dist.df)
meta.sp <- meta[-which(meta$Season == "Winter"),]
adonis(cog.sp.dist ~ Season, data=meta.sp, permutations= 99999)
  # p-value = 0.2

# Spring vs. Winter
cog.wp.dist.df <- cog.dist.df[,-3:-4]
cog.wp.dist.df <- cog.wp.dist.df[,-5]
cog.wp.dist.df <- cog.wp.dist.df[-3:-4,]
cog.wp.dist.df <- cog.wp.dist.df[-5,]
cog.wp.dist <- as.dist(cog.wp.dist.df)
meta.wp <- meta[-which(meta$Season == "Summer"),]
adonis(cog.wp.dist ~ Season, data=meta.wp, permutations= 99999)
  # p-value = 0.2

```

