---
title: "Squirrel_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---

```{r}
library('readr')
#source('http://portal.nersc.gov/dna/RD/Metagenome_RD/MetaBAT/Files/benchmark.R')
  # https://bitbucket.org/berkeleylab/metabat/wiki/Best%20Binning%20Practices
#library('limma')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
theme_set(theme_bw())
set.seed(1)

deSEQ <- function(data, valuetest, cutoff = 0, alpha = 0.01){
  data_pruned <- prune_taxa(taxa_sums(data) > cutoff, data)
  de_data <- phyloseq_to_deseq2(data_pruned, valuetest)
  de_data2 <- DESeq(de_data, test="Wald", fitType="local")
  res_data <- results(de_data2, cooksCutoff = FALSE)
  sig_data <- res_data[which(res_data$padj < alpha), ]
  sigtab_sherm <- cbind(as(sig_data, "data.frame"), as(tax_table(data_pruned)[rownames(sig_data), ], "matrix"))
}
```


### KEGG Brite
```{r}
ko.comp <- read.csv('KEGG/ko_comp.csv', fileEncoding="UTF-8-BOM")
rownames(ko.comp) <- ko.comp[,1]
ko.comp <- ko.comp[,-1]

# Remove SPAdes
kegg <- ko.comp[-which(ko.comp$Assembler == "SPAdes"),]


### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
kegg.norm <- kegg[,-26:-29]
for(i in 1:9){
  kegg.norm[i,] <- round(kegg.norm[i,]*norm.factor[i,1])
}
kegg.norm$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")




### Summarize KEGG
sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
kegg.norm.sum <- sum[4:6,]
kegg.norm.sum[4:6,] <- win[4:6,]
kegg.norm.sum[7:9,] <- spr[4:6,]
colnames(kegg.norm.sum) <- colnames(sum)
kegg.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(kegg.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")

### Check for Normal distribution
norm.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.test <- shapiro.test(kegg.norm[,i])
  norm.pvals[i,1] <- temp.test$p.value
  rownames(norm.pvals)[i] <- colnames(kegg.norm)[i]
}
length(which(norm.pvals[,1] < 0.06))
rownames(norm.pvals)[which(norm.pvals[,1] < 0.06)]

### Overall Summary
kegg.all.sum <- data.frame(999)
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.all.sum[1,i] <- sum(kegg.norm[1:9,i])/9
  kegg.all.sum[2,i] <- sd(kegg.norm[1:9,i])
  kegg.all.sum[3,i] <- sd(kegg.norm[1:9,i])/(sqrt(9))
}
colnames(kegg.all.sum) <- colnames(kegg.norm)[1:25]
rownames(kegg.all.sum) <- c("Mean", "SD", "SE")

### Overall Rel Abund
kegg.relabund <- kegg.norm[,-1]
kegg.relabund <- kegg.relabund[,-25]
for(i in 1:9){
  kegg.relabund[i+9,] <- (kegg.relabund[i,] / kegg.relabund[i,24])*100
}
kegg.relabund <- kegg.relabund[-1:-9,]
rownames(kegg.relabund) <- rownames(kegg.norm)[1:9]

kegg.relabund.sum <- data.frame(999)
for(i in 1:(ncol(kegg.relabund))){
  kegg.relabund.sum[1,i] <- sum(kegg.relabund[1:3,i])/3
  kegg.relabund.sum[2,i] <- sd(kegg.relabund[1:3,i])
  kegg.relabund.sum[3,i] <- sd(kegg.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.relabund.sum) <- colnames(kegg.relabund)[1:24]
rownames(kegg.relabund.sum) <- c("Mean", "SD", "SE")
kegg.relabund.sum <- kegg.relabund.sum[,order(kegg.relabund.sum[1,], decreasing=T)]




### Seasonal Summary
kegg.sum.sum <- data.frame(999)
kegg.sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.sum.sum[1,i] <- sum(kegg.sum[1:3,i])/3
  kegg.sum.sum[2,i] <- sd(kegg.sum[1:3,i])
  kegg.sum.sum[3,i] <- sd(kegg.sum[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.sum) <- colnames(kegg.sum)[1:25]
rownames(kegg.sum.sum) <- c("Mean", "SD", "SE")
# Winter
kegg.win.sum <- data.frame(999)
kegg.win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.win.sum[1,i] <- sum(kegg.win[1:3,i])/3
  kegg.win.sum[2,i] <- sd(kegg.win[1:3,i])
  kegg.win.sum[3,i] <- sd(kegg.win[1:3,i])/(sqrt(3))
}
colnames(kegg.win.sum) <- colnames(kegg.win)[1:25]
rownames(kegg.win.sum) <- c("Mean", "SD", "SE")
#Spring
kegg.spr.sum <- data.frame(999)
kegg.spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.spr.sum[1,i] <- sum(kegg.spr[1:3,i])/3
  kegg.spr.sum[2,i] <- sd(kegg.spr[1:3,i])
  kegg.spr.sum[3,i] <- sd(kegg.spr[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.sum) <- colnames(kegg.spr)[1:25]
rownames(kegg.spr.sum) <- c("Mean", "SD", "SE")

### Seasonal Rel Abund
kegg.seas.relabund <- kegg.norm[,-1]
kegg.seas.relabund <- kegg.seas.relabund[,-25]
for(i in 1:9){
  kegg.seas.relabund[i+9,] <- (kegg.seas.relabund[i,] / kegg.seas.relabund[i,24])*100
}
kegg.seas.relabund <- kegg.seas.relabund[-1:-9,]
rownames(kegg.seas.relabund) <- rownames(kegg.norm)[1:9]

sum <- which(kegg.norm$Season == "Summer")
win <- which(kegg.norm$Season == "Winter")
spr <- which(kegg.norm$Season == "Spring")
sum.relabund <- kegg.seas.relabund[sum,]
win.relabund <- kegg.seas.relabund[win,]
spr.relabund <- kegg.seas.relabund[spr,]

# Summer
kegg.sum.relabund.sum <- data.frame(999)
for(i in 1:(ncol(sum.relabund))){
  kegg.sum.relabund.sum[1,i] <- sum(sum.relabund[1:3,i])/3
  kegg.sum.relabund.sum[2,i] <- sd(sum.relabund[1:3,i])
  kegg.sum.relabund.sum[3,i] <- sd(sum.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.relabund.sum) <- colnames(sum.relabund)[1:24]
rownames(kegg.sum.relabund.sum) <- c("Mean", "SD", "SE")
kegg.sum.relabund.sum <- kegg.sum.relabund.sum[,order(kegg.sum.relabund.sum[1,], decreasing=T)]

# Winter
kegg.win.relabund.sum <- data.frame(999)
for(i in 1:(ncol(win.relabund))){
  kegg.win.relabund.sum[1,i] <- sum(win.relabund[1:3,i])/3
  kegg.win.relabund.sum[2,i] <- sd(win.relabund[1:3,i])
  kegg.win.relabund.sum[3,i] <- sd(win.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.win.relabund.sum) <- colnames(win.relabund)[1:24]
rownames(kegg.win.relabund.sum) <- c("Mean", "SD", "SE")
kegg.win.relabund.sum <- kegg.win.relabund.sum[,order(kegg.win.relabund.sum[1,], decreasing=T)]

# Spring
kegg.spr.relabund.sum <- data.frame(999)
for(i in 1:(ncol(spr.relabund))){
  kegg.spr.relabund.sum[1,i] <- sum(spr.relabund[1:3,i])/3
  kegg.spr.relabund.sum[2,i] <- sd(spr.relabund[1:3,i])
  kegg.spr.relabund.sum[3,i] <- sd(spr.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.relabund.sum) <- colnames(spr.relabund)[1:24]
rownames(kegg.spr.relabund.sum) <- c("Mean", "SD", "SE")
kegg.spr.relabund.sum <- kegg.spr.relabund.sum[,order(kegg.spr.relabund.sum[1,], decreasing=T)]


### Prep for barplot
sum.trim <- kegg.sum.relabund.sum
sum.trim <- data.frame(t(sum.trim))
sum.trim <- sum.trim[-1:-2,]
sum.trim <- sum.trim[-3,]
sum.trim <- sum.trim[-4,]
sum.trim <- sum.trim[-6:-20,]
sum.trim$Season <- rep("Summer",5)
sum.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
sum.trim$se.max <- sum.trim$Mean + sum.trim$SE
sum.trim$se.min <- sum.trim$Mean - sum.trim$SE
win.trim <- kegg.win.relabund.sum
win.trim <- data.frame(t(win.trim))
win.trim <- win.trim[-1:-2,]
win.trim <- win.trim[-3,]
win.trim <- win.trim[-4,]
win.trim <- win.trim[-6:-20,]
win.trim$Season <- rep("Winter",5)
win.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
win.trim$se.max <- win.trim$Mean + win.trim$SE
win.trim$se.min <- win.trim$Mean - win.trim$SE
spr.trim <- kegg.spr.relabund.sum
spr.trim <- data.frame(t(spr.trim))
spr.trim <- spr.trim[-1:-2,]
spr.trim <- spr.trim[-3,]
spr.trim <- spr.trim[-4,]
spr.trim <- spr.trim[-6:-20,]
spr.trim$Season <- rep("Spring",5)
spr.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
spr.trim$se.max <- spr.trim$Mean + spr.trim$SE
spr.trim$se.min <- spr.trim$Mean - spr.trim$SE
kegg.barplot <- sum.trim
kegg.barplot[6:10,] <- win.trim
kegg.barplot[11:15,] <- spr.trim
kegg.barplot$Pathway <- ordered(kegg.barplot$Pathway, levels=c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism"))
kegg.barplot$Season <- ordered(kegg.barplot$Season, levels=c("Summer", "Winter", "Spring"))

##### BARPLOT! #####
tiff("kegg.barplot.tiff", width = 13, height = 7, units = "in", res = 300)
ggplot(kegg.barplot, aes(x=Pathway, y=Mean, fill=Pathway)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(. ~ Season) +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values = c("lightcoral", "lightskyblue", "palegreen", "plum", "khaki1")) +
  xlab("Pathway") +
  ylab("Mean Relative Abundance") +
  scale_y_continuous(limits=c(0,55), expand=c(0,0))+
  theme(axis.text.x = element_blank(),
        panel.spacing=unit(1, "lines"),
        axis.title = element_text(size=20),
        strip.text.x = element_text(size=14),
        strip.text.y = element_text(size=14),
        axis.text.y = element_text(size=12),
        legend.title = element_text(size = 14),
        legend.text = element_text(size=12))
dev.off()



### ANOVA
kegg.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.aov <- summary(aov(kegg.norm[,i] ~ Season, data=kegg.norm))
  kegg.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(kegg.pvals)[i] <- colnames(kegg)[i]
}
sig <- rownames(kegg.pvals)[which(kegg.pvals[,1] < 0.06)]
# 
# # Tukey HSD
# tukey.pvals <- data.frame(rep(999,4))
# cat <- which(colnames(kegg) == sig[1])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,1] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[2])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,2] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[3])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,3] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[4])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,4] <- temp.tukey$Season[,4]
# colnames(tukey.pvals) <- sig
# rownames(tukey.pvals) <- rownames(temp.tukey$Season)

# tukey.pvals <- data.frame(rep(999,4))
# for(i in 1:length(sig)){
#   cat <- which(colnames(kegg) == sig[i])
#   temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
#   temp.tukey <- TukeyHSD(temp.aov)
#   tukey.pvals[(1:3),i] <- temp.tukey$Season[,4]
#   rownames(tukey.pvals)[1:3,i] <- rownames(temp.tukey$Season)
# }

```


### dbcan
```{r}
dbcan.comp <- read.csv('dbcan/dbcan_comp.csv', fileEncoding="UTF-8-BOM")
rownames(dbcan.comp) <- dbcan.comp[,1]
dbcan.comp <- dbcan.comp[,-1]

# Remove SPAdes
dbcan <- dbcan.comp[-which(dbcan.comp$Assembler == "SPAdes"),]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
dbcan.norm <- dbcan[,-9:-10]
for(i in 1:8){
  dbcan.norm[i,] <- round(dbcan.norm[i,]*norm.factor[i,1])
}
dbcan.norm$Season <- dbcan$Season

### Check for Normal distribution
dbcan.norm.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.test <- shapiro.test(dbcan.norm[,i])
  dbcan.norm.pvals[i,1] <- temp.test$p.value
  rownames(dbcan.norm.pvals)[i] <- colnames(dbcan.norm)[i]
}
  # All normal

### Summarize dbcan
sum <- dbcan.norm[which(dbcan.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- dbcan.norm[which(dbcan.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- dbcan.norm[which(dbcan.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
dbcan.norm.sum <- sum[4:6,]
dbcan.norm.sum[4:6,] <- win[4:6,]
dbcan.norm.sum[7:9,] <- spr[4:6,]
colnames(dbcan.norm.sum) <- colnames(sum)
dbcan.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(dbcan.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")


### Prep for barplot ###
# barplot.df <- dbcan.norm.sum[,-1]
# barplot.df <- barplot.df[,-4]
# barplot.df <- barplot.df[,-5:-6]
# dbcan.barplot <- t(barplot.df[1,])
# dbcan.barplot[6:10,] <- t(barplot.df[4,])

##### BARPLOT! #####
dbcan.barplot <- read.csv('dbcan/barplot_df.csv', fileEncoding="UTF-8-BOM")
dbcan.barplot$Season <- ordered(dbcan.barplot$Season, levels=c("Summer", "Winter", "Spring"))
cbm <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate-Binding Modules"),]
ce <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate Esterases"),]
gh <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Glycoside Hydrolases"),]
pl <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Polysaccharide Lyases"),]

cbm.barplot <- ggplot(cbm, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate-Binding Modules") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) + #,
        #legend.title = element_text(size=16),
        #legend.text = element_text(size=14)) +
        guides(fill = F)

g_legend <- function(a.gplot){
   tmp <- ggplot_gtable(ggplot_build(a.gplot))
   leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
   legend <- tmp$grobs[[leg]]
   return(legend)}
legend <- g_legend(cbm.barplot)

ce.barplot <- ggplot(ce, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate Esterases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

gh.barplot <- ggplot(gh, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Glycoside Hydrolases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

pl.barplot <- ggplot(pl, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Polysaccharide Lyases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)



tiff("dbcan.barplot.tiff", width = 13, height = 8, units = "in", res = 300)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(2,3, widths=c(1,1,0.3), heights=c(1,1)))))
print(cbm.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(ce.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
print(gh.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=1))
print(pl.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=2))
legend$vp <- viewport(layout.pos.row = 1:2, layout.pos.col = 3)
grid.draw(legend)
dev.off()

### Overall Summary
dbcan.all.sum <- data.frame(999)
for(i in 1:(ncol(dbcan.norm)-1)){
  dbcan.all.sum[1,i] <- sum(dbcan.norm[1:9,i])/9
  dbcan.all.sum[2,i] <- sd(dbcan.norm[1:9,i])
  dbcan.all.sum[3,i] <- sd(dbcan.norm[1:9,i])/(sqrt(9))
}
colnames(dbcan.all.sum) <- colnames(dbcan.norm)[1:6]
rownames(dbcan.all.sum) <- c("Mean", "SD", "SE")

### ANOVA
dbcan.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.aov <- summary(aov(dbcan.norm[,i] ~ Season, data=dbcan.norm))
  dbcan.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(dbcan.pvals)[i] <- colnames(dbcan)[i]
}
sig <- rownames(dbcan.pvals)[which(dbcan.pvals[,1] < 0.06)]

```































### KEGG Pathways
```{r}
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
meta <- read.csv('KEGG/metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]
#meta <- meta[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
path.otu <- otu_table(path, taxa_are_rows = T)
path.meta <- sample_data(meta)
path.physeq <- merge_phyloseq(path.otu, path.meta)



```