---
title: "Squirrel_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---

```{r}
library('readr')
#source('http://portal.nersc.gov/dna/RD/Metagenome_RD/MetaBAT/Files/benchmark.R')
  # https://bitbucket.org/berkeleylab/metabat/wiki/Best%20Binning%20Practices
#library('limma')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
library('phangorn')
library('vegan')
theme_set(theme_bw())
set.seed(1)

deSEQ <- function(data, valuetest, cutoff = 0, alpha = 0.01){
  data_pruned <- prune_taxa(taxa_sums(data) > cutoff, data)
  de_data <- phyloseq_to_deseq2(data_pruned, valuetest)
  de_data2 <- DESeq(de_data, test="Wald", fitType="local")
  res_data <- results(de_data2, cooksCutoff = FALSE)
  sig_data <- res_data[which(res_data$padj < alpha), ]
  sigtab_sherm <- cbind(as(sig_data, "data.frame"), as(tax_table(data_pruned)[rownames(sig_data), ], "matrix"))
}
```


### KEGG Brite
```{r}
ko.comp <- read.csv('KEGG/ko_comp.csv', fileEncoding="UTF-8-BOM")
rownames(ko.comp) <- ko.comp[,1]
ko.comp <- ko.comp[,-1]

# Remove SPAdes
kegg <- ko.comp[-which(ko.comp$Assembler == "SPAdes"),]


### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
kegg.norm <- kegg[,-26:-29]
for(i in 1:9){
  kegg.norm[i,] <- round(kegg.norm[i,]*norm.factor[i,1])
}
kegg.norm$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")




### Summarize KEGG
sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
kegg.norm.sum <- sum[4:6,]
kegg.norm.sum[4:6,] <- win[4:6,]
kegg.norm.sum[7:9,] <- spr[4:6,]
colnames(kegg.norm.sum) <- colnames(sum)
kegg.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(kegg.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")

### Check for Normal distribution
norm.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.test <- shapiro.test(kegg.norm[,i])
  norm.pvals[i,1] <- temp.test$p.value
  rownames(norm.pvals)[i] <- colnames(kegg.norm)[i]
}
length(which(norm.pvals[,1] < 0.06))
rownames(norm.pvals)[which(norm.pvals[,1] < 0.06)]

### Overall Summary
kegg.all.sum <- data.frame(999)
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.all.sum[1,i] <- sum(kegg.norm[1:9,i])/9
  kegg.all.sum[2,i] <- sd(kegg.norm[1:9,i])
  kegg.all.sum[3,i] <- sd(kegg.norm[1:9,i])/(sqrt(9))
}
colnames(kegg.all.sum) <- colnames(kegg.norm)[1:25]
rownames(kegg.all.sum) <- c("Mean", "SD", "SE")

### Overall Rel Abund
kegg.relabund <- kegg.norm[,-1]
kegg.relabund <- kegg.relabund[,-25]
for(i in 1:9){
  kegg.relabund[i+9,] <- (kegg.relabund[i,] / kegg.relabund[i,24])*100
}
kegg.relabund <- kegg.relabund[-1:-9,]
rownames(kegg.relabund) <- rownames(kegg.norm)[1:9]

kegg.relabund.sum <- data.frame(999)
for(i in 1:(ncol(kegg.relabund))){
  kegg.relabund.sum[1,i] <- sum(kegg.relabund[1:3,i])/3
  kegg.relabund.sum[2,i] <- sd(kegg.relabund[1:3,i])
  kegg.relabund.sum[3,i] <- sd(kegg.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.relabund.sum) <- colnames(kegg.relabund)[1:24]
rownames(kegg.relabund.sum) <- c("Mean", "SD", "SE")
kegg.relabund.sum <- kegg.relabund.sum[,order(kegg.relabund.sum[1,], decreasing=T)]




### Seasonal Summary
kegg.sum.sum <- data.frame(999)
kegg.sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.sum.sum[1,i] <- sum(kegg.sum[1:3,i])/3
  kegg.sum.sum[2,i] <- sd(kegg.sum[1:3,i])
  kegg.sum.sum[3,i] <- sd(kegg.sum[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.sum) <- colnames(kegg.sum)[1:25]
rownames(kegg.sum.sum) <- c("Mean", "SD", "SE")
# Winter
kegg.win.sum <- data.frame(999)
kegg.win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.win.sum[1,i] <- sum(kegg.win[1:3,i])/3
  kegg.win.sum[2,i] <- sd(kegg.win[1:3,i])
  kegg.win.sum[3,i] <- sd(kegg.win[1:3,i])/(sqrt(3))
}
colnames(kegg.win.sum) <- colnames(kegg.win)[1:25]
rownames(kegg.win.sum) <- c("Mean", "SD", "SE")
#Spring
kegg.spr.sum <- data.frame(999)
kegg.spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.spr.sum[1,i] <- sum(kegg.spr[1:3,i])/3
  kegg.spr.sum[2,i] <- sd(kegg.spr[1:3,i])
  kegg.spr.sum[3,i] <- sd(kegg.spr[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.sum) <- colnames(kegg.spr)[1:25]
rownames(kegg.spr.sum) <- c("Mean", "SD", "SE")

### Seasonal Rel Abund
kegg.seas.relabund <- kegg.norm[,-1]
kegg.seas.relabund <- kegg.seas.relabund[,-25]
for(i in 1:9){
  kegg.seas.relabund[i+9,] <- (kegg.seas.relabund[i,] / kegg.seas.relabund[i,24])*100
}
kegg.seas.relabund <- kegg.seas.relabund[-1:-9,]
rownames(kegg.seas.relabund) <- rownames(kegg.norm)[1:9]

sum <- which(kegg.norm$Season == "Summer")
win <- which(kegg.norm$Season == "Winter")
spr <- which(kegg.norm$Season == "Spring")
sum.relabund <- kegg.seas.relabund[sum,]
win.relabund <- kegg.seas.relabund[win,]
spr.relabund <- kegg.seas.relabund[spr,]

# Summer
kegg.sum.relabund.sum <- data.frame(999)
for(i in 1:(ncol(sum.relabund))){
  kegg.sum.relabund.sum[1,i] <- sum(sum.relabund[1:3,i])/3
  kegg.sum.relabund.sum[2,i] <- sd(sum.relabund[1:3,i])
  kegg.sum.relabund.sum[3,i] <- sd(sum.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.relabund.sum) <- colnames(sum.relabund)[1:24]
rownames(kegg.sum.relabund.sum) <- c("Mean", "SD", "SE")
kegg.sum.relabund.sum <- kegg.sum.relabund.sum[,order(kegg.sum.relabund.sum[1,], decreasing=T)]

# Winter
kegg.win.relabund.sum <- data.frame(999)
for(i in 1:(ncol(win.relabund))){
  kegg.win.relabund.sum[1,i] <- sum(win.relabund[1:3,i])/3
  kegg.win.relabund.sum[2,i] <- sd(win.relabund[1:3,i])
  kegg.win.relabund.sum[3,i] <- sd(win.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.win.relabund.sum) <- colnames(win.relabund)[1:24]
rownames(kegg.win.relabund.sum) <- c("Mean", "SD", "SE")
kegg.win.relabund.sum <- kegg.win.relabund.sum[,order(kegg.win.relabund.sum[1,], decreasing=T)]

# Spring
kegg.spr.relabund.sum <- data.frame(999)
for(i in 1:(ncol(spr.relabund))){
  kegg.spr.relabund.sum[1,i] <- sum(spr.relabund[1:3,i])/3
  kegg.spr.relabund.sum[2,i] <- sd(spr.relabund[1:3,i])
  kegg.spr.relabund.sum[3,i] <- sd(spr.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.relabund.sum) <- colnames(spr.relabund)[1:24]
rownames(kegg.spr.relabund.sum) <- c("Mean", "SD", "SE")
kegg.spr.relabund.sum <- kegg.spr.relabund.sum[,order(kegg.spr.relabund.sum[1,], decreasing=T)]


### Prep for barplot
sum.trim <- kegg.sum.relabund.sum
sum.trim <- data.frame(t(sum.trim))
sum.trim <- sum.trim[-1:-2,]
sum.trim <- sum.trim[-3,]
sum.trim <- sum.trim[-4,]
sum.trim <- sum.trim[-6:-20,]
sum.trim$Season <- rep("Summer",5)
sum.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
sum.trim$se.max <- sum.trim$Mean + sum.trim$SE
sum.trim$se.min <- sum.trim$Mean - sum.trim$SE
win.trim <- kegg.win.relabund.sum
win.trim <- data.frame(t(win.trim))
win.trim <- win.trim[-1:-2,]
win.trim <- win.trim[-3,]
win.trim <- win.trim[-4,]
win.trim <- win.trim[-6:-20,]
win.trim$Season <- rep("Winter",5)
win.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
win.trim$se.max <- win.trim$Mean + win.trim$SE
win.trim$se.min <- win.trim$Mean - win.trim$SE
spr.trim <- kegg.spr.relabund.sum
spr.trim <- data.frame(t(spr.trim))
spr.trim <- spr.trim[-1:-2,]
spr.trim <- spr.trim[-3,]
spr.trim <- spr.trim[-4,]
spr.trim <- spr.trim[-6:-20,]
spr.trim$Season <- rep("Spring",5)
spr.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
spr.trim$se.max <- spr.trim$Mean + spr.trim$SE
spr.trim$se.min <- spr.trim$Mean - spr.trim$SE
kegg.barplot <- sum.trim
kegg.barplot[6:10,] <- win.trim
kegg.barplot[11:15,] <- spr.trim
kegg.barplot$Pathway <- ordered(kegg.barplot$Pathway, levels=c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism"))
kegg.barplot$Season <- ordered(kegg.barplot$Season, levels=c("Summer", "Winter", "Spring"))

##### BARPLOT! #####
tiff("kegg.barplot.tiff", width = 13, height = 7, units = "in", res = 300)
ggplot(kegg.barplot, aes(x=Pathway, y=Mean, fill=Pathway)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(. ~ Season) +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values = c("lightcoral", "lightskyblue", "palegreen", "plum", "khaki1")) +
  xlab("Pathway") +
  ylab("Mean Relative Abundance") +
  scale_y_continuous(limits=c(0,55), expand=c(0,0))+
  theme(axis.text.x = element_blank(),
        panel.spacing=unit(1, "lines"),
        axis.title = element_text(size=20),
        strip.text.x = element_text(size=14),
        strip.text.y = element_text(size=14),
        axis.text.y = element_text(size=12),
        legend.title = element_text(size = 14),
        legend.text = element_text(size=12))
dev.off()



### ANOVA
kegg.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.aov <- summary(aov(kegg.norm[,i] ~ Season, data=kegg.norm))
  kegg.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(kegg.pvals)[i] <- colnames(kegg)[i]
}
sig <- rownames(kegg.pvals)[which(kegg.pvals[,1] < 0.06)]
# 
# # Tukey HSD
# tukey.pvals <- data.frame(rep(999,4))
# cat <- which(colnames(kegg) == sig[1])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,1] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[2])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,2] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[3])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,3] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[4])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,4] <- temp.tukey$Season[,4]
# colnames(tukey.pvals) <- sig
# rownames(tukey.pvals) <- rownames(temp.tukey$Season)

# tukey.pvals <- data.frame(rep(999,4))
# for(i in 1:length(sig)){
#   cat <- which(colnames(kegg) == sig[i])
#   temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
#   temp.tukey <- TukeyHSD(temp.aov)
#   tukey.pvals[(1:3),i] <- temp.tukey$Season[,4]
#   rownames(tukey.pvals)[1:3,i] <- rownames(temp.tukey$Season)
# }

```


### dbcan
```{r}
dbcan.comp <- read.csv('dbcan/dbcan_comp.csv', fileEncoding="UTF-8-BOM")
rownames(dbcan.comp) <- dbcan.comp[,1]
dbcan.comp <- dbcan.comp[,-1]

# Remove SPAdes
dbcan <- dbcan.comp[-which(dbcan.comp$Assembler == "SPAdes"),]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
dbcan.norm <- dbcan[,-9:-10]
for(i in 1:8){
  dbcan.norm[i,] <- round(dbcan.norm[i,]*norm.factor[i,1])
}
dbcan.norm$Season <- dbcan$Season

### Check for Normal distribution
dbcan.norm.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.test <- shapiro.test(dbcan.norm[,i])
  dbcan.norm.pvals[i,1] <- temp.test$p.value
  rownames(dbcan.norm.pvals)[i] <- colnames(dbcan.norm)[i]
}
  # All normal

### Summarize dbcan
sum <- dbcan.norm[which(dbcan.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- dbcan.norm[which(dbcan.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- dbcan.norm[which(dbcan.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
dbcan.norm.sum <- sum[4:6,]
dbcan.norm.sum[4:6,] <- win[4:6,]
dbcan.norm.sum[7:9,] <- spr[4:6,]
colnames(dbcan.norm.sum) <- colnames(sum)
dbcan.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(dbcan.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")


### Prep for barplot ###
# barplot.df <- dbcan.norm.sum[,-1]
# barplot.df <- barplot.df[,-4]
# barplot.df <- barplot.df[,-5:-6]
# dbcan.barplot <- t(barplot.df[1,])
# dbcan.barplot[6:10,] <- t(barplot.df[4,])

##### BARPLOT! #####
dbcan.barplot <- read.csv('dbcan/barplot_df.csv', fileEncoding="UTF-8-BOM")
dbcan.barplot$Season <- ordered(dbcan.barplot$Season, levels=c("Summer", "Winter", "Spring"))
cbm <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate-Binding Modules"),]
ce <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Carbohydrate Esterases"),]
gh <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Glycoside Hydrolases"),]
pl <- dbcan.barplot[which(dbcan.barplot$Cazyme=="Polysaccharide Lyases"),]

cbm.barplot <- ggplot(cbm, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate-Binding Modules") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) + #,
        #legend.title = element_text(size=16),
        #legend.text = element_text(size=14)) +
        guides(fill = F)

g_legend <- function(a.gplot){
   tmp <- ggplot_gtable(ggplot_build(a.gplot))
   leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
   legend <- tmp$grobs[[leg]]
   return(legend)}
legend <- g_legend(cbm.barplot)

ce.barplot <- ggplot(ce, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Carbohydrate Esterases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

gh.barplot <- ggplot(gh, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Glycoside Hydrolases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)

pl.barplot <- ggplot(pl, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Polysaccharide Lyases") +
  ylab("Mean Count") +
  theme(plot.title=element_text(hjust=0.5, size=16, face = "bold"),
        axis.text.x = element_text(size=12),
        axis.title = element_text(size=14),
        axis.text.y = element_text(size=12),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm")) +
        guides(fill = F)



tiff("dbcan.barplot.tiff", width = 13, height = 8, units = "in", res = 300)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(2,3, widths=c(1,1,0.3), heights=c(1,1)))))
print(cbm.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(ce.barplot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
print(gh.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=1))
print(pl.barplot, vp=viewport(layout.pos.row=2, layout.pos.col=2))
legend$vp <- viewport(layout.pos.row = 1:2, layout.pos.col = 3)
grid.draw(legend)
dev.off()

### Overall Summary
dbcan.all.sum <- data.frame(999)
for(i in 1:(ncol(dbcan.norm)-1)){
  dbcan.all.sum[1,i] <- sum(dbcan.norm[1:9,i])/9
  dbcan.all.sum[2,i] <- sd(dbcan.norm[1:9,i])
  dbcan.all.sum[3,i] <- sd(dbcan.norm[1:9,i])/(sqrt(9))
}
colnames(dbcan.all.sum) <- colnames(dbcan.norm)[1:6]
rownames(dbcan.all.sum) <- c("Mean", "SD", "SE")

### ANOVA
dbcan.pvals <- data.frame(rep(999, 8))
for(i in 1:(ncol(dbcan.norm)-1)){
  temp.aov <- summary(aov(dbcan.norm[,i] ~ Season, data=dbcan.norm))
  dbcan.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(dbcan.pvals)[i] <- colnames(dbcan)[i]
}
sig <- rownames(dbcan.pvals)[which(dbcan.pvals[,1] < 0.06)]

```































### KEGG Pathways
```{r}
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('KEGG/metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(class) <- class[,1]
class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

### Calculate proportions
path.prop <- path.norm
for(i in 1:9){
  path.prop[,i] <- (path.prop[,i] / colSums(path.prop)[i])
}
rem <- which(rowSums(path.prop) < 0.001)
path.prop <- path.prop[-rem,]

kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(kend) <- kend[,1]
kend[,2:9] <- NA
colnames(kend) <- kend[,1]
kend[,1] <- NA

kend[1,1] <- cor(path.prop[,1], path.prop[,1], method = "kendall")
kend[1,2] <- cor(path.prop[,1], path.prop[,2], method = "kendall")
kend[1,3] <- cor(path.prop[,1], path.prop[,3], method = "kendall")
kend[1,4] <- cor(path.prop[,1], path.prop[,4], method = "kendall")
kend[1,5] <- cor(path.prop[,1], path.prop[,5], method = "kendall")
kend[1,6] <- cor(path.prop[,1], path.prop[,6], method = "kendall")
kend[1,7] <- cor(path.prop[,1], path.prop[,7], method = "kendall")
kend[1,8] <- cor(path.prop[,1], path.prop[,8], method = "kendall")
kend[1,9] <- cor(path.prop[,1], path.prop[,9], method = "kendall")
kend[2,1] <- cor(path.prop[,2], path.prop[,1], method = "kendall")
kend[2,2] <- cor(path.prop[,2], path.prop[,2], method = "kendall")
kend[2,3] <- cor(path.prop[,2], path.prop[,3], method = "kendall")
kend[2,4] <- cor(path.prop[,2], path.prop[,4], method = "kendall")
kend[2,5] <- cor(path.prop[,2], path.prop[,5], method = "kendall")
kend[2,6] <- cor(path.prop[,2], path.prop[,6], method = "kendall")
kend[2,7] <- cor(path.prop[,2], path.prop[,7], method = "kendall")
kend[2,8] <- cor(path.prop[,2], path.prop[,8], method = "kendall")
kend[2,9] <- cor(path.prop[,2], path.prop[,9], method = "kendall")
kend[3,1] <- cor(path.prop[,3], path.prop[,1], method = "kendall")
kend[3,2] <- cor(path.prop[,3], path.prop[,2], method = "kendall")
kend[3,3] <- cor(path.prop[,3], path.prop[,3], method = "kendall")
kend[3,4] <- cor(path.prop[,3], path.prop[,4], method = "kendall")
kend[3,5] <- cor(path.prop[,3], path.prop[,5], method = "kendall")
kend[3,6] <- cor(path.prop[,3], path.prop[,6], method = "kendall")
kend[3,7] <- cor(path.prop[,3], path.prop[,7], method = "kendall")
kend[3,8] <- cor(path.prop[,3], path.prop[,8], method = "kendall")
kend[3,9] <- cor(path.prop[,3], path.prop[,9], method = "kendall")
kend[4,1] <- cor(path.prop[,4], path.prop[,1], method = "kendall")
kend[4,2] <- cor(path.prop[,4], path.prop[,2], method = "kendall")
kend[4,3] <- cor(path.prop[,4], path.prop[,3], method = "kendall")
kend[4,4] <- cor(path.prop[,4], path.prop[,4], method = "kendall")
kend[4,5] <- cor(path.prop[,4], path.prop[,5], method = "kendall")
kend[4,6] <- cor(path.prop[,4], path.prop[,6], method = "kendall")
kend[4,7] <- cor(path.prop[,4], path.prop[,7], method = "kendall")
kend[4,8] <- cor(path.prop[,4], path.prop[,8], method = "kendall")
kend[4,9] <- cor(path.prop[,4], path.prop[,9], method = "kendall")
kend[5,1] <- cor(path.prop[,5], path.prop[,1], method = "kendall")
kend[5,2] <- cor(path.prop[,5], path.prop[,2], method = "kendall")
kend[5,3] <- cor(path.prop[,5], path.prop[,3], method = "kendall")
kend[5,4] <- cor(path.prop[,5], path.prop[,4], method = "kendall")
kend[5,5] <- cor(path.prop[,5], path.prop[,5], method = "kendall")
kend[5,6] <- cor(path.prop[,5], path.prop[,6], method = "kendall")
kend[5,7] <- cor(path.prop[,5], path.prop[,7], method = "kendall")
kend[5,8] <- cor(path.prop[,5], path.prop[,8], method = "kendall")
kend[5,9] <- cor(path.prop[,5], path.prop[,9], method = "kendall")
kend[6,1] <- cor(path.prop[,6], path.prop[,1], method = "kendall")
kend[6,2] <- cor(path.prop[,6], path.prop[,2], method = "kendall")
kend[6,3] <- cor(path.prop[,6], path.prop[,3], method = "kendall")
kend[6,4] <- cor(path.prop[,6], path.prop[,4], method = "kendall")
kend[6,5] <- cor(path.prop[,6], path.prop[,5], method = "kendall")
kend[6,6] <- cor(path.prop[,6], path.prop[,6], method = "kendall")
kend[6,7] <- cor(path.prop[,6], path.prop[,7], method = "kendall")
kend[6,8] <- cor(path.prop[,6], path.prop[,8], method = "kendall")
kend[6,9] <- cor(path.prop[,6], path.prop[,9], method = "kendall")
kend[7,1] <- cor(path.prop[,7], path.prop[,1], method = "kendall")
kend[7,2] <- cor(path.prop[,7], path.prop[,2], method = "kendall")
kend[7,3] <- cor(path.prop[,7], path.prop[,3], method = "kendall")
kend[7,4] <- cor(path.prop[,7], path.prop[,4], method = "kendall")
kend[7,5] <- cor(path.prop[,7], path.prop[,5], method = "kendall")
kend[7,6] <- cor(path.prop[,7], path.prop[,6], method = "kendall")
kend[7,7] <- cor(path.prop[,7], path.prop[,7], method = "kendall")
kend[7,8] <- cor(path.prop[,7], path.prop[,8], method = "kendall")
kend[7,9] <- cor(path.prop[,7], path.prop[,9], method = "kendall")
kend[8,1] <- cor(path.prop[,8], path.prop[,1], method = "kendall")
kend[8,2] <- cor(path.prop[,8], path.prop[,2], method = "kendall")
kend[8,3] <- cor(path.prop[,8], path.prop[,3], method = "kendall")
kend[8,4] <- cor(path.prop[,8], path.prop[,4], method = "kendall")
kend[8,5] <- cor(path.prop[,8], path.prop[,5], method = "kendall")
kend[8,6] <- cor(path.prop[,8], path.prop[,6], method = "kendall")
kend[8,7] <- cor(path.prop[,8], path.prop[,7], method = "kendall")
kend[8,8] <- cor(path.prop[,8], path.prop[,8], method = "kendall")
kend[8,9] <- cor(path.prop[,8], path.prop[,9], method = "kendall")
kend[9,1] <- cor(path.prop[,9], path.prop[,1], method = "kendall")
kend[9,2] <- cor(path.prop[,9], path.prop[,2], method = "kendall")
kend[9,3] <- cor(path.prop[,9], path.prop[,3], method = "kendall")
kend[9,4] <- cor(path.prop[,9], path.prop[,4], method = "kendall")
kend[9,5] <- cor(path.prop[,9], path.prop[,5], method = "kendall")
kend[9,6] <- cor(path.prop[,9], path.prop[,6], method = "kendall")
kend[9,7] <- cor(path.prop[,9], path.prop[,7], method = "kendall")
kend[9,8] <- cor(path.prop[,9], path.prop[,8], method = "kendall")
kend[9,9] <- cor(path.prop[,9], path.prop[,9], method = "kendall")

#write.table(kend, file="kegg.kendall.txt", sep="\t")

```

```{r}
kend.fix <- kend
kend.fix[1,1] <- 0
kend.fix[2,2] <- 0
kend.fix[3,3] <- 0
kend.fix[4,4] <- 0
kend.fix[5,5] <- 0
kend.fix[6,6] <- 0
kend.fix[7,7] <- 0
kend.fix[8,8] <- 0
kend.fix[9,9] <- 0
path.upgma <- upgma(kend.fix)
plot(path.upgma)


path.nj <- NJ(data.frame(kend.fix))

```




### Pathway Bray-Curtis
```{r}
#Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('KEGG/metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(class) <- class[,1]
class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
path.otu <- otu_table(path.norm, taxa_are_rows = T)
path.samp <- sample_data(meta)
path.tax <- tax_table(class)
row.names(path.tax) <- row.names(class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

nmds <- ordinate(physeq=path.physeq, method="NMDS", distance="bray")
  # stress = 0.01481376

plot_ordination(physeq=path.physeq, ordination=nmds, color="Season")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.bray<- phyloseq::distance(physeq=path.physeq, method="bray")
adonis(path.bray~ Season, data=path.sampdf)
  # p-value = 0.333
beta.tax = betadisper(d=path.bray, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.304

### ANOSIM
anosim(path.bray, meta$Season, permutations=1000)
  # p-value = 0.13986
```

### Pathway Jaccard
```{r}
jac <- ordinate(physeq=path.physeq, method="NMDS", distance="jaccard")
  # stress = 0.02570319
plot_ordination(physeq=path.physeq, ordination=jac, color="Season")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.jac <- phyloseq::distance(physeq=path.physeq, method="jaccard")
adonis(path.jac~ Season, data=path.sampdf)
  # p-value = 0.235
beta.tax = betadisper(d=path.jac, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.367

### ANOSIM
anosim(path.jac, path.sampdf$Season, permutations=1000)
  # p-value = 0.14386
```

### Pathway SIMPER
```{r}
### Separate out pairw
simper(path.norm, path.sampdf$Season, permutations=100)

simper.pretty(path.norm, metrics = meta, interesting = 'Season', perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'pathway')

path.simper <- read.csv("pathway_clean_simper.csv")

kruskal.pretty(otu = path.norm, metrics = meta, csv =  path.simper, interesting = 'Season', output_name = 'path.kw')

path.simper.kw <- read.csv("path.kw_krusk_simper.csv")
```

### Simper Functions
```{r}
simper.pretty = function(x, metrics, interesting, perc_cutoff, low_cutoff, low_val, output_name){
  library(vegan)
  #handling otu tables for taxa levels
  save=list(0)
  if(grepl("Otu", colnames(x)[1])!=TRUE){
    #converts output from A.Neumann Taxonomy script
    save=list(58)
    x=as.data.frame(t(x))
    orig_names=colnames(x)
    new_names=list()
    l=1
    for(n in colnames(x)){
      ifelse((l<10), (colnames(x)[l]=c(paste0('Otu000',c(l)))), (colnames(x)[l]=c(paste0('Otu00',c(l))))) 
      new_names=append(new_names, colnames(x)[l])
      l=l+1
    }
    orig_names=as.data.frame(orig_names, row.names = as.character(new_names))
  }
  #running simper
  for(variables in interesting){
    test_1=with(metrics, simper(x, metrics[[variables]]))
    #parsing through simper output, saving desired info to table
    for(name in names(test_1)){
      testmx=matrix(ncol=length(interesting))
      testmx=cbind(test_1[[name]]$ord,test_1[[name]]$cusum)
      sorted=testmx[order(testmx[,1]),]
      sorted=cbind(sorted,test_1[[name]]$species)
      sorted=sorted[order(sorted[,2]),]
      t=matrix(sorted[sorted[,2]<=perc_cutoff,],ncol=3)
      i=nrow(t)
      #converting percents to percent of whole
      while(i>1){
        t[i,2]=as.character(as.numeric(t[i,2])-as.numeric(t[i-1,2]))
        i=i-1
      }
      t[,1]=name
      write.table(t,file=paste(output_name,'_simper.csv',sep=""), append=TRUE, sep=",", col.names = FALSE)
    }}
  y=read.table(paste(output_name,'_simper.csv',sep=""), header=FALSE,sep=",",fill = TRUE,row.names = NULL)
  file.remove(paste(output_name,'_simper.csv',sep = ""))
  y=y[-c(1)]
  colnames(y) = c("Comparison", "SIMPER", "OTU")
  #removing results lower than low cutoff
  if(low_cutoff=='y'){
    y=y[!(as.numeric(as.character(y$SIMPER))<low_val),]
  }
  #prevents changing of colnames if OTU table
  if(58 %in% save){
    y$OTU=orig_names[match(y$OTU, rownames(orig_names)),1]
  }
  write.csv(y,file=paste(output_name,'_clean_simper.csv', sep=''))
}






kruskal.pretty = function(otu, metrics, csv, interesting, output_name, taxonomy){
  library(vegan)
  library(dplyr)
  if(grepl("Otu", colnames(otu)[1])!=TRUE){
    #converts output from A.Neuman Taxonomy script
    otu=as.data.frame(t(otu))
  }
  #changing csv$X to rownames to allow proper splitting of comparisons
  csv$X=as.integer(rownames(csv))
  L=list()
  R=list()
  mean_L=c()
  sd_L=c()
  mean_R=c()
  sd_R=c()
  L_mean=c()
  R_mean=c()
  L_sd=c()
  R_sd=c()
  krusk=c()
  tax=c()
  L_abund=c()
  R_abund=c()
  L_abund_sd=c()
  R_abund_sd=c()
  abund=as.matrix(otu)
  abund=abund/rowSums(abund)
  for(b in levels(csv$Comparison)){
    otu_list=dplyr::filter(csv, Comparison==b) #saves otu list for current comparison
    for(i in csv$X){
      if(as.character(csv$Comparison[i])==b){  ##splitting comparisons so can call individually for table generation
        splt=as.data.frame(matrix(unlist(strsplit(as.character(csv$Comparison[i]),'_')), nrow=1, byrow=T))
        cola=as.character(splt[1,1])
        colb=as.character(splt[1,2])
        break
      }
    }
    #saving topic containing var of interest (cola/colb) (less memory intensive)
    for(topic in interesting){
      #preventing crash if there is only one topic in interesting
      if(is.null(levels(metrics[[topic]]))==TRUE){
        topic1=topic
        break
      }
      for(sbtpic in levels(metrics[[topic]])){
        if(sbtpic==cola){
          topic1=topic
          break
        }
      } 
    }
    #iterate thru rows in tpics of intrst til matches cola and colb, generates otu and metrics tbl  ##!Processing can be reduced!##
    for(rowe1 in metrics[[topic1]]){
      for(rowe2 in metrics[[topic1]]){ 
        if(rowe1==cola & rowe2==colb){ 
          listbact=otu[c(metrics[[topic1]]==cola|metrics[[topic1]]==colb),]
          listmet=metrics[c(metrics[[topic1]]==cola|metrics[[topic1]]==colb),]
          break
        }
      }
    }
    #collecting differential abundances
    sample_L=row.names(subset(metrics, metrics[[topic1]] == c(cola)))
    sample_R=row.names(subset(metrics, metrics[[topic1]] == c(colb)))
    #collecting abund values, perform/save mean and stdev calculations
    for(otus in otu_list$OTU){
      for(sample in sample_L){
        L=append(L,abund[sample,otus])
        mean_L[[otus]]=mean(as.numeric(L))
        sd_L[[otus]]=sd(as.numeric(L))
      }
      for(sample in sample_R){
        R=append(R,abund[sample,otus])
        mean_R[[otus]]=mean(as.numeric(R))
        sd_R[[otus]]=sd(as.numeric(R))
      }
      L=list()
      R=list()
    }
    #runs kruskal.test for each otu in simper csv, stores as list, also stores abundances
    for(otus in otu_list$OTU){
      result=kruskal.test(listbact[[otus]]~listmet[[topic1]])
      krusk=append(krusk, result$p.value)
      #stores taxonomic classification for each otu as list
      if(missing(taxonomy)){
        tax=append(tax, c("NA"))
      } else {
        tax=append(tax, as.character(taxonomy[otus, "Taxonomy"]))
      }
      L_mean=append(L_mean, as.character(mean_L[[otus]]))
      R_mean=append(R_mean, as.character(mean_R[[otus]]))
      L_sd=append(L_sd, as.character(sd_L[[otus]]))
      R_sd=append(R_sd, as.character(sd_R[[otus]]))
      
    }
  }
  #adjusted p-values for multiple comparisons
  fdr=p.adjust(krusk, method='fdr')
  #order csv to match 'krusk'/'fdr' list, add p.val, add taxonomy, re-ord to match orig csv, write to csv
  o_csv=dplyr::arrange(csv, Comparison)
  o_csv[,5]=krusk
  o_csv[,6]=fdr
  o_csv[,7]=tax
  o_csv[,8]=L_mean
  o_csv[,9]=L_sd
  o_csv[,10]=R_mean
  o_csv[,11]=R_sd
  o_csv=dplyr::arrange(o_csv, X)
  colnames(o_csv)[which(names(o_csv) == "V5")] <- "krusk_p.val" #changes column header
  colnames(o_csv)[which(names(o_csv) == "V6")] <- "fdr_krusk_p.val"
  colnames(o_csv)[which(names(o_csv) == "V7")] <- "Taxonomy"
  colnames(o_csv)[which(names(o_csv) == "V8")] <- "Left mean abund"
  colnames(o_csv)[which(names(o_csv) == "V9")] <- "Left stdev"
  colnames(o_csv)[which(names(o_csv) == "V10")] <- "Right mean abund"
  colnames(o_csv)[which(names(o_csv) == "V11")] <- "Right stdev"
  o_csv[,1]=NULL
  write.csv(o_csv, file=paste(output_name,"_krusk_simper.csv", sep=""))
}

```


### Pathway DESeq
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
path.otu <- otu_table(path, taxa_are_rows = T)
path.otu <- path.otu + 1
path.samp <- sample_data(meta)
path.tax <- tax_table(class)
row.names(path.tax) <- row.names(class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.path <- rownames(test[sum,])
  # ko04145
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])

# Spring vs. Summer
ss.physeq <- subset_samples(path.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
sum <- which(test$log2FoldChange > 0)
sum.path <- rownames(test[sum,])
  # ko00540
  # ko00130

# Spring vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])
  # ko00540
  # ko00130

```


### dbcan family prep
```{r}
### 3715
dbcan.3715 <- read.csv("dbcan/3715.idba.dbCAN.parsed.csv")
dbcan.3717 <- read.csv("dbcan/3717.idba.dbCAN.parsed.csv")
dbcan.3723 <- read.csv("dbcan/3723.idba.dbCAN.parsed.csv")
dbcan.3733 <- read.csv("dbcan/3733.idba.dbCAN.parsed.csv")
dbcan.3734 <- read.csv("dbcan/3734.idba.dbCAN.parsed.csv")
dbcan.3744 <- read.csv("dbcan/3744.idba.dbCAN.parsed.csv")
dbcan.3772 <- read.csv("dbcan/3772.idba.dbCAN.parsed.csv")
dbcan.3773 <- read.csv("dbcan/3773.idba.dbCAN.parsed.csv")
dbcan.3775 <- read.csv("dbcan/3775.idba.dbCAN.parsed.csv")


# Isolate family names
fam.3715 <- substr(dbcan.3715$predicted_protein, 1, 5)
fam.3717 <- substr(dbcan.3717$predicted_protein, 1, 5)
fam.3723 <- substr(dbcan.3723$predicted_protein, 1, 5)
fam.3733 <- substr(dbcan.3733$predicted_protein, 1, 5)
fam.3734 <- substr(dbcan.3734$predicted_protein, 1, 5)
fam.3744 <- substr(dbcan.3744$predicted_protein, 1, 5)
fam.3772 <- substr(dbcan.3772$predicted_protein, 1, 5)
fam.3773 <- substr(dbcan.3773$predicted_protein, 1, 5)
fam.3775 <- substr(dbcan.3775$predicted_protein, 1, 5)


# Remove unwanted categories
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "AA")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "GT")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "SL")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "co")]
fam.3715 <- fam.3715[-which(substr(fam.3715, 1, 2) == "do")]

fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "AA")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "GT")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "SL")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "co")]
fam.3717 <- fam.3717[-which(substr(fam.3717, 1, 2) == "do")]

fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "AA")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "GT")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "SL")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "co")]
fam.3723 <- fam.3723[-which(substr(fam.3723, 1, 2) == "do")]

fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "AA")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "GT")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "SL")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "co")]
fam.3733 <- fam.3733[-which(substr(fam.3733, 1, 2) == "do")]

fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "AA")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "GT")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "SL")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "co")]
fam.3734 <- fam.3734[-which(substr(fam.3734, 1, 2) == "do")]

fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "AA")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "GT")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "SL")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "co")]
fam.3744 <- fam.3744[-which(substr(fam.3744, 1, 2) == "do")]

fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "AA")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "GT")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "SL")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "co")]
fam.3772 <- fam.3772[-which(substr(fam.3772, 1, 2) == "do")]

fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "AA")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "GT")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "SL")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "co")]
fam.3773 <- fam.3773[-which(substr(fam.3773, 1, 2) == "do")]

fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "AA")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "GT")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "SL")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "co")]
fam.3775 <- fam.3775[-which(substr(fam.3775, 1, 2) == "do")]

# Pull out unique families
fam.names.3715 <- unique(fam.3715)
fam.names.3717 <- unique(fam.3717)
fam.names.3723 <- unique(fam.3723)
fam.names.3733 <- unique(fam.3733)
fam.names.3734 <- unique(fam.3734)
fam.names.3744 <- unique(fam.3744)
fam.names.3772 <- unique(fam.3772)
fam.names.3773 <- unique(fam.3773)
fam.names.3775 <- unique(fam.3775)

# Create df
fam.df.3715 <- data.frame(fam.names.3715)
fam.df.3715$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3715)){
  x <- length(which(substr(fam.3715, 1, 5) == fam.names.3715[i]))
  fam.df.3715[i,2] <- x
}

fam.df.3717 <- data.frame(fam.names.3717)
fam.df.3717$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3717)){
  x <- length(which(substr(fam.3717, 1, 5) == fam.names.3717[i]))
  fam.df.3717[i,2] <- x
}

fam.df.3723 <- data.frame(fam.names.3723)
fam.df.3723$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3723)){
  x <- length(which(substr(fam.3723, 1, 5) == fam.names.3723[i]))
  fam.df.3723[i,2] <- x
}

fam.df.3733 <- data.frame(fam.names.3733)
fam.df.3733$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3733)){
  x <- length(which(substr(fam.3733, 1, 5) == fam.names.3733[i]))
  fam.df.3733[i,2] <- x
}

fam.df.3734 <- data.frame(fam.names.3734)
fam.df.3734$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3734)){
  x <- length(which(substr(fam.3734, 1, 5) == fam.names.3734[i]))
  fam.df.3734[i,2] <- x
}

fam.df.3744 <- data.frame(fam.names.3744)
fam.df.3744$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3744)){
  x <- length(which(substr(fam.3744, 1, 5) == fam.names.3744[i]))
  fam.df.3744[i,2] <- x
}

fam.df.3772 <- data.frame(fam.names.3772)
fam.df.3772$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3772)){
  x <- length(which(substr(fam.3772, 1, 5) == fam.names.3772[i]))
  fam.df.3772[i,2] <- x
}

fam.df.3773 <- data.frame(fam.names.3773)
fam.df.3773$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3773)){
  x <- length(which(substr(fam.3773, 1, 5) == fam.names.3773[i]))
  fam.df.3773[i,2] <- x
}

fam.df.3775 <- data.frame(fam.names.3775)
fam.df.3775$count <- 9999
# Add in family counts
for(i in 1:length(fam.names.3775)){
  x <- length(which(substr(fam.3775, 1, 5) == fam.names.3775[i]))
  fam.df.3775[i,2] <- x
}

# Save dataframe
#write.csv(fam.df.3715, "3715.dbcan.fam.csv")
#write.csv(fam.df.3717, "3717.dbcan.fam.csv")
#write.csv(fam.df.3723, "3723.dbcan.fam.csv")
#write.csv(fam.df.3733, "3733.dbcan.fam.csv")
#write.csv(fam.df.3734, "3734.dbcan.fam.csv")
#write.csv(fam.df.3744, "3744.dbcan.fam.csv")
#write.csv(fam.df.3772, "3772.dbcan.fam.csv")
#write.csv(fam.df.3773, "3773.dbcan.fam.csv")
#write.csv(fam.df.3775, "3775.dbcan.fam.csv")


```
