---
title: "OG_MetaG"
author: "Edna Chiang"
date: "March 3, 2017"
output: html_document
---
### Load Libraries
```{r}
library('readr')
library('pgirmess')
library('ggplot2')
library('grid')
library('phyloseq')
library('DESeq2')
library('phangorn')
library('vegan')
library('pheatmap')
library('RColorBrewer')
library('gplots')
library('tidyr')
library('dplyr')
library('stringr')
library('lme4')
library('picante')
theme_set(theme_bw())
set.seed(1)
```

### Load Functions
```{r}
source('scripts/misc.R')
source('scripts/import.phylum.sum.R')
source('scripts/import.class.sum.R')
source('scripts/import.order.sum.R')
source('scripts/import.family.sum.R')
source('scripts/import.genus.sum.R')
source('scripts/import.species.sum.R')
source('scripts/simper.R')
source('scripts/pairwise.adonis.R')
```

### Kaiju - Taxonomic Classification
```{r}
########## Domain ##########
# Load files
domain <- read.csv('kaiju/nohost/domain/domain.csv', header=T)
domain$Season <- ordered(domain$Season, levels=c("Summer", "Winter", "Spring"))
meta <- read.csv('metadata.csv', header=T)
meta$Season <- ordered(meta$Season, levels=c("Summer", "Winter", "Spring"))

# Calculate relative abundance for total sequences
domain.ra <- domain
domain.ra$Archaea <- (domain.ra$Archaea / domain.ra$Total_Sequences)
domain.ra$Bacteria <- (domain.ra$Bacteria / domain.ra$Total_Sequences)
domain.ra$Eukaryota <- (domain.ra$Eukaryota / domain.ra$Total_Sequences)
domain.ra$Viruses <- (domain.ra$Viruses / domain.ra$Total_Sequences)
domain.ra$Unclassified <- (domain.ra$Total_Unclassified / domain.ra$Total_Sequences)

# Calculate relative abundance for classified sequences
domain.clas.ra <- domain
domain.clas.ra$Archaea <- (domain.clas.ra$Archaea / domain.clas.ra$Total_Classified)
domain.clas.ra$Bacteria <- (domain.clas.ra$Bacteria / domain.clas.ra$Total_Classified)
domain.clas.ra$Eukaryota <- (domain.clas.ra$Eukaryota / domain.clas.ra$Total_Classified)
domain.clas.ra$Viruses <- (domain.clas.ra$Viruses / domain.clas.ra$Total_Classified)

# Summarize all sequences
domain.sum <- domain.ra %>%
  group_by(Season) %>%
  summarize(Archaea_Mean = mean(Archaea)*100,
            Bacteria_Mean = mean(Bacteria)*100,
            Eukaryota_Mean = mean(Eukaryota)*100,
            Viruses_Mean = mean(Viruses)*100,
            Unclassified_Mean = mean(Unclassified)*100,
            Archaea_SD = sd(Archaea)*100,
            Bacteria_SD = sd(Bacteria)*100,
            Eukaryota_SD = sd(Eukaryota)*100,
            Viruses_SD = sd(Viruses)*100,
            Unclassified_SD = sd(Unclassified)*100,
            Archaea_SE = se(Archaea)*100,
            Bacteria_SE = se(Bacteria)*100,
            Eukaryota_SE = se(Eukaryota)*100,
            Viruses_SE = se(Viruses)*100,
            Unclassified_SE = se(Unclassified)*100)

# Summarize only classified sequences
domain.clas.sum <- domain.clas.ra %>%
  group_by(Season) %>%
  summarize(Archaea_Mean = mean(Archaea)*100,
            Bacteria_Mean = mean(Bacteria)*100,
            Eukaryota_Mean = mean(Eukaryota)*100,
            Viruses_Mean = mean(Viruses)*100,
            Archaea_SD = sd(Archaea)*100,
            Bacteria_SD = sd(Bacteria)*100,
            Eukaryota_SD = sd(Eukaryota)*100,
            Viruses_SD = sd(Viruses)*100,
            Archaea_SE = se(Archaea)*100,
            Bacteria_SE = se(Bacteria)*100,
            Eukaryota_SE = se(Eukaryota)*100,
            Viruses_SE = se(Viruses)*100)

# Create domain all seqs dataframe for plotting
domain.df <- data.frame(matrix(NA, nrow=15, ncol=5))
colnames(domain.df) <- c("Season", "Domain", "Mean", "SD", "SE")
domain.df$Season <- c(rep("Summer", 5), rep("Winter", 5), rep("Spring", 5))
domain.df$Season <- ordered(domain.df$Season, levels=c("Summer", "Winter", "Spring"))
domain.df$Domain <- c(rep(c("Archaea", "Bacteria", "Eukaryota", "Viruses", "Unclassified"), 3))
domain.df$Domain <- ordered(domain.df$Domain, levels=c("Archaea", "Bacteria", "Eukaryota", "Viruses",
                                                       "Unclassified"))
domain.df$Mean <- as.numeric(c(domain.sum[1,2:6], domain.sum[2, 2:6], domain.sum[3, 2:6]))
domain.df$SD <- as.numeric(c(domain.sum[1, 7:11], domain.sum[2, 7:11], domain.sum[3, 7:11]))
domain.df$SE <- as.numeric(c(domain.sum[1, 12:16], domain.sum[2, 12:16], domain.sum[3, 12:16]))

# Create domain classified seqs dataframe for plotting
domain.clas.df <- data.frame(matrix(NA, nrow=12, ncol=5))
colnames(domain.clas.df) <- c("Season", "Domain", "Mean", "SD", "SE")
domain.clas.df$Season <- c(rep("Summer", 4), rep("Winter", 4), rep("Spring", 4))
domain.clas.df$Season <- ordered(domain.clas.df$Season, levels=c("Summer", "Winter", "Spring"))
domain.clas.df$Domain <- c(rep(c("Archaea", "Bacteria", "Eukaryota", "Viruses"), 3))
domain.clas.df$Domain <- ordered(domain.clas.df$Domain, levels=c("Archaea", "Bacteria", "Eukaryota",
                                                                "Viruses"))
domain.clas.df$Mean <- as.numeric(c(domain.clas.sum[1,2:5], domain.clas.sum[2, 2:5],
                                    domain.clas.sum[3, 2:5]))
domain.clas.df$SD <- as.numeric(c(domain.clas.sum[1, 6:9], domain.clas.sum[2, 6:9],
                                  domain.clas.sum[3, 6:9]))
domain.clas.df$SE <- as.numeric(c(domain.clas.sum[1, 10:13], domain.clas.sum[2, 10:13],
                                  domain.clas.sum[3, 10:13]))

# Barplot for all sequences
ggplot(domain.df, aes(x=Domain, y=Mean, fill=Domain)) +
  geom_bar(stat="identity") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.5,60)) +
  facet_grid(Season ~ .) +
  scale_fill_brewer(palette="Set1") +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))

# Barplot for classified sequences
ggplot(domain.clas.df, aes(x=Domain, y=Mean, fill=Domain)) +
  geom_bar(stat="identity") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.5,100)) +
  facet_grid(Season ~ .) +
  scale_fill_brewer(palette="Set1") +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))



########## Phylum ##########
phy.sum <- import.phylum.sum('kaiju/nohost/phylum')

# Summarize phyla to get top 10
phy.sum.sum <- phy.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))
  # Top 10 phyla = Unclassified, Bacteroidetes, Firmicutes, Verrucomicrobia, Proteobacteria, Actinobacteria, Candidatus Melainabacteria, Spirochaetes, Tenericutes, Elusimicrobia (and Fusobacteria)

# Only look at top 10 phyla
top.total <- c("Unclassified", "Bacteroidetes", "Firmicutes", "Verrucomicrobia", "Proteobacteria", "Actinobacteria", "Candidatus Melainabacteria", "Spirochaetes", "Tenericutes", "Elusimicrobia")
clas.total <- c("Bacteroidetes", "Firmicutes", "Verrucomicrobia", "Proteobacteria", "Actinobacteria", "Candidatus Melainabacteria", "Spirochaetes", "Tenericutes", "Elusimicrobia", "Fusobacteria")

# Pull out top 10 phyla - Including Unclassified
phy.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(phy.total) <- c("Season", "Phylum", "Mean", "SD", "SE")
for (i in 1:length(top.total)){
  top <- which(phy.sum$Phylum == top.total[i])
  for (j in 1:length(top)){
    phy.total$Season[(j + 3*(i-1))] <- as.character(phy.sum$Season[top[j]])
    phy.total$Phylum[(j + 3*(i-1))] <- phy.sum$Phylum[top[j]]
    phy.total$Mean[(j + 3*(i-1))] <- phy.sum$Mean[top[j]]
    phy.total$SD[(j + 3*(i-1))] <- phy.sum$SD[top[j]]
    phy.total$SE[(j + 3*(i-1))] <- phy.sum$SE[top[j]]
  }
}
phy.total$Phylum <- ordered(phy.total$Phylum, levels=c("Unclassified", "Bacteroidetes", "Firmicutes", "Verrucomicrobia", "Proteobacteria", "Actinobacteria", "Candidatus Melainabacteria", "Spirochaetes", "Tenericutes", "Elusimicrobia"))
phy.total$Season <- ordered(phy.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 10 phyla (including unclassified)
ggplot(phy.total, aes(x=Phylum, y=Mean, fill=Phylum)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.1,63)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 10 phyla (Only classified)
phy.clas.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(phy.clas.total) <- c("Season", "Phylum", "Mean", "SD", "SE")
for (i in 1:length(clas.total)){
  top <- which(phy.sum$Phylum == clas.total[i])
  for (j in 1:length(top)){
    phy.clas.total$Season[(j + 3*(i-1))] <- as.character(phy.sum$Season[top[j]])
    phy.clas.total$Phylum[(j + 3*(i-1))] <- phy.sum$Phylum[top[j]]
    phy.clas.total$Mean[(j + 3*(i-1))] <- phy.sum$Mean[top[j]]
    phy.clas.total$SD[(j + 3*(i-1))] <- phy.sum$SD[top[j]]
    phy.clas.total$SE[(j + 3*(i-1))] <- phy.sum$SE[top[j]]
  }
}
phy.clas.total$Phylum <- ordered(phy.clas.total$Phylum, levels=c("Bacteroidetes", "Firmicutes", "Verrucomicrobia", "Proteobacteria", "Actinobacteria", "Candidatus Melainabacteria", "Spirochaetes", "Tenericutes", "Elusimicrobia", "Fusobacteria"))
phy.clas.total$Season <- ordered(phy.clas.total$Season, levels=c("Summer", "Winter", "Spring"))


# Barplot for top 10 phyla (only classified)
ggplot(phy.clas.total, aes(x=Phylum, y=Mean, fill=Phylum)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.1,45)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Phyla") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))




########## Class ##########
cla.sum <- import.class.sum('kaiju/nohost/class')

# Summarize classes to get top 10
cla.sum.sum <- cla.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))

# Only look at top 10 classes
top.class <- c("Bacteroidia", "Clostridia", "Verrucomicrobiae", "Deltaproteobacteria", "Bacilli", "Alphaproteobacteria", "Erysipelotrichia", "Negativicutes", "Coriobacteriia", "Gammaproteobacteria")

# Pull out top 10 classes - Including Unclassified
cla.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(cla.total) <- c("Season", "Class", "Mean", "SD", "SE")
for (i in 1:length(top.class)){
  top <- which(cla.sum$Class == top.class[i])
  for (j in 1:length(top)){
    cla.total$Season[(j + 3*(i-1))] <- as.character(cla.sum$Season[top[j]])
    cla.total$Class[(j + 3*(i-1))] <- cla.sum$Class[top[j]]
    cla.total$Mean[(j + 3*(i-1))] <- cla.sum$Mean[top[j]]
    cla.total$SD[(j + 3*(i-1))] <- cla.sum$SD[top[j]]
    cla.total$SE[(j + 3*(i-1))] <- cla.sum$SE[top[j]]
  }
}
cla.total$Class <- ordered(cla.total$Class, levels=c("Bacteroidia", "Clostridia", "Verrucomicrobiae", "Deltaproteobacteria", "Bacilli", "Alphaproteobacteria", "Erysipelotrichia", "Negativicutes", "Coriobacteriia", "Gammaproteobacteria"))
cla.total$Season <- ordered(cla.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 10 classes (no unclassified)
ggplot(cla.total, aes(x=Class, y=Mean, fill=Class)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.1,42)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Classes") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))




########## Order ##########
ord.sum <- import.order.sum('kaiju/nohost/order')

# Summarize orders to get top 10
ord.sum.sum <- ord.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))

# Only look at top 10 orders
top.order <- c("Bacteroidales", "Clostridiales", "Verrucomicrobiales", "Desulfovibrionales", "Lactobacillales", "Erysipelotrichales", "Rhodospirillales", "Bacillales", "Acidaminococcales", "Flavobacteriales")

# Pull out top 10 orders - Including Unclassified
ord.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(ord.total) <- c("Season", "Order", "Mean", "SD", "SE")
for (i in 1:length(top.order)){
  top <- which(ord.sum$Order == top.order[i])
  for (j in 1:length(top)){
    ord.total$Season[(j + 3*(i-1))] <- as.character(ord.sum$Season[top[j]])
    ord.total$Order[(j + 3*(i-1))] <- ord.sum$Order[top[j]]
    ord.total$Mean[(j + 3*(i-1))] <- ord.sum$Mean[top[j]]
    ord.total$SD[(j + 3*(i-1))] <- ord.sum$SD[top[j]]
    ord.total$SE[(j + 3*(i-1))] <- ord.sum$SE[top[j]]
  }
}
ord.total$Order <- ordered(ord.total$Order, levels=c("Bacteroidales", "Clostridiales", "Verrucomicrobiales", "Desulfovibrionales", "Lactobacillales", "Erysipelotrichales", "Rhodospirillales", "Bacillales", "Acidaminococcales", "Flavobacteriales"))
ord.total$Season <- ordered(ord.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 10 orders (no unclassified)
ggplot(ord.total, aes(x=Order, y=Mean, fill=Order)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.1,42)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Orders") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))





########## Family ##########
fam.sum <- import.family.sum('kaiju/nohost/family')

# Summarize families to get top 10
fam.sum.sum <- fam.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

# Only look at top 10 orders
top.family <- c("Bacteroidaceae", "Lachnospiraceae", "Prevotellaceae", "Akkermansiaceae", "Tannerellaceae", "Ruminococcaceae", "Muribaculaceae", "Rikenellaceae", "Clostridiaceae", "Eubacteriaceae")

# Pull out top 10 families - Including Unclassified
fam.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(fam.total) <- c("Season", "Family", "Mean", "SD", "SE")
for (i in 1:length(top.family)){
  top <- which(fam.sum$Family == top.family[i])
  for (j in 1:length(top)){
    fam.total$Season[(j + 3*(i-1))] <- as.character(fam.sum$Season[top[j]])
    fam.total$Family[(j + 3*(i-1))] <- fam.sum$Family[top[j]]
    fam.total$Mean[(j + 3*(i-1))] <- fam.sum$Mean[top[j]]
    fam.total$SD[(j + 3*(i-1))] <- fam.sum$SD[top[j]]
    fam.total$SE[(j + 3*(i-1))] <- fam.sum$SE[top[j]]
  }
}
fam.total$Family <- ordered(fam.total$Family, levels=c("Bacteroidaceae", "Lachnospiraceae", "Prevotellaceae", "Akkermansiaceae", "Tannerellaceae", "Ruminococcaceae", "Muribaculaceae", "Rikenellaceae", "Clostridiaceae", "Eubacteriaceae"))
fam.total$Season <- ordered(fam.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 10 orders (no unclassified)
ggplot(fam.total, aes(x=Family, y=Mean, fill=Family)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.3,16)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Families") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))






########## Genus ##########
gen.sum <- import.genus.sum('kaiju/nohost/genus')

# Summarize genera to get top 10
gen.sum.sum <- gen.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 10 genera
top.genus <- c("Bacteroides", "Prevotella", "Akkermansia", "Parabacteroides", "Alistipes", "Clostridium", "Eubacterium", "Ruminococcus", "Oscillibacter", "Roseburia")

# Pull out top 10 genera - Including Unclassified
gen.total <- data.frame(matrix(nrow=30, ncol=5))
colnames(gen.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.genus)){
  top <- which(gen.sum$Genus == top.genus[i])
  for (j in 1:length(top)){
    gen.total$Season[(j + 3*(i-1))] <- as.character(gen.sum$Season[top[j]])
    gen.total$Genus[(j + 3*(i-1))] <- gen.sum$Genus[top[j]]
    gen.total$Mean[(j + 3*(i-1))] <- gen.sum$Mean[top[j]]
    gen.total$SD[(j + 3*(i-1))] <- gen.sum$SD[top[j]]
    gen.total$SE[(j + 3*(i-1))] <- gen.sum$SE[top[j]]
  }
}
gen.total$Genus <- ordered(gen.total$Genus, levels=c("Bacteroides", "Prevotella", "Akkermansia", "Parabacteroides", "Alistipes", "Clostridium", "Eubacterium", "Ruminococcus", "Oscillibacter", "Roseburia"))
gen.total$Season <- ordered(gen.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 10 orders (no unclassified)
ggplot(gen.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.2,16)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))
```









### MicrobeCensus - Average Genome Size
```{r}
# Load data
mc <- read.csv('MicrobeCensus/Summary.csv', row.names = 1)
mc.ma <- as.matrix(mc)

# Test normality
hist(mc.ma[2,])
qqnorm(mc.ma[2,])
qqline(mc.ma[2,])
shapiro.test(mc.ma[2,])
  # p-value = 0.03487
  # Not Normal

# Format for Kruskal-Wallis
mc <- t(mc)
mc <- data.frame(mc)
mc$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
mc$Season <- ordered(mc$Season, levels=c("Summer", "Winter", "Spring"))

# Kruskal-Wallis
kruskal.test(formula = AGS ~ Season, data = mc)
  # p-value = 0.4298
```

### dbCAN - Proportion CAZymes from ORFs
```{r}
prodigalCounts <- read.csv('dbcan/prodigalCounts.csv', row.names=1)
prodigalCounts <- as.matrix(prodigalCounts)

# Test normality
hist(prodigalCounts[3,])
qqnorm(prodigalCounts[3,])
qqline(prodigalCounts[3,])
shapiro.test(prodigalCounts[3,])
  # p-value = 0.001575
  # Not Normal

# Format for Kruskal-Wallis
prodigalCounts <- read.csv('dbcan/prodigalCounts.csv', row.names=1)
prodigalCounts <- t(prodigalCounts)
prodigalCounts <- data.frame(prodigalCounts)
prodigalCounts$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
prodigalCounts$Season <- ordered(prodigalCounts$Season, levels=c("Summer", "Winter", "Spring"))
prodigalCounts$Diet <- c("Feeding", "Feeding", "Fasting", "Fasting", "Feeding", "Feeding", "Fasting", "Feeding", "Feeding")
prodigalCounts$Diet <- ordered(prodigalCounts$Diet, levels=c("Feeding", "Fasting"))
colnames(prodigalCounts) <- c("Total_ORFs", "Total_CAZymes", "CAZymePercentage", "Season", "Diet")


# Test w/ Kruskal-Wallis
kruskal.test(formula = CAZymePercentage ~ Season, data = prodigalCounts)
  # p-value = 0.5611
kruskal.test(formula = CAZymePercentage ~ Diet, data = prodigalCounts)
  # p-value = 0.4386

### Dot plot
#tiff("dbcan.prop.tiff", width=5, height=4, units="in", res= 300)
ggplot(prodigalCounts, aes(x=Season, y=CAZymePercentage, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as CAZyme (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(2.0, 2.6, 3.2, 3.8, 4.4), limits=c(2.0, 4.4)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
#dev.off()
```

### dbCAN - Create phyloseq object
```{r}
# Load data
otu <- read.csv('dbcan/dbcan_otu_table.csv', row.names=1)
colnames(otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
otu.tab <- otu_table(otu, taxa_are_rows=T)
tax <- read.csv('dbcan/dbcan_tax_table.csv', row.names=1)
tax <- tax_table(tax)
colnames(tax) <- c("Class", "Family")
taxa_names(tax) <- tax[,2]
meta <- read.csv('dbcan/dbcan.spades.metadata.csv', row.names=1)
meta.ord <- meta

# Create phyloseq object
dbcan.phy <- merge_phyloseq(otu.tab, tax, sample_data(meta.ord))

# Trim CAZymes w/ count < 4
# 3 samples per season, so 4 seemed like a somewhat reasonable arbitrary cutoff
# Cutoff = 4 also ensures that no normalized counts < 3, so no further cutoffs needed
dbcan.cut.phy <- prune_taxa(taxa_sums(dbcan.phy) > 3, dbcan.phy)

# Identify trimmed CAZymes (count < 4)
#all <- rownames(tax_table(dbcan.phy))
#remain <- rownames(tax_table(dbcan.cut.phy))
#for (i in all){
#  if (i %in% remain == F){
#    print(i)
#  }
#}
# GH12, GH17, GH34, GH52, GH100, GH107, GH118, GH149, GH153, GT29, GT37, GT42, GT52, GT58, GT60, GT67, GT68, GT77, GT85, GT93, GT97, CE5, CE13, PL2, PL14, PL16, PL18, PL28


# Normalized Counts At Family-level
# https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
#otu.cut <- data.frame(otu_table(dbcan.cut.phy))
#dbcan.dds <- DESeqDataSetFromMatrix(countData = otu.cut, colData = meta, design = ~ Season)
#dbcan.dds <- estimateSizeFactors(dbcan.dds)
#dbcan.norm <- counts(dbcan.dds, normalized=T)
#write.csv(dbcan.norm, file="dbcan/dbcan.Normalized.Counts.Family.csv")

dbcan.fam.norm <- read.csv('dbcan/dbcan.Normalized.Counts.Family.csv', row.names=1)
dbcan.fam.norm.otu <- otu_table(dbcan.fam.norm, taxa_are_rows=T)
sample_names(dbcan.fam.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
dbcan.fam.norm.phy <- merge_phyloseq(dbcan.fam.norm.otu, sample_data(dbcan.phy), tax_table(dbcan.phy))

# Normalize Counts to Class-Level
#dbcan.class <- tax_glom(dbcan.cut.phy, taxrank="Class")
#dbcan.class.otu <- data.frame(otu_table(dbcan.class))
#dbcan.dds <- DESeqDataSetFromMatrix(countData = dbcan.class.otu, colData = meta, design = ~ Season)
#dbcan.dds <- estimateSizeFactors(dbcan.dds)
#dbcan.norm <- counts(dbcan.dds, normalized=T)
#write.csv(dbcan.norm, file="dbcan/dbcan.Normalized.Counts.Class.csv")

dbcan.class.norm <- read.csv('dbcan/dbcan.Normalized.Counts.Class.csv', row.names=1)
rownames(dbcan.class.norm) <- c("Polysaccharide Lyase", "Glycosyltransferase", "Glycoside Hydrolase", "Carbohydrate Esterase")

# Pairwise season physeq objects
#dbcan.sw <- subset_samples(dbcan99, Season != "Spring")
  # Summer & Winter
#dbcan.sp <- subset_samples(dbcan99, Season != "Winter")
  # Summer & Spring
#dbcan.wp <- subset_samples(dbcan99, Season != "Summer")
  # Winter & Spring
```
### dbCAN - Count CAZyme families
```{r}
# Subset out samples by season
phy.sum <- subset_samples(dbcan.cut.phy, Season == "Summer")
phy.win <- subset_samples(dbcan.cut.phy, Season == "Winter")
phy.spr <- subset_samples(dbcan.cut.phy, Season == "Spring")

# Remove CAZyme families with no counts
phy.sum <- prune_taxa(taxa_sums(phy.sum) > 0, phy.sum)
phy.win <- prune_taxa(taxa_sums(phy.win) > 0, phy.win)
phy.spr <- prune_taxa(taxa_sums(phy.spr) > 0, phy.spr)


# Pull out unique CAZyme families in each season and order alphabetically
sum.fam <- get_taxa_unique(phy.sum, "Family")
sum.fam <- sort(sum.fam)
win.fam <- get_taxa_unique(phy.win, "Family")
win.fam <- sort(win.fam)
spr.fam <- get_taxa_unique(phy.spr, "Family")
spr.fam <- sort(spr.fam)


# Summer - Count # of CAZymes in each class
class <- substr(sum.fam,1,2)
length(which(class == "GH"))
  # 122
length(which(class == "GT"))
  # 60
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 26


# Winter - Count # of CAZymes in each class
class <- substr(win.fam,1,2)
length(which(class == "GH"))
  # 117
length(which(class == "GT"))
  # 60
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 25


# Spring - Count # of CAZymes in each class
class <- substr(spr.fam,1,2)
length(which(class == "GH"))
  # 124
length(which(class == "GT"))
  # 63
length(which(class == "CE"))
  # 13
length(which(class == "PL"))
  # 26

```

### dbCAN - Class-level bar plot
```{r}
# Create phyloseq object to summarize data
class.tax <- data.frame(c("Carbohydrate Esterase", "Glycoside Hydrolase", "Glycosyltransferase", 
                          "Polysaccharide Lyase"))
class.tax <- tax_table(class.tax)
row.names(class.tax) <- class.tax[,1]
colnames(class.tax) <- "Class"
dbcan.class.norm.otu <- otu_table(dbcan.class.norm, taxa_are_rows=T)
colnames(dbcan.class.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
dbcan.class.phy <- merge_phyloseq(dbcan.class.norm.otu, class.tax, sample_data(meta.ord))

# Summarize data
dbcan.class.df <- psmelt(dbcan.class.phy)
dbcan.class.sum <- dbcan.class.df %>%
  group_by(Class, Season) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
dbcan.class.sum$Class <- ordered(dbcan.class.sum$Class, c("Glycoside Hydrolase",
                                                                  "Glycosyltransferase",
                                                                  "Carbohydrate Esterase",
                                                                  "Polysaccharide Lyase"))
dbcan.class.sum$Season <- ordered(dbcan.class.sum$Season, levels = c("Summer", "Winter", "Spring"))

# Plot barplot
#tiff("dbcan.class.tiff", width=10, height=4, units="in", res= 300)
ggplot(dbcan.class.sum, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,9500)) +
  facet_grid(. ~ Class) +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab("CAZyme Class") +
  ylab("Mean Normalized Counts") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"),
        panel.spacing = unit(0.5, "lines"),
        axis.title.x = element_text(margin = margin(t = 20)))
#dev.off()




# # Summarize data
# dbcan.class.diet.sum <- dbcan.class.df %>%
#   group_by(Class, Diet) %>%
#   summarize(Mean = mean(Abundance),
#             SD = sd(Abundance),
#             SE = se(Abundance))
# dbcan.class.diet.sum$Class <- ordered(dbcan.class.diet.sum$Class, c("Glycoside Hydrolase",
#                                                                   "Glycosyltransferase",
#                                                                   "Carbohydrate Esterase",
#                                                                   "Polysaccharide Lyase"))
# dbcan.class.diet.sum$Diet <- ordered(dbcan.class.diet.sum$Diet, levels=c("Feeding", "Fasting"))
# 
# # Plot barplot
# ggplot(dbcan.class.diet.sum, aes(x=Diet, y=Mean, fill=Diet)) +
#   geom_bar(stat="identity", color="black") +
#   geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
#   scale_y_continuous(limits=c(0,9500)) +
#   facet_grid(. ~ Class) +
#   scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3")) +
#   xlab("CAZyme Class") +
#   ylab("Mean Normalized Counts") +
#   theme(panel.grid.major = element_blank(),
#         panel.grid.minor = element_blank(),
#         strip.background = element_blank(),
#         strip.text = element_text(face="bold"),
#         strip.text.x = element_text(size=11),
#         legend.title = element_text(face="bold"),
#         panel.spacing = unit(0.5, "lines"),
#         axis.title.x = element_text(margin = margin(t = 15)))
```

### dbCAN - Test class-level
```{r}
# Same results if you compare Diet (Feeding vs. Fasting)
### CE
ce <- dbcan.class.df[which(dbcan.class.df$Class == "Carbohydrate Esterase"),]

# Test normality
hist(ce$Abundance)
qqnorm(ce$Abundance)
qqline(ce$Abundance)
shapiro.test(ce$Abundance)
  # p-value = 0.6811
  # Normal

# Test CE w/ ANOVA
summary(aov(Abundance ~ Season, data = ce))
  # p-value = 0.332



### GH
gh <- dbcan.class.df[which(dbcan.class.df$Class == "Glycoside Hydrolase"),]

# Test normality
hist(gh$Abundance)
qqnorm(gh$Abundance)
qqline(gh$Abundance)
shapiro.test(gh$Abundance)
  # p-value = 0.4749
  # Normal

# Test GH w/ ANOVA
summary(aov(Abundance ~ Season, data = gh))
  # p-value = 0.00789

# GH posthoc test w/ Tukey's
TukeyHSD(aov(Abundance ~ Season, data = gh))
  # Summer-Spring = 0.0297554*
  # Winter-Spring = 0.0076492**
  # Winter-Summer = 0.4762540



### GT
gt <- dbcan.class.df[which(dbcan.class.df$Class == "Glycosyltransferase"),]

# Test normality
hist(gt$Abundance)
qqnorm(gt$Abundance)
qqline(gt$Abundance)
shapiro.test(gt$Abundance)
  # p-value = 0.4515
  # Normal

# Test GT w/ ANOVA
summary(aov(Abundance ~ Season, data = gt))
  # p-value = 0.0074

# GT posthoc test w/ Tukey's
TukeyHSD(aov(Abundance ~ Season, data = gt))
  # Summer-Spring = 0.3591323
  # Winter-Spring = 0.0067560**
  # Winter-Summer = 0.0347394*



### PL
pl <- dbcan.class.df[which(dbcan.class.df$Class == "Polysaccharide Lyase"),]

# Test normality
hist(pl$Abundance)
qqnorm(pl$Abundance)
qqline(pl$Abundance)
shapiro.test(pl$Abundance)
  # p-value = 0.08667
  # Normal

# Test PL w/ ANOVA
summary(aov(Abundance ~ Season, data = pl))
  # p-value = 0.116




### Adjust Pvals
p.adjust(c(0.332, 0.00789, 0.0074, 0.116), method="BH")
  # CE = 0.33200
  # GH = 0.01578 *
  # GT = 0.01578 *
  # PL = 0.15467


```












### dbCAN - Family Bray-Curtis NMDS
```{r}
# Create family-level physeq object
colnames(dbcan.fam.norm) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
dbcan.fam.phy <- merge_phyloseq(otu_table(dbcan.fam.norm, taxa_are_rows=T), tax, sample_data(meta.ord))
sample_data(dbcan.fam.phy)$Season <- ordered(sample_data(dbcan.fam.phy)$Season, levels=c("Summer", "Winter", "Spring"))

#dbcan.nmds <- ordinate(physeq = dbcan.fam.phy, method = "NMDS", distance = "bray")
  # stress = 0.0731986
#save(dbcan.nmds, file="dbcan.nmds.RData")
load("dbcan.nmds.RData")

### NMDS Ellipses
sample_data(dbcan.fam.phy)$Elip <- paste(sample_data(dbcan.fam.phy)$Season)
dbcan.elip <- data.frame(MDS1 = dbcan.nmds$points[,1], MDS2 = dbcan.nmds$points[,2],
                           Group = sample_data(dbcan.fam.phy)$Season)
plot.new()
dbcan.elip.ord <- ordiellipse(dbcan.nmds, sample_data(dbcan.fam.phy)$Elip, display="sites", kind="se", conf=0.95, label=T)
dbcan.elip.df <- data.frame()
for(i in levels(dbcan.elip$Group)){
  dbcan.elip.df <- rbind(dbcan.elip.df,
                           cbind(as.data.frame(with(dbcan.elip[dbcan.elip$Group==i,],
                                                    veganCovEllipse(dbcan.elip.ord[[i]]$cov,
                                                                    dbcan.elip.ord[[i]]$center,
                                                                    dbcan.elip.ord[[i]]$scale))), group=i
                                 ))
}


### NMDS Ordination
#tiff("dbcan.nmds.tiff", width=8, height=6, units="in", res=300)
plot_ordination(physeq = dbcan.fam.phy, ordination = dbcan.nmds, color = "Season") +
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  geom_path(data=dbcan.elip.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  ggtitle("CAZyme Composition") +
  geom_point(size=3) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=20),
        axis.text.y = element_text(size=20),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        )
#dev.off()



```

### dbCAN - Family Bray-Curtis PCoA
```{r}
dbcan.pcoa <- ordinate(physeq = dbcan.fam.phy, method="PCoA", distance="bray")
plot_ordination(physeq = dbcan.fam.phy, ordination = dbcan.pcoa, axes = c(1,2),
              color = "Season")
  # Axis 1 + 2 = 41.9 + 29.7 = 71.6%

```


### dbCAN - Family Bray-Curtis PERMANOVA
```{r}
### PERMANOVA
dbcan.bray <- phyloseq::distance(dbcan.fam.phy, method="bray")

# Season
adonis(dbcan.bray~Season, data=meta, permutations = 99999)
  # p-value = 0.00725
  # R2 = 0.53003
beta.tax = betadisper(d=dbcan.bray, group=meta$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.546


# Diet
adonis(dbcan.bray~Diet, data=meta, permutations = 99999)
  # p-value = 0.01251
  # R2 = 0.32065
beta.tax = betadisper(d=dbcan.bray, group=meta$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.149


# Sex
adonis(dbcan.bray~Sex, data=meta, permutations = 99999)
  # p-value = 0.3818
  # R2 = 0.13182
beta.tax = betadisper(d=dbcan.bray, group=meta$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.894



### Pairwise PERMANOVA - Season
pairwise.adonis(as.matrix(t(otu_table(dbcan.fam.phy))), meta$Season, sim.method="bray", p.adjust.m="fdr")
  # Summer-Winter: adj-pval=0.1, R2 = 0.15
  # Summer-Spring: adj-pval=0.2, R2 = 0.20
  # Winter-Spring: adj-pval=0.1, R2 = 0.15


### Global PERMANOVA is sig, but not pairwise PERMANOVA-- why?
# https://help.xlstat.com/s/article/how-to-interpret-contradictory-results-between-anova-and-multiple-pairwise-comparisons?language=en_US
# Likely due to lack of statistical power (small n)
```


### dbCAN - Family UPGMA
```{r}
### Create Kendall's Tau Correlation Matrix
# Using Kendall's instead of Spearman's because Kendall is more accurate w/ smaller n's
# https://www.statisticssolutions.com/kendalls-tau-and-spearmans-rank-correlation-coefficient/
# http://www.r-tutor.com/gpu-computing/correlation/kendall-rank-coefficient
dbcan.cor <- cor(data.frame(otu_table(dbcan.fam.phy)), method = "kendall", use="pairwise")

### UPGMA
dbcan.upgma <- upgma(dbcan.cor)
# Rename sample names to make interpreting tree easier
dbcan.upgma$tip.label <- c("Summer_F_1", "Summer_F_1", "Winter_M_2", "Winter_F_3", "Spring_M_3", "Summer_M_4", "Winter_M_5", "Spring_F_5", "Spring_M_5")
  # Last number refers to litter

### Plot
# http://www.phytools.org/Cordoba2017/ex/2/Intro-to-phylogenies.html
plot(dbcan.upgma, edge.width=2, type="cladogram")
```


### dbCAN DESeq
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
# https://compbiocore.github.io/deseq-workshop-1/assets/deseq_workshop_1.html
# https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html
# Use UN-NORMALIZED data!!!
  # DESeq accounts for library size


### Test DESeq2 to determine fit type
# https://support.bioconductor.org/p/81094/
#dbcan.dds <- phyloseq_to_deseq2(dbcan.cut.phy, ~Season)
#dbcan.dds$Season <- factor(dbcan.dds$Season, levels=c("Summer", "Winter", "Spring"))
#dbcan.dds <- dbcan.dds[rowSums(counts(dbcan.dds)) > 1,]
#dbcan.local <- DESeq(dbcan.dds, fitType = "local")
#dbcan.par <- DESeq(dbcan.dds, fitType = "parametric")
#mad(log(mcols(dbcan.local)$dispGeneEst) - log(mcols(dbcan.local)$dispFit))
  # 9.744673
#mad(log(mcols(dbcan.par)$dispGeneEst) - log(mcols(dbcan.par)$dispFit))
  # 7.288668
# Parametric is a better fit (lower median absolute residual) than local

### Convert phyloseq object to deseq2 object
dbcan.dds <- phyloseq_to_deseq2(dbcan.cut.phy, ~Season)
  # Design variable must be unordered otherwise you get an error
dbcan.dds$Season <- factor(dbcan.dds$Season, levels=c("Summer", "Winter", "Spring"))

# DESeq DataStructure for analyses
dbcan.dds <- DESeq(dbcan.dds)
#dbcan.rl <- rlog(dbcan.dds, blind=F)

# Pull out results from pairwise comparison
dbcan.sw <- results(dbcan.dds, contrast=c("Season", "Summer", "Winter"))
dbcan.sw.sig <- dbcan.sw[which(dbcan.sw$padj < 0.05),]
  # Enriched in Summer compared to Winter (positive log2FoldChange)
    # GT101, PL1, GH43, GH115, CE12, GH10, GH59
dbcan.sp <- results(dbcan.dds, contrast=c("Season", "Summer", "Spring"))
dbcan.sp.sig <- dbcan.sp[which(dbcan.sp$padj < 0.05),]
  # Enriched in Summer compared to Spring (positive log2FoldChange)
    # GT101
dbcan.wp <- results(dbcan.dds, contrast=c("Season", "Winter", "Spring"))
dbcan.wp.sig <- dbcan.wp[which(dbcan.wp$padj < 0.05),]
dbcan.wp.sig[which(dbcan.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
      # GT41, GH20, GT4, GT9, CE11, GT3, GH84
dbcan.wp.sig[which(dbcan.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH3, GH32, GH43, GH5, GH78, GH115, GH42, CE12, GH106, GH10, GH39
```

### dbCAN - DESeq Heatmap
```{r}
# Note: I tried merging the samples by season so the heatmap only displays Summer, Winter, and Spring, but I get an error that say I have insufficient data. Therefore, I display each sample separately.

# Excluding some families because they has such high counts that it makes the heatmap hard to interpret visually
# I'll make a separate plot for just GH43, GH3, GT4, and GT41


dbcan.sig <- subset_taxa(dbcan.fam.norm.phy, Family == "GT101" | Family == "GH59" | Family == "PL1" |
                           Family == "GH10" | Family == "GH115" | Family == "CE12" | Family == "GH5" |
                           Family == "GH32" | Family == "GH39" | Family == "GH42" | Family == "GH78" |
                           Family == "GH106" | Family == "GH20" | Family == "GH84" | Family == "GT3" |
                           Family == "GH9" | Family == "CE11")

#tiff("dbcan.heatmap.tiff", width=9, height=10, units="in", res=300)
plot_heatmap(dbcan.sig, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("CE11", "GH9", "GT3", "GH84", "GH20", "GH106", "GH78", "GH42", "GH39", "GH32", "GH5", "CE12", "GH115", "GH10", "PL1", "GH59", "GT101")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("CAZyme Family\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"), limits=c(0,300),
                       breaks=c(0, 100, 200, 300), 
                       labels=c("0", "100", "200", "300"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(7,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
#dev.off()

### GH43, GH3, GT4, and GT41 plot
dbcan.high <- subset_taxa(dbcan.fam.norm.phy, Family == "GH43" | Family == "GH3" | Family == "GT4" |
                            Family == "GT41")
tiff("dbcan.high.heatmap.tiff", width=9, height=3.3, units="in", res=300)
plot_heatmap(dbcan.high, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("GT41", "GT4", "GH3", "GH43")) +
  geom_tile(colour="gray40") +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("CAZyme Family\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"), limits=c(300,1200),
                       breaks=c(300, 600, 900, 1200), 
                       labels=c("300", "600", "900", "1200"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
   theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(2,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
dev.off()

```



### dbCAN - Kaiju on GH10
```{r}
########## Phylum ##########
phy.GH10.sum <- import.phylum.sum('kaiju/sep_sig_cazymes/GH10/phylum',4)

# Summarize phyla to get top 10
phy.GH10.sum.sum <- phy.GH10.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
cla.GH10.sum <- import.class.sum('kaiju/sep_sig_cazymes/GH10/class', 7)

# Summarize classes to get top 10
cla.GH10.sum.sum <- cla.GH10.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
ord.GH10.sum <- import.order.sum('kaiju/sep_sig_cazymes/GH10/order', 8)

# Summarize orders to get top 10
ord.GH10.sum.sum <- ord.GH10.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
fam.GH10.sum <- import.family.sum('kaiju/sep_sig_cazymes/GH10/family', 12)

# Summarize families to get top 10
fam.GH10.sum.sum <- fam.GH10.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
gen.GH10.sum <- import.genus.sum('kaiju/sep_sig_cazymes/GH10/genus',11)

# Summarize genera to get top 10
gen.GH10.sum.sum <- gen.GH10.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))


########## Species ##########
spe.GH10.sum <- import.species.sum('kaiju/sep_sig_cazymes/GH10/species',16)

# Summarize species to get top 10
spe.GH10.sum.sum <- spe.GH10.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```


### dbCAN - Kaiju on GH20
```{r}
########## Phylum ##########
phy.GH20.sum <- import.phylum.sum('kaiju/sep_sig_cazymes/GH20/phylum',6)

# Summarize phyla to get top 10
phy.GH20.sum.sum <- phy.GH20.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
cla.GH20.sum <- import.class.sum('kaiju/sep_sig_cazymes/GH20/class', 9)

# Summarize classes to get top 10
cla.GH20.sum.sum <- cla.GH20.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
ord.GH20.sum <- import.order.sum('kaiju/sep_sig_cazymes/GH20/order', 10)

# Summarize orders to get top 10
ord.GH20.sum.sum <- ord.GH20.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
fam.GH20.sum <- import.family.sum('kaiju/sep_sig_cazymes/GH20/family', 17)

# Summarize families to get top 10
fam.GH20.sum.sum <- fam.GH20.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
gen.GH20.sum <- import.genus.sum('kaiju/sep_sig_cazymes/GH20/genus',19)

# Summarize genera to get top 10
gen.GH20.sum.sum <- gen.GH20.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 5 genera (excluding unclassified, but using the % calculated with unclassified
top.GH20.genus <- c("Bacteroides", "Parabacteroides", "Akkermansia", "Alistipes", "Prevotella")


# Pull out top 5 genera
gen.GH20.total <- data.frame(matrix(nrow=15, ncol=5))
colnames(gen.GH20.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.GH20.genus)){
  top <- which(gen.GH20.sum$Genus == top.GH20.genus[i])
  for (j in 1:length(top)){
    gen.GH20.total$Season[(j + 3*(i-1))] <- as.character(gen.GH20.sum$Season[top[j]])
    gen.GH20.total$Genus[(j + 3*(i-1))] <- gen.GH20.sum$Genus[top[j]]
    gen.GH20.total$Mean[(j + 3*(i-1))] <- gen.GH20.sum$Mean[top[j]]
    gen.GH20.total$SD[(j + 3*(i-1))] <- gen.GH20.sum$SD[top[j]]
    gen.GH20.total$SE[(j + 3*(i-1))] <- gen.GH20.sum$SE[top[j]]
  }
}
gen.GH20.total$Genus <- ordered(gen.GH20.total$Genus, levels=c("Bacteroides", "Parabacteroides", "Akkermansia", "Alistipes", "Prevotella"))
gen.GH20.total$Season <- ordered(gen.GH20.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 5 genera
ggplot(gen.GH20.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-0.2,41)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 5 genera - All seasons combined
gen.GH20.combined <- gen.GH20.total %>%
  group_by(Genus) %>%
    summarize(Mean = mean(Mean),
              SD = sd(Mean),
              SE = se(Mean))
gen.GH20.combined$SD[1] = sd(gen.GH20.total$Mean[1:3])
gen.GH20.combined$SD[2] = sd(gen.GH20.total$Mean[4:6])
gen.GH20.combined$SD[3] = sd(gen.GH20.total$Mean[7:9])
gen.GH20.combined$SD[4] = sd(gen.GH20.total$Mean[10:12])
gen.GH20.combined$SD[5] = sd(gen.GH20.total$Mean[13:15])
gen.GH20.combined$SE[1] = (sd(gen.GH20.total$Mean[1:3])/3)
gen.GH20.combined$SE[2] = (sd(gen.GH20.total$Mean[4:6])/3)
gen.GH20.combined$SE[3] = (sd(gen.GH20.total$Mean[7:9])/3)
gen.GH20.combined$SE[4] = (sd(gen.GH20.total$Mean[10:12])/3)
gen.GH20.combined$SE[5] = (sd(gen.GH20.total$Mean[13:15])/3)

  
# Barplot for top 5 genera - All seasons combined
#tiff("GH20.genus.tiff", width=7, height=5, units="in", res= 300)
ggplot(gen.GH20.combined, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,35), breaks=c(0,8.75,17.5,26.25,35)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance (%)") +
  ggtitle("GH20 Genus-level Classification") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"),
        plot.title=element_text(hjust=0.5, face="bold"))
#dev.off()

########## Species ##########
spe.GH20.sum <- import.species.sum('kaiju/sep_sig_cazymes/GH20/species',42)

# Summarize species to get top 10
spe.GH20.sum.sum <- spe.GH20.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```

### dbCAN - Kaiju on GH84
```{r}
########## Phylum ##########
phy.GH84.sum <- import.phylum.sum('kaiju/sep_sig_cazymes/GH84/phylum',4)

# Summarize phyla to get top 10
phy.GH84.sum.sum <- phy.GH84.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
cla.GH84.sum <- import.class.sum('kaiju/sep_sig_cazymes/GH84/class', 4)

# Summarize classes to get top 10
cla.GH84.sum.sum <- cla.GH84.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
ord.GH84.sum <- import.order.sum('kaiju/sep_sig_cazymes/GH84/order', 4)

# Summarize orders to get top 10
ord.GH84.sum.sum <- ord.GH84.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
fam.GH84.sum <- import.family.sum('kaiju/sep_sig_cazymes/GH84/family', 12)

# Summarize families to get top 10
fam.GH84.sum.sum <- fam.GH84.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
gen.GH84.sum <- import.genus.sum('kaiju/sep_sig_cazymes/GH84/genus',10)

# Summarize genera to get top 10
gen.GH84.sum.sum <- gen.GH84.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 5 genera (excluding unclassified, but using the % calculated with unclassified
top.GH84.genus <- c("Bacteroides", "Parabacteroides", "Akkermansia", "Prevotella", "Alloprevotella")


# Pull out top 5 genera
gen.GH84.total <- data.frame(matrix(nrow=15, ncol=5))
colnames(gen.GH84.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.GH84.genus)){
  top <- which(gen.GH84.sum$Genus == top.GH84.genus[i])
  for (j in 1:length(top)){
    gen.GH84.total$Season[(j + 3*(i-1))] <- as.character(gen.GH84.sum$Season[top[j]])
    gen.GH84.total$Genus[(j + 3*(i-1))] <- gen.GH84.sum$Genus[top[j]]
    gen.GH84.total$Mean[(j + 3*(i-1))] <- gen.GH84.sum$Mean[top[j]]
    gen.GH84.total$SD[(j + 3*(i-1))] <- gen.GH84.sum$SD[top[j]]
    gen.GH84.total$SE[(j + 3*(i-1))] <- gen.GH84.sum$SE[top[j]]
  }
}
gen.GH84.total$Genus <- ordered(gen.GH84.total$Genus, levels=c("Bacteroides", "Parabacteroides", "Akkermansia", "Prevotella", "Alloprevotella"))
gen.GH84.total$Season <- ordered(gen.GH84.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 5 genera
ggplot(gen.GH84.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-5,45)) +
  facet_grid(Season ~ .) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#ffff33", "#a65628")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 5 genera - All seasons combined
gen.GH84.combined <- gen.GH84.total %>%
  group_by(Genus) %>%
    summarize(Mean = mean(Mean),
              SD = sd(Mean),
              SE = se(Mean))
gen.GH84.combined$SD[1] = sd(gen.GH84.total$Mean[1:3])
gen.GH84.combined$SD[2] = sd(gen.GH84.total$Mean[4:6])
gen.GH84.combined$SD[3] = sd(gen.GH84.total$Mean[7:9])
gen.GH84.combined$SD[4] = sd(gen.GH84.total$Mean[10:12])
gen.GH84.combined$SD[5] = sd(gen.GH84.total$Mean[13:15])
gen.GH84.combined$SE[1] = (sd(gen.GH84.total$Mean[1:3])/3)
gen.GH84.combined$SE[2] = (sd(gen.GH84.total$Mean[4:6])/3)
gen.GH84.combined$SE[3] = (sd(gen.GH84.total$Mean[7:9])/3)
gen.GH84.combined$SE[4] = (sd(gen.GH84.total$Mean[10:12])/3)
gen.GH84.combined$SE[5] = (sd(gen.GH84.total$Mean[13:15])/3)

  
# Barplot for top 5 genera - All seasons combined
#tiff("GH84.genus.tiff", width=7, height=5, units="in", res= 300)
ggplot(gen.GH84.combined, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,33), breaks=c(0,8.5,17,25.5,33)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#ffff33", "#a65628")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance (%)") +
  ggtitle("GH84 Genus-level Classification") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"),
        plot.title=element_text(hjust=0.5, face="bold"))
#dev.off()



########## Species ##########
spe.GH84.sum <- import.species.sum('kaiju/sep_sig_cazymes/GH84/species',10)

# Summarize species to get top 10
spe.GH84.sum.sum <- spe.GH84.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```

### dbCAN - Kaiju on GT3
```{r}
########## Phylum ##########
phy.GT3.sum <- import.phylum.sum('kaiju/sep_sig_cazymes/GT3/phylum',4)

# Summarize phyla to get top 10
phy.GT3.sum.sum <- phy.GT3.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
cla.GT3.sum <- import.class.sum('kaiju/sep_sig_cazymes/GT3/class', 4)

# Summarize classes to get top 10
cla.GT3.sum.sum <- cla.GT3.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
ord.GT3.sum <- import.order.sum('kaiju/sep_sig_cazymes/GT3/order', 4)

# Summarize orders to get top 10
ord.GT3.sum.sum <- ord.GT3.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
fam.GT3.sum <- import.family.sum('kaiju/sep_sig_cazymes/GT3/family', 11)

# Summarize families to get top 10
fam.GT3.sum.sum <- fam.GT3.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
gen.GT3.sum <- import.genus.sum('kaiju/sep_sig_cazymes/GT3/genus',10)

# Summarize genera to get top 10
gen.GT3.sum.sum <- gen.GT3.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 5 genera (excluding unclassified, but using the % calculated with unclassified
top.GT3.genus <- c("Alistipes", "Bacteroides", "Prevotella", "Parabacteroides", "Odoribacter")


# Pull out top 5 genera
gen.GT3.total <- data.frame(matrix(nrow=15, ncol=5))
colnames(gen.GT3.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.GT3.genus)){
  top <- which(gen.GT3.sum$Genus == top.GT3.genus[i])
  for (j in 1:length(top)){
    gen.GT3.total$Season[(j + 3*(i-1))] <- as.character(gen.GT3.sum$Season[top[j]])
    gen.GT3.total$Genus[(j + 3*(i-1))] <- gen.GT3.sum$Genus[top[j]]
    gen.GT3.total$Mean[(j + 3*(i-1))] <- gen.GT3.sum$Mean[top[j]]
    gen.GT3.total$SD[(j + 3*(i-1))] <- gen.GT3.sum$SD[top[j]]
    gen.GT3.total$SE[(j + 3*(i-1))] <- gen.GT3.sum$SE[top[j]]
  }
}
gen.GT3.total$Genus <- ordered(gen.GT3.total$Genus, levels=c("Alistipes", "Bacteroides", "Prevotella", "Parabacteroides", "Odoribacter"))
gen.GT3.total$Season <- ordered(gen.GT3.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 5 genera
ggplot(gen.GT3.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-5,45)) +
  scale_fill_manual(values=c("#984ea3", "#e41a1c", "#ffff33", "#377eb8", "gainsboro")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 5 genera - All seasons combined
gen.GT3.combined <- gen.GT3.total %>%
  group_by(Genus) %>%
    summarize(Mean = mean(Mean),
              SD = sd(Mean),
              SE = se(Mean))
gen.GT3.combined$SD[1] = sd(gen.GT3.total$Mean[1:3])
gen.GT3.combined$SD[2] = sd(gen.GT3.total$Mean[4:6])
gen.GT3.combined$SD[3] = sd(gen.GT3.total$Mean[7:9])
gen.GT3.combined$SD[4] = sd(gen.GT3.total$Mean[10:12])
gen.GT3.combined$SD[5] = sd(gen.GT3.total$Mean[13:15])
gen.GT3.combined$SE[1] = (sd(gen.GT3.total$Mean[1:3])/3)
gen.GT3.combined$SE[2] = (sd(gen.GT3.total$Mean[4:6])/3)
gen.GT3.combined$SE[3] = (sd(gen.GT3.total$Mean[7:9])/3)
gen.GT3.combined$SE[4] = (sd(gen.GT3.total$Mean[10:12])/3)
gen.GT3.combined$SE[5] = (sd(gen.GT3.total$Mean[13:15])/3)

  
# Barplot for top 5 genera - All seasons combined
#tiff("GT3.genus.tiff", width=7, height=5, units="in", res= 300)
ggplot(gen.GT3.combined, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,13), breaks=c(0,3.25,6.5,9.75,13)) +
  scale_fill_manual(values=c("#984ea3", "#e41a1c", "#ffff33", "#377eb8", "gainsboro")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance (%)") +
  ggtitle("GT3 Genus-level Classification") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"),
        plot.title=element_text(hjust=0.5, face="bold"))
#dev.off()


########## Species ##########
spe.GT3.sum <- import.species.sum('kaiju/sep_sig_cazymes/GT3/species',22)

# Summarize species to get top 10
spe.GT3.sum.sum <- spe.GT3.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```







### dbCAN - Mucin-degrading CAZymes
```{r}
# List of mucin-degrading CAZymes
muc.enz <- c("GH2", "GH3", "GH4", "GH16", "GH18", "GH20", "GH27", "GH29", "GH31", "GH33", "GH35", "GH36", "GH38", "GH42", "GH67", "GH76", "GH84", "GH85", "GH89", "GH92", "GH95", "GH97", "GH98", "GH101", "GH109", "GH110", "GH112", "GH123", "GH125", "GH129", "GH130", "GH163")



# Pull out mucin-degrading CAZymes
for (i in 1:length(muc.enz)){
  
  if (i == 1){
    enz <- which(rownames(dbcan.fam.norm) == muc.enz[i])
    muc.df <- dbcan.fam.norm[enz,]
  }
  
  enz <- which(rownames(dbcan.fam.norm) == muc.enz[i])
  muc.df[i,] <- dbcan.fam.norm[enz,]
}
colnames(muc.df) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

# Sum of mucin-degrading CAZymes
for (i in 1:ncol(muc.df)) {
  if (i == 1){
    muc.sum <- sum(muc.df[,i])
    muc.sum.df <- data.frame(muc.sum)
    colnames(muc.sum.df)[i] <- colnames(muc.df)[i]
  }
  muc.sum <- sum(muc.df[,i])
  muc.sum.df[,i] <- muc.sum
  colnames(muc.sum.df)[i] <- colnames(muc.df)[i]
}
muc.sum.df <- data.frame(t(muc.sum.df))
colnames(muc.sum.df) <- "Total_MucinDeg"
muc.sum.df$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
muc.sum.df$Season <- ordered(muc.sum.df$Season, levels=c("Summer", "Winter", "Spring"))


# Test normality
hist(muc.sum.df[,1])
qqnorm(muc.sum.df[,1])
qqline(muc.sum.df[,1])
shapiro.test(muc.sum.df[,1])
  # p-value = 0.979
  # Normal

# Test w/ ANOVA
summary(aov(Total_MucinDeg ~ Season, data = muc.sum.df))
  # p-value = 0.0263

# Posthoc test w/ Tukey's
TukeyHSD(aov(Total_MucinDeg ~ Season, data = muc.sum.df))
  # Winter-Summer = 0.3489136
  # Spring-Summer = 0.0223984*
  # Spring-Winter = 0.1447197


### Boxplot
ggplot(muc.sum.df, aes(x=Season, y=Total_MucinDeg, fill=Season)) +
  geom_boxplot(position=position_dodge(1)) +
  geom_point(position=position_dodge(width = 0.2), size=2) +
  ylab("Normalized Counts of Mucin-degrading CAZymes") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(3400, 3600, 3800, 4000, 4200), limits=c(3400, 4200)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())


### DESeq2
dbcan.muc <- subset_taxa(dbcan.phy, Family == "GH2" | Family == "GH3" | Family == "GH4" | 
                           Family == "GH16" | Family == "GH18" | Family == "GH20" | Family == "GH27" |
                           Family == "GH29" | Family == "GH31" | Family == "GH33" | Family == "GH35" |
                           Family == "GH36" | Family == "GH38" | Family == "GH42" | Family == "GH67" |
                           Family == "GH76" | Family == "GH84" | Family == "GH85" | Family == "GH89" |
                           Family == "GH92" | Family == "GH95" | Family == "GH97" | Family == "GH98" |
                           Family == "GH101" | Family == "GH109" | Family == "GH110" |
                           Family == "GH112" | Family == "GH123" | Family == "GH125" |
                           Family == "GH129" | Family == "GH130" | Family == "GH163")




dbcan.muc.dds <- phyloseq_to_deseq2(dbcan.muc, ~Season)
  # Design variable must be unordered otherwise you get an error
dbcan.muc.dds$Season <- factor(dbcan.muc.dds$Season, levels=c("Summer", "Winter", "Spring"))
dbcan.muc.dds <- dbcan.muc.dds[rowSums(counts(dbcan.muc.dds)) > 1,]
dbcan.muc.dds <- DESeq(dbcan.muc.dds)

# Pull out results from pairwise comparisons
dbcan.muc.sw <- results(dbcan.muc.dds, contrast=c("Season", "Summer", "Winter"))
dbcan.muc.sw.sig <- dbcan.muc.sw[which(dbcan.muc.sw$padj < 0.05),]
  # None
dbcan.muc.sp <- results(dbcan.muc.dds, contrast=c("Season", "Summer", "Spring"))
dbcan.muc.sp.sig <- dbcan.muc.sp[which(dbcan.muc.sp$padj < 0.05),]
  # None
dbcan.muc.wp <- results(dbcan.muc.dds, contrast=c("Season", "Winter", "Spring"))
dbcan.muc.wp.sig <- dbcan.muc.wp[which(dbcan.muc.wp$padj < 0.05),]
dbcan.muc.wp.sig[which(dbcan.muc.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # GH20, GH84
dbcan.muc.wp.sig[which(dbcan.muc.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH3, GH36, GH42, GH129
```



### Mucin - Proportion mucin-degrading genes from ORFs
```{r}
prodigalCounts <- read.csv('mucin/prodigalCounts.csv', row.names=1)
prodigalCounts <- as.matrix(prodigalCounts)

# Test normality
hist(prodigalCounts[3,])
qqnorm(prodigalCounts[3,])
qqline(prodigalCounts[3,])
shapiro.test(prodigalCounts[3,])
  # p-value = 0.05926
  # Normal

# Format for ANOVA
prodigalCounts <- read.csv('mucin/prodigalCounts.csv', row.names=1)
prodigalCounts <- t(prodigalCounts)
prodigalCounts <- data.frame(prodigalCounts)
prodigalCounts$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")
prodigalCounts$Season <- ordered(prodigalCounts$Season, levels=c("Summer", "Winter", "Spring"))
colnames(prodigalCounts) <- c("Total_ORFs", "Total_MucinGenes", "MucinGenePercentage", "Season")

# Test w/ ANOVA
summary(aov(MucinGenePercentage ~ Season, data = prodigalCounts))
  # p-value = 0.224




### Dot plot
#tiff("mucin.prop.tiff", width=5, height=4, units="in", res= 300)
ggplot(prodigalCounts, aes(x=Season, y=MucinGenePercentage, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, stackratio=1.15) +
  ylab("% ORFs annotated as Mucin-Degrading Gene (%)") +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_y_continuous(breaks = c(0.06, 0.07, 0.08, 0.09), limits=c(0.06, 0.09)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"))
#dev.off()
```



### Mucin - Create phyloseq object
```{r}
# Load data
muc.otu <- read.csv('mucin/mucinCount.csv', row.names=1)
colnames(muc.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
muc.otu.tab <- otu_table(muc.otu, taxa_are_rows=T)
muc.tax <- read.csv('mucin/mucinClassification.csv', row.names=1)
muc.tax <- tax_table(muc.tax)
colnames(muc.tax) <- c("GH_Class", "GH_Family", "EC1", "EC2", "EC3", "EC4", "FullName")
taxa_names(muc.tax) <- muc.tax[,7]
meta <- read.csv('dbcan/dbcan.spades.metadata.csv', row.names=1)
meta.ord <- meta


# Create phyloseq object
mucin.phy <- merge_phyloseq(muc.otu.tab, muc.tax, sample_data(meta.ord))

# Remove CAZymes w/ count < 4
# 4 to stay consistent with how I trimmed all CAZymes
mucin.cut.phy <- prune_taxa(taxa_sums(mucin.phy) > 3, mucin.phy)

# Identify trimmed CAZymes (count < 4)
all <- rownames(tax_table(mucin.phy))
remain <- rownames(tax_table(mucin.cut.phy))
for (i in all){
  if (i %in% remain == F){
    print(i)
  }
}
# GH16_3.2.1.23, GH16_3.2.1.103, GH4_3.2.1.24, 3.1.1.2, 3.1.6.3, 3.1.6.4, 3.1.6.8, 3.1.6.14


# Normalized Counts
#muc.cut.otu <- data.frame(otu_table(mucin.cut.phy))
#mucin.dds <- DESeqDataSetFromMatrix(countData = muc.cut.otu, colData = meta, design = ~ Season)
#mucin.dds <- estimateSizeFactors(mucin.dds)
#mucin.norm <- counts(mucin.dds, normalized=T)
#write.csv(mucin.norm, file = "mucin/mucin.normalized.counts.csv")
mucin.norm <- read.csv('mucin/mucin.normalized.counts.csv', row.names=1)
muc.norm.otu <- otu_table(mucin.norm, taxa_are_rows=T)
sample_names(muc.norm.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
muc.norm.phy <- merge_phyloseq(muc.norm.otu, sample_data(mucin.phy), tax_table(mucin.phy))

# Relative abundance (relative to # ORFs)
mucin.ra <- read.csv('mucin/mucinCountRA.csv', row.names=1)
mucin.ra.otu <- otu_table(mucin.ra, taxa_are_rows=T)
sample_names(mucin.ra.otu) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
muc.ra.phy <- merge_phyloseq(mucin.ra.otu, sample_data(mucin.phy), tax_table(mucin.phy))
```

### Mucin - Kaiju
```{r}
########## Phylum ##########
muc.phy.sum <- import.phylum.sum('kaiju/mucin_all/phylum',7)

# Summarize phyla to get top 10
muc.phy.sum.sum <- muc.phy.sum %>%
  group_by(Phylum) %>%
  summarize(Mean = mean(Mean))



########## Class ##########
muc.cla.sum <- import.class.sum('kaiju/mucin_all/class', 11)

# Summarize classes to get top 10
muc.cla.sum.sum <- muc.cla.sum %>%
  group_by(Class) %>%
  summarize(Mean = mean(Mean))



########## Order ##########
muc.ord.sum <- import.order.sum('kaiju/mucin_all/order', 14)

# Summarize orders to get top 10
muc.ord.sum.sum <- muc.ord.sum %>%
  group_by(Order) %>%
  summarize(Mean = mean(Mean))



########## Family ##########
muc.fam.sum <- import.family.sum('kaiju/mucin_all/family', 27)

# Summarize families to get top 10
muc.fam.sum.sum <- muc.fam.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))


########## Genus ##########
muc.gen.sum <- import.genus.sum('kaiju/mucin_all/genus',35)

# Summarize genera to get top 10
muc.gen.sum.sum <- muc.gen.sum %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Mean))

# Only look at top 11 genera (excluding unclassified, but using the % calculated with unclassified
top.muc.genus <- c("Bacteroides", "Prevotella", "Parabacteroides", "Clostridium", "Alistipes", "Roseburia", "Blautia", "Paenibacillus", "Faecalibacterium", "Ruminococcus", "Akkermansia")



# Pull out top 11 genera
muc.gen.total <- data.frame(matrix(nrow=33, ncol=5))
colnames(muc.gen.total) <- c("Season", "Genus", "Mean", "SD", "SE")
for (i in 1:length(top.muc.genus)){
  top <- which(muc.gen.sum$Genus == top.muc.genus[i])
  for (j in 1:length(top)){
    muc.gen.total$Season[(j + 3*(i-1))] <- as.character(muc.gen.sum$Season[top[j]])
    muc.gen.total$Genus[(j + 3*(i-1))] <- muc.gen.sum$Genus[top[j]]
    muc.gen.total$Mean[(j + 3*(i-1))] <- muc.gen.sum$Mean[top[j]]
    muc.gen.total$SD[(j + 3*(i-1))] <- muc.gen.sum$SD[top[j]]
    muc.gen.total$SE[(j + 3*(i-1))] <- muc.gen.sum$SE[top[j]]
  }
}
muc.gen.total$Genus <- ordered(muc.gen.total$Genus, levels=c("Bacteroides", "Prevotella", "Parabacteroides", "Clostridium", "Alistipes", "Roseburia", "Blautia", "Paenibacillus", "Faecalibacterium", "Ruminococcus", "Akkermansia"))
muc.gen.total$Season <- ordered(muc.gen.total$Season, levels=c("Summer", "Winter", "Spring"))

# Barplot for top 11 genera
ggplot(muc.gen.total, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(Season ~ .) +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(-5,45)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush", "lightpink")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"))


# Pull out top 5 genera - All seasons combined
muc.gen.combined <- muc.gen.total %>%
  group_by(Genus) %>%
    summarize(Mean = mean(Mean),
              SD = sd(Mean),
              SE = se(Mean))
muc.gen.combined$SD[1] = sd(muc.gen.total$Mean[1:3])
muc.gen.combined$SD[2] = sd(muc.gen.total$Mean[4:6])
muc.gen.combined$SD[3] = sd(muc.gen.total$Mean[7:9])
muc.gen.combined$SD[4] = sd(muc.gen.total$Mean[10:12])
muc.gen.combined$SD[5] = sd(muc.gen.total$Mean[13:15])
muc.gen.combined$SD[6] = sd(muc.gen.total$Mean[16:18])
muc.gen.combined$SD[7] = sd(muc.gen.total$Mean[19:21])
muc.gen.combined$SD[8] = sd(muc.gen.total$Mean[22:24])
muc.gen.combined$SD[9] = sd(muc.gen.total$Mean[25:27])
muc.gen.combined$SD[10] = sd(muc.gen.total$Mean[28:30])
muc.gen.combined$SD[11] = sd(muc.gen.total$Mean[31:33])
muc.gen.combined$SE[1] = (sd(muc.gen.total$Mean[1:3])/3)
muc.gen.combined$SE[2] = (sd(muc.gen.total$Mean[4:6])/3)
muc.gen.combined$SE[3] = (sd(muc.gen.total$Mean[7:9])/3)
muc.gen.combined$SE[4] = (sd(muc.gen.total$Mean[10:12])/3)
muc.gen.combined$SE[5] = (sd(muc.gen.total$Mean[13:15])/3)
muc.gen.combined$SE[6] = (sd(muc.gen.total$Mean[16:18])/3)
muc.gen.combined$SE[7] = (sd(muc.gen.total$Mean[19:21])/3)
muc.gen.combined$SE[8] = (sd(muc.gen.total$Mean[22:24])/3)
muc.gen.combined$SE[9] = (sd(muc.gen.total$Mean[25:27])/3)
muc.gen.combined$SE[10] = (sd(muc.gen.total$Mean[28:30])/3)
muc.gen.combined$SE[11] = (sd(muc.gen.total$Mean[31:33])/3)
  
# Barplot for top 5 genera - All seasons combined
tiff("muc.gen.tiff", width=9, height=5, units="in", res= 300)
ggplot(muc.gen.combined, aes(x=Genus, y=Mean, fill=Genus)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,26), breaks=c(0,6.5,13,19.5,26)) +
  scale_fill_manual(values=c("#e41a1c", "#377eb8", "#4daf4a", "#984ea3", "#ffff33", "#a65628", "gainsboro", "gray47", "black", "lavenderblush", "lightpink")) +
  xlab("Genera") +
  ylab("Mean Relative Abundance (%)") +
  ggtitle("Mucin-Degradation Genes Genus-level Classification") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        legend.title = element_text(face="bold"),
        plot.title=element_text(hjust=0.5, face="bold"),
        axis.text.x = element_text(angle=30, hjust=1))
dev.off()


########## Species ##########
muc.spe.sum <- import.species.sum('kaiju/mucin_all/species',87)

# Summarize species to get top 10
muc.spe.sum.sum <- muc.spe.sum %>%
  group_by(Species) %>%
  summarize(Mean = mean(Mean))
```
### Mucin - Enzyme bar plots
```{r}
# Summarize data
mucin.df <- psmelt(muc.ra.phy)
mucin.sum <- mucin.df %>%
  group_by(FullName, Season) %>%
  summarize(Mean = mean(Abundance),
            SD= sd(Abundance),
            SE = se(Abundance))
mucin.sum$FullName = ordered(mucin.sum$FullName, c("GH2_3.2.1.23", "GH2_3.2.1.25", "GH3_3.2.1.52", "GH4_3.2.1.22", "GH20_3.2.1.52", "GH27_3.2.1.22", "GH29_3.2.1.51", "GH31_3.2.1.20", "GH35_3.2.1.23", "GH36_3.2.1.22", "GH38_3.2.1.24", "GH42_3.2.1.23", "GH67_3.2.1.139", "GH84_3.2.1.169", "GH95_3.2.1.51", "GH101_3.2.1.97", "GH110_3.2.1.22", "GH112_2.4.1.211", "GH130_2.4.1.281", "GH130_2.4.1.320", "4.1.3.3"))
mucin.sum$Season = ordered(mucin.sum$Season, levels=c("Summer", "Winter", "Spring"))


# Plot barplot
#tiff("mucin.family.tiff", width=33, height=7, units="in", res= 300)
ggplot(mucin.sum, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,2e-2)) +
  facet_grid(. ~ FullName) +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  xlab("Mucin-Degrading Gene Classification") +
  ylab("Mean Relative Abundance") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"),
        panel.spacing = unit(0.5, "lines"),
        axis.title.x = element_text(margin = margin(t = 15)))
#dev.off()

```

### Mucin - Compare Rel Abund
```{r}
### GH2_3.2.1.23
muc.GH2_3.2.1.23 <- subset_taxa(muc.ra.phy, FullName == "GH2_3.2.1.23")
muc.GH2_3.2.1.23.df <- data.frame(sample_data(muc.GH2_3.2.1.23))
muc.GH2_3.2.1.23.df$RelAbund <- t(data.frame(otu_table(muc.GH2_3.2.1.23)))

# Test normality
hist(muc.GH2_3.2.1.23.df$RelAbund, breaks=10)
qqnorm(muc.GH2_3.2.1.23.df$RelAbund)
qqline(muc.GH2_3.2.1.23.df$RelAbund)
shapiro.test(muc.GH2_3.2.1.23.df$RelAbund)
  # p-value = 0.4094
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH2_3.2.1.23.df))
  # p-value = 0.0145




### GH2_3.2.1.25
muc.GH2_3.2.1.25 <- subset_taxa(muc.ra.phy, FullName == "GH2_3.2.1.25")
muc.GH2_3.2.1.25.df <- data.frame(sample_data(muc.GH2_3.2.1.25))
muc.GH2_3.2.1.25.df$RelAbund <- t(data.frame(otu_table(muc.GH2_3.2.1.25)))

# Test normality
hist(muc.GH2_3.2.1.25.df$RelAbund, breaks=10)
qqnorm(muc.GH2_3.2.1.25.df$RelAbund)
qqline(muc.GH2_3.2.1.25.df$RelAbund)
shapiro.test(muc.GH2_3.2.1.25.df$RelAbund)
  # p-value = 0.7485
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH2_3.2.1.25.df))
  # p-value = 0.0411





### GH3_3.2.1.52
muc.GH3_3.2.1.52 <- subset_taxa(muc.ra.phy, FullName == "GH3_3.2.1.52")
muc.GH3_3.2.1.52.df <- data.frame(sample_data(muc.GH3_3.2.1.52))
muc.GH3_3.2.1.52.df$RelAbund <- t(data.frame(otu_table(muc.GH3_3.2.1.52)))

# Test normality
hist(muc.GH3_3.2.1.52.df$RelAbund, breaks=10)
qqnorm(muc.GH3_3.2.1.52.df$RelAbund)
qqline(muc.GH3_3.2.1.52.df$RelAbund)
shapiro.test(muc.GH3_3.2.1.52.df$RelAbund)
  # p-value = 0.09048
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH3_3.2.1.52.df))
  # p-value = 0.189



### GH4_3.2.1.22
muc.GH4_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH4_3.2.1.22")
muc.GH4_3.2.1.22.df <- data.frame(sample_data(muc.GH4_3.2.1.22))
muc.GH4_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH4_3.2.1.22)))

# Test normality
hist(muc.GH4_3.2.1.22.df$RelAbund, breaks=10)
qqnorm(muc.GH4_3.2.1.22.df$RelAbund)
qqline(muc.GH4_3.2.1.22.df$RelAbund)
shapiro.test(muc.GH4_3.2.1.22.df$RelAbund)
  # p-value = 0.6583
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH4_3.2.1.22.df))
  # p-value = 0.0877



### GH20_3.2.1.52
muc.GH20_3.2.1.52 <- subset_taxa(muc.ra.phy, FullName == "GH20_3.2.1.52")
muc.GH20_3.2.1.52.df <- data.frame(sample_data(muc.GH20_3.2.1.52))
muc.GH20_3.2.1.52.df$RelAbund <- t(data.frame(otu_table(muc.GH20_3.2.1.52)))

# Test normality
hist(muc.GH20_3.2.1.52.df$RelAbund, breaks=10)
qqnorm(muc.GH20_3.2.1.52.df$RelAbund)
qqline(muc.GH20_3.2.1.52.df$RelAbund)
shapiro.test(muc.GH20_3.2.1.52.df$RelAbund)
  # p-value = 0.2594
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH20_3.2.1.52.df))
  # p-value = 0.215



### GH27_3.2.1.22
muc.GH27_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH27_3.2.1.22")
muc.GH27_3.2.1.22.df <- data.frame(sample_data(muc.GH27_3.2.1.22))
muc.GH27_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH27_3.2.1.22)))

# Test normality
hist(muc.GH27_3.2.1.22.df$RelAbund, breaks=10)
qqnorm(muc.GH27_3.2.1.22.df$RelAbund)
qqline(muc.GH27_3.2.1.22.df$RelAbund)
shapiro.test(muc.GH27_3.2.1.22.df$RelAbund)
  # p-value = 0.2055
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH27_3.2.1.22.df))
  # p-value = 0.00774



### GH29_3.2.1.51
muc.GH29_3.2.1.51 <- subset_taxa(muc.ra.phy, FullName == "GH29_3.2.1.51")
muc.GH29_3.2.1.51.df <- data.frame(sample_data(muc.GH29_3.2.1.51))
muc.GH29_3.2.1.51.df$RelAbund <- t(data.frame(otu_table(muc.GH29_3.2.1.51)))

# Test normality
hist(muc.GH29_3.2.1.51.df$RelAbund, breaks=10)
qqnorm(muc.GH29_3.2.1.51.df$RelAbund)
qqline(muc.GH29_3.2.1.51.df$RelAbund)
shapiro.test(muc.GH29_3.2.1.51.df$RelAbund)
  # p-value = 0.9895
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH29_3.2.1.51.df))
  # p-value = 0.0302




### GH31_3.2.1.20
muc.GH31_3.2.1.20 <- subset_taxa(muc.ra.phy, FullName == "GH31_3.2.1.20")
muc.GH31_3.2.1.20.df <- data.frame(sample_data(muc.GH31_3.2.1.20))
muc.GH31_3.2.1.20.df$RelAbund <- t(data.frame(otu_table(muc.GH31_3.2.1.20)))

# Test normality
hist(muc.GH31_3.2.1.20.df$RelAbund, breaks=10)
qqnorm(muc.GH31_3.2.1.20.df$RelAbund)
qqline(muc.GH31_3.2.1.20.df$RelAbund)
shapiro.test(muc.GH31_3.2.1.20.df$RelAbund)
  # p-value = 0.1951
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH31_3.2.1.20.df))
  # p-value = 0.133



### GH35_3.2.1.23
muc.GH35_3.2.1.23 <- subset_taxa(muc.ra.phy, FullName == "GH35_3.2.1.23")
muc.GH35_3.2.1.23.df <- data.frame(sample_data(muc.GH35_3.2.1.23))
muc.GH35_3.2.1.23.df$RelAbund <- t(data.frame(otu_table(muc.GH35_3.2.1.23)))

# Test normality
hist(muc.GH35_3.2.1.23.df$RelAbund, breaks=10)
qqnorm(muc.GH35_3.2.1.23.df$RelAbund)
qqline(muc.GH35_3.2.1.23.df$RelAbund)
shapiro.test(muc.GH35_3.2.1.23.df$RelAbund)
  # p-value = 0.4842
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH35_3.2.1.23.df))
  # p-value = 0.0753



### GH36_3.2.1.22
muc.GH36_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH36_3.2.1.22")
muc.GH36_3.2.1.22.df <- data.frame(sample_data(muc.GH36_3.2.1.22))
muc.GH36_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH36_3.2.1.22)))

# Test normality
hist(muc.GH36_3.2.1.22.df$RelAbund, breaks=10)
qqnorm(muc.GH36_3.2.1.22.df$RelAbund)
qqline(muc.GH36_3.2.1.22.df$RelAbund)
shapiro.test(muc.GH36_3.2.1.22.df$RelAbund)
  # p-value = 0.557
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH36_3.2.1.22.df))
  # p-value = 0.00819



### GH38_3.2.1.24
muc.GH38_3.2.1.24 <- subset_taxa(muc.ra.phy, FullName == "GH38_3.2.1.24")
muc.GH38_3.2.1.24.df <- data.frame(sample_data(muc.GH38_3.2.1.24))
muc.GH38_3.2.1.24.df$RelAbund <- t(data.frame(otu_table(muc.GH38_3.2.1.24)))

# Test normality
hist(muc.GH38_3.2.1.24.df$RelAbund, breaks=10)
qqnorm(muc.GH38_3.2.1.24.df$RelAbund)
qqline(muc.GH38_3.2.1.24.df$RelAbund)
shapiro.test(muc.GH38_3.2.1.24.df$RelAbund)
  # p-value = 0.8813
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH38_3.2.1.24.df))
  # p-value = 0.136



### GH42_3.2.1.23
muc.GH42_3.2.1.23 <- subset_taxa(muc.ra.phy, FullName == "GH42_3.2.1.23")
muc.GH42_3.2.1.23.df <- data.frame(sample_data(muc.GH42_3.2.1.23))
muc.GH42_3.2.1.23.df$RelAbund <- t(data.frame(otu_table(muc.GH42_3.2.1.23)))

# Test normality
hist(muc.GH42_3.2.1.23.df$RelAbund, breaks=10)
qqnorm(muc.GH42_3.2.1.23.df$RelAbund)
qqline(muc.GH42_3.2.1.23.df$RelAbund)
shapiro.test(muc.GH42_3.2.1.23.df$RelAbund)
  # p-value = 0.4227
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH42_3.2.1.23.df))
  # p-value = 0.208



### GH67_3.2.1.139
muc.GH67_3.2.1.139 <- subset_taxa(muc.ra.phy, FullName == "GH67_3.2.1.139")
muc.GH67_3.2.1.139.df <- data.frame(sample_data(muc.GH67_3.2.1.139))
muc.GH67_3.2.1.139.df$RelAbund <- t(data.frame(otu_table(muc.GH67_3.2.1.139)))

# Test normality
hist(muc.GH67_3.2.1.139.df$RelAbund, breaks=10)
qqnorm(muc.GH67_3.2.1.139.df$RelAbund)
qqline(muc.GH67_3.2.1.139.df$RelAbund)
shapiro.test(muc.GH67_3.2.1.139.df$RelAbund)
  # p-value = 0.6543
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH67_3.2.1.139.df))
  # p-value = 0.0227



### GH84_3.2.1.169
muc.GH84_3.2.1.169 <- subset_taxa(muc.ra.phy, FullName == "GH84_3.2.1.169")
muc.GH84_3.2.1.169.df <- data.frame(sample_data(muc.GH84_3.2.1.169))
muc.GH84_3.2.1.169.df$RelAbund <- t(data.frame(otu_table(muc.GH84_3.2.1.169)))

# Test normality
hist(muc.GH84_3.2.1.169.df$RelAbund, breaks=10)
qqnorm(muc.GH84_3.2.1.169.df$RelAbund)
qqline(muc.GH84_3.2.1.169.df$RelAbund)
shapiro.test(muc.GH84_3.2.1.169.df$RelAbund)
  # p-value = 0.4944
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH84_3.2.1.169.df))
  # p-value = 0.00195




### GH95_3.2.1.51
muc.GH95_3.2.1.51 <- subset_taxa(muc.ra.phy, FullName == "GH95_3.2.1.51")
muc.GH95_3.2.1.51.df <- data.frame(sample_data(muc.GH95_3.2.1.51))
muc.GH95_3.2.1.51.df$RelAbund <- t(data.frame(otu_table(muc.GH95_3.2.1.51)))

# Test normality
hist(muc.GH95_3.2.1.51.df$RelAbund, breaks=10)
qqnorm(muc.GH95_3.2.1.51.df$RelAbund)
qqline(muc.GH95_3.2.1.51.df$RelAbund)
shapiro.test(muc.GH95_3.2.1.51.df$RelAbund)
  # p-value = 0.283
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH95_3.2.1.51.df))
  # p-value = 0.0789



### GH101_3.2.1.97
muc.GH101_3.2.1.97 <- subset_taxa(muc.ra.phy, FullName == "GH101_3.2.1.97")
muc.GH101_3.2.1.97.df <- data.frame(sample_data(muc.GH101_3.2.1.97))
muc.GH101_3.2.1.97.df$RelAbund <- t(data.frame(otu_table(muc.GH101_3.2.1.97)))

# Test normality
hist(muc.GH101_3.2.1.97.df$RelAbund, breaks=10)
qqnorm(muc.GH101_3.2.1.97.df$RelAbund)
qqline(muc.GH101_3.2.1.97.df$RelAbund)
shapiro.test(muc.GH101_3.2.1.97.df$RelAbund)
  # p-value = 2.513e-05
  # Not Normal

# Run Kruskal-Wallis
kruskal.test(RelAbund ~ Season, data = muc.GH101_3.2.1.97.df)
  # p-value = 0.558



### GH110_3.2.1.22
muc.GH110_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH110_3.2.1.22")
muc.GH110_3.2.1.22.df <- data.frame(sample_data(muc.GH110_3.2.1.22))
muc.GH110_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH110_3.2.1.22)))

# Test normality
hist(muc.GH110_3.2.1.22.df$RelAbund, breaks=10)
qqnorm(muc.GH110_3.2.1.22.df$RelAbund)
qqline(muc.GH110_3.2.1.22.df$RelAbund)
shapiro.test(muc.GH110_3.2.1.22.df$RelAbund)
  # p-value = 0.9366
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH110_3.2.1.22.df))
  # p-value = 0.0082



### GH112_2.4.1.211
muc.GH112_2.4.1.211 <- subset_taxa(muc.ra.phy, FullName == "GH112_2.4.1.211")
muc.GH112_2.4.1.211.df <- data.frame(sample_data(muc.GH112_2.4.1.211))
muc.GH112_2.4.1.211.df$RelAbund <- t(data.frame(otu_table(muc.GH112_2.4.1.211)))

# Test normality
hist(muc.GH112_2.4.1.211.df$RelAbund, breaks=10)
qqnorm(muc.GH112_2.4.1.211.df$RelAbund)
qqline(muc.GH112_2.4.1.211.df$RelAbund)
shapiro.test(muc.GH112_2.4.1.211.df$RelAbund)
  # p-value = 0.7092
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH112_2.4.1.211.df))
  # p-value = 0.326



### GH130_2.4.1.281
muc.GH130_2.4.1.281 <- subset_taxa(muc.ra.phy, FullName == "GH130_2.4.1.281")
muc.GH130_2.4.1.281.df <- data.frame(sample_data(muc.GH130_2.4.1.281))
muc.GH130_2.4.1.281.df$RelAbund <- t(data.frame(otu_table(muc.GH130_2.4.1.281)))

# Test normality
hist(muc.GH130_2.4.1.281.df$RelAbund, breaks=10)
qqnorm(muc.GH130_2.4.1.281.df$RelAbund)
qqline(muc.GH130_2.4.1.281.df$RelAbund)
shapiro.test(muc.GH130_2.4.1.281.df$RelAbund)
  # p-value = 0.1855
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH130_2.4.1.281.df))
  # p-value = 0.747



### GH130_2.4.1.320
muc.GH130_2.4.1.320 <- subset_taxa(muc.ra.phy, FullName == "GH130_2.4.1.320")
muc.GH130_2.4.1.320.df <- data.frame(sample_data(muc.GH130_2.4.1.320))
muc.GH130_2.4.1.320.df$RelAbund <- t(data.frame(otu_table(muc.GH130_2.4.1.320)))

# Test normality
hist(muc.GH130_2.4.1.320.df$RelAbund, breaks=10)
qqnorm(muc.GH130_2.4.1.320.df$RelAbund)
qqline(muc.GH130_2.4.1.320.df$RelAbund)
shapiro.test(muc.GH130_2.4.1.320.df$RelAbund)
  # p-value = 0.2561
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.GH130_2.4.1.320.df))
  # p-value = 0.501



### 4.1.3.3
muc.4.1.3.3 <- subset_taxa(muc.ra.phy, FullName == "4.1.3.3")
muc.4.1.3.3.df <- data.frame(sample_data(muc.4.1.3.3))
muc.4.1.3.3.df$RelAbund <- t(data.frame(otu_table(muc.4.1.3.3)))

# Test normality
hist(muc.4.1.3.3.df$RelAbund, breaks=10)
qqnorm(muc.4.1.3.3.df$RelAbund)
qqline(muc.4.1.3.3.df$RelAbund)
shapiro.test(muc.4.1.3.3.df$RelAbund)
  # p-value = 0.08095
  # Normal

# Run ANOVA
summary(aov(RelAbund ~ Season, data = muc.4.1.3.3.df))
  # p-value = 0.224


### Adjust pval
# Listed in order of mucinEnzymes.xlsx
p.adjust(c(0.224, 0.558, 0.0082, 0.326, 0.747, 0.501, 0.0145, 0.0411, 0.215, 0.00774, 0.0302, 0.189, 0.133, 0.0753, 0.00819, 0.136, 0.0877, 0.208, 0.0227, 0.00195, 0.0789), "fdr")
 #  [1] 0.2767059 0.5859000 0.0430500* 0.3803333 0.7470000 0.5537368 0.0609000 0.1078875 0.2767059
 # [10] 0.0430500* 0.0906000 0.2767059 0.2196923 0.1656900 0.0430500* 0.2196923 0.1674273 0.2767059
 # [19] 0.0794500 0.0409500* 0.1656900
  # Sig = GH110_3.2.1.22, GH27_3.2.1.22, GH36_3.2.1.22, GH84_3.2.1.169







### GH27_3.2.1.22
muc.GH27_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH27_3.2.1.22")
muc.GH27_3.2.1.22.df <- data.frame(sample_data(muc.GH27_3.2.1.22))
muc.GH27_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH27_3.2.1.22)))

# Run Tukey Posthoc
TukeyHSD(aov(RelAbund ~ Season, data = muc.GH27_3.2.1.22.df))
  # Summer-Spring = 0.0101625*
  # Winter-Spring = 0.9142216
  # Winter-Summer = 0.0157542*


### GH36_3.2.1.22
muc.GH36_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH36_3.2.1.22")
muc.GH36_3.2.1.22.df <- data.frame(sample_data(muc.GH36_3.2.1.22))
muc.GH36_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH36_3.2.1.22)))

# Run Tukey Posthoc
TukeyHSD(aov(RelAbund ~ Season, data = muc.GH36_3.2.1.22.df))
  # Summer-Spring = 0.9583155
  # Winter-Spring = 0.0113784*
  # Winter-Summer = 0.0153971*


### GH84_3.2.1.169
muc.GH84_3.2.1.169 <- subset_taxa(muc.ra.phy, FullName == "GH84_3.2.1.169")
muc.GH84_3.2.1.169.df <- data.frame(sample_data(muc.GH84_3.2.1.169))
muc.GH84_3.2.1.169.df$RelAbund <- t(data.frame(otu_table(muc.GH84_3.2.1.169)))

# Run Tukey Posthoc
TukeyHSD(aov(RelAbund ~ Season, data = muc.GH84_3.2.1.169.df))
  # Summer-Spring = 0.1104771
  # Winter-Spring = 0.0016364**
  # Winter-Summer = 0.0170359*



### GH110_3.2.1.22
muc.GH110_3.2.1.22 <- subset_taxa(muc.ra.phy, FullName == "GH110_3.2.1.22")
muc.GH110_3.2.1.22.df <- data.frame(sample_data(muc.GH110_3.2.1.22))
muc.GH110_3.2.1.22.df$RelAbund <- t(data.frame(otu_table(muc.GH110_3.2.1.22)))
# Run Tukey Posthoc
TukeyHSD(aov(RelAbund ~ Season, data = muc.GH110_3.2.1.22.df))
  # Summer-Spring = 0.1208541
  # Winter-Spring = 0.0066596*
  # Winter-Summer = 0.1012993

```
### Mucin - Sig enzymes barplot
```{r}
# Subset sig genes
sig1 = which(mucin.sum$FullName == "GH27_3.2.1.22")
sig2 = which(mucin.sum$FullName == "GH36_3.2.1.22")
sig3 = which(mucin.sum$FullName == "GH84_3.2.1.169")
sig4 = which(mucin.sum$FullName == "GH110_3.2.1.22")
mucSig = c(sig1,sig2,sig3,sig4)
muc.sig.sum <- mucin.sum[mucSig,]


# Plot barplot
tiff("mucin.sig.tiff", width=8, height=4, units="in", res= 300)
ggplot(muc.sig.sum, aes(x=Season, y=Mean, fill=Season)) +
  geom_bar(stat="identity", color="black") +
  geom_errorbar(aes(ymax=Mean+SE, ymin=Mean-SE), width=0.25) +
  scale_y_continuous(limits=c(0,1.8e-2), breaks=c(0,0.0045, 0.009, 0.0135, 0.018)) +
  facet_grid(. ~ FullName) +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ylab("Mean Relative Abundance (%)") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(face="bold"),
        strip.text.x = element_text(size=11),
        legend.title = element_text(face="bold"),
        panel.spacing = unit(0.5, "lines"),
        axis.title.x = element_text(margin = margin(t = 15)))
dev.off()
```

### Mucin - DESeq
```{r}
# https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
# https://compbiocore.github.io/deseq-workshop-1/assets/deseq_workshop_1.html
# Use UN-NORMALIZED data!!!
  # DESeq accounts for library size


### Test DESeq2 to determine fit type
# https://support.bioconductor.org/p/81094/
#mucin.dds <- phyloseq_to_deseq2(mucin.cut.phy, ~Season)
#mucin.dds$Season <- factor(mucin.dds$Season, levels=c("Summer", "Winter", "Spring"))
#mucin.dds <- mucin.dds[rowSums(counts(mucin.dds)) > 1,]
#mucin.local <- DESeq(mucin.dds, fitType = "local")
#mucin.par <- DESeq(mucin.dds, fitType = "parametric")
#mad(log(mcols(mucin.local)$dispGeneEst) - log(mcols(mucin.local)$dispFit))
  # 0.8548652
#mad(log(mcols(mucin.par)$dispGeneEst) - log(mcols(mucin.par)$dispFit))
  # 0.8548652
# No difference in fit type, but DESeq says local is better



### Convert phyloseq object to deseq2 object
mucin.dds <- phyloseq_to_deseq2(mucin.cut.phy, ~Season)
  # Design variable must be unordered otherwise you get an error
mucin.dds$Season <- factor(mucin.dds$Season, levels=c("Summer", "Winter", "Spring"))
mucin.dds <- mucin.dds[rowSums(counts(mucin.dds)) > 1,]


# DESeq DataStructure for analyses
mucin.dds <- DESeq(mucin.dds)
  # DESeq choose fitType = local


# Pull out results from pairwise comparison
mucin.sw <- results(mucin.dds, contrast=c("Season", "Summer", "Winter"))
mucin.sw.sig <- mucin.sw[which(mucin.sw$padj < 0.05),]
  # None
mucin.sp <- results(mucin.dds, contrast=c("Season", "Summer", "Spring"))
mucin.sp.sig <- mucin.sp[which(mucin.sp$padj < 0.05),]
  # Enriched in Spring compared to Summer (negative log2FoldChange)
    # GH112_2.4.1.211
mucin.wp <- results(mucin.dds, contrast=c("Season", "Winter", "Spring"))
mucin.wp.sig <- mucin.wp[which(mucin.wp$padj < 0.05),]
mucin.wp.sig[which(mucin.wp.sig$log2FoldChange > 0),]
  # Enriched in Winter compared to Spring (positive log2FoldChange)
    # GH110_3.2.1.22, GH84_3.2.1.169
mucin.wp.sig[which(mucin.wp.sig$log2FoldChange < 0),]
  # Enriched in Spring compared to Winter (negative log2FoldChange)
    # GH36_3.2.1.22, GH42_3.2.1.23
```



### Mucin - Heatmap
```{r}
mucin.sig <- subset_taxa(muc.norm.phy, FullName == "GH112_2.4.1.211" | FullName == "GH36_3.2.1.22" |
                           FullName == "GH42_3.2.1.23" | FullName == "GH84_3.2.1.169" |
                           FullName == "GH110_3.2.1.22")
                           
  


tiff("mucin.heatmap.tiff", width=15, height=7, units="in", res=300)
plot_heatmap(mucin.sig, sample.label = "Season", sample.order = c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"), taxa.order = c("GH110_3.2.1.22", "GH84_3.2.1.169", "GH42_3.2.1.23", "GH36_3.2.1.22", "GH112_2.4.1.211")) +
  geom_tile(colour="gray40") +
  geom_text(aes(label=sprintf("%0.2f", round(Abundance, digits=4))), size = 7) +
  scale_x_discrete(position = "top", labels=c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring")) +
  xlab("Season\n\n") +
  ylab("Mucin-Degrading Enzyme Gene\n\n") +
  scale_fill_gradientn(name=expression("Normalized\nCounts\n"), limits=c(0, 100),
                       breaks=c(0, 25, 50, 75, 100), 
                       labels=c("0", "25", "50", "75", "100"),
                       colours = c("#FFFFFF", "#ffffcc", "#ffeda0", "#fed976", "#feb24c", "#fd8d3c",
                                   "#fc4e2a", "#e31a1c", "#bd0026", "#800026")) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=20),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        panel.border=element_blank(),
        axis.title.x = element_text(size=20, face="bold"),
        axis.title.y = element_text(size=20, face="bold"),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20)) +
  guides(fill = guide_colorbar(barheight = unit(5,"in"),
                               frame.colour="black",
                               frame.linewidth=1))
dev.off()


```





### Mucin - Family-Level for each mucin enzyme
```{r}
# I chose Family-level because that was the taxon that gave the most taxonomic resolution without too many being unclassified

### 4.1.3.3
fam.4.1.3.3.sum <- import.family.sum('kaiju/mucin_sep/4.1.3.3/family',2)
# Summarize families
fam.4.1.3.3.sum.sum <- fam.4.1.3.3.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH2_3.2.1.23
fam.GH2_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH2_3.2.1.23/family',10)
# Summarize families
fam.GH2_3.2.1.23.sum.sum <- fam.GH2_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH2_3.2.1.25
fam.GH2_3.2.1.25.sum <- import.family.sum('kaiju/mucin_sep/GH2_3.2.1.25/family',3)
# Summarize families
fam.GH2_3.2.1.25.sum.sum <- fam.GH2_3.2.1.25.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH3_3.2.1.52
# Not found in some samples

### GH4_3.2.1.22
# Not found in some samples

### GH20_3.2.1.52
# Not found in some samples

### GH27_3.2.1.22
fam.GH27_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH27_3.2.1.22/family',4)
# Summarize families
fam.GH27_3.2.1.22.sum.sum <- fam.GH27_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH29_3.2.1.51
fam.GH29_3.2.1.51.sum <- import.family.sum('kaiju/mucin_sep/GH29_3.2.1.51/family',7)
# Summarize families
fam.GH29_3.2.1.51.sum.sum <- fam.GH29_3.2.1.51.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH31_3.2.1.20
fam.GH31_3.2.1.20.sum <- import.family.sum('kaiju/mucin_sep/GH31_3.2.1.20/family',6)
# Summarize families
fam.GH31_3.2.1.20.sum.sum <- fam.GH31_3.2.1.20.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH35_3.2.1.23
fam.GH35_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH35_3.2.1.23/family',5)
# Summarize families
fam.GH35_3.2.1.23.sum.sum <- fam.GH35_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH36_3.2.1.22
fam.GH36_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH36_3.2.1.22/family',13)
# Summarize families
fam.GH36_3.2.1.22.sum.sum <- fam.GH36_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH38_3.2.1.24
# Not found in some samples

### GH42_3.2.1.23
fam.GH42_3.2.1.23.sum <- import.family.sum('kaiju/mucin_sep/GH42_3.2.1.23/family',4)
# Summarize families
fam.GH42_3.2.1.23.sum.sum <- fam.GH42_3.2.1.23.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH67_3.2.1.139
fam.GH67_3.2.1.139.sum <- import.family.sum('kaiju/mucin_sep/GH67_3.2.1.139/family',7)
# Summarize families
fam.GH67_3.2.1.139.sum.sum <- fam.GH67_3.2.1.139.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH84_3.2.1.169
fam.GH84_3.2.1.169.sum <- import.family.sum('kaiju/mucin_sep/GH84_3.2.1.169/family',10)
# Summarize families
fam.GH84_3.2.1.169.sum.sum <- fam.GH84_3.2.1.169.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH95_3.2.1.51
fam.GH95_3.2.1.51.sum <- import.family.sum('kaiju/mucin_sep/GH95_3.2.1.51/family',7)
# Summarize families
fam.GH95_3.2.1.51.sum.sum <- fam.GH95_3.2.1.51.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH101_3.2.1.97
# Not found in some samples

### GH110_3.2.1.22
fam.GH110_3.2.1.22.sum <- import.family.sum('kaiju/mucin_sep/GH110_3.2.1.22/family',6)
# Summarize families
fam.GH110_3.2.1.22.sum.sum <- fam.GH110_3.2.1.22.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH112_2.4.1.211
fam.GH112_2.4.1.211.sum <- import.family.sum('kaiju/mucin_sep/GH112_2.4.1.211/family',6)
# Summarize families
fam.GH112_2.4.1.211.sum.sum <- fam.GH112_2.4.1.211.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH130_2.4.1.281
fam.GH130_2.4.1.281.sum <- import.family.sum('kaiju/mucin_sep/GH130_2.4.1.281/family',8)
# Summarize families
fam.GH130_2.4.1.281.sum.sum <- fam.GH130_2.4.1.281.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))

### GH130_2.4.1.320
fam.GH130_2.4.1.320.sum <- import.family.sum('kaiju/mucin_sep/GH130_2.4.1.320/family',7)
# Summarize families
fam.GH130_2.4.1.320.sum.sum <- fam.GH130_2.4.1.320.sum %>%
  group_by(Family) %>%
  summarize(Mean = mean(Mean))
```



### KEGG Brite
```{r}
ko.comp <- read.csv('KEGG/ko_comp.csv', fileEncoding="UTF-8-BOM")
rownames(ko.comp) <- ko.comp[,1]
ko.comp <- ko.comp[,-1]

# Remove SPAdes
kegg <- ko.comp[-which(ko.comp$Assembler == "SPAdes"),]


### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
rownames(norm.factor) <- c(3715,3717,3723,3733,3734,3744,3772,3773,3775)
colnames(norm.factor) <- "Factor"
kegg.norm <- kegg[,-26:-29]
for(i in 1:9){
  kegg.norm[i,] <- round(kegg.norm[i,]*norm.factor[i,1])
}
kegg.norm$Season <- c("Summer", "Summer", "Winter", "Winter", "Spring", "Summer", "Winter", "Spring", "Spring")




### Summarize KEGG
sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(sum)-1)){
  sum[4,i] <- sum(sum[1:3,i])/3
  sum[5,i] <- sd(sum[1:3,i])
  sum[6,i] <- sd(sum[1:3,i])/(sqrt(3))
}
rownames(sum)[4:6] <- c("Mean", "SD", "SE")
spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(spr)-1)){
  spr[4,i] <- sum(spr[1:3,i])/3
  spr[5,i] <- sd(spr[1:3,i])
  spr[6,i] <- sd(spr[1:3,i])/(sqrt(3))
}
rownames(spr)[4:6] <- c("Mean", "SD", "SE")
win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(win)-1)){
  win[4,i] <- sum(win[1:3,i])/3
  win[5,i] <- sd(win[1:3,i])
  win[6,i] <- sd(win[1:3,i])/(sqrt(3))
}
rownames(win)[4:6] <- c("Mean", "SD", "SE")
kegg.norm.sum <- sum[4:6,]
kegg.norm.sum[4:6,] <- win[4:6,]
kegg.norm.sum[7:9,] <- spr[4:6,]
colnames(kegg.norm.sum) <- colnames(sum)
kegg.norm.sum$Season <- c(rep("Summer",3), rep("Winter",3), rep("Spring",3))
rownames(kegg.norm.sum) <- c("Summer Mean", "Summer SD", "Summer SE", "Winter Mean", "Winter SD", "Winter SE", "Spring Mean", "Spring SD", "Spring SE")

### Check for Normal distribution
norm.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.test <- shapiro.test(kegg.norm[,i])
  norm.pvals[i,1] <- temp.test$p.value
  rownames(norm.pvals)[i] <- colnames(kegg.norm)[i]
}
length(which(norm.pvals[,1] < 0.06))
rownames(norm.pvals)[which(norm.pvals[,1] < 0.06)]

### Overall Summary
kegg.all.sum <- data.frame(999)
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.all.sum[1,i] <- sum(kegg.norm[1:9,i])/9
  kegg.all.sum[2,i] <- sd(kegg.norm[1:9,i])
  kegg.all.sum[3,i] <- sd(kegg.norm[1:9,i])/(sqrt(9))
}
colnames(kegg.all.sum) <- colnames(kegg.norm)[1:25]
rownames(kegg.all.sum) <- c("Mean", "SD", "SE")

### Overall Rel Abund
kegg.relabund <- kegg.norm[,-1]
kegg.relabund <- kegg.relabund[,-25]
for(i in 1:9){
  kegg.relabund[i+9,] <- (kegg.relabund[i,] / kegg.relabund[i,24])*100
}
kegg.relabund <- kegg.relabund[-1:-9,]
rownames(kegg.relabund) <- rownames(kegg.norm)[1:9]

kegg.relabund.sum <- data.frame(999)
for(i in 1:(ncol(kegg.relabund))){
  kegg.relabund.sum[1,i] <- sum(kegg.relabund[1:3,i])/3
  kegg.relabund.sum[2,i] <- sd(kegg.relabund[1:3,i])
  kegg.relabund.sum[3,i] <- sd(kegg.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.relabund.sum) <- colnames(kegg.relabund)[1:24]
rownames(kegg.relabund.sum) <- c("Mean", "SD", "SE")
kegg.relabund.sum <- kegg.relabund.sum[,order(kegg.relabund.sum[1,], decreasing=T)]




### Seasonal Summary
kegg.sum.sum <- data.frame(999)
kegg.sum <- kegg.norm[which(kegg.norm$Season == "Summer"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.sum.sum[1,i] <- sum(kegg.sum[1:3,i])/3
  kegg.sum.sum[2,i] <- sd(kegg.sum[1:3,i])
  kegg.sum.sum[3,i] <- sd(kegg.sum[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.sum) <- colnames(kegg.sum)[1:25]
rownames(kegg.sum.sum) <- c("Mean", "SD", "SE")
# Winter
kegg.win.sum <- data.frame(999)
kegg.win <- kegg.norm[which(kegg.norm$Season == "Winter"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.win.sum[1,i] <- sum(kegg.win[1:3,i])/3
  kegg.win.sum[2,i] <- sd(kegg.win[1:3,i])
  kegg.win.sum[3,i] <- sd(kegg.win[1:3,i])/(sqrt(3))
}
colnames(kegg.win.sum) <- colnames(kegg.win)[1:25]
rownames(kegg.win.sum) <- c("Mean", "SD", "SE")
#Spring
kegg.spr.sum <- data.frame(999)
kegg.spr <- kegg.norm[which(kegg.norm$Season == "Spring"),]
for(i in 1:(ncol(kegg.norm)-1)){
  kegg.spr.sum[1,i] <- sum(kegg.spr[1:3,i])/3
  kegg.spr.sum[2,i] <- sd(kegg.spr[1:3,i])
  kegg.spr.sum[3,i] <- sd(kegg.spr[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.sum) <- colnames(kegg.spr)[1:25]
rownames(kegg.spr.sum) <- c("Mean", "SD", "SE")

### Seasonal Rel Abund
kegg.seas.relabund <- kegg.norm[,-1]
kegg.seas.relabund <- kegg.seas.relabund[,-25]
for(i in 1:9){
  kegg.seas.relabund[i+9,] <- (kegg.seas.relabund[i,] / kegg.seas.relabund[i,24])*100
}
kegg.seas.relabund <- kegg.seas.relabund[-1:-9,]
rownames(kegg.seas.relabund) <- rownames(kegg.norm)[1:9]

sum <- which(kegg.norm$Season == "Summer")
win <- which(kegg.norm$Season == "Winter")
spr <- which(kegg.norm$Season == "Spring")
sum.relabund <- kegg.seas.relabund[sum,]
win.relabund <- kegg.seas.relabund[win,]
spr.relabund <- kegg.seas.relabund[spr,]

# Summer
kegg.sum.relabund.sum <- data.frame(999)
for(i in 1:(ncol(sum.relabund))){
  kegg.sum.relabund.sum[1,i] <- sum(sum.relabund[1:3,i])/3
  kegg.sum.relabund.sum[2,i] <- sd(sum.relabund[1:3,i])
  kegg.sum.relabund.sum[3,i] <- sd(sum.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.sum.relabund.sum) <- colnames(sum.relabund)[1:24]
rownames(kegg.sum.relabund.sum) <- c("Mean", "SD", "SE")
kegg.sum.relabund.sum <- kegg.sum.relabund.sum[,order(kegg.sum.relabund.sum[1,], decreasing=T)]

# Winter
kegg.win.relabund.sum <- data.frame(999)
for(i in 1:(ncol(win.relabund))){
  kegg.win.relabund.sum[1,i] <- sum(win.relabund[1:3,i])/3
  kegg.win.relabund.sum[2,i] <- sd(win.relabund[1:3,i])
  kegg.win.relabund.sum[3,i] <- sd(win.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.win.relabund.sum) <- colnames(win.relabund)[1:24]
rownames(kegg.win.relabund.sum) <- c("Mean", "SD", "SE")
kegg.win.relabund.sum <- kegg.win.relabund.sum[,order(kegg.win.relabund.sum[1,], decreasing=T)]

# Spring
kegg.spr.relabund.sum <- data.frame(999)
for(i in 1:(ncol(spr.relabund))){
  kegg.spr.relabund.sum[1,i] <- sum(spr.relabund[1:3,i])/3
  kegg.spr.relabund.sum[2,i] <- sd(spr.relabund[1:3,i])
  kegg.spr.relabund.sum[3,i] <- sd(spr.relabund[1:3,i])/(sqrt(3))
}
colnames(kegg.spr.relabund.sum) <- colnames(spr.relabund)[1:24]
rownames(kegg.spr.relabund.sum) <- c("Mean", "SD", "SE")
kegg.spr.relabund.sum <- kegg.spr.relabund.sum[,order(kegg.spr.relabund.sum[1,], decreasing=T)]


### Prep for barplot
sum.trim <- kegg.sum.relabund.sum
sum.trim <- data.frame(t(sum.trim))
sum.trim <- sum.trim[-1:-2,]
sum.trim <- sum.trim[-3,]
sum.trim <- sum.trim[-4,]
sum.trim <- sum.trim[-6:-20,]
sum.trim$Season <- rep("Summer",5)
sum.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
sum.trim$se.max <- sum.trim$Mean + sum.trim$SE
sum.trim$se.min <- sum.trim$Mean - sum.trim$SE
win.trim <- kegg.win.relabund.sum
win.trim <- data.frame(t(win.trim))
win.trim <- win.trim[-1:-2,]
win.trim <- win.trim[-3,]
win.trim <- win.trim[-4,]
win.trim <- win.trim[-6:-20,]
win.trim$Season <- rep("Winter",5)
win.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
win.trim$se.max <- win.trim$Mean + win.trim$SE
win.trim$se.min <- win.trim$Mean - win.trim$SE
spr.trim <- kegg.spr.relabund.sum
spr.trim <- data.frame(t(spr.trim))
spr.trim <- spr.trim[-1:-2,]
spr.trim <- spr.trim[-3,]
spr.trim <- spr.trim[-4,]
spr.trim <- spr.trim[-6:-20,]
spr.trim$Season <- rep("Spring",5)
spr.trim$Pathway <- c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism")
spr.trim$se.max <- spr.trim$Mean + spr.trim$SE
spr.trim$se.min <- spr.trim$Mean - spr.trim$SE
kegg.barplot <- sum.trim
kegg.barplot[6:10,] <- win.trim
kegg.barplot[11:15,] <- spr.trim
kegg.barplot$Pathway <- ordered(kegg.barplot$Pathway, levels=c("Other", "Carbohydrate Metabolism", "Amino Acid Metabolism", "Nucleotide Metabolism", "Energy Metabolism"))
kegg.barplot$Season <- ordered(kegg.barplot$Season, levels=c("Summer", "Winter", "Spring"))

##### BARPLOT! #####
tiff("kegg.barplot.tiff", width = 13, height = 7, units = "in", res = 300)
ggplot(kegg.barplot, aes(x=Pathway, y=Mean, fill=Pathway)) +
  geom_bar(stat="identity", color="black") +
  facet_grid(. ~ Season) +
  geom_errorbar(aes(ymax=se.max, ymin=se.min), width=0.25) +
  scale_fill_manual(values = c("lightcoral", "lightskyblue", "palegreen", "plum", "khaki1")) +
  xlab("Pathway") +
  ylab("Mean Relative Abundance") +
  scale_y_continuous(limits=c(0,55), expand=c(0,0))+
  theme(axis.text.x = element_blank(),
        panel.spacing=unit(1, "lines"),
        axis.title = element_text(size=20),
        strip.text.x = element_text(size=14),
        strip.text.y = element_text(size=14),
        axis.text.y = element_text(size=12),
        legend.title = element_text(size = 14),
        legend.text = element_text(size=12))
dev.off()



### ANOVA
kegg.pvals <- data.frame(rep(999, 25))
for(i in 1:(ncol(kegg.norm)-1)){
  temp.aov <- summary(aov(kegg.norm[,i] ~ Season, data=kegg.norm))
  kegg.pvals[i,1] <- temp.aov[[1]][1,5]
  rownames(kegg.pvals)[i] <- colnames(kegg)[i]
}
sig <- rownames(kegg.pvals)[which(kegg.pvals[,1] < 0.06)]
# 
# # Tukey HSD
# tukey.pvals <- data.frame(rep(999,4))
# cat <- which(colnames(kegg) == sig[1])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,1] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[2])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,2] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[3])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,3] <- temp.tukey$Season[,4]
# 
# cat <- which(colnames(kegg) == sig[4])
# temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
# temp.tukey <- TukeyHSD(temp.aov)
# tukey.pvals[1:3,4] <- temp.tukey$Season[,4]
# colnames(tukey.pvals) <- sig
# rownames(tukey.pvals) <- rownames(temp.tukey$Season)

# tukey.pvals <- data.frame(rep(999,4))
# for(i in 1:length(sig)){
#   cat <- which(colnames(kegg) == sig[i])
#   temp.aov <- aov(kegg[,cat] ~ Season, data=kegg)
#   temp.tukey <- TukeyHSD(temp.aov)
#   tukey.pvals[(1:3),i] <- temp.tukey$Season[,4]
#   rownames(tukey.pvals)[1:3,i] <- rownames(temp.tukey$Season)
# }

```





### KEGG Pathways
```{r}
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(class) <- class[,1]
class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

### Calculate proportions
path.prop <- path.norm
for(i in 1:9){
  path.prop[,i] <- (path.prop[,i] / colSums(path.prop)[i])
}
rem <- which(rowSums(path.prop) < 0.001)
path.prop <- path.prop[-rem,]

kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(kend) <- kend[,1]
kend[,2:9] <- NA
colnames(kend) <- kend[,1]
kend[,1] <- NA

kend[1,1] <- cor(path.prop[,1], path.prop[,1], method = "kendall")
kend[1,2] <- cor(path.prop[,1], path.prop[,2], method = "kendall")
kend[1,3] <- cor(path.prop[,1], path.prop[,3], method = "kendall")
kend[1,4] <- cor(path.prop[,1], path.prop[,4], method = "kendall")
kend[1,5] <- cor(path.prop[,1], path.prop[,5], method = "kendall")
kend[1,6] <- cor(path.prop[,1], path.prop[,6], method = "kendall")
kend[1,7] <- cor(path.prop[,1], path.prop[,7], method = "kendall")
kend[1,8] <- cor(path.prop[,1], path.prop[,8], method = "kendall")
kend[1,9] <- cor(path.prop[,1], path.prop[,9], method = "kendall")
kend[2,1] <- cor(path.prop[,2], path.prop[,1], method = "kendall")
kend[2,2] <- cor(path.prop[,2], path.prop[,2], method = "kendall")
kend[2,3] <- cor(path.prop[,2], path.prop[,3], method = "kendall")
kend[2,4] <- cor(path.prop[,2], path.prop[,4], method = "kendall")
kend[2,5] <- cor(path.prop[,2], path.prop[,5], method = "kendall")
kend[2,6] <- cor(path.prop[,2], path.prop[,6], method = "kendall")
kend[2,7] <- cor(path.prop[,2], path.prop[,7], method = "kendall")
kend[2,8] <- cor(path.prop[,2], path.prop[,8], method = "kendall")
kend[2,9] <- cor(path.prop[,2], path.prop[,9], method = "kendall")
kend[3,1] <- cor(path.prop[,3], path.prop[,1], method = "kendall")
kend[3,2] <- cor(path.prop[,3], path.prop[,2], method = "kendall")
kend[3,3] <- cor(path.prop[,3], path.prop[,3], method = "kendall")
kend[3,4] <- cor(path.prop[,3], path.prop[,4], method = "kendall")
kend[3,5] <- cor(path.prop[,3], path.prop[,5], method = "kendall")
kend[3,6] <- cor(path.prop[,3], path.prop[,6], method = "kendall")
kend[3,7] <- cor(path.prop[,3], path.prop[,7], method = "kendall")
kend[3,8] <- cor(path.prop[,3], path.prop[,8], method = "kendall")
kend[3,9] <- cor(path.prop[,3], path.prop[,9], method = "kendall")
kend[4,1] <- cor(path.prop[,4], path.prop[,1], method = "kendall")
kend[4,2] <- cor(path.prop[,4], path.prop[,2], method = "kendall")
kend[4,3] <- cor(path.prop[,4], path.prop[,3], method = "kendall")
kend[4,4] <- cor(path.prop[,4], path.prop[,4], method = "kendall")
kend[4,5] <- cor(path.prop[,4], path.prop[,5], method = "kendall")
kend[4,6] <- cor(path.prop[,4], path.prop[,6], method = "kendall")
kend[4,7] <- cor(path.prop[,4], path.prop[,7], method = "kendall")
kend[4,8] <- cor(path.prop[,4], path.prop[,8], method = "kendall")
kend[4,9] <- cor(path.prop[,4], path.prop[,9], method = "kendall")
kend[5,1] <- cor(path.prop[,5], path.prop[,1], method = "kendall")
kend[5,2] <- cor(path.prop[,5], path.prop[,2], method = "kendall")
kend[5,3] <- cor(path.prop[,5], path.prop[,3], method = "kendall")
kend[5,4] <- cor(path.prop[,5], path.prop[,4], method = "kendall")
kend[5,5] <- cor(path.prop[,5], path.prop[,5], method = "kendall")
kend[5,6] <- cor(path.prop[,5], path.prop[,6], method = "kendall")
kend[5,7] <- cor(path.prop[,5], path.prop[,7], method = "kendall")
kend[5,8] <- cor(path.prop[,5], path.prop[,8], method = "kendall")
kend[5,9] <- cor(path.prop[,5], path.prop[,9], method = "kendall")
kend[6,1] <- cor(path.prop[,6], path.prop[,1], method = "kendall")
kend[6,2] <- cor(path.prop[,6], path.prop[,2], method = "kendall")
kend[6,3] <- cor(path.prop[,6], path.prop[,3], method = "kendall")
kend[6,4] <- cor(path.prop[,6], path.prop[,4], method = "kendall")
kend[6,5] <- cor(path.prop[,6], path.prop[,5], method = "kendall")
kend[6,6] <- cor(path.prop[,6], path.prop[,6], method = "kendall")
kend[6,7] <- cor(path.prop[,6], path.prop[,7], method = "kendall")
kend[6,8] <- cor(path.prop[,6], path.prop[,8], method = "kendall")
kend[6,9] <- cor(path.prop[,6], path.prop[,9], method = "kendall")
kend[7,1] <- cor(path.prop[,7], path.prop[,1], method = "kendall")
kend[7,2] <- cor(path.prop[,7], path.prop[,2], method = "kendall")
kend[7,3] <- cor(path.prop[,7], path.prop[,3], method = "kendall")
kend[7,4] <- cor(path.prop[,7], path.prop[,4], method = "kendall")
kend[7,5] <- cor(path.prop[,7], path.prop[,5], method = "kendall")
kend[7,6] <- cor(path.prop[,7], path.prop[,6], method = "kendall")
kend[7,7] <- cor(path.prop[,7], path.prop[,7], method = "kendall")
kend[7,8] <- cor(path.prop[,7], path.prop[,8], method = "kendall")
kend[7,9] <- cor(path.prop[,7], path.prop[,9], method = "kendall")
kend[8,1] <- cor(path.prop[,8], path.prop[,1], method = "kendall")
kend[8,2] <- cor(path.prop[,8], path.prop[,2], method = "kendall")
kend[8,3] <- cor(path.prop[,8], path.prop[,3], method = "kendall")
kend[8,4] <- cor(path.prop[,8], path.prop[,4], method = "kendall")
kend[8,5] <- cor(path.prop[,8], path.prop[,5], method = "kendall")
kend[8,6] <- cor(path.prop[,8], path.prop[,6], method = "kendall")
kend[8,7] <- cor(path.prop[,8], path.prop[,7], method = "kendall")
kend[8,8] <- cor(path.prop[,8], path.prop[,8], method = "kendall")
kend[8,9] <- cor(path.prop[,8], path.prop[,9], method = "kendall")
kend[9,1] <- cor(path.prop[,9], path.prop[,1], method = "kendall")
kend[9,2] <- cor(path.prop[,9], path.prop[,2], method = "kendall")
kend[9,3] <- cor(path.prop[,9], path.prop[,3], method = "kendall")
kend[9,4] <- cor(path.prop[,9], path.prop[,4], method = "kendall")
kend[9,5] <- cor(path.prop[,9], path.prop[,5], method = "kendall")
kend[9,6] <- cor(path.prop[,9], path.prop[,6], method = "kendall")
kend[9,7] <- cor(path.prop[,9], path.prop[,7], method = "kendall")
kend[9,8] <- cor(path.prop[,9], path.prop[,8], method = "kendall")
kend[9,9] <- cor(path.prop[,9], path.prop[,9], method = "kendall")

#write.table(kend, file="kegg.kendall.txt", sep="\t")

```

### Pathway UPGMA
```{r}
kend.fix <- kend
kend.fix[1,1] <- 0
kend.fix[2,2] <- 0
kend.fix[3,3] <- 0
kend.fix[4,4] <- 0
kend.fix[5,5] <- 0
kend.fix[6,6] <- 0
kend.fix[7,7] <- 0
kend.fix[8,8] <- 0
kend.fix[9,9] <- 0
path.upgma <- upgma(kend.fix)
plot(path.upgma)


path.nj <- NJ(data.frame(kend.fix))

```




### Pathway Bray-Curtis
```{r}
# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]
kegg.class <- kegg.class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
path.norm <- path
for(i in 1:9){
  path.norm[,i] <- round(path.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(path.norm)){
  print(sum(path.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
path.otu <- otu_table(path.norm, taxa_are_rows = T)
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

nmds.kegg <- ordinate(physeq=path.physeq, method="NMDS", distance="bray")
  # stress = 0.01481376

plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Season")
plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Diet")
plot_ordination(physeq=path.physeq, ordination=nmds.kegg, color="Sex")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.bray<- phyloseq::distance(physeq=path.physeq, method="bray")

# Season
adonis(path.bray~ Season, data=path.sampdf)
  # p-value = 0.333
beta.tax = betadisper(d=path.bray, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.304

# Diet
adonis(path.bray~ Diet, data=path.sampdf)
  # p-value = 0.515
beta.tax = betadisper(d=path.bray, group=path.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.368

# Sex
adonis(path.bray~ Sex, data=path.sampdf)
  # p-value = 0.241
beta.tax = betadisper(d=path.bray, group=path.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.592

### ANOSIM
anosim(path.bray, meta$Season, permutations=1000)
  # p-value = 0.13986
anosim(path.bray, meta$Diet, permutations=1000)
  # p-value =  0.50849
anosim(path.bray, meta$Sex, permutations=1000)
  # p-value = 0.52747
```

### Pathway Jaccard
```{r}
jac.kegg <- ordinate(physeq=path.physeq, method="NMDS", distance="jaccard")
  # stress = 0.02570319
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Season")
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Diet")
plot_ordination(physeq=path.physeq, ordination=jac.kegg, color="Sex")

### PERMANOVA
path.sampdf <- data.frame(sample_data(path.physeq))
path.jac <- phyloseq::distance(physeq=path.physeq, method="jaccard")

# Season
adonis(path.jac~ Season, data=path.sampdf)
  # p-value = 0.235
beta.tax = betadisper(d=path.jac, group=path.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.367

# Diet
adonis(path.jac~ Diet, data=path.sampdf)
  # p-value = 0.456
beta.tax = betadisper(d=path.jac, group=path.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.372

# Diet
adonis(path.jac~ Sex, data=path.sampdf)
  # p-value = 0.248
beta.tax = betadisper(d=path.jac, group=path.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.61

### ANOSIM
anosim(path.jac, path.sampdf$Season, permutations=1000)
  # p-value = 0.14386
anosim(path.jac, path.sampdf$Diet, permutations=1000)
  # p-value = 0.5045
anosim(path.jac, path.sampdf$Sex, permutations=1000)
  # 0.52348
```

### Pathway SIMPER
```{r}
simper.pretty(path.norm, metrics = meta, interesting = c('Season','Diet', 'Sex'), perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'kegg')
path.simper <- read.csv("kegg_clean_simper.csv")
kruskal.pretty(otu = path.norm, metrics = meta, csv =  path.simper, interesting = c('Season','Diet', 'Sex'), output_name = 'kegg')
path.simper.kw <- read.csv("kegg_krusk_simper.csv")
```

### Pathway DESeq
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]

path.otu <- otu_table(path, taxa_are_rows = T)
#path.otu <- path.otu + 1
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
path.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.path <- rownames(test[sum,])
  # "ko04370" "ko04145"
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])

# Spring vs. Summer
ss.physeq <- subset_samples(path.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
  # "ko00650" "ko01051" "ko04122" "ko02010"
sum <- which(test$log2FoldChange > 0)
sum.path <- rownames(test[sum,])
  # "ko00061" "ko00540" "ko00740" "ko00130"

# Spring vs. Winter
sw.physeq <- subset_samples(path.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.path <- rownames(test[spr,])
  # "ko00650" "ko01051" "ko04122" "ko02010"
win <- which(test$log2FoldChange > 0)
win.path <- rownames(test[win,])
  # "ko00061" "ko00540" "ko00740" "ko00130"

# Feed vs Fast
test <- deSEQ(data=path.physeq, valuetest= ~Diet)
fast <- which(test$log2FoldChange < 0)
fast.path <- rownames(test[spr,])
  # "ko00040"
feed <- which(test$log2FoldChange > 0)
feed.path <- rownames(test[win,])
  # "ko00190" "ko00940" "ko00405" "ko04145"

### Identify Overrep kegg families
kegg.feed.id <- data.frame(feed.path)
for(i in 1:length(feed.path)){
  test <- which(feed.path[i] == kegg.class[,1])
  kegg.feed.id[i,2] <- kegg.class[test,2]
  kegg.feed.id[i,3] <- kegg.class[test,3]
  kegg.feed.id[i,4] <- kegg.class[test,4]
}
#write.csv(kegg.feed.id, 'KEGG/kegg.feed.id.csv')
kegg.fast.id <- data.frame(fast.path[1])
test <- which(fast.path[1] == kegg.class[,1])
kegg.fast.id[1,2] <- kegg.class[test,2]
kegg.fast.id[1,3] <- kegg.class[test,3]
kegg.fast.id[1,4] <- kegg.class[test,4]
#write.csv(kegg.fast.id, 'KEGG/kegg.fast.id.csv')
  
# for(i in 1:length(fast.path[1])){
#   test <- which(fast.path[i] == kegg.class[,1])
#   kegg.fast.id[i,2] <- kegg.class[test,2]
#   kegg.fast.id[i,3] <- kegg.class[test,3]
#   kegg.fast.id[i,4] <- kegg.class[test,4]
# }
#write.csv(kegg.fast.id, 'kegg/kegg.fast.id.csv')

# Female vs Male
test <- deSEQ(data=path.physeq, valuetest= ~Sex)
fem <- which(test$log2FoldChange < 0)
fem.path <- rownames(test[fem,])
  # "ko00232" "ko03040" "ko03013" "ko03015" "ko03008" "ko04141" "ko03460" "ko04014" "ko04015" "ko04010" "ko04011" "ko04310" "ko04340" "ko04390" "ko04392" "ko04371" "ko04064" "ko04668" "ko04068" "ko04020" "ko04072" "ko04071" "ko04024" "ko04022" "ko04150" "ko04144" "ko04145" "ko04139" "ko04110" "ko04111" "ko04113" "ko04210" "ko04810"
mal <- which(test$log2FoldChange > 0)
mal.path <- rownames(test[mal,])
  # "ko00190" "ko00940" "ko00405" "ko04145"
```


### KEGG Heatmap
```{r}
# Prepare physeq object
path <- read.csv('KEGG/kegg_counts.csv', fileEncoding="UTF-8-BOM")
row.names(path) <- path[,1]
path <- path[,-1]
colnames(path) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

kegg.class <- read.csv('KEGG/classification.csv', fileEncoding="UTF-8-BOM")
row.names(kegg.class) <- kegg.class[,1]

path.otu <- otu_table(path, taxa_are_rows = T)
path.samp <- sample_data(meta)
path.tax <- tax_table(kegg.class)
row.names(path.tax) <- row.names(kegg.class)
kegg.physeq <- merge_phyloseq(path.otu, path.samp, path.tax)


kegg.de.data <- phyloseq_to_deseq2(kegg.physeq, ~Season)
kegg.de <- DESeq(kegg.de.data, test="Wald", fitType="local")

kegg.res <- results(kegg.de, pAdjustMethod = "BH")

### Extracting results: https://support.bioconductor.org/p/72765/

# Summer vs. Winter
kegg.res.su.wi <- results(kegg.de, contrast=c("Season", "Summer", "Winter"), pAdjustMethod = "BH")
kegg.su.wi.sig <- kegg.res.su.wi[which(kegg.res.su.wi$padj < 0.05),]
  # No sig!!!
# kegg.su.wi.sig.df <- cbind(as(kegg.su.wi.sig, "data.frame"),
#                           as(tax_table(kegg.physeq)[rownames(kegg.su.wi.sig), ], "matrix"))
# kegg.su.wi.sum.sig <- which(kegg.su.wi.sig.df$log2FoldChange < 0)
# kegg.su.wi.sum <- rownames(kegg.su.wi.sig[kegg.su.wi.sum.sig,])
#   # Summer
# kegg.su.wi.win.sig <- which(kegg.su.wi.sig.df$log2FoldChange > 0)
# kegg.su.wi.win <- rownames(kegg.su.wi.sig[kegg.su.wi.win.sig,])
#   # Winter

# Summer vs. Spring
kegg.res.su.sp <- results(kegg.de, contrast=c("Season", "Spring", "Summer"), pAdjustMethod = "BH")
kegg.su.sp.sig <- kegg.res.su.sp[which(kegg.res.su.sp$padj < 0.05),]
kegg.su.sp.sig.df <- cbind(as(kegg.su.sp.sig, "data.frame"),
                          as(tax_table(kegg.physeq)[rownames(kegg.su.sp.sig), ], "matrix"))
kegg.su.sp.spr.sig <- which(kegg.su.sp.sig.df$log2FoldChange < 0)
kegg.su.sp.spr <- rownames(kegg.su.sp.sig[kegg.su.sp.spr.sig,])
  # Spring
  # "ko00061" "ko00540" "ko00740" "ko00130"
kegg.su.sp.sum.sig <- which(kegg.su.sp.sig.df$log2FoldChange > 0)
kegg.su.sp.sum <- rownames(kegg.su.sp.sig[kegg.su.sp.sum.sig,])
  # Summer
  #  "ko00030" "ko00650" "ko01051" "ko04122" "ko02010"

# Spring vs Winter
kegg.res.sp.wi <- results(kegg.de, contrast=c("Season", "Spring", "Winter"), pAdjustMethod = "BH")
kegg.sp.wi.sig <- kegg.res.sp.wi[which(kegg.res.sp.wi$padj < 0.05),]
kegg.sp.wi.sig.df <- cbind(as(kegg.sp.wi.sig, "data.frame"),
                          as(tax_table(kegg.physeq)[rownames(kegg.sp.wi.sig), ], "matrix"))
kegg.sp.wi.spr.sig <- which(kegg.sp.wi.sig.df$log2FoldChange < 0)
kegg.sp.wi.spr <- rownames(kegg.sp.wi.sig[kegg.sp.wi.spr.sig,])
  # Spring
  # "ko00190" "ko00410" "ko00531" "ko00604" "ko00540" "ko00780" "ko00790" "ko00130"
kegg.sp.wi.win.sig <- which(kegg.sp.wi.sig.df$log2FoldChange > 0)
kegg.sp.wi.win <- rownames(kegg.sp.wi.sig[kegg.sp.wi.win.sig,])
  # Winter
  # "ko00040" "ko00051" "ko00500" "ko00660" "ko00290" "ko00460" "ko00940" "ko00405" "ko02010" "ko02060" "ko04145"

# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

### Pull out top 2 (largest log fold change while sig) for each comparison
# kegg.su.wi.sum.df <- kegg.su.wi.sig.df[kegg.su.wi.sum.sig,]
# kegg.su.wi.sum.df <- kegg.su.wi.sum.df[order(kegg.su.wi.sum.df$log2FoldChange),]
# kegg.su.wi.win.df <- kegg.su.wi.sig.df[kegg.su.wi.win.sig,]
# kegg.su.wi.win.df <- kegg.su.wi.win.df[order(kegg.su.wi.win.df$log2FoldChange),]

kegg.su.sp.spr.df <- kegg.su.sp.sig.df[kegg.su.sp.spr.sig,]
kegg.su.sp.spr.df <- kegg.su.sp.spr.df[order(kegg.su.sp.spr.df$log2FoldChange),]
kegg.su.sp.sum.df <- kegg.su.sp.sig.df[kegg.su.sp.sum.sig,]
kegg.su.sp.sum.df <- kegg.su.sp.sum.df[order(kegg.su.sp.sum.df$log2FoldChange),]
kegg.sp.wi.spr.df <- kegg.sp.wi.sig.df[kegg.sp.wi.spr.sig,]
kegg.sp.wi.spr.df <- kegg.sp.wi.spr.df[order(kegg.sp.wi.spr.df$log2FoldChange),]
kegg.sp.wi.win.df <- kegg.sp.wi.sig.df[kegg.sp.wi.win.sig,]
kegg.sp.wi.win.df <- kegg.sp.wi.win.df[order(kegg.sp.wi.win.df$log2FoldChange),]

kegg.sig <- rbind(kegg.su.sp.spr.df, kegg.su.sp.sum.df, kegg.sp.wi.spr.df, kegg.sp.wi.win.df)
kegg.sig$log2FoldChange[which(kegg.sig$log2FoldChange < 0)] <-
  kegg.sig$log2FoldChange[which(kegg.sig$log2FoldChange < 0)] * -1
kegg.sig <- kegg.sig[order(kegg.sig$log2FoldChange, decreasing=T),]
#kegg.sig <- kegg.sig[order(kegg.sig$padj),]
kegg.use <- rownames(kegg.sig)[1:10] 
kegg.use[5] <- "ko00540"
kegg.use <- kegg.use[-6]
kegg.use <- kegg.use[-7]
kegg.use[9:10] <- rownames(kegg.sig)[11:12]
kegg.use[10] <- rownames(kegg.sig)[13]
kegg.use <- kegg.use[-9]
kegg.use[10] <- rownames(kegg.sig)[14]

#select <- order(rowMeans(counts(kegg.de, normalized=T)), decreasing=T)[1:20]
  # For top 20 keggs
kegg.rows <- rownames(kegg.de)
kegg.select <- rep(1,10)
for(i in 1:length(kegg.use)){
  save <- which(kegg.rows == kegg.use[i])
  kegg.select[i] <- save
}
kegg.de.df <- data.frame(c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"))
rownames(kegg.de.df) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")
colnames(kegg.de.df) <- "Season"
kegg.de.df$Season <- ordered(kegg.de.df$Season, levels=c("Summer", "Winter", "Spring"))
kegg.de$X <- ordered(kegg.de$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))
kegg.ntd <- normTransform(kegg.de)
  # DESeq test automatically transforms data
  # To visualize our data, we need to run this step separately
  # I use normTransform because I run DESeq with local fit
      # Local fit does a transformation similar to normTransform
  # Otherwise you should use vst instead
  # See link above for more info


kegg.ntd.assay <- data.frame(assay(kegg.ntd)[kegg.select,])
kegg.ntd.assay.order <- kegg.ntd.assay
kegg.ntd.assay.order[,1] <- kegg.ntd.assay$X3715
kegg.ntd.assay.order[,2] <- kegg.ntd.assay$X3717
kegg.ntd.assay.order[,3] <- kegg.ntd.assay$X3744
kegg.ntd.assay.order[,4] <- kegg.ntd.assay$X3723
kegg.ntd.assay.order[,5] <- kegg.ntd.assay$X3733
kegg.ntd.assay.order[,6] <- kegg.ntd.assay$X3772
kegg.ntd.assay.order[,7] <- kegg.ntd.assay$X3734
kegg.ntd.assay.order[,8] <- kegg.ntd.assay$X3773
kegg.ntd.assay.order[,9] <- kegg.ntd.assay$X3775
colnames(kegg.ntd.assay.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

tiff("kegg_DESeq.tiff", width=6, height=5.5, units="in", res=600)
pheatmap(kegg.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, treeheight_col = 0,
         show_colnames=F, border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255),
         labels_col = c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"),
         annotation_col=kegg.de.df, annotation_legend = F,
         annotation_colors = list(Season = c(Summer = "#00cccc", Winter = "#008484", Spring = "#004f4f"))
)
dev.off()


tiff("kegg_DESeq1.tiff", width=3, height=2.25, units="in", res=600)
pheatmap(kegg.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, show_colnames=F, 
         border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
)
dev.off()


pheatmap(counts(cog.de)[kegg.select,], cluster_rows=F, show_rownames=T, cluster_cols=F, show_colnames=T, 
+          border_color = "gray40", fontsize = 14, width = 3, height=3,
+          color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
+ )



































# Prepare phyloseq object
kegg.size.fac <- estimateSizeFactors(kegg.de)
kegg.norm.deseq <- counts(kegg.size.fac, normalized=TRUE)
kegg.otu.deseq <- otu_table(kegg.norm.deseq, taxa_are_rows = T)
kegg.samp <- sample_data(meta)
kegg.tax <- tax_table(kegg.class)
rownames(kegg.tax) <- rownames(kegg.class)
kegg.deseq <- merge_phyloseq(kegg.otu.deseq, kegg.samp, kegg.tax)
sample_data(kegg.deseq)$Season <- ordered(sample_data(kegg.deseq)$Season, levels=c("Summer", "Winter", "Spring"))

nmds.kegg.deseq <- ordinate(physeq=kegg.deseq, method="NMDS", distance="bray")
  # stress = 0.03492425

### SEASON ELLIPSES
sample_data(kegg.deseq)$Elip <- paste(sample_data(kegg.deseq)$Season)
kegg.deseq.elip.sum <- data.frame(MDS1 = nmds.kegg.deseq$points[,1], MDS2 = nmds.kegg.deseq$points[,2],
                           Group = sample_data(kegg.deseq)$Season)
plot.new()
kegg.deseq.elip.ord <- ordiellipse(nmds.kegg.deseq, sample_data(kegg.deseq)$Elip, display="sites", kind="se", conf=0.95, label=T)
kegg.deseq.elip.sum.df <- data.frame()
for(i in levels(kegg.deseq.elip.sum$Group)){
  kegg.deseq.elip.sum.df <- rbind(kegg.deseq.elip.sum.df,
                           cbind(as.data.frame(with(kegg.deseq.elip.sum[kegg.deseq.elip.sum$Group==i,],
                                                    veganCovEllipse(kegg.deseq.elip.ord[[i]]$cov,
                                                                    kegg.deseq.elip.ord[[i]]$center,
                                                                    kegg.deseq.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
tiff("kegg_NMDS.tiff", width=3.5, height=3.5, units="in", res=600)
plot_ordination(physeq=kegg.deseq, ordination=nmds.kegg.deseq, color="Season") +
  geom_path(data=kegg.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
    theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
dev.off()

kegg.nmds.plot <- plot_ordination(physeq=kegg.deseq, ordination=nmds.kegg.deseq, color="Season") +
  geom_path(data=kegg.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
    theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5),
        plot.margin = unit(c(0.1, 0.1, 0.1, 1), "cm")) +
  guides(color=F)


### PERMANOVA
kegg.deseq.sampdf <- data.frame(sample_data(kegg.deseq))
kegg.deseq.bray<- phyloseq::distance(physeq=kegg.deseq, method="bray")

## Season
adonis(kegg.deseq.bray~ Season, data=kegg.deseq.sampdf)
  # p-value = 0.075
beta.tax = betadisper(d=kegg.deseq.bray, group=kegg.deseq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.457

## Diet
adonis(kegg.deseq.bray~ Diet, data=kegg.deseq.sampdf)
  # p-value = 0.371
beta.tax = betadisper(d=kegg.deseq.bray, group=kegg.deseq.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.052

```


### COG Count Prep
```{r}
### Count Counts
# Load tables
cog.3715 <- read.table("COG/3715.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3717 <- read.table("COG/3717.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3723 <- read.table("COG/3723.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3733 <- read.table("COG/3733.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3734 <- read.table("COG/3734.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3744 <- read.table("COG/3744.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3772 <- read.table("COG/3772.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3773 <- read.table("COG/3773.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)
cog.3775 <- read.table("COG/3775.results/cog_stats.txt", sep="\t", quote="", stringsAsFactors = F)

# Create COG count table
cog.names <- paste(c(cog.3715[,1], cog.3717[,1], cog.3723[,1], cog.3733[,1], cog.3734[,1], cog.3744[,1], cog.3772[,1], cog.3773[,1], cog.3775[,1]))
cog.uniq <- unique(cog.names)
cog.all <- data.frame(cog.uniq)
for(i in 1:nrow(cog.3715)){
  test <- which(cog.all[,1] == cog.3715[i,1])
  cog.all[test,2] <- cog.3715[i,3]
}
for(i in 1:nrow(cog.3717)){
  test <- which(cog.all[,1] == cog.3717[i,1])
  cog.all[test,3] <- cog.3717[i,3]
}
for(i in 1:nrow(cog.3723)){
  test <- which(cog.all[,1] == cog.3723[i,1])
  cog.all[test,4] <- cog.3723[i,3]
}
for(i in 1:nrow(cog.3733)){
  test <- which(cog.all[,1] == cog.3733[i,1])
  cog.all[test,5] <- cog.3733[i,3]
}
for(i in 1:nrow(cog.3734)){
  test <- which(cog.all[,1] == cog.3734[i,1])
  cog.all[test,6] <- cog.3734[i,3]
}
for(i in 1:nrow(cog.3744)){
  test <- which(cog.all[,1] == cog.3744[i,1])
  cog.all[test,7] <- cog.3744[i,3]
}
for(i in 1:nrow(cog.3772)){
  test <- which(cog.all[,1] == cog.3772[i,1])
  cog.all[test,8] <- cog.3772[i,3]
}
for(i in 1:nrow(cog.3773)){
  test <- which(cog.all[,1] == cog.3773[i,1])
  cog.all[test,9] <- cog.3773[i,3]
}
for(i in 1:nrow(cog.3775)){
  test <- which(cog.all[,1] == cog.3775[i,1])
  cog.all[test,10] <- cog.3775[i,3]
}
rownames(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")
cog.all[is.na(cog.all)] <- 0
  # Replaces all NA's w/ 0


# Save COG count table
#write.csv(cog.all, "COG/cog_counts.csv")

### Create COG Classification table
# Temp version
cog.class <- data.frame(cog.uniq)
#write.csv(cog.class, "COG/cog_classification.csv")


# Load tables
cog.prot.3715 <- read.table("COG/3715.results/protein-id_cog.edit.txt", sep="\t", quote="", stringsAsFactors = F, na = NA)
test <- unique(cog.prot.3715)
test1 <- cog.prot.3715[,2]
test2 <- which(rownames(cog.all) == test1[1])
test3 <- cog.all

cog..prot.3717 <- read.table("COG/3717.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3723 <- read.table("COG/3723.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3733 <- read.table("COG/3733.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3734 <- read.table("COG/3734.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3744 <- read.table("COG/3744.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3772 <- read.table("COG/3772.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3773 <- read.table("COG/3773.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
cog..prot.3775 <- read.table("COG/3775.results/protein-id_cog.txt", sep="\t", quote="", stringsAsFactors = F)
```

### COG Kendall Correlation
```{r}
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

class <- read.csv('COG/cog_classification.csv')
rownames(class) <- class[,1]
#class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
cog.all.norm <- cog.all
for(i in 1:9){
  cog.all.norm[,i] <- round(cog.all.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(cog.all.norm)){
  print(sum(cog.all.norm[i,]))
}
  # All > 0

### Calculate proportions
cog.all.prop <- cog.all.norm
for(i in 1:9){
  cog.all.prop[,i] <- (cog.all.prop[,i] / colSums(cog.all.prop)[i])
}
rem <- which(rowSums(cog.all.prop) < 0.001)
cog.all.prop <- cog.all.prop[-rem,]

cog.kend <- data.frame(c(3715, 3717, 3723, 3733, 3734, 3744, 3772, 3773, 3775))
rownames(cog.kend) <- cog.kend[,1]
cog.kend[,2:9] <- NA
colnames(cog.kend) <- cog.kend[,1]
cog.kend[,1] <- NA

cog.kend[1,1] <- cor(cog.all.prop[,1], cog.all.prop[,1], method = "kendall")
cog.kend[1,2] <- cor(cog.all.prop[,1], cog.all.prop[,2], method = "kendall")
cog.kend[1,3] <- cor(cog.all.prop[,1], cog.all.prop[,3], method = "kendall")
cog.kend[1,4] <- cor(cog.all.prop[,1], cog.all.prop[,4], method = "kendall")
cog.kend[1,5] <- cor(cog.all.prop[,1], cog.all.prop[,5], method = "kendall")
cog.kend[1,6] <- cor(cog.all.prop[,1], cog.all.prop[,6], method = "kendall")
cog.kend[1,7] <- cor(cog.all.prop[,1], cog.all.prop[,7], method = "kendall")
cog.kend[1,8] <- cor(cog.all.prop[,1], cog.all.prop[,8], method = "kendall")
cog.kend[1,9] <- cor(cog.all.prop[,1], cog.all.prop[,9], method = "kendall")
cog.kend[2,1] <- cor(cog.all.prop[,2], cog.all.prop[,1], method = "kendall")
cog.kend[2,2] <- cor(cog.all.prop[,2], cog.all.prop[,2], method = "kendall")
cog.kend[2,3] <- cor(cog.all.prop[,2], cog.all.prop[,3], method = "kendall")
cog.kend[2,4] <- cor(cog.all.prop[,2], cog.all.prop[,4], method = "kendall")
cog.kend[2,5] <- cor(cog.all.prop[,2], cog.all.prop[,5], method = "kendall")
cog.kend[2,6] <- cor(cog.all.prop[,2], cog.all.prop[,6], method = "kendall")
cog.kend[2,7] <- cor(cog.all.prop[,2], cog.all.prop[,7], method = "kendall")
cog.kend[2,8] <- cor(cog.all.prop[,2], cog.all.prop[,8], method = "kendall")
cog.kend[2,9] <- cor(cog.all.prop[,2], cog.all.prop[,9], method = "kendall")
cog.kend[3,1] <- cor(cog.all.prop[,3], cog.all.prop[,1], method = "kendall")
cog.kend[3,2] <- cor(cog.all.prop[,3], cog.all.prop[,2], method = "kendall")
cog.kend[3,3] <- cor(cog.all.prop[,3], cog.all.prop[,3], method = "kendall")
cog.kend[3,4] <- cor(cog.all.prop[,3], cog.all.prop[,4], method = "kendall")
cog.kend[3,5] <- cor(cog.all.prop[,3], cog.all.prop[,5], method = "kendall")
cog.kend[3,6] <- cor(cog.all.prop[,3], cog.all.prop[,6], method = "kendall")
cog.kend[3,7] <- cor(cog.all.prop[,3], cog.all.prop[,7], method = "kendall")
cog.kend[3,8] <- cor(cog.all.prop[,3], cog.all.prop[,8], method = "kendall")
cog.kend[3,9] <- cor(cog.all.prop[,3], cog.all.prop[,9], method = "kendall")
cog.kend[4,1] <- cor(cog.all.prop[,4], cog.all.prop[,1], method = "kendall")
cog.kend[4,2] <- cor(cog.all.prop[,4], cog.all.prop[,2], method = "kendall")
cog.kend[4,3] <- cor(cog.all.prop[,4], cog.all.prop[,3], method = "kendall")
cog.kend[4,4] <- cor(cog.all.prop[,4], cog.all.prop[,4], method = "kendall")
cog.kend[4,5] <- cor(cog.all.prop[,4], cog.all.prop[,5], method = "kendall")
cog.kend[4,6] <- cor(cog.all.prop[,4], cog.all.prop[,6], method = "kendall")
cog.kend[4,7] <- cor(cog.all.prop[,4], cog.all.prop[,7], method = "kendall")
cog.kend[4,8] <- cor(cog.all.prop[,4], cog.all.prop[,8], method = "kendall")
cog.kend[4,9] <- cor(cog.all.prop[,4], cog.all.prop[,9], method = "kendall")
cog.kend[5,1] <- cor(cog.all.prop[,5], cog.all.prop[,1], method = "kendall")
cog.kend[5,2] <- cor(cog.all.prop[,5], cog.all.prop[,2], method = "kendall")
cog.kend[5,3] <- cor(cog.all.prop[,5], cog.all.prop[,3], method = "kendall")
cog.kend[5,4] <- cor(cog.all.prop[,5], cog.all.prop[,4], method = "kendall")
cog.kend[5,5] <- cor(cog.all.prop[,5], cog.all.prop[,5], method = "kendall")
cog.kend[5,6] <- cor(cog.all.prop[,5], cog.all.prop[,6], method = "kendall")
cog.kend[5,7] <- cor(cog.all.prop[,5], cog.all.prop[,7], method = "kendall")
cog.kend[5,8] <- cor(cog.all.prop[,5], cog.all.prop[,8], method = "kendall")
cog.kend[5,9] <- cor(cog.all.prop[,5], cog.all.prop[,9], method = "kendall")
cog.kend[6,1] <- cor(cog.all.prop[,6], cog.all.prop[,1], method = "kendall")
cog.kend[6,2] <- cor(cog.all.prop[,6], cog.all.prop[,2], method = "kendall")
cog.kend[6,3] <- cor(cog.all.prop[,6], cog.all.prop[,3], method = "kendall")
cog.kend[6,4] <- cor(cog.all.prop[,6], cog.all.prop[,4], method = "kendall")
cog.kend[6,5] <- cor(cog.all.prop[,6], cog.all.prop[,5], method = "kendall")
cog.kend[6,6] <- cor(cog.all.prop[,6], cog.all.prop[,6], method = "kendall")
cog.kend[6,7] <- cor(cog.all.prop[,6], cog.all.prop[,7], method = "kendall")
cog.kend[6,8] <- cor(cog.all.prop[,6], cog.all.prop[,8], method = "kendall")
cog.kend[6,9] <- cor(cog.all.prop[,6], cog.all.prop[,9], method = "kendall")
cog.kend[7,1] <- cor(cog.all.prop[,7], cog.all.prop[,1], method = "kendall")
cog.kend[7,2] <- cor(cog.all.prop[,7], cog.all.prop[,2], method = "kendall")
cog.kend[7,3] <- cor(cog.all.prop[,7], cog.all.prop[,3], method = "kendall")
cog.kend[7,4] <- cor(cog.all.prop[,7], cog.all.prop[,4], method = "kendall")
cog.kend[7,5] <- cor(cog.all.prop[,7], cog.all.prop[,5], method = "kendall")
cog.kend[7,6] <- cor(cog.all.prop[,7], cog.all.prop[,6], method = "kendall")
cog.kend[7,7] <- cor(cog.all.prop[,7], cog.all.prop[,7], method = "kendall")
cog.kend[7,8] <- cor(cog.all.prop[,7], cog.all.prop[,8], method = "kendall")
cog.kend[7,9] <- cor(cog.all.prop[,7], cog.all.prop[,9], method = "kendall")
cog.kend[8,1] <- cor(cog.all.prop[,8], cog.all.prop[,1], method = "kendall")
cog.kend[8,2] <- cor(cog.all.prop[,8], cog.all.prop[,2], method = "kendall")
cog.kend[8,3] <- cor(cog.all.prop[,8], cog.all.prop[,3], method = "kendall")
cog.kend[8,4] <- cor(cog.all.prop[,8], cog.all.prop[,4], method = "kendall")
cog.kend[8,5] <- cor(cog.all.prop[,8], cog.all.prop[,5], method = "kendall")
cog.kend[8,6] <- cor(cog.all.prop[,8], cog.all.prop[,6], method = "kendall")
cog.kend[8,7] <- cor(cog.all.prop[,8], cog.all.prop[,7], method = "kendall")
cog.kend[8,8] <- cor(cog.all.prop[,8], cog.all.prop[,8], method = "kendall")
cog.kend[8,9] <- cor(cog.all.prop[,8], cog.all.prop[,9], method = "kendall")
cog.kend[9,1] <- cor(cog.all.prop[,9], cog.all.prop[,1], method = "kendall")
cog.kend[9,2] <- cor(cog.all.prop[,9], cog.all.prop[,2], method = "kendall")
cog.kend[9,3] <- cor(cog.all.prop[,9], cog.all.prop[,3], method = "kendall")
cog.kend[9,4] <- cor(cog.all.prop[,9], cog.all.prop[,4], method = "kendall")
cog.kend[9,5] <- cor(cog.all.prop[,9], cog.all.prop[,5], method = "kendall")
cog.kend[9,6] <- cor(cog.all.prop[,9], cog.all.prop[,6], method = "kendall")
cog.kend[9,7] <- cor(cog.all.prop[,9], cog.all.prop[,7], method = "kendall")
cog.kend[9,8] <- cor(cog.all.prop[,9], cog.all.prop[,8], method = "kendall")
cog.kend[9,9] <- cor(cog.all.prop[,9], cog.all.prop[,9], method = "kendall")

#write.table(cog.kend, file="COG/cog.kendall.txt", sep="\t")
```

### COG UPGMA
```{r}
cog.kend <- read.table("COG/cog.kendall.txt", sep="\t")
cog.kend.fix <- cog.kend
cog.kend.fix[1,1] <- 0
cog.kend.fix[2,2] <- 0
cog.kend.fix[3,3] <- 0
cog.kend.fix[4,4] <- 0
cog.kend.fix[5,5] <- 0
cog.kend.fix[6,6] <- 0
cog.kend.fix[7,7] <- 0
cog.kend.fix[8,8] <- 0
cog.kend.fix[9,9] <- 0
cog.upgma <- upgma(cog.kend.fix)
plot(cog.upgma)


cog.nj <- NJ(data.frame(cog.kend.fix))
```
### COG DESeq2
```{r}
##### DESeq
# Use UN-NORMALIZED data!!!
# DESeq accounts for library size!!!

# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv')
row.names(meta) <- meta[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

cog.otu <- otu_table(cog.all, taxa_are_rows = T)
#cog.otu <- cog.otu + 1
cog.samp <- sample_data(meta)
rownames(cog.class) <- cog.class[,2]
cog.tax <- tax_table(cog.class)
row.names(cog.tax) <- row.names(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)

# Summer vs. Winter
sw.physeq <- subset_samples(cog.physeq, Season != "Spring")
test <- deSEQ(data=sw.physeq, valuetest= ~Season)
sum <- which(test$log2FoldChange < 0)
sum.cog <- rownames(test[sum,])
  # "COG0243" "COG1874" "COG2207" "COG2730" "COG3507" "COG3534" "COG3693" "COG3757" "COG3866" "COG4677" "COG5434"
win <- which(test$log2FoldChange > 0)
win.cog <- rownames(test[win,])
  # "COG1775" "COG3246"

# Spring vs. Summer
ss.physeq <- subset_samples(cog.physeq, Season != "Winter")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.cog <- rownames(test[spr,])
  # "COG0395" "COG0444" "COG1129" "COG1172" "COG1175" "COG3958" "COG3959" "COG4166"
sum <- which(test$log2FoldChange > 0)
sum.cog <- rownames(test[sum,])
  # "COG0243" "COG0437" "COG0457" "COG0810" "COG1409" "COG1538" "COG1596" "COG1629" "COG2885"

# Spring vs. Winter
sw.physeq <- subset_samples(cog.physeq, Season != "Summer")
test <- deSEQ(data=ss.physeq, valuetest= ~Season)
spr <- which(test$log2FoldChange < 0)
spr.cog <- rownames(test[spr,])
  # "COG0395" "COG0444" "COG1129" "COG1172" "COG1175" "COG3958" "COG3959" "COG4166"
win <- which(test$log2FoldChange > 0)
win.cog <- rownames(test[win,])
  # "COG0243" "COG0437" "COG0457" "COG0810" "COG1538" "COG1596" "COG1629" "COG2885"

# Fasting vs Feeding
test <- deSEQ(data=sw.physeq, valuetest= ~Diet)
fast <- which(test$log2FoldChange < 0)
fast.cog <- rownames(test[fast,])
  # "COG0156" "COG0247" "COG0339" "COG0457" "COG0612" "COG0659" "COG0760" "COG0767" "COG0774" "COG0776" "COG0795" "COG0810" "COG0811" "COG0841" "COG0845" "COG0847" "COG0848" "COG0859" "COG1043" "COG1137" "COG1409" "COG1452" "COG1463" "COG1538" "COG1566" "COG1596" "COG1629" "COG1663" "COG2067" "COG2095" "COG2259" "COG2814" "COG2825" "COG2834" "COG2885" "COG2982" "COG3169" "COG3176" "COG3187" "COG3203" "COG3468" "COG3525" "COG3637" "COG3712" "COG4105" "COG4409" "COG4566" "COG4775" "COG4783" "COG5010"
feed <- which(test$log2FoldChange > 0)
feed.cog <- rownames(test[feed,])
  # "COG0004" "COG0028" "COG0246" "COG0372" "COG0395" "COG0407" "COG0448" "COG0456" "COG0554" "COG0561" "COG0675" "COG0722" "COG0745" "COG0855" "COG1026" "COG1061" "COG1070" "COG1104" "COG1105" "COG1131" "COG1132" "COG1134" "COG1136" "COG1175" "COG1302" "COG1307" "COG1349" "COG1461" "COG1472" "COG1511" "COG1609" "COG1621" "COG1653" "COG1662" "COG1686" "COG1874" "COG1882" "COG1925" "COG2182" "COG2190" "COG2199" "COG2200" "COG2207" "COG2252" "COG2357" "COG2407" "COG2508" "COG2730" "COG2972" "COG3044" "COG3279" "COG3331" "COG3345" "COG3437" "COG3459" "COG3481" "COG3507" "COG3534" "COG3635" "COG3664" "COG3693" "COG3757" "COG3835" "COG3857" "COG3887" "COG3894" "COG3940" "COG3958" "COG3959" "COG4166" "COG4189" "COG4209" "COG4284" "COG4465" "COG4468" "COG4485" "COG4509" "COG4626" "COG4709" "COG4753" "COG4805" "COG4870" "COG5578"

### Identify Overrep COGs
cog.feed.id <- data.frame(feed.cog)
cog.feed.id$ID <- NA
for(i in 1:length(feed.cog)){
  test <- which(feed.cog[i] == cog.3715[,1])
  cog.feed.id[i,2] <- cog.3715[test,2]
}
#write.csv(cog.feed.id, 'COG/cog.feed.id.csv')
cog.fast.id <- data.frame(fast.cog)
for(i in 1:length(fast.cog)){
  test <- which(fast.cog[i] == cog.3723[,1])
  cog.fast.id[i,2] <- cog.3723[test,2]
}
#write.csv(cog.fast.id, 'COG/cog.fast.id.csv')

# Female vs Male
test <- deSEQ(data=cog.physeq, valuetest= ~Sex)
fem <- which(test$log2FoldChange < 0)
fem.cog <- rownames(test[fem,])
  # "COG0639" "COG0724" "COG1025" "COG1224" "COG1643" "COG2319" "COG5049" "COG5077" "COG5078" "COG5099" "COG5206" "COG5207" "COG5272" "COG5307" "COG5647"
feed <- which(test$log2FoldChange > 0)
feed.cog <- rownames(test[feed,])
  # "COG5184" "COG5279"
```





### COG Heatmap
```{r}
# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
cog.all.order <- cog.all
cog.all.order$X3715 <- cog.all$X3715
cog.all.order$X3717 <- cog.all$X3717
cog.all.order$X3744 <- cog.all$X3744
cog.all.order$X3723 <- cog.all$X3723
cog.all.order$X3733 <- cog.all$X3733
cog.all.order$X3772 <- cog.all$X3772
cog.all.order$X3734 <- cog.all$X3734
cog.all.order$X3773 <- cog.all$X3773
cog.all.order$X3775 <- cog.all$X3775
colnames(cog.all.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

meta.order <- read.csv('metadata.order.csv')
row.names(meta.order) <- meta.order[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

cog.otu <- otu_table(cog.all.order, taxa_are_rows = T)
#cog.otu <- cog.otu + 1
cog.samp <- sample_data(meta.order)
rownames(cog.class) <- cog.class[,2]
cog.tax <- tax_table(cog.class)
row.names(cog.tax) <- row.names(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)



cog.de.data <- phyloseq_to_deseq2(cog.physeq, ~Season)
cog.de <- DESeq(cog.de.data, test="Wald", fitType="local")

cog.res <- results(cog.de, pAdjustMethod = "BH")

### Extracting results: https://support.bioconductor.org/p/72765/

# Summer vs. Winter
cog.res.su.wi <- results(cog.de, contrast=c("Season", "Summer", "Winter"), pAdjustMethod = "BH")
cog.su.wi.sig <- cog.res.su.wi[which(cog.res.su.wi$padj < 0.05),]
cog.su.wi.sig.df <- cbind(as(cog.su.wi.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.su.wi.sig), ], "matrix"))
cog.su.wi.sum.sig <- which(cog.su.wi.sig.df$log2FoldChange < 0)
cog.su.wi.sum <- rownames(cog.su.wi.sig[cog.su.wi.sum.sig,])
  # Summer
  # "COG1775" "COG3246"
cog.su.wi.win.sig <- which(cog.su.wi.sig.df$log2FoldChange > 0)
cog.su.wi.win <- rownames(cog.su.wi.sig[cog.su.wi.win.sig,])
  # Winter
  # "COG1472" "COG1874" "COG2207" "COG2730" "COG3507" "COG3534" "COG3693" "COG3757" "COG3866" "COG3940" "COG4677" "COG5434"

# Summer vs. Spring
cog.res.su.sp <- results(cog.de, contrast=c("Season", "Spring", "Summer"), pAdjustMethod = "BH")
cog.su.sp.sig <- cog.res.su.sp[which(cog.res.su.sp$padj < 0.05),]
cog.su.sp.sig.df <- cbind(as(cog.su.sp.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.su.sp.sig), ], "matrix"))
cog.su.sp.spr.sig <- which(cog.su.sp.sig.df$log2FoldChange < 0)
cog.su.sp.spr <- rownames(cog.su.sp.sig[cog.su.sp.spr.sig,])
  # Spring
  # "COG0243" "COG0437" "COG0457" "COG0545" "COG0810" "COG0811" "COG1409" "COG1452" "COG1538" "COG1566" "COG1596" "COG1629" "COG2067" "COG2814" "COG2825" "COG2885" "COG3176" "COG3187" "COG4206" "COG4235" "COG4775"
cog.su.sp.sum.sig <- which(cog.su.sp.sig.df$log2FoldChange > 0)
cog.su.sp.sum <- rownames(cog.su.sp.sig[cog.su.sp.sum.sig,])
  # Summer
  # "COG0183" "COG0395" "COG0411" "COG0444" "COG0448" "COG1126" "COG1172" "COG1175" "COG1744" "COG1775" "COG3958" "COG3959" "COG4166" "COG4905"

# Spring vs Winter
cog.res.sp.wi <- results(cog.de, contrast=c("Season", "Spring", "Winter"), pAdjustMethod = "BH")
cog.sp.wi.sig <- cog.res.sp.wi[which(cog.res.sp.wi$padj < 0.05),]
cog.sp.wi.sig.df <- cbind(as(cog.sp.wi.sig, "data.frame"),
                          as(tax_table(cog.physeq)[rownames(cog.sp.wi.sig), ], "matrix"))
cog.sp.wi.spr.sig <- which(cog.sp.wi.sig.df$log2FoldChange < 0)
cog.sp.wi.spr <- rownames(cog.sp.wi.sig[cog.sp.wi.spr.sig,])
  # Spring
  # "COG0156" "COG0247" "COG0306" "COG0457" "COG0612" "COG0659" "COG0760" "COG0767" "COG0774" "COG0795" "COG0810" "COG0811" "COG0841" "COG0845" "COG0847" "COG0848" "COG0859" "COG1043" "COG1127" "COG1137" "COG1409" "COG1452" "COG1463" "COG1538" "COG1596" "COG1629" "COG1663" "COG1678" "COG2067" "COG2095" "COG2151" "COG2209" "COG2259" "COG2825" "COG2834" "COG2885" "COG2911" "COG2982" "COG3169" "COG3176" "COG3187" "COG3203" "COG3468" "COG3525" "COG3636" "COG3637" "COG3657" "COG4105" "COG4232" "COG4235" "COG4409" "COG4566" "COG4775" "COG4783"
cog.sp.wi.win.sig <- which(cog.sp.wi.sig.df$log2FoldChange > 0)
cog.sp.wi.win <- rownames(cog.sp.wi.sig[cog.sp.wi.win.sig,])
  # Winter
  # "COG0004" "COG0028" "COG0119" "COG0246" "COG0395" "COG0407" "COG0554" "COG0561" "COG0722" "COG0745" "COG0855" "COG1026" "COG1051" "COG1061" "COG1070" "COG1104" "COG1105" "COG1131" "COG1132" "COG1136" "COG1161" "COG1172" "COG1175" "COG1263" "COG1302" "COG1307" "COG1349" "COG1472" "COG1511" "COG1609" "COG1621" "COG1653" "COG1874" "COG1879" "COG1882" "COG1925" "COG2182" "COG2190" "COG2199" "COG2207" "COG2252" "COG2357" "COG2407" "COG2723" "COG2730" "COG2816" "COG2972" "COG3044" "COG3153" "COG3279" "COG3331" "COG3345" "COG3481" "COG3507" "COG3534" "COG3635" "COG3664" "COG3693" "COG3757" "COG3835" "COG3857" "COG3866" "COG3887" "COG3894" "COG3940" "COG3959" "COG3973" "COG4086" "COG4166" "COG4189" "COG4209" "COG4284" "COG4465" "COG4468" "COG4485" "COG4509" "COG4709" "COG4753" "COG4805" "COG4862" "COG4870" "COG4962" "COG5578"

# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

### Pull out top 2 (largest log fold change while sig) for each comparison
cog.su.wi.sum.df <- cog.su.wi.sig.df[cog.su.wi.sum.sig,]
cog.su.wi.sum.df <- cog.su.wi.sum.df[order(cog.su.wi.sum.df$log2FoldChange),]

cog.su.wi.win.df <- cog.su.wi.sig.df[cog.su.wi.win.sig,]
cog.su.wi.win.df <- cog.su.wi.win.df[order(cog.su.wi.win.df$log2FoldChange),]

cog.su.sp.spr.df <- cog.su.sp.sig.df[cog.su.sp.spr.sig,]
cog.su.sp.spr.df <- cog.su.sp.spr.df[order(cog.su.sp.spr.df$log2FoldChange),]
cog.su.sp.sum.df <- cog.su.sp.sig.df[cog.su.sp.sum.sig,]
cog.su.sp.sum.df <- cog.su.sp.sum.df[order(cog.su.sp.sum.df$log2FoldChange),]
cog.sp.wi.spr.df <- cog.sp.wi.sig.df[cog.sp.wi.spr.sig,]
cog.sp.wi.spr.df <- cog.sp.wi.spr.df[order(cog.sp.wi.spr.df$log2FoldChange),]
cog.sp.wi.win.df <- cog.sp.wi.sig.df[cog.sp.wi.win.sig,]
cog.sp.wi.win.df <- cog.sp.wi.win.df[order(cog.sp.wi.win.df$log2FoldChange),]

cog.sig <- rbind(cog.su.wi.sum.df, cog.su.wi.win.df, cog.su.sp.spr.df, cog.su.sp.sum.df, cog.sp.wi.spr.df, cog.sp.wi.win.df)
cog.sig$log2FoldChange[which(cog.sig$log2FoldChange < 0)] <-
  cog.sig$log2FoldChange[which(cog.sig$log2FoldChange < 0)] * -1
cog.sig <- cog.sig[order(cog.sig$log2FoldChange, decreasing=T),]
cog.use <-rownames(cog.sig)[1:10] 


#select <- order(rowMeans(counts(cog.de, normalized=T)), decreasing=T)[1:20]
  # For top 20 COGs
cog.rows <- rownames(cog.de)
select <- rep(1,10)
for(i in 1:length(cog.use)){
  save <- which(cog.rows == cog.use[i])
  select[i] <- save
}
cog.de.df <- as.data.frame(colData(cog.de)[,c("Season")])
rownames(cog.de.df) <- meta$X
colnames(cog.de.df) <- "Season"
cog.de.df$Season <- ordered(cog.de.df$Season, levels=c("Summer", "Winter", "Spring"))


cog.de.df <- data.frame(c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"))
rownames(cog.de.df) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")
colnames(cog.de.df) <- "Season"
cog.de.df$Season <- ordered(cog.de.df$Season, levels=c("Summer", "Winter", "Spring"))
cog.de$X <- ordered(cog.de$X, levels=c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775"))


cog.ntd <- normTransform(cog.de)
  # DESeq test automatically transforms data
  # To visualize our data, we need to run this step separately
  # I use normTransform because I run DESeq with local fit
      # Local fit does a transformation similar to normTransform
  # Otherwise you should use vst instead
  # See link above for more info

cog.ntd.assay <- data.frame(assay(cog.ntd)[select,])
cog.ntd.assay.order <- cog.ntd.assay
cog.ntd.assay.order[,1] <- cog.ntd.assay$X3715
cog.ntd.assay.order[,2] <- cog.ntd.assay$X3717
cog.ntd.assay.order[,3] <- cog.ntd.assay$X3744
cog.ntd.assay.order[,4] <- cog.ntd.assay$X3723
cog.ntd.assay.order[,5] <- cog.ntd.assay$X3733
cog.ntd.assay.order[,6] <- cog.ntd.assay$X3772
cog.ntd.assay.order[,7] <- cog.ntd.assay$X3734
cog.ntd.assay.order[,8] <- cog.ntd.assay$X3773
cog.ntd.assay.order[,9] <- cog.ntd.assay$X3775
colnames(cog.ntd.assay.order) <- c("3715", "3717", "3744", "3723", "3733", "3772", "3734", "3773", "3775")

tiff("COG_DESeq.tiff", width=6.2, height=5.5, units="in", res=600)
pheatmap(cog.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, treeheight_col = 0,
         show_colnames=F, border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255),
         labels_col = c("Summer", "Summer", "Summer", "Winter", "Winter", "Winter", "Spring", "Spring", "Spring"), annotation_legend = F,
         annotation_col=cog.de.df,
         annotation_colors = list(Season = c(Summer = "#00cccc", Winter = "#008484", Spring = "#004f4f"))
)
dev.off()


tiff("COG_DESeq1.tiff", width=3, height=2.25, units="in", res=600)
pheatmap(cog.ntd.assay.order, cluster_rows=F, show_rownames=F, cluster_cols=F, show_colnames=F, 
         border_color = "gray40", fontsize = 14, width = 3, height=3,
         color = colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(255)
)
dev.off()




### Sample Clustering: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

cog.kend <- read.table("COG/cog.kendall.txt", sep="\t")
cog.kend.fix <- cog.kend
cog.kend.fix[1,1] <- 0
cog.kend.fix[2,2] <- 0
cog.kend.fix[3,3] <- 0
cog.kend.fix[4,4] <- 0
cog.kend.fix[5,5] <- 0
cog.kend.fix[6,6] <- 0
cog.kend.fix[7,7] <- 0
cog.kend.fix[8,8] <- 0
cog.kend.fix[9,9] <- 0
pheatmap(cog.kend.fix, clustering_distance_rows=cog.kend.fix, clustering_distance_cols=cog.kend.fix)

cog.dist <- dist(t(assay(cog.ntd)))
cog.dist.matrix <- as.matrix(cog.dist)
rownames(cog.dist.matrix) <- paste(cog.ntd$Season)
colnames(cog.dist.matrix) <- NULL


pheatmap(cog.dist.matrix, clustering_distance_rows=cog.dist, clustering_distance_cols=cog.dist,
         col=colorRampPalette(rev(brewer.pal(9, "YlOrRd")))(100), annotation_rows=T, annotation_cols=T)
  # Default clustering method = complete


cog.pca <- plotPCA(cog.ntd, intgroup="Season", returnData=T)
cog.pca.var <- round(100 * attr(cog.pca, "Percent_Variation"))
ggplot(cog.pca, aes(x=PC1, y=PC2, color=Season)) +
  geom_point(size=2) +
  xlab(paste0("PC1: ", cog.pca.var[1], "% variance")) +
  ylab(paste0("PC2: ", cog.pca.var[2], "% variance")) +
  coord_fixed()

# Prepare phyloseq object
cog.size.fac <- estimateSizeFactors(cog.de)
cog.norm.deseq <- counts(cog.size.fac, normalized=TRUE)
cog.otu.deseq <- otu_table(cog.norm.deseq, taxa_are_rows = T)
cog.samp <- sample_data(meta)
cog.tax <- tax_table(cog.class)
rownames(cog.tax) <- rownames(cog.class)
cog.deseq <- merge_phyloseq(cog.otu.deseq, cog.samp, cog.tax)
sample_data(cog.deseq)$Season <- ordered(sample_data(cog.deseq)$Season, levels=c("Summer", "Winter", "Spring"))

nmds.cog.deseq <- ordinate(physeq=cog.deseq, method="NMDS", distance="bray")
  # Run 20 stress 0.06470567

### SEASON ELLIPSES
sample_data(cog.deseq)$Elip <- paste(sample_data(cog.deseq)$Season)
cog.deseq.elip.sum <- data.frame(MDS1 = nmds.cog.deseq$points[,1], MDS2 = nmds.cog.deseq$points[,2],
                           Group = sample_data(cog.deseq)$Season)
plot.new()
cog.deseq.elip.ord <- ordiellipse(nmds.cog.deseq, sample_data(cog.deseq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.deseq.elip.sum.df <- data.frame()
for(i in levels(cog.deseq.elip.sum$Group)){
  cog.deseq.elip.sum.df <- rbind(cog.deseq.elip.sum.df,
                           cbind(as.data.frame(with(cog.deseq.elip.sum[cog.deseq.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.deseq.elip.ord[[i]]$cov,
                                                                    cog.deseq.elip.ord[[i]]$center,
                                                                    cog.deseq.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
#tiff("COG_NMDS.tiff", width=3.5, height=3.5, units="in", res=600)
plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
#dev.off()


cog.nmds.plot <- plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5))
cog.nmds.legend <- g_legend(cog.nmds.plot)
cog.nmds.plot <- plot_ordination(physeq=cog.deseq, ordination=nmds.cog.deseq, color="Season") +
  geom_path(data=cog.deseq.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  theme(axis.title.x = element_text(size=10),
        axis.title.y = element_text(size=10),
        legend.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.margin=margin(0,0,0,0),
        legend.box.margin=margin(-5,-5,-5,-5),
        plot.margin = unit(c(0.1, 1, 0.1, 0.1), "cm")) +
  guides(color=F)


### PERMANOVA
cog.deseq.sampdf <- data.frame(sample_data(cog.deseq))
cog.deseq.bray<- phyloseq::distance(physeq=cog.deseq, method="bray")

## Season
adonis(cog.deseq.bray~ Season, data=cog.deseq.sampdf)
  # p-value = 0.003
beta.tax = betadisper(d=cog.deseq.bray, group=cog.deseq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.721


## Diet
adonis(cog.deseq.bray~ Diet, data=cog.deseq.sampdf)
  # p-value = 0.137
beta.tax = betadisper(d=cog.deseq.bray, group=cog.deseq.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.12




```

### COG Bray-Curtis
```{r}
# Prepare physeq object
cog.all <- read.csv('COG/cog_counts.csv')
row.names(cog.all) <- cog.all[,1]
cog.all <- cog.all[,-1]
colnames(cog.all) <- c("3715", "3717", "3723", "3733", "3734", "3744", "3772", "3773", "3775")

meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")
row.names(meta) <- meta[,1]

cog.class <- read.csv('COG/cog_classification.csv')
rownames(cog.class) <- cog.class[,2]
#class <- class[,-1]

### Normalize
norm.factor <- data.frame(c(0.903, 0.715, 0.936, 0.904, 0.662, 0.902, 1.000, 0.849, 0.826))
cog.all.norm <- cog.all
for(i in 1:9){
  cog.all.norm[,i] <- round(cog.all.norm[,i]*norm.factor[i,1])
}
# Remove pathways = 0
for(i in 1:nrow(cog.all.norm)){
  print(sum(cog.all.norm[i,]))
}
  # All > 0

# Prepare phyloseq object
cog.otu <- otu_table(cog.all, taxa_are_rows = T)
cog.samp <- sample_data(meta)
cog.tax <- tax_table(cog.class)
rownames(cog.tax) <- rownames(cog.class)
cog.physeq <- merge_phyloseq(cog.otu, cog.samp, cog.tax)
sample_data(cog.physeq)$Season <- ordered(sample_data(cog.physeq)$Season, levels=c("Summer", "Winter", "Spring"))


nmds.cog <- ordinate(physeq=cog.physeq, method="NMDS", distance="bray")
  # stress = 0.05592207 

plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Season")
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Diet")
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Sex")

### DIET ELLIPSES
sample_data(cog.physeq)$Elip <- paste(sample_data(cog.physeq)$Diet)
cog.elip.sum <- data.frame(MDS1 = nmds.cog$points[,1], MDS2 = nmds.cog$points[,2],
                           Group = sample_data(cog.physeq)$Elip)
plot.new()
cog.elip.ord <- ordiellipse(nmds.cog, sample_data(cog.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.elip.sum.df <- data.frame()
for(i in levels(cog.elip.sum$Group)){
  cog.elip.sum.df <- rbind(cog.elip.sum.df,
                           cbind(as.data.frame(with(cog.elip.sum[cog.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.elip.ord[[i]]$cov,
                                                                    cog.elip.ord[[i]]$center,
                                                                    cog.elip.ord[[i]]$scale))), group=i
                                 ))

}


### SEASON ELLIPSES
sample_data(cog.physeq)$Elip <- paste(sample_data(cog.physeq)$Season)
cog.elip.sum <- data.frame(MDS1 = nmds.cog$points[,1], MDS2 = nmds.cog$points[,2],
                           Group = sample_data(cog.physeq)$Season)
plot.new()
cog.elip.ord <- ordiellipse(nmds.cog, sample_data(cog.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
cog.elip.sum.df <- data.frame()
for(i in levels(cog.elip.sum$Group)){
  cog.elip.sum.df <- rbind(cog.elip.sum.df,
                           cbind(as.data.frame(with(cog.elip.sum[cog.elip.sum$Group==i,],
                                                    veganCovEllipse(cog.elip.ord[[i]]$cov,
                                                                    cog.elip.ord[[i]]$center,
                                                                    cog.elip.ord[[i]]$scale))), group=i
                                 ))

}


### Plot
plot_ordination(physeq=cog.physeq, ordination=nmds.cog, color="Season") +
  geom_path(data=cog.elip.sum.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  scale_color_manual(values=c("sd", "asdf", "Asdf"))




### PERMANOVA
cog.sampdf <- data.frame(sample_data(cog.physeq))
cog.bray<- phyloseq::distance(physeq=cog.physeq, method="bray")

## Season
adonis(cog.bray~ Season, data=cog.sampdf)
  # p-value = 0.164
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.877

## Diet
adonis(cog.bray~ Diet, data=cog.sampdf)
  # p-value = 0.237
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.426

## Sex
adonis(cog.bray~ Sex, data=cog.sampdf)
  # p-value = 0.372
beta.tax = betadisper(d=cog.bray, group=cog.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.58

### ANOSIM
anosim(cog.bray, meta$Season, permutations=1000)
  # p-value = 0.1852
anosim(cog.bray, meta$Diet, permutations=1000)
  # p-value = 0.1049
anosim(cog.bray, meta$Sex, permutations=1000)
  # p-value = 0.92308
```

### COG Jaccard
```{r}
jac.cog <- ordinate(physeq=cog.physeq, method="NMDS", distance="jaccard")
  # stress = 0.04125187
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Season")
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Diet")
plot_ordination(physeq=cog.physeq, ordination=jac.cog, color="Sex")

### PERMANOVA
cog.sampdf <- data.frame(sample_data(cog.physeq))
cog.jac <- phyloseq::distance(physeq=cog.physeq, method="jaccard")

## Season
adonis(cog.jac~ Season, data=cog.sampdf)
  # p-value = 0.148
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.867

## Diet
adonis(cog.jac~ Diet, data=cog.sampdf)
  # p-value = 0.17
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.398

## Sex
adonis(cog.jac~ Sex, data=cog.sampdf)
  # p-value = 0.509
beta.tax = betadisper(d=cog.jac, group=cog.sampdf$Sex)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.585

### ANOSIM
anosim(cog.jac, cog.sampdf$Season, permutations=1000)
  # p-value = 0.14386
anosim(cog.jac, cog.sampdf$Diet, permutations=1000)
  # p-value = 0.25874
anosim(cog.jac, cog.sampdf$Sex, permutations=1000)
  # p-value = 0.9001
```

### COG SIMPER
```{r}
### Separate out pairw
simper.pretty(cog.all.norm, metrics = meta, interesting = c('Season', 'Diet', 'Sex'), perc_cutoff = 0.5, low_cutoff = 'y', low_val = 0.01, 'cog')

cog.simper <- read.csv("cog_clean_simper.csv")

kruskal.pretty(otu = cog.all.norm, metrics = meta, csv = cog.simper, interesting = c('Season', 'Diet', 'Sex'), output_name = 'cog.kw')

cog.simper.kw <- read.csv("cog.kw_krusk_simper.csv")
```



### Poster combined NMDS (COG & KEGG)
```{r}
tiff("NMDS_cog_kegg.tiff", width = 7.5, height = 3, units = "in", res = 600)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(1,3, widths=c(1,1,0.3)))))
print(cog.nmds.plot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(kegg.nmds.plot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
cog.nmds.legend$vp <- viewport(layout.pos.row = 1, layout.pos.col = 3)
grid.draw(cog.nmds.legend)
dev.off()
```


